// @beeline/design-tokens/js/tokens/components/chips/chips.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const chipsHeight = "32px";
export const chipsPaddingLeft = "12px";
export const chipsPaddingRight = "12px";
export const chipsBorderWidth = "1px";
export const chipsBorderStyle = "solid";
export const chipsBorderRadius = "30px";
export const chipsBorderColorFocus = "var(--color-border-focus, #212121)";
export const chipsOpacityDisabled = 0.48;
export const chipsShadowDragged = "0px 6px 38px rgba(0, 0, 0, 0.16), 0px 0px 10px rgba(0, 0, 0, 0.08)";
export const chipsFontSize = "15px";
export const chipsFontWeight = 400;
export const chipsFontLineHeight = "18px";
export const chipsFontLetterSpacing = "0.2px";
export const chipsIconSize = "20px";
export const chipsIconSpacing = "8px";
export const chipsAvatarSpacing = "8px";
export const chipsRemoveSpacing = "8px";
export const chipsTextFontSize = "15px";
export const chipsTextFontWeight = 400;
export const chipsTextFontLineHeight = "18px";
export const chipsTextFontLetterSpacing = "0.2px";
export const chipsBaseBackgroundColor = "var(--color-control-background, rgba(25, 28, 52, 0.1))";
export const chipsBaseBackgroundColorHover = "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))";
export const chipsBaseBackgroundColorPressed = "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))";
export const chipsBaseBackgroundColorDragged = "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))";
export const chipsBaseHoverBackgroundColor = "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))";
export const chipsBasePressedBackgroundColor = "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))";
export const chipsBaseDraggedBackgroundColor = "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))";
export const chipsSelectedBackgroundColor = "var(--color-background-brand, #fdd835)";
export const chipsSelectedBackgroundColorHover = "var(--color-background-brand-hover, #fdc435)";
export const chipsSelectedBackgroundColorPressed = "var(--color-background-brand-pressed, #fdb435)";
export const chipsSelectedBackgroundColorDragged = "var(--color-background-brand-pressed, #fdb435)";
export const chipsSelectedHoverBackgroundColor = "var(--color-background-brand-hover, #fdc435)";
export const chipsSelectedPressedBackgroundColor = "var(--color-background-brand-pressed, #fdb435)";
export const chipsSelectedDraggedBackgroundColor = "var(--color-background-brand-pressed, #fdb435)";
export const chipsOnlyTextPaddingLeft = "12px";
export const chipsOnlyTextPaddingRight = "12px";
export const chipsWithIconPaddingLeft = "8px";
export const chipsWithAvatarPaddingLeft = "8px";
export const chipsRemovablePaddingRight = "8px";
export const chipsFocusedBorderWidth = "1px";
export const chipsFocusedBorderStyle = "solid";
export const chipsFocusedBorderColor = "var(--color-border-focus, #212121)";
export const chipsDisabledOpacity = 0.48;
export const chipsDraggedShadow = "0px 6px 38px rgba(0, 0, 0, 0.16), 0px 0px 10px rgba(0, 0, 0, 0.08)";
export const chipsGroupSpacing = "8px";
