// @beeline/design-tokens/js/tokens/components/chips/chips.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	chipsHeight: "32px",
	chipsPaddingLeft: "12px",
	chipsPaddingRight: "12px",
	chipsBorderWidth: "1px",
	chipsBorderStyle: "solid",
	chipsBorderRadius: "30px",
	chipsBorderColorFocus: "var(--color-border-focus, #212121)",
	chipsOpacityDisabled: 0.48,
	chipsShadowDragged: "0px 6px 38px rgba(0, 0, 0, 0.16), 0px 0px 10px rgba(0, 0, 0, 0.08)",
	chipsFontSize: "15px",
	chipsFontWeight: 400,
	chipsFontLineHeight: "18px",
	chipsFontLetterSpacing: "0.2px",
	chipsIconSize: "20px",
	chipsIconSpacing: "8px",
	chipsAvatarSpacing: "8px",
	chipsRemoveSpacing: "8px",
	chipsTextFontSize: "15px",
	chipsTextFontWeight: 400,
	chipsTextFontLineHeight: "18px",
	chipsTextFontLetterSpacing: "0.2px",
	chipsBaseBackgroundColor: "var(--color-control-background, rgba(25, 28, 52, 0.1))",
	chipsBaseBackgroundColorHover: "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))",
	chipsBaseBackgroundColorPressed: "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))",
	chipsBaseBackgroundColorDragged: "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))",
	chipsBaseHoverBackgroundColor: "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))",
	chipsBasePressedBackgroundColor: "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))",
	chipsBaseDraggedBackgroundColor: "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))",
	chipsSelectedBackgroundColor: "var(--color-background-brand, #fdd835)",
	chipsSelectedBackgroundColorHover: "var(--color-background-brand-hover, #fdc435)",
	chipsSelectedBackgroundColorPressed: "var(--color-background-brand-pressed, #fdb435)",
	chipsSelectedBackgroundColorDragged: "var(--color-background-brand-pressed, #fdb435)",
	chipsSelectedHoverBackgroundColor: "var(--color-background-brand-hover, #fdc435)",
	chipsSelectedPressedBackgroundColor: "var(--color-background-brand-pressed, #fdb435)",
	chipsSelectedDraggedBackgroundColor: "var(--color-background-brand-pressed, #fdb435)",
	chipsOnlyTextPaddingLeft: "12px",
	chipsOnlyTextPaddingRight: "12px",
	chipsWithIconPaddingLeft: "8px",
	chipsWithAvatarPaddingLeft: "8px",
	chipsRemovablePaddingRight: "8px",
	chipsFocusedBorderWidth: "1px",
	chipsFocusedBorderStyle: "solid",
	chipsFocusedBorderColor: "var(--color-border-focus, #212121)",
	chipsDisabledOpacity: 0.48,
	chipsDraggedShadow: "0px 6px 38px rgba(0, 0, 0, 0.16), 0px 0px 10px rgba(0, 0, 0, 0.08)",
	chipsGroupSpacing: "8px",
};
