// @beeline/design-system-react/extracted-styles/esm/components/Store/Store.js
import { jsx } from 'react/jsx-runtime';
import { classNames } from '../../helpers/classnames.js';
import AppGalleryBlackEn from './icons/AppGalleryBlackEn.js';
import AppGalleryBlackRu from './icons/AppGalleryBlackRu.js';
import AppGalleryWhiteEn from './icons/AppGalleryWhiteEn.js';
import AppGalleryWhiteRu from './icons/AppGalleryWhiteRu.js';
import AppStoreBlackEn from './icons/AppStoreBlackEn.js';
import AppStoreBlackRu from './icons/AppStoreBlackRu.js';
import AppStoreWhiteEn from './icons/AppStoreWhiteEn.js';
import AppStoreWhiteRu from './icons/AppStoreWhiteRu.js';
import GooglePlayBlackEn from './icons/GooglePlayBlackEn.js';
import GooglePlayBlackRu from './icons/GooglePlayBlackRu.js';
import GooglePlayWhiteEn from './icons/GooglePlayWhiteEn.js';
import GooglePlayWhiteRu from './icons/GooglePlayWhiteRu.js';

const ariaLabels = {
    appstore: {
        ru: 'Скачать в App Store',
        en: 'Download on the App Store',
    },
    googleplay: {
        ru: 'Скачать в Google Play',
        en: 'Get it on Google Play',
    },
    appgallery: {
        ru: 'Скачать в AppGallery',
        en: 'Get it on AppGallery',
    },
};
const iconsMap = {
    appstore: {
        black: {
            ru: AppStoreBlackRu,
            en: AppStoreBlackEn,
        },
        white: {
            ru: AppStoreWhiteRu,
            en: AppStoreWhiteEn,
        },
    },
    googleplay: {
        black: {
            ru: GooglePlayBlackRu,
            en: GooglePlayBlackEn,
        },
        white: {
            ru: GooglePlayWhiteRu,
            en: GooglePlayWhiteEn,
        },
    },
    appgallery: {
        black: {
            ru: AppGalleryBlackRu,
            en: AppGalleryBlackEn,
        },
        white: {
            ru: AppGalleryWhiteRu,
            en: AppGalleryWhiteEn,
        },
    },
};
const Store = ({ store, theme, lang, href, className, dataTestId = 'Store', ariaLabel, ...rest }) => {
    const StoreIcon = iconsMap[store][theme][lang];
    const label = ariaLabel || ariaLabels[store][lang];
    return (jsx("a", { ...rest, href: href, className: classNames('dsb_store', `dsb_store--${store}`, `dsb_store--${theme}`, `dsb_store--${lang}`, className), "data-testid": dataTestId, "aria-label": label, children: jsx(StoreIcon, { "aria-hidden": "true" }) }));
};

export { Store };
