// @beeline/design-system-react/extracted-styles/esm/components/Button/Button.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { forwardRef } from 'react';
import { classNames } from '../../helpers/classnames.js';

const VARIANT_ALIAS_MAP = {
    primary: 'contained',
    secondary: 'outlined',
    ghost: 'plain',
    inverse: 'overlay',
};
function resolveVariant(variant) {
    return VARIANT_ALIAS_MAP[variant] ?? variant;
}
const Button = forwardRef(({ className, children, size = 'small', variant = 'secondary', disabled, startIcon, endIcon, fullWidth = false, href, dataTestId = 'Button', ...props }, ref) => {
    const resolvedVariant = resolveVariant(variant);
    const buttonClassNames = classNames('dsb_button', `dsb_button__${size}`, `dsb_button__${resolvedVariant}`, {
        'dsb_button__start-icon': !!startIcon,
        'dsb_button__end-icon': !!endIcon,
        'dsb_button__icon-button': !children,
        dsb_button__disabled: disabled,
        'dsb_button__full-width': fullWidth,
        dsb_button__link: !!href,
    }, className);
    if (href) {
        return (jsxs("a", { ...props, href: href, className: buttonClassNames, ref: ref, role: "button", "data-testid": dataTestId, children: [startIcon && (jsx("div", { className: "dsb_button-start-icon", "aria-hidden": "true", children: startIcon })), children, endIcon && (jsx("div", { className: "dsb_button-end-icon", "aria-hidden": "true", children: endIcon }))] }));
    }
    return (jsxs("button", { "data-testid": dataTestId, ...props, className: buttonClassNames, disabled: disabled, ref: ref, children: [startIcon && (jsx("div", { className: "dsb_button-start-icon", "aria-hidden": "true", children: startIcon })), children, endIcon && (jsx("div", { className: "dsb_button-end-icon", "aria-hidden": "true", children: endIcon }))] }));
});

export { Button, resolveVariant };
