// @beeline/design-tokens/js/tokens/components/banner/banner.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const bannerPadding = "16px";
export const bannerBorderRadius = "16px";
export const bannerTextFontSize = "15px";
export const bannerTextFontWeight = 400;
export const bannerTextFontLineHeight = "18px";
export const bannerTextFontLetterSpacing = "0.2px";
export const bannerTextColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const bannerIconSize = "20px";
export const bannerIconSpacing = "16px";
export const bannerActionsSpacing = "8px";
export const bannerActionsSideMarginLeft = "24px";
export const bannerActionsBelowMarginTop = "16px";
export const bannerCloseButtonMarginLeft = "24px";
export const bannerNeutralBackgroundColor = "var(--color-status-neutral-background, #f1f1f3)";
export const bannerNeutralIconColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const bannerErrorBackgroundColor = "var(--color-status-error-background, #ffecef)";
export const bannerErrorIconColor = "var(--color-status-error, #ff5555)";
export const bannerInfoBackgroundColor = "var(--color-status-info-background, #e3f2ff)";
export const bannerInfoIconColor = "var(--color-status-info, #1a73e8)";
export const bannerSuccessBackgroundColor = "var(--color-status-success-background, #e7f6eb)";
export const bannerSuccessIconColor = "var(--color-status-success, #2ca853)";
export const bannerWarningBackgroundColor = "var(--color-status-warning-background, #fff4e1)";
export const bannerWarningIconColor = "var(--color-status-warning, #ff9419)";
