// @beeline/design-tokens/js/tokens/components/banner/banner.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	bannerPadding: "16px",
	bannerBorderRadius: "16px",
	bannerTextFontSize: "15px",
	bannerTextFontWeight: 400,
	bannerTextFontLineHeight: "18px",
	bannerTextFontLetterSpacing: "0.2px",
	bannerTextColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	bannerIconSize: "20px",
	bannerIconSpacing: "16px",
	bannerActionsSpacing: "8px",
	bannerActionsSideMarginLeft: "24px",
	bannerActionsBelowMarginTop: "16px",
	bannerCloseButtonMarginLeft: "24px",
	bannerNeutralBackgroundColor: "var(--color-status-neutral-background, #f1f1f3)",
	bannerNeutralIconColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	bannerErrorBackgroundColor: "var(--color-status-error-background, #ffecef)",
	bannerErrorIconColor: "var(--color-status-error, #ff5555)",
	bannerInfoBackgroundColor: "var(--color-status-info-background, #e3f2ff)",
	bannerInfoIconColor: "var(--color-status-info, #1a73e8)",
	bannerSuccessBackgroundColor: "var(--color-status-success-background, #e7f6eb)",
	bannerSuccessIconColor: "var(--color-status-success, #2ca853)",
	bannerWarningBackgroundColor: "var(--color-status-warning-background, #fff4e1)",
	bannerWarningIconColor: "var(--color-status-warning, #ff9419)",
};
