// @beeline/design-tokens/js/tokens/components/fileUploader/fileUploader.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const fileUploaderHeight = "104px";
export const fileUploaderPadding = "16px";
export const fileUploaderBorderWidth = "1px";
export const fileUploaderBorderRadius = "12px";
export const fileUploaderBorderColor = "var(--color-border, rgba(25, 28, 52, 0.18))";
export const fileUploaderTextFontSize = "15px";
export const fileUploaderTextFontWeight = 400;
export const fileUploaderTextFontLineHeight = "18px";
export const fileUploaderTextFontLetterSpacing = "0.2px";
export const fileUploaderHelperTextPaddingLeft = "16px";
export const fileUploaderHelperTextMarginTop = "4px";
export const fileUploaderHelperTextColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const fileUploaderHelperTextFontSize = "13px";
export const fileUploaderHelperTextFontWeight = 400;
export const fileUploaderHelperTextFontLineHeight = "16px";
export const fileUploaderHelperTextFontLetterSpacing = "0.2px";
export const fileUploaderHoverBackgroundColor = "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))";
export const fileUploaderDragOverBorderColor = "var(--color-status-info, #1a73e8)";
export const fileUploaderDragOverBackgroundColor = "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))";
export const fileUploaderDisabledBackgroundColor = "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))";
export const fileUploaderDisabledOpacity = 0.48;
export const fileUploaderErrorBackgroundColor = "var(--color-control-background-error, rgba(255, 85, 85, 0.08))";
export const fileUploaderErrorBorderColor = "var(--color-status-error, #ff5555)";
export const fileUploaderErrorHelperTextColor = "var(--color-status-error, #ff5555)";
