// @beeline/design-tokens/js/tokens/components/fileUploader/fileUploader.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	fileUploaderHeight: "104px",
	fileUploaderPadding: "16px",
	fileUploaderBorderWidth: "1px",
	fileUploaderBorderRadius: "12px",
	fileUploaderBorderColor: "var(--color-border, rgba(25, 28, 52, 0.18))",
	fileUploaderTextFontSize: "15px",
	fileUploaderTextFontWeight: 400,
	fileUploaderTextFontLineHeight: "18px",
	fileUploaderTextFontLetterSpacing: "0.2px",
	fileUploaderHelperTextPaddingLeft: "16px",
	fileUploaderHelperTextMarginTop: "4px",
	fileUploaderHelperTextColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	fileUploaderHelperTextFontSize: "13px",
	fileUploaderHelperTextFontWeight: 400,
	fileUploaderHelperTextFontLineHeight: "16px",
	fileUploaderHelperTextFontLetterSpacing: "0.2px",
	fileUploaderHoverBackgroundColor: "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))",
	fileUploaderDragOverBorderColor: "var(--color-status-info, #1a73e8)",
	fileUploaderDragOverBackgroundColor: "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))",
	fileUploaderDisabledBackgroundColor: "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))",
	fileUploaderDisabledOpacity: 0.48,
	fileUploaderErrorBackgroundColor: "var(--color-control-background-error, rgba(255, 85, 85, 0.08))",
	fileUploaderErrorBorderColor: "var(--color-status-error, #ff5555)",
	fileUploaderErrorHelperTextColor: "var(--color-status-error, #ff5555)",
};
