// @beeline/design-system-react/build/components/DatePicker/utils.js
import { MaskedRange } from 'imask';
import { formatWithTimeZone, formatDateOnly } from './convertor.js';
import parse from '../../node_modules/date-fns/esm/parse/index.js';
import isValid from '../../node_modules/date-fns/esm/isValid/index.js';

const EMPTY_DATE = '';
const DATE_LENGTH = 8;
const RANGE_DELIMETER = ' – ';
const ISO_FORMAT_PATTERN = /^(?<year>\d{4})-(?<month>\d{2})-(?<day>\d{2}.*)$/;
const RU_FORMAT_PATTERN_F = (enableTimePicker) => new RegExp('^' + // Начало строки
    '(?<day>\\d{2})\\.' + // День
    '(?<month>\\d{2})\\.' + // Месяц
    '(?<year>\\d{4})' + // Год
    (enableTimePicker // Если время включено
        ? '\\s+' + // Пробел
            '(?<hours>\\d{2}):' + // Часы
            '(?<minutes>\\d{2})' // Минуты
        : '') +
    '$');
const isRangeMode = (value) => Array.isArray(value);
const isFilledDate = (value, options = {}) => {
    const { pattern = ISO_FORMAT_PATTERN, isPartial = false } = options;
    if (isRangeMode(value)) {
        return isPartial ? value.some(date => date && pattern.test(date)) : value.every(date => date && pattern.test(date));
    }
    return pattern.test(value);
};
const isEmptyDate = (value) => {
    if (isRangeMode(value)) {
        return !value.join('');
    }
    return !value;
};
const formatDateToISOWithTimeZone = (value, currentHour = '00', currentMinute = '00', currentSecond = '00', enableTimePicker) => {
    const formatToISO = (date) => {
        const pattern = RU_FORMAT_PATTERN_F(enableTimePicker);
        const match = pattern.exec(date);
        if (!match || !match.groups) {
            return '';
        }
        const { day, month, year, hours, minutes } = match.groups;
        // Используем время из ввода или переданное время
        const hour = hours || currentHour;
        const minute = minutes || currentMinute;
        const second = currentSecond;
        // Создаем строку даты в формате, понятном для парсинга
        const dateString = `${day}.${month}.${year}`;
        const dateTimeString = `${day}.${month}.${year} ${hour}:${minute}:${second}`;
        // Парсим дату с учетом текущей временной зоны
        const parsedDate = parse(enableTimePicker ? dateTimeString : dateString, enableTimePicker ? 'dd.MM.yyyy HH:mm:ss' : 'dd.MM.yyyy', new Date());
        if (!isValid(parsedDate)) {
            return '';
        }
        // Преобразуем дату в ISO-строку с учетом временной зоны
        const result = enableTimePicker
            ? parsedDate.toISOString()
            : new Date(parsedDate.getTime() - parsedDate.getTimezoneOffset() * 60_000).toISOString().split('T')[0];
        return result;
    };
    return value && formatToISO(value);
};
const getInputValue = (value, enableTimePicker = false, timeZone) => {
    const formatDate = (dateStr) => {
        if (!dateStr || dateStr === EMPTY_DATE) {
            return '';
        }
        let iso = dateStr;
        if (typeof iso !== 'string') {
            if (iso instanceof Date) {
                iso = iso.toISOString();
            }
            else {
                return '';
            }
        }
        const date = new Date(iso);
        if (isValid(date)) {
            const hasTime = iso.includes('T') && iso.length > 10;
            if (enableTimePicker && hasTime) {
                return formatWithTimeZone(iso, timeZone);
            }
            return formatDateOnly(iso);
        }
        return '';
    };
    if (isRangeMode(value)) {
        const delimiter = isEmptyDate(value) ? EMPTY_DATE : RANGE_DELIMETER;
        return value
            .map(rangeVal => {
            if (rangeVal !== EMPTY_DATE) {
                return formatDate(rangeVal);
            }
            return EMPTY_DATE;
        })
            .join(delimiter);
    }
    if (value !== EMPTY_DATE) {
        return formatDate(value);
    }
    return EMPTY_DATE;
};
const maskDate = (value) => {
    const date = value.replaceAll(/\D/g, '').slice(0, DATE_LENGTH);
    if (date.length >= 5) {
        return `${date.slice(0, 2)}.${date.slice(2, 4)}.${date.slice(4)}`;
    }
    if (date.length >= 3) {
        return `${date.slice(0, 2)}.${date.slice(2)}`;
    }
    return date;
};
const maskDateTime = (value, enableTime = true) => {
    const dateTime = value.replaceAll(/\D/g, '').slice(0, 12);
    const day = dateTime.slice(0, 2);
    const month = dateTime.slice(2, 4);
    const year = dateTime.slice(4, 8);
    const hour = dateTime.slice(8, 10);
    const minute = dateTime.slice(10, 12);
    let result = '';
    if (day) {
        result += day;
    }
    if (month) {
        result += `.${month}`;
    }
    if (year) {
        result += `.${year}`;
    }
    if (hour && enableTime) {
        result += ` ${hour}`;
    }
    if (minute && enableTime) {
        result += `:${minute}`;
    }
    return result.trim();
};
const parseTimeFromInput = (dateStr, enableTimePicker) => {
    const match = RU_FORMAT_PATTERN_F(enableTimePicker).exec(dateStr);
    if (match && match.groups) {
        const { day, month, year, hours, minutes } = match.groups;
        const date = `${day}.${month}.${year}`;
        return {
            date,
            hours,
            minutes,
        };
    }
    return {
        date: dateStr,
    };
};
const maskConfig = {
    mask: Date,
    pattern: 'd.`m.`Y',
    lazy: false,
    placeholderChar: '_',
    overwrite: false,
    autofix: true,
    blocks: {
        d: {
            mask: MaskedRange,
            from: 1,
            to: 31,
            maxLength: 2,
        },
        m: {
            mask: MaskedRange,
            from: 1,
            to: 12,
            maxLength: 2,
        },
        Y: {
            mask: MaskedRange,
            from: 1900,
            to: 9999,
            maxLength: 4,
        },
    },
    format: (date) => {
        const day = date.getDate().toString().padStart(2, '0');
        const month = (date.getMonth() + 1).toString().padStart(2, '0');
        const year = date.getFullYear();
        return `${day}.${month}.${year}`;
    },
    parse: (str) => {
        const [day, month, year] = str.split('.').map(Number);
        return new Date(year, month - 1, day);
    },
};
const maskConfigTime = {
    mask: Date,
    pattern: 'd.`m.`Y `H:`M',
    lazy: false,
    placeholderChar: '_',
    overwrite: false,
    autofix: true,
    blocks: {
        d: {
            mask: MaskedRange,
            from: 1,
            to: 31,
            maxLength: 2,
        },
        m: {
            mask: MaskedRange,
            from: 1,
            to: 12,
            maxLength: 2,
        },
        Y: {
            mask: MaskedRange,
            from: 1900,
            to: 9999,
            maxLength: 4,
        },
        H: {
            mask: MaskedRange,
            from: 0,
            to: 23,
            maxLength: 2,
        },
        M: {
            mask: MaskedRange,
            from: 0,
            to: 59,
            maxLength: 2,
        },
    },
    format: (date) => {
        const day = date.getDate().toString().padStart(2, '0');
        const month = (date.getMonth() + 1).toString().padStart(2, '0');
        const year = date.getFullYear();
        const hours = date.getHours().toString().padStart(2, '0');
        const minutes = date.getMinutes().toString().padStart(2, '0');
        return `${day}.${month}.${year} ${hours}:${minutes}`;
    },
    parse: (str) => {
        const [datePart, timePart] = str.split(' ');
        const [day, month, year] = datePart.split('.').map(Number);
        const [hours, minutes] = timePart ? timePart.split(':').map(Number) : [0, 0];
        return new Date(year, month - 1, day, hours, minutes);
    },
};
//TODO переписать для ренжа валидацию
const maskConfigDateRange = {
    mask: Date,
    blocks: {
        // Блоки для первой даты
        d: {
            mask: MaskedRange,
            from: 1,
            to: 31,
            maxLength: 2,
        },
        m: {
            mask: MaskedRange,
            from: 1,
            to: 12,
            maxLength: 2,
        },
        Y: {
            mask: MaskedRange,
            from: 1900,
            to: 9999,
            maxLength: 4,
        },
        // Блоки для второй даты
        d1: {
            mask: MaskedRange,
            from: 1,
            to: 31,
            maxLength: 2,
        },
        m1: {
            mask: MaskedRange,
            from: 1,
            to: 12,
            maxLength: 2,
        },
        Y1: {
            mask: MaskedRange,
            from: 1900,
            to: 9999,
            maxLength: 4,
        },
    },
    // Обновленный шаблон с использованием новых блоков
    pattern: 'd.`m.`Y – d1.`m1.`Y1',
    format: value => {
        const [startDate, endDate] = value;
        const formatDate = date => {
            const day = date.getDate().toString().padStart(2, '0');
            const month = (date.getMonth() + 1).toString().padStart(2, '0');
            const year = date.getFullYear();
            return `${day}.${month}.${year}`;
        };
        return `${formatDate(startDate)} – ${formatDate(endDate)}`;
    },
    parse: str => {
        const [startStr, endStr] = str.split(' – ');
        const parseDate = dateStr => {
            const [day, month, year] = dateStr.trim().split('.').map(Number);
            if (isNaN(day) || isNaN(month) || isNaN(year)) {
                return null;
            }
            return new Date(year, month - 1, day);
        };
        const startDate = parseDate(startStr);
        const endDate = parseDate(endStr);
        return [startDate, endDate];
    },
    placeholderChar: '_',
    overwrite: true,
    autofix: false,
    lazy: false,
};
const maskConfigTimeRange = {
    mask: Date,
    blocks: {
        // Блоки для первой даты и времени
        d: {
            mask: MaskedRange,
            from: 1,
            to: 31,
            maxLength: 2,
        },
        m: {
            mask: MaskedRange,
            from: 1,
            to: 12,
            maxLength: 2,
        },
        Y: {
            mask: MaskedRange,
            from: 1900,
            to: 9999,
            maxLength: 4,
        },
        H: {
            mask: MaskedRange,
            from: 0,
            to: 23,
            maxLength: 2,
        },
        M: {
            mask: MaskedRange,
            from: 0,
            to: 59,
            maxLength: 2,
        },
        // Блоки для второй даты и времени
        d2: {
            mask: MaskedRange,
            from: 1,
            to: 31,
            maxLength: 2,
        },
        m2: {
            mask: MaskedRange,
            from: 1,
            to: 12,
            maxLength: 2,
        },
        Y2: {
            mask: MaskedRange,
            from: 1900,
            to: 9999,
            maxLength: 4,
        },
        H2: {
            mask: MaskedRange,
            from: 0,
            to: 23,
            maxLength: 2,
        },
        M2: {
            mask: MaskedRange,
            from: 0,
            to: 59,
            maxLength: 2,
        },
    },
    pattern: 'd.`m.`Y `H:`M – d2.`m2.`Y2 `H2:`M2',
    format: value => {
        const [startDate, endDate] = value;
        const formatDateTime = date => {
            const day = date.getDate().toString().padStart(2, '0');
            const month = (date.getMonth() + 1).toString().padStart(2, '0');
            const year = date.getFullYear();
            const hours = date.getHours().toString().padStart(2, '0');
            const minutes = date.getMinutes().toString().padStart(2, '0');
            return `${day}.${month}.${year} ${hours}:${minutes}`;
        };
        return `${formatDateTime(startDate)} – ${formatDateTime(endDate)}`;
    },
    parse: str => {
        const [startStr, endStr] = str.split(' – ');
        const parseDateTime = dateTimeStr => {
            if (!dateTimeStr)
                return new Date();
            const [datePart, timePart] = dateTimeStr.trim().split(' ');
            const [day, month, year] = datePart.split('.').map(Number);
            const [hours, minutes] = timePart ? timePart.split(':').map(Number) : [0, 0];
            return new Date(year, month - 1, day, hours, minutes);
        };
        const startDate = parseDateTime(startStr);
        const endDate = parseDateTime(endStr);
        return [startDate, endDate];
    },
    placeholderChar: '_',
    overwrite: true,
    autofix: false,
    lazy: false,
};
function formatValue(value, enableTimePicker, timeZone, mode) {
    if (isRangeMode(value) && Array.isArray(value)) {
        const hasUndefined = value.includes(undefined);
        const hasEmptyString = value.every(item => item === '');
        const isEmpty = value.length === 0;
        // Для rangeSplit режима возвращаем массив
        if (mode === 'rangeSplit') {
            // Если массив пустой, возвращаем массив с двумя пустыми строками
            if (isEmpty) {
                return ['', ''];
            }
            return value.map(item => getInputValue(item, enableTimePicker, timeZone));
        }
        if (hasEmptyString) {
            return value.map(item => getInputValue(item, enableTimePicker, timeZone)).join(EMPTY_DATE);
        }
        if (!hasUndefined) {
            return value.map(item => getInputValue(item, enableTimePicker, timeZone)).join(RANGE_DELIMETER);
        }
        return '';
    }
    return getInputValue(value, enableTimePicker, timeZone);
}

export { DATE_LENGTH, EMPTY_DATE, ISO_FORMAT_PATTERN, RANGE_DELIMETER, RU_FORMAT_PATTERN_F, formatDateToISOWithTimeZone, formatValue, getInputValue, isEmptyDate, isFilledDate, isRangeMode, maskConfig, maskConfigDateRange, maskConfigTime, maskConfigTimeRange, maskDate, maskDateTime, parseTimeFromInput };
