// @beeline/design-system-react/extracted-styles/esm/components/PhoneInput/PhoneInput.types.js
var Countries;
(function (Countries) {
    Countries["AD"] = "AD";
    Countries["EE"] = "EE";
    Countries["KW"] = "KW";
    Countries["PY"] = "PY";
    Countries["AE"] = "AE";
    Countries["EG"] = "EG";
    Countries["KY"] = "KY";
    Countries["QA"] = "QA";
    Countries["AF"] = "AF";
    Countries["EH"] = "EH";
    Countries["KZ"] = "KZ";
    Countries["RE"] = "RE";
    Countries["AG"] = "AG";
    Countries["ER"] = "ER";
    Countries["LA"] = "LA";
    Countries["RO"] = "RO";
    Countries["AI"] = "AI";
    Countries["ES"] = "ES";
    Countries["LB"] = "LB";
    Countries["RS"] = "RS";
    Countries["AL"] = "AL";
    Countries["ET"] = "ET";
    Countries["LC"] = "LC";
    Countries["RU"] = "RU";
    Countries["AM"] = "AM";
    Countries["FI"] = "FI";
    Countries["LI"] = "LI";
    Countries["RW"] = "RW";
    Countries["AO"] = "AO";
    Countries["FJ"] = "FJ";
    Countries["LK"] = "LK";
    Countries["SA"] = "SA";
    Countries["AQ"] = "AQ";
    Countries["FK"] = "FK";
    Countries["LR"] = "LR";
    Countries["SB"] = "SB";
    Countries["AR"] = "AR";
    Countries["FM"] = "FM";
    Countries["LS"] = "LS";
    Countries["SC"] = "SC";
    Countries["AS"] = "AS";
    Countries["FO"] = "FO";
    Countries["LT"] = "LT";
    Countries["SD"] = "SD";
    Countries["AT"] = "AT";
    Countries["FR"] = "FR";
    Countries["LU"] = "LU";
    Countries["SE"] = "SE";
    Countries["AU"] = "AU";
    Countries["GA"] = "GA";
    Countries["LV"] = "LV";
    Countries["SG"] = "SG";
    Countries["AW"] = "AW";
    Countries["GB_ENG"] = "GB-ENG";
    Countries["LY"] = "LY";
    Countries["SH"] = "SH";
    Countries["AX"] = "AX";
    Countries["GB_NIR"] = "GB-NIR";
    Countries["MA"] = "MA";
    Countries["SI"] = "SI";
    Countries["AZ"] = "AZ";
    Countries["GB_SCT"] = "GB-SCT";
    Countries["MC"] = "MC";
    Countries["SJ"] = "SJ";
    Countries["BA"] = "BA";
    Countries["GB_UKM"] = "GB-UKM";
    Countries["MD"] = "MD";
    Countries["SK"] = "SK";
    Countries["BB"] = "BB";
    Countries["GB_WLS"] = "GB-WLS";
    Countries["ME"] = "ME";
    Countries["SL"] = "SL";
    Countries["BD"] = "BD";
    Countries["GD"] = "GD";
    Countries["MF"] = "MF";
    Countries["SM"] = "SM";
    Countries["BE"] = "BE";
    Countries["GE"] = "GE";
    Countries["MG"] = "MG";
    Countries["SN"] = "SN";
    Countries["BF"] = "BF";
    Countries["GF"] = "GF";
    Countries["MH"] = "MH";
    Countries["SO"] = "SO";
    Countries["BG"] = "BG";
    Countries["GG"] = "GG";
    Countries["MK"] = "MK";
    Countries["SR"] = "SR";
    Countries["BH"] = "BH";
    Countries["GH"] = "GH";
    Countries["ML"] = "ML";
    Countries["SS"] = "SS";
    Countries["BI"] = "BI";
    Countries["GI"] = "GI";
    Countries["MM"] = "MM";
    Countries["ST"] = "ST";
    Countries["BJ"] = "BJ";
    Countries["GL"] = "GL";
    Countries["MN"] = "MN";
    Countries["SV"] = "SV";
    Countries["BL"] = "BL";
    Countries["GM"] = "GM";
    Countries["MO"] = "MO";
    Countries["SX"] = "SX";
    Countries["BM"] = "BM";
    Countries["GN"] = "GN";
    Countries["MP"] = "MP";
    Countries["SY"] = "SY";
    Countries["BN"] = "BN";
    Countries["GP"] = "GP";
    Countries["MQ"] = "MQ";
    Countries["SZ"] = "SZ";
    Countries["BO"] = "BO";
    Countries["GQ"] = "GQ";
    Countries["MR"] = "MR";
    Countries["TC"] = "TC";
    Countries["BQ_BO"] = "BQ-BO";
    Countries["GR"] = "GR";
    Countries["MS"] = "MS";
    Countries["TF"] = "TF";
    Countries["BQ_SA"] = "BQ-SA";
    Countries["MT"] = "MT";
    Countries["TG"] = "TG";
    Countries["BQ_SE"] = "BQ-SE";
    Countries["MU"] = "MU";
    Countries["TH"] = "TH";
    Countries["BR"] = "BR";
    Countries["GT"] = "GT";
    Countries["MV"] = "MV";
    Countries["TJ"] = "TJ";
    Countries["BS"] = "BS";
    Countries["GU"] = "GU";
    Countries["MW"] = "MW";
    Countries["TK"] = "TK";
    Countries["BT"] = "BT";
    Countries["GW"] = "GW";
    Countries["MX"] = "MX";
    Countries["TL"] = "TL";
    Countries["BV"] = "BV";
    Countries["GY"] = "GY";
    Countries["MY"] = "MY";
    Countries["TM"] = "TM";
    Countries["BW"] = "BW";
    Countries["HK"] = "HK";
    Countries["MZ"] = "MZ";
    Countries["TN"] = "TN";
    Countries["BY"] = "BY";
    Countries["HM"] = "HM";
    Countries["NA"] = "NA";
    Countries["TO"] = "TO";
    Countries["BZ"] = "BZ";
    Countries["HN"] = "HN";
    Countries["NC"] = "NC";
    Countries["TR"] = "TR";
    Countries["CA"] = "CA";
    Countries["HR"] = "HR";
    Countries["NE"] = "NE";
    Countries["TT"] = "TT";
    Countries["CC"] = "CC";
    Countries["HT"] = "HT";
    Countries["NF"] = "NF";
    Countries["TV"] = "TV";
    Countries["CD"] = "CD";
    Countries["HU"] = "HU";
    Countries["NG"] = "NG";
    Countries["TW"] = "TW";
    Countries["CF"] = "CF";
    Countries["ID"] = "ID";
    Countries["NI"] = "NI";
    Countries["TZ"] = "TZ";
    Countries["CG"] = "CG";
    Countries["IE"] = "IE";
    Countries["NL"] = "NL";
    Countries["UA"] = "UA";
    Countries["CH"] = "CH";
    Countries["IL"] = "IL";
    Countries["NO"] = "NO";
    Countries["UG"] = "UG";
    Countries["CI"] = "CI";
    Countries["IM"] = "IM";
    Countries["NP"] = "NP";
    Countries["UM"] = "UM";
    Countries["CK"] = "CK";
    Countries["IN"] = "IN";
    Countries["NR"] = "NR";
    Countries["US"] = "US";
    Countries["CL"] = "CL";
    Countries["INT"] = "INT";
    Countries["NU"] = "NU";
    Countries["UY"] = "UY";
    Countries["CM"] = "CM";
    Countries["IO"] = "IO";
    Countries["NZ"] = "NZ";
    Countries["UZ"] = "UZ";
    Countries["CN"] = "CN";
    Countries["IQ"] = "IQ";
    Countries["OM"] = "OM";
    Countries["VA"] = "VA";
    Countries["CR"] = "CR";
    Countries["IS"] = "IS";
    Countries["PA"] = "PA";
    Countries["VC"] = "VC";
    Countries["CU"] = "CU";
    Countries["IT"] = "IT";
    Countries["PE"] = "PE";
    Countries["VE"] = "VE";
    Countries["CV"] = "CV";
    Countries["JE"] = "JE";
    Countries["PF"] = "PF";
    Countries["VG"] = "VG";
    Countries["CW"] = "CW";
    Countries["JM"] = "JM";
    Countries["PG"] = "PG";
    Countries["CX"] = "CX";
    Countries["JO"] = "JO";
    Countries["PH"] = "PH";
    Countries["VN"] = "VN";
    Countries["CY"] = "CY";
    Countries["JP"] = "JP";
    Countries["PK"] = "PK";
    Countries["VU"] = "VU";
    Countries["CZ"] = "CZ";
    Countries["KE"] = "KE";
    Countries["PL"] = "PL";
    Countries["WF"] = "WF";
    Countries["DE"] = "DE";
    Countries["KG"] = "KG";
    Countries["PM"] = "PM";
    Countries["WS"] = "WS";
    Countries["DJ"] = "DJ";
    Countries["KH"] = "KH";
    Countries["YE"] = "YE";
    Countries["DK"] = "DK";
    Countries["KI"] = "KI";
    Countries["PN"] = "PN";
    Countries["YT_UNF"] = "YT-UNF";
    Countries["DM"] = "DM";
    Countries["KM"] = "KM";
    Countries["PR"] = "PR";
    Countries["ZA"] = "ZA";
    Countries["DO"] = "DO";
    Countries["KN"] = "KN";
    Countries["PS"] = "PS";
    Countries["ZM"] = "ZM";
    Countries["DZ"] = "DZ";
    Countries["KP"] = "KP";
    Countries["PT"] = "PT";
    Countries["ZW"] = "ZW";
    Countries["EC"] = "EC";
    Countries["KR"] = "KR";
    Countries["PW"] = "PW";
})(Countries || (Countries = {}));
Object.values(Countries);
const CIS_COUNTRIES = [
    Countries.AM,
    Countries.AZ,
    Countries.BY,
    Countries.KZ,
    Countries.KG,
    Countries.MD,
    Countries.RU,
    Countries.TJ,
];

export { CIS_COUNTRIES, Countries };
