// @beeline/design-system-react/build/components/Preview/components/ImagePreview/ImagePreview.js
import { jsx } from 'react/jsx-runtime';
import './ImagePreview2.js';
import { useRef, useState, useEffect, useCallback } from 'react';

const IMAGE_FIT_PADDING = 0.9;
const EDGE_PADDING = 40;
const ImagePreview = ({ value, scale, rotation, onInitialScale }) => {
    const containerRef = useRef(null);
    const imgRef = useRef(null);
    const [imageDimensions, setImageDimensions] = useState(null);
    const [isReady, setIsReady] = useState(false);
    const calculatedScaleRef = useRef(null);
    const onInitialScaleRef = useRef(onInitialScale);
    const translateRef = useRef({ x: 0, y: 0 });
    const [isDragging, setIsDragging] = useState(false);
    const dragStartRef = useRef({ mouseX: 0, mouseY: 0, translateX: 0, translateY: 0 });
    const [currentSrc, setCurrentSrc] = useState(value.src);
    // Синхронный сброс при смене картинки (до коммита в DOM — без glitch)
    if (value.src !== currentSrc) {
        setCurrentSrc(value.src);
        setIsReady(false);
        setImageDimensions(null);
        calculatedScaleRef.current = null;
        translateRef.current = { x: 0, y: 0 };
    }
    const isRotated90 = rotation % 180 !== 0;
    const getVisualSize = () => {
        if (!imageDimensions || !containerRef.current)
            return null;
        const w = isRotated90 ? imageDimensions.height : imageDimensions.width;
        const h = isRotated90 ? imageDimensions.width : imageDimensions.height;
        return { width: w * scale, height: h * scale };
    };
    const visualSize = getVisualSize();
    const canDrag = !!visualSize &&
        !!containerRef.current &&
        (visualSize.width > containerRef.current.clientWidth || visualSize.height > containerRef.current.clientHeight);
    useEffect(() => {
        onInitialScaleRef.current = onInitialScale;
    }, [onInitialScale]);
    // Загружаем картинку и получаем размеры через new Image()
    useEffect(() => {
        const img = new Image();
        img.src = value.src;
        const handleLoad = () => {
            setImageDimensions({ width: img.naturalWidth, height: img.naturalHeight });
        };
        if (img.complete && img.naturalWidth > 0) {
            handleLoad();
        }
        else {
            img.addEventListener('load', handleLoad);
        }
        return () => {
            img.removeEventListener('load', handleLoad);
        };
    }, [value.src]);
    // Вычисляем scale когда есть размеры картинки и контейнера
    useEffect(() => {
        if (!imageDimensions || !containerRef.current || calculatedScaleRef.current !== null)
            return;
        const containerWidth = containerRef.current.clientWidth;
        const containerHeight = containerRef.current.clientHeight;
        if (containerWidth === 0 || containerHeight === 0)
            return;
        const widthRatio = containerWidth / imageDimensions.width;
        const heightRatio = containerHeight / imageDimensions.height;
        const newScale = Math.min(widthRatio, heightRatio) * IMAGE_FIT_PADDING;
        calculatedScaleRef.current = newScale;
        onInitialScaleRef.current(newScale);
    }, [imageDimensions]);
    // Показываем когда scale обновился до вычисленного
    useEffect(() => {
        if (calculatedScaleRef.current !== null && !isReady && imageDimensions) {
            if (Math.abs(scale - calculatedScaleRef.current) < 0.1) {
                setIsReady(true);
            }
        }
    }, [scale, isReady, imageDimensions]);
    // Сбрасываем translate при уменьшении до initial или ниже
    useEffect(() => {
        if (!canDrag) {
            translateRef.current = { x: 0, y: 0 };
            applyTransform(0, 0);
        }
    }, [canDrag]);
    const getClampedTranslate = (tx, ty) => {
        if (!visualSize || !containerRef.current)
            return { x: 0, y: 0 };
        const containerWidth = containerRef.current.clientWidth;
        const containerHeight = containerRef.current.clientHeight;
        const maxX = Math.max(0, (visualSize.width - containerWidth) / 2 + EDGE_PADDING);
        const maxY = Math.max(0, (visualSize.height - containerHeight) / 2 + EDGE_PADDING);
        return {
            x: Math.max(-maxX, Math.min(maxX, tx)),
            y: Math.max(-maxY, Math.min(maxY, ty)),
        };
    };
    const applyTransform = (tx, ty) => {
        if (!imgRef.current)
            return;
        imgRef.current.style.transform = `translate(${tx}px, ${ty}px) scale(${scale}) rotate(${rotation}deg)`;
    };
    const handleMouseDown = useCallback((e) => {
        if (!canDrag)
            return;
        e.preventDefault();
        setIsDragging(true);
        dragStartRef.current = {
            mouseX: e.clientX,
            mouseY: e.clientY,
            translateX: translateRef.current.x,
            translateY: translateRef.current.y,
        };
    }, [canDrag]);
    const handleMouseMove = useCallback((e) => {
        if (!isDragging)
            return;
        const dx = e.clientX - dragStartRef.current.mouseX;
        const dy = e.clientY - dragStartRef.current.mouseY;
        const clamped = getClampedTranslate(dragStartRef.current.translateX + dx, dragStartRef.current.translateY + dy);
        translateRef.current = clamped;
        applyTransform(clamped.x, clamped.y);
    }, [isDragging, imageDimensions, scale, rotation]);
    const handleMouseUp = useCallback(() => {
        setIsDragging(false);
    }, []);
    // Не рендерим картинку пока не готова
    if (!isReady || !imageDimensions) {
        return jsx("div", { className: "dsb_preview__content-img", ref: containerRef });
    }
    const { x: tx, y: ty } = translateRef.current;
    const containerClassName = [
        'dsb_preview__content-img',
        canDrag && 'dsb_preview__content-img--draggable',
        isDragging && 'dsb_preview__content-img--dragging',
    ]
        .filter(Boolean)
        .join(' ');
    return (jsx("div", { className: containerClassName, ref: containerRef, onMouseDown: handleMouseDown, onMouseMove: handleMouseMove, onMouseUp: handleMouseUp, onMouseLeave: handleMouseUp, children: jsx("img", { ref: imgRef, src: value.src, alt: value.alt, draggable: false, style: {
                transform: `translate(${tx}px, ${ty}px) scale(${scale}) rotate(${rotation}deg)`,
                transformOrigin: 'center center',
            } }) }));
};

export { ImagePreview };
