// @beeline/design-tokens/script/generate-exports.js
#!/usr/bin/env node
/**
 * Генерирует поле `exports` для корневого package.json по структуре dist/web.
 * 
 * Сканирует:
 * - JS модули: dist/web/js (по package.json в подкаталогах)
 * - CSS файлы: dist/web/css (font-face.css, iconfont/iconfont.css, scrollbar.css + wildcard)
 * - SCSS файлы: dist/web/scss (font-face.scss, iconfont/iconfont.scss, scrollbar.scss + wildcard)
 * - Assets: dist/web/assets (wildcard для всех файлов: SVG, шрифты, изображения)
 * 
 * Нужен для резолва подпутей в бандлерах (webpack и др.), которые требуют явного
 * объявления экспортов в корневом package.json.
 *
 * Запуск: после build, до prerelease. По умолчанию сканирует dist/web/js (CSS/SCSS находятся рядом).
 *   node script/generate-exports.js [--stdout] [путь/к/js]
 */

const fs = require('fs');
const path = require('path');

const ROOT_DIR = path.resolve(__dirname, '..');
const DEFAULT_JS_DIR = path.join(ROOT_DIR, 'dist', 'web', 'js');
const ROOT_PACKAGE = path.join(ROOT_DIR, 'package.json');

function parseArgs() {
	const args = process.argv.slice(2);
	let jsDir = DEFAULT_JS_DIR;
	let stdout = false;
	for (const arg of args) {
		if (arg === '--stdout') stdout = true;
		else if (!arg.startsWith('-')) jsDir = path.resolve(arg);
	}
	return { jsDir, stdout };
}

function getEntryFromPkg(pkg) {
	if (pkg.exports && pkg.exports['.'] && typeof pkg.exports['.'] === 'object') {
		const e = pkg.exports['.'];
		const strip = (s) => (s && s.replace(/^\.\//, '')) || '';
		const imp = strip(e.import);
		const req = strip(e.require);
		const types = strip(e.types) || imp.replace(/\.esm\.js$/, '.d.ts');
		if (imp && req) return { esm: imp, common: req, types };
	}
	if (pkg.main && pkg.module) {
		const common = pkg.main.replace(/^\.\//, '');
		const esm = pkg.module.replace(/^\.\//, '');
		const types = (pkg.types && pkg.types.replace(/^\.\//, '')) || esm.replace(/\.esm\.js$/, '.d.ts');
		return { esm, common, types };
	}
	return null;
}

function walk(dir, base = 'js') {
	const entries = [];
	if (!fs.existsSync(dir) || !fs.statSync(dir).isDirectory()) return entries;
	const pkgPath = path.join(dir, 'package.json');
	if (fs.existsSync(pkgPath)) {
		try {
			const pkg = JSON.parse(fs.readFileSync(pkgPath, 'utf8'));
			const names = getEntryFromPkg(pkg);
			if (names) entries.push([base.replace(/\\/g, '/'), names]);
		} catch (_) {}
	}
	for (const name of fs.readdirSync(dir, { withFileTypes: true })) {
		if (name.isDirectory()) {
			const sub = path.join(dir, name.name);
			const subBase = base + '/' + name.name;
			entries.push(...walk(sub, subBase));
		}
	}
	return entries;
}

function scanCssScssAssets(baseDir) {
	const cssDir = path.join(baseDir, '..', 'css');
	const scssDir = path.join(baseDir, '..', 'scss');
	const assetsDir = path.join(baseDir, '..', 'assets');
	const entries = [];
	
	// CSS файлы
	if (fs.existsSync(cssDir)) {
		const cssFiles = [
			'font-face.css',
			'iconfont/iconfont.css',
			'scrollbar.css'
		];
		for (const file of cssFiles) {
			const fullPath = path.join(cssDir, file);
			if (fs.existsSync(fullPath)) {
				entries.push(['./css/' + file, './css/' + file]);
			}
		}
		// Wildcard для всех CSS файлов
		entries.push(['./css/*', './css/*']);
	}
	
	// SCSS файлы
	if (fs.existsSync(scssDir)) {
		const scssFiles = [
			'font-face.scss',
			'iconfont/iconfont.scss',
			'scrollbar.scss'
		];
		for (const file of scssFiles) {
			const fullPath = path.join(scssDir, file);
			if (fs.existsSync(fullPath)) {
				entries.push(['./scss/' + file, './scss/' + file]);
			}
		}
		// Wildcard для всех SCSS файлов
		entries.push(['./scss/*', './scss/*']);
	}
	
	// Assets (SVG, шрифты, изображения и т.д.) - wildcard для всех файлов
	if (fs.existsSync(assetsDir)) {
		// Wildcard для всех assets файлов (SVG, PNG, TTF, WOFF, WOFF2 и т.д.)
		entries.push(['./assets/*', './assets/*']);
	}
	
	return entries;
}

function buildExports(entries) {
	const exp = {};
	
	// Группируем для лучшей сортировки
	const cssScssAssets = [];
	const jsIndex = [];
	const jsComponents = [];
	const jsGlobals = [];
	const jsThemes = [];
	const jsOther = [];
	
	for (const [subpath, names] of entries) {
		const key = subpath.startsWith('./') ? subpath : './' + subpath;
		
		if (typeof names === 'string') {
			// CSS/SCSS/Assets файлы
			cssScssAssets.push([key, names]);
		} else {
			// JS модули - группируем по категориям
			if (key === './js/tokens' || key === './js/iconfont' || key === './js/iconfont/icons') {
				jsIndex.push([key, names]);
			} else if (key.includes('/components/')) {
				jsComponents.push([key, names]);
			} else if (key.includes('/globals/')) {
				jsGlobals.push([key, names]);
			} else if (key.includes('/themes/')) {
				jsThemes.push([key, names]);
			} else {
				jsOther.push([key, names]);
			}
		}
	}
	
	// Сортируем компоненты по имени
	jsComponents.sort((a, b) => a[0].localeCompare(b[0]));
	
	// Собираем в правильном порядке
	const ordered = [
		...cssScssAssets,
		...jsIndex,
		...jsComponents,
		...jsGlobals,
		...jsThemes,
		...jsOther
	];
	
	// Строим exports с сохранением порядка
	for (const [key, names] of ordered) {
		if (typeof names === 'string') {
			exp[key] = names;
		} else {
			const slash = key.endsWith('/') ? '' : '/';
			const esmPath = key + slash + names.esm;
			const commonPath = key + slash + names.common;
			const typesPath = key + slash + names.types;
			
			// Основной экспорт для подпути без расширения
			exp[key] = {
				import: esmPath,
				require: commonPath,
				types: typesPath,
			};
			
			// Явные пути к файлам с расширениями (для прямых импортов)
			exp[esmPath] = esmPath;
			exp[commonPath] = commonPath;
			exp[typesPath] = typesPath;
		}
	}
	
	return exp;
}

function main() {
	const { jsDir, stdout } = parseArgs();
	const jsEntries = walk(jsDir);
	const cssScssAssetsEntries = scanCssScssAssets(jsDir);
	const entries = [...jsEntries, ...cssScssAssetsEntries];
	const exports = buildExports(entries);
	if (entries.length === 0) {
		console.warn('generate-exports: no entries (run build first?).');
	}
	if (stdout) {
		console.log(JSON.stringify({ exports }, null, 2));
		return;
	}
	let pkg;
	try {
		pkg = JSON.parse(fs.readFileSync(ROOT_PACKAGE, 'utf8'));
	} catch (e) {
		console.error('generate-exports: failed to read package.json', e.message);
		process.exit(1);
	}
	pkg.exports = exports;
	const order = ['name', 'scope', 'version', 'description', 'bin', 'scripts', 'config', 'repository', 'author', 'license', 'exports', 'devDependencies', 'publishConfig'];
	const out = {};
	for (const k of order) if (k in pkg) out[k] = pkg[k];
	for (const k of Object.keys(pkg)) if (!(k in out)) out[k] = pkg[k];
	fs.writeFileSync(ROOT_PACKAGE, JSON.stringify(out, null, 2) + '\n');
	console.log('generate-exports: updated package.json with', Object.keys(exports).length, 'subpaths');
}

main();
