// @beeline/design-tokens/js/tokens/components/bottomSheet/bottomSheet.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	bottomSheetBorderRadius: "12px",
	bottomSheetBackgroundColor: "var(--color-background-medium, #ffffff)",
	bottomSheetShadow: "0px 6px 38px rgba(0, 0, 0, 0.16), 0px 0px 10px rgba(0, 0, 0, 0.08)",
	bottomSheetTop: "64px",
	bottomSheetSwiperHeight: "12px",
	bottomSheetSwiperTrumbHeight: "4px",
	bottomSheetSwiperTrumbWidth: "40px",
	bottomSheetSwiperTrumbBackgroundColor: "var(--color-utilities-scroll, #b6b7bf)",
	bottomSheetSwiperTrumbBorderRadius: "2px",
	bottomSheetHeaderGap: "16px",
	bottomSheetHeaderPadding: "16px",
	bottomSheetHeaderTitleFontWeight: 500,
	bottomSheetHeaderSubtitleColor: "var(--color-text-grey-inactive, rgba(25, 28, 52, 0.7))",
	bottomSheetHeaderSubtitleMarginTop: "2px",
	bottomSheetHeaderActionsGap: "8px",
	bottomSheetHeaderActionsButtonHeight: "24px",
	bottomSheetHeaderActionsButtonColor: "var(--color-button-plain, #1a73e8)",
	bottomSheetActionsGap: "8px",
	bottomSheetActionsPaddingLeft: "16px",
	bottomSheetActionsPaddingRight: "16px",
	bottomSheetActionsPaddingTop: "16px",
	bottomSheetActionsPaddingBottom: "24px",
	bottomSheetModalBackgroundColor: "var(--color-background-backdrop, rgba(0, 0, 0, 0.48))",
};
