// @beeline/design-tokens/js/tokens/components/bottomSheet/bottomSheet.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const bottomSheetBorderRadius = "12px";
export const bottomSheetBackgroundColor = "var(--color-background-medium, #ffffff)";
export const bottomSheetShadow = "0px 6px 38px rgba(0, 0, 0, 0.16), 0px 0px 10px rgba(0, 0, 0, 0.08)";
export const bottomSheetTop = "64px";
export const bottomSheetSwiperHeight = "12px";
export const bottomSheetSwiperTrumbHeight = "4px";
export const bottomSheetSwiperTrumbWidth = "40px";
export const bottomSheetSwiperTrumbBackgroundColor = "var(--color-utilities-scroll, #b6b7bf)";
export const bottomSheetSwiperTrumbBorderRadius = "2px";
export const bottomSheetHeaderGap = "16px";
export const bottomSheetHeaderPadding = "16px";
export const bottomSheetHeaderTitleFontWeight = 500;
export const bottomSheetHeaderSubtitleColor = "var(--color-text-grey-inactive, rgba(25, 28, 52, 0.7))";
export const bottomSheetHeaderSubtitleMarginTop = "2px";
export const bottomSheetHeaderActionsGap = "8px";
export const bottomSheetHeaderActionsButtonHeight = "24px";
export const bottomSheetHeaderActionsButtonColor = "var(--color-button-plain, #1a73e8)";
export const bottomSheetActionsGap = "8px";
export const bottomSheetActionsPaddingLeft = "16px";
export const bottomSheetActionsPaddingRight = "16px";
export const bottomSheetActionsPaddingTop = "16px";
export const bottomSheetActionsPaddingBottom = "24px";
export const bottomSheetModalBackgroundColor = "var(--color-background-backdrop, rgba(0, 0, 0, 0.48))";
