// @beeline/design-system-react/extracted-styles/esm/components/Select/Select.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { BottomSheet } from '../BottomSheet/BottomSheet.js';
import { Icon } from '../Icon/Icon.js';
import { NewDropdownMenu } from '../NewDropdownMenu/NewDropdownMenu.js';
import { PopoverContext } from '../Popover/Popover.js';
import '../Popover/PopoverHelper/PopoverHelper.js';
import { classNames } from '../../helpers/classnames.js';
import '@beeline/design-tokens/js/iconfont/icons';
import '../Button/Button.js';
import '../Button/ProgressButton/ProgressButton.js';
import '../Typography/Typography.js';
import { useOverflowText } from './components/hooks/useOverflowText.js';
import { Tooltip } from '../Tooltip/Tooltip.js';
import { getBEMClasses } from '../../helpers/get-bem.js';
import { useDeviceSize } from '../../helpers/use-device-size.js';
import { Icons } from '@beeline/design-tokens/js/iconfont';
import React__default, { useRef, useMemo } from 'react';
import { TextField } from '../TextField/TextField.js';
import { DropDownContent } from './components/DropDownContent/index.js';
import { OverflowCount } from './components/OverflowCount/index.js';
import { defaultMakeOption } from './helpers.js';
import { useSelect } from './useSelect.js';

const [selectClassName, , selectElement] = getBEMClasses('dsb__select');
const [optionsClassName, , optionsElement] = selectElement('options');
const [itemClassName] = optionsElement('item');
const Select = ({ label, helperText, multiple, filter, filterPlaceholder = 'Поиск', filterFunction, size, error, values, renderValue, onChange, onClose, onOpen, onOutsideClick, onManualClickOption, open, onFocus, onBlur, options, fullWidth, className, disabled, makeOption, loading, loadingText, loadingContent, applicationRootElementID, dropdownElementID, dropdownClassName, dataTestId = 'Select', selectAll = false, selectAllText = 'Выбрать все', showTooltip = false, noOptionsText = 'Нет результатов', enableKeyboardNavigation = false, isShowCountOfValues = false, overlayScroll = true, shouldRenderOverlay, overlayClassName, alignDropDown, makeOptionWrapper, compareBy, hideDropDown = false, disableMobileBottomSheet = false, groups, onClick, onKeyDown: externalOnKeyDown, ...props }) => {
    const inputRef = useRef(null);
    const filterRef = useRef(null);
    const selectRef = useRef(null);
    const switchButtonRef = useRef(null);
    const optionsRef = useRef(null);
    const { isExtraSmallDevice } = useDeviceSize();
    const useMobileBottomSheet = isExtraSmallDevice && !disableMobileBottomSheet;
    const selectAllIndex = -2;
    const { isOpen, setOpen, filterValue, disableTooltip, privateOptions, privateValues, inputLabel, handleOpen, handleClose, onSelectItem, onMultiselectItem, focusedIndex, focusedOptionId, handleSelectAll, getSelectAllStatus, changeFilterValue, } = useSelect({
        open,
        onOpen,
        onClose,
        onChange,
        options,
        values,
        renderValue,
        filter,
        filterFunction,
        enableKeyboardNavigation,
        multiple,
        selectAll,
        inputRef,
        filterRef,
        dropdownRef: selectRef,
        showTooltip,
        onFocus,
        onBlur,
        selectAllText,
        noOptionsText,
        disabled,
        onManualClickOption,
        makeOption,
        loading,
        loadingText,
        loadingContent,
        optionsRef,
        compareBy,
        groups,
    });
    const handleInputBlur = (event) => {
        if (!isOpen) {
            onBlur?.(event);
            inputRef.current?.blur();
        }
    };
    const handleInputFocus = (event) => {
        onFocus?.(event);
        inputRef.current?.focus();
    };
    const handleInputClick = (e) => {
        onClick?.(e);
        isOpen ? handleClose() : handleOpen();
    };
    const handleInputKeyDown = (event) => {
        externalOnKeyDown?.(event);
        if (!enableKeyboardNavigation || disabled || isOpen || event.defaultPrevented) {
            return;
        }
        if (event.key === 'Enter' || event.key === ' ' || event.key === 'ArrowDown') {
            event.preventDefault();
            handleOpen();
        }
    };
    const handleSetOpenWrapper = (newOpen) => {
        if (newOpen) {
            handleOpen();
        }
        else {
            handleClose();
        }
    };
    React__default.useEffect(() => {
        if (!isOpen) {
            return;
        }
        const handleEsc = (event) => {
            if (event.key === 'Escape') {
                onOutsideClick?.();
            }
        };
        document.addEventListener('keydown', handleEsc);
        return () => document.removeEventListener('keydown', handleEsc);
    }, [isOpen, onOutsideClick]);
    const dropdownMenuProps = {
        targetElement: inputRef,
        verticalOffset: 1,
        applicationRootElementID,
        dropdownElementID,
        className: dropdownClassName,
        onBlur,
        isOpen,
        handleSetOpen: handleSetOpenWrapper,
        onOutsideClick,
        overlayScroll,
        shouldRenderOverlay,
        overlayClassName,
        align: alignDropDown,
    };
    const isIconName = isOpen ? Icons.NavArrowUp : Icons.NavArrowDown;
    const rightIcon = jsx(Icon, { ref: switchButtonRef, size: size, iconName: isIconName });
    const rootClassName = classNames(selectClassName, Boolean(className) && className);
    const popoverContextValue = useMemo(() => ({
        applicationRootElementId: 'root',
        popoverRootElementID: 'select-popover',
    }), []);
    const hiddenRef = useRef(null);
    const countRef = useRef(null);
    const { count, className: overflowClassName } = useOverflowText({
        hiddenRef,
        inputRef,
    });
    const counter = (jsxs("p", { className: overflowClassName, ref: countRef, children: ["(+", count, ")"] }));
    const optionsClassNameWithBottomSheet = classNames(optionsClassName, {
        dsb__select__options_in_bottom_sheet: useMobileBottomSheet,
    });
    const dropDownContentProps = {
        selectedItems: privateValues,
        optionsData: privateOptions,
        selectAll,
        filter,
        filterValue,
        filterFunction,
        selectAllIndex,
        optionsClassName: optionsClassNameWithBottomSheet,
        multiple,
        itemClassName,
        onMultiselectItem,
        onManualClickOption,
        onSelectItem,
        focusedIndex,
        handleSelectAll,
        makeOption,
        showTooltip,
        defaultMakeOption,
        selectAllText,
        loading,
        loadingText,
        loadingContent,
        noOptionsText,
        optionsRef,
        selectStatus: getSelectAllStatus(),
        filterPlaceholder,
        filterRef,
        changeFilterValue,
        dataTestId,
        selectRef,
        makeOptionWrapper,
        groups,
        unstyled: useMobileBottomSheet,
    };
    return (jsxs("div", { className: rootClassName, "data-testid": dataTestId, children: [isShowCountOfValues && (jsx(OverflowCount, { inputRef: inputRef, hiddenRef: hiddenRef, values: values, countRef: countRef })), jsx(PopoverContext.Provider, { value: popoverContextValue, children: jsx(Tooltip, { title: inputLabel, disableHover: disableTooltip, children: jsx(TextField, { ref: inputRef, label: label, helperText: helperText, size: size, error: error, "aria-invalid": error, role: "combobox" // NOSONAR S6819 - WAI-ARIA combobox pattern, native select doesn't support filter/groups/custom options
                        , "aria-expanded": isOpen, "aria-haspopup": "listbox", "aria-controls": `${dataTestId}-listbox`, "aria-activedescendant": focusedOptionId !== undefined ? `${dataTestId}-option-${focusedOptionId}` : undefined, fullWidth: fullWidth, endAdornment: rightIcon, onBlur: handleInputBlur, onFocus: handleInputFocus, onClick: handleInputClick, onKeyDown: handleInputKeyDown, disabled: disabled, value: inputLabel, contentEditable: true, isClickableAdornment: false, readOnly: true, dataTestId: dataTestId + '-textfield', isShowCountOfValues: isShowCountOfValues, counter: counter, isFocused: isOpen, ...props }) }) }), isOpen && !hideDropDown && !useMobileBottomSheet && (jsx(NewDropdownMenu, { ...dropdownMenuProps, children: jsx(DropDownContent, { ...dropDownContentProps }) })), isOpen && !hideDropDown && useMobileBottomSheet && (jsx(BottomSheet, { isOpen: isOpen, handleSetOpen: setOpen, modal: true, onOutsideClick: onOutsideClick, header: {
                    title: label || 'Выберите значение',
                    closeButton: true,
                }, className: dropdownClassName, children: jsx(DropDownContent, { ...dropDownContentProps }) }))] }));
};
Select.propTypes = {
    open: (props, propName, componentName) => {
        if (Object.prototype.hasOwnProperty.call(props, propName) && (!props.onClose || !props.onOpen)) {
            return new Error(`It seems you forgot pass 'onClose' and 'onOpen' props, please check.
					'open' prop should be using both with 'onClose' and 'onOpen' props in '${componentName}'.`);
        }
    },
};

export { Select };
