// @beeline/design-system-react/extracted-styles/esm/components/Select/useSelect.js
import { CheckboxType } from './Select.types.js';
import { useState, useRef, useEffect, useCallback } from 'react';
import { useAddArrayIndex } from './components/hooks/useAddArrayIndex.js';
import { useSelectKeyBoardNavigation } from './components/hooks/useSelectKeyBoardNavigation.js';
import { defaultRenderValue, getOptionsInDisplayOrder, getPreparedOptions, compareValues, getCheckboxStatus } from './helpers.js';

const useSelect = (props) => {
    const { open, onOpen, onClose, onChange, options, values, renderValue, filter, filterFunction, enableKeyboardNavigation, multiple, selectAll, inputRef, filterRef, showTooltip, onFocus, onBlur, optionsRef, compareBy, onManualClickOption, dropdownRef, groups, } = props;
    const [isOpen, setOpen] = useState(open);
    const [filterValue, setFilterValue] = useState('');
    const [disableTooltip, setDisableTooltip] = useState(true);
    const [privateOptions, privateValues] = useAddArrayIndex({ options, values });
    const lastFocusedRef = useRef(null);
    useEffect(() => {
        const handleFocusIn = (event) => {
            if (filterRef.current && filterRef.current.contains(event.target)) {
                lastFocusedRef.current = filterRef.current;
            }
            else if (inputRef.current && inputRef.current.contains(event.target)) {
                lastFocusedRef.current = inputRef.current;
            }
        };
        document.addEventListener('focusin', handleFocusIn);
        return () => {
            document.removeEventListener('focusin', handleFocusIn);
        };
    }, [filterRef, inputRef]);
    useEffect(() => {
        if (open !== undefined) {
            setOpen(open);
        }
    }, [open]);
    useEffect(() => {
        setDisableTooltip(showTooltip ? inputRef?.current?.offsetWidth >= inputRef?.current?.scrollWidth : true);
    }, [inputRef, renderValue, values, showTooltip]);
    const handleOpen = () => {
        setFilterValue('');
        setOpen(true);
        onOpen?.();
        inputRef.current?.focus();
        setTimeout(() => filter && filterRef.current?.focus(), 100);
    };
    const handleClose = useCallback(() => {
        if (filter && filterRef.current && lastFocusedRef.current === filterRef.current) {
            const event = {
                target: inputRef.current,
                currentTarget: inputRef.current,
                bubbles: true,
                cancelable: false,
                type: 'blur',
            };
            onBlur?.(event);
        }
        setFilterValue('');
        setOpen(false);
        onClose?.();
        inputRef.current?.blur();
    }, [setFilterValue, setOpen, onClose, inputRef, filter, filterRef, onBlur]);
    const inputLabel = renderValue ? renderValue(values) : defaultRenderValue(values);
    const onSelectItem = (item) => () => {
        onChange([item.value]);
        handleClose();
    };
    const onMultiselectItem = (item) => () => {
        // Используем compareBy для поиска элемента
        const itemIndex = privateValues.findIndex(value => compareValues(item.value, value.value, compareBy));
        const isAddition = itemIndex === -1;
        const updItems = isAddition ? [...privateValues, item] : privateValues.filter((_value, i) => itemIndex !== i);
        const userValues = updItems.map(item => item.value);
        onChange(userValues, item.value);
    };
    const [focusedIndex, setFocusedIndex] = useSelectKeyBoardNavigation({
        enableKeyboardNavigation,
        isOpen,
        options: privateOptions,
        handleClose,
        multiple,
        optionsRef,
        onMultiselectItem,
        onSelectItem,
        onManualClickOption,
        selectAll,
        inputRef,
        dropdownRef,
        filter,
        filterValue,
        filterFunction,
        groups,
    });
    const keyboardOptions = getOptionsInDisplayOrder(getPreparedOptions(privateOptions, filter, filterValue, filterFunction), groups);
    const focusedOptionId = focusedIndex >= 0 ? keyboardOptions[focusedIndex]?.id : undefined;
    const handleSelectAll = () => {
        const filteredOptions = getPreparedOptions(privateOptions, filter, filterValue, filterFunction);
        const enabledFilteredOptions = filteredOptions.filter(opt => !opt.disabled);
        const isChecked = getSelectAllStatus() === CheckboxType.CHECKED;
        const checkedFilterFunction = (val) => !enabledFilteredOptions.some(opt => compareValues(opt.value, val.value, compareBy));
        const updatedValues = isChecked
            ? privateValues.filter(element => checkedFilterFunction(element))
            : [
                ...privateValues,
                ...enabledFilteredOptions.filter(opt => !privateValues.some(val => compareValues(opt.value, val.value, compareBy))),
            ];
        onChange(updatedValues.map(item => item.value));
    };
    const getSelectAllStatus = () => getCheckboxStatus(privateOptions, privateValues, filterValue, filter, filterFunction, compareBy);
    const changeFilterValue = (value) => {
        setFilterValue(value);
    };
    return {
        isOpen,
        setOpen,
        filterValue,
        setFilterValue,
        disableTooltip,
        privateOptions,
        privateValues,
        inputLabel,
        handleOpen,
        handleClose,
        onSelectItem,
        onMultiselectItem,
        focusedIndex,
        focusedOptionId,
        setFocusedIndex,
        handleSelectAll,
        getSelectAllStatus,
        changeFilterValue,
    };
};

export { useSelect };
