// @beeline/design-system-react/build/components/Skeleton/Skeleton.js
import { jsx } from 'react/jsx-runtime';
import './Skeleton2.js';
import { useMemo } from 'react';
import { classNames } from '../../helpers/classnames.js';

/**
 * @description Skeleton
 * @param props - props
 * @param props.animated - animated or not animated skeleton. Default value: true
 * @param props.variant - variant of skeleton
 * @param props.className - custom class name
 * @param props.radius - border radius style
 * @param props.style - styles
 * @param props.height - height style
 * @param props.width - width style
 * @function
 */
const Skeleton = ({ variant = 'text', animated = true, className, radius, style, height, width, margin, ...rest }) => {
    /**
     * Memoized width. It set height like width, if it is square or circle variant and haven't width.
     */
    const localWidth = useMemo(() => ((variant === 'square' || variant === 'circle') && !width ? height : width), [height, variant, width]);
    /**
     * Memoized margins
     */
    const localMargins = useMemo(() => {
        if (typeof margin === 'object') {
            return {
                marginBottom: margin.bottom,
                marginLeft: margin.left,
                marginRight: margin.right,
                marginTop: margin.top,
            };
        }
        return {
            margin,
        };
    }, [margin]);
    /**
     * Memoized class names. Depends on variant, animation and custom class name.
     */
    const classes = useMemo(() => classNames('dsb_skeleton', `dsb_skeleton__${variant}`, { 'dsb_skeleton__no-animation': !animated }, className), [animated, className, variant]);
    /**
     * Memoized styles. It adds a borderRadius, height and width, if it has they.
     */
    const localStyle = useMemo(() => ({
        ...style,
        ...localMargins,
        borderRadius: radius,
        height,
        width: localWidth,
    }), [style, localMargins, radius, height, localWidth]);
    return (jsx("div", { "data-testid": "Skeleton", role: "progressbar", "aria-valuetext": "Loading", "aria-busy": "true", className: classes, style: localStyle, ...rest }));
};

export { Skeleton };
