// @beeline/design-tokens/js/tokens/components/label/label.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	labelHeight: "24px",
	labelBorderRadius: "12px",
	labelIconSize: "20px",
	labelTextFontSize: "15px",
	labelTextFontWeight: 500,
	labelTextFontLineHeight: "20px",
	labelTextFontLetterSpacing: "0.2px",
	labelContainedPaddingHorizontal: "8px",
	labelContainedFontSize: "15px",
	labelContainedFontWeight: 500,
	labelContainedFontLineHeight: "20px",
	labelContainedFontLetterSpacing: "0.2px",
	labelContainedTextColor: "var(--color-text-active-inverse, #ffffff)",
	labelContainedErrorBackgroundColor: "var(--color-status-error, #ff5555)",
	labelContainedSuccessBackgroundColor: "var(--color-status-success, #2ca853)",
	labelContainedInfoBackgroundColor: "var(--color-status-info, #1a73e8)",
	labelContainedWarningBackgroundColor: "var(--color-status-warning, #ff9419)",
	labelContainedNeutralBackgroundColor: "var(--color-status-neutral, #5e6071)",
	labelContainedMagentaBackgroundColor: "var(--color-accent-magenta, #bc00b8)",
	labelContainedPurpleBackgroundColor: "var(--color-accent-purple, #7e00ed)",
	labelContainedTealBackgroundColor: "var(--color-accent-teal, #00adc7)",
	labelContainedIconOnlyPadding: "2px",
	labelContainedIconOnlyWidth: "24px",
	labelOutlinePaddingLeft: "4px",
	labelOutlinePaddingRight: "8px",
	labelOutlineBorderWidth: "1px",
	labelOutlineBorderStyle: "solid",
	labelOutlineFontSize: "15px",
	labelOutlineFontWeight: 400,
	labelOutlineFontLineHeight: "18px",
	labelOutlineFontLetterSpacing: "0.2px",
	labelOutlineIconMarginRight: "4px",
	labelOutlineErrorTextColor: "var(--color-status-error, #ff5555)",
	labelOutlineErrorBorderColor: "var(--color-status-error, #ff5555)",
	labelOutlineSuccessTextColor: "var(--color-status-success, #2ca853)",
	labelOutlineSuccessBorderColor: "var(--color-status-success, #2ca853)",
	labelOutlineInfoTextColor: "var(--color-status-info, #1a73e8)",
	labelOutlineInfoBorderColor: "var(--color-status-info, #1a73e8)",
	labelOutlineWarningTextColor: "var(--color-status-warning, #ff9419)",
	labelOutlineWarningBorderColor: "var(--color-status-warning, #ff9419)",
	labelOutlineNeutralTextColor: "var(--color-status-neutral, #5e6071)",
	labelOutlineNeutralBorderColor: "var(--color-status-neutral, #5e6071)",
	labelOutlineMagentaTextColor: "var(--color-accent-magenta, #bc00b8)",
	labelOutlineMagentaBorderColor: "var(--color-accent-magenta, #bc00b8)",
	labelOutlinePurpleTextColor: "var(--color-accent-purple, #7e00ed)",
	labelOutlinePurpleBorderColor: "var(--color-accent-purple, #7e00ed)",
	labelOutlineTealTextColor: "var(--color-accent-teal, #00adc7)",
	labelOutlineTealBorderColor: "var(--color-accent-teal, #00adc7)",
	labelOnlyTextPaddingHorizontal: "8px",
	labelWithIconPaddingLeft: "4px",
	labelWithIconPaddingRight: "8px",
	labelWithIconSpacing: "4px",
	labelOnlyIconPadding: "2px",
	labelOnlyIconWidth: "24px",
};
