// @beeline/design-tokens/js/tokens/components/label/label.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const labelHeight = "24px";
export const labelBorderRadius = "12px";
export const labelIconSize = "20px";
export const labelTextFontSize = "15px";
export const labelTextFontWeight = 500;
export const labelTextFontLineHeight = "20px";
export const labelTextFontLetterSpacing = "0.2px";
export const labelContainedPaddingHorizontal = "8px";
export const labelContainedFontSize = "15px";
export const labelContainedFontWeight = 500;
export const labelContainedFontLineHeight = "20px";
export const labelContainedFontLetterSpacing = "0.2px";
export const labelContainedTextColor = "var(--color-text-active-inverse, #ffffff)";
export const labelContainedErrorBackgroundColor = "var(--color-status-error, #ff5555)";
export const labelContainedSuccessBackgroundColor = "var(--color-status-success, #2ca853)";
export const labelContainedInfoBackgroundColor = "var(--color-status-info, #1a73e8)";
export const labelContainedWarningBackgroundColor = "var(--color-status-warning, #ff9419)";
export const labelContainedNeutralBackgroundColor = "var(--color-status-neutral, #5e6071)";
export const labelContainedMagentaBackgroundColor = "var(--color-accent-magenta, #bc00b8)";
export const labelContainedPurpleBackgroundColor = "var(--color-accent-purple, #7e00ed)";
export const labelContainedTealBackgroundColor = "var(--color-accent-teal, #00adc7)";
export const labelContainedIconOnlyPadding = "2px";
export const labelContainedIconOnlyWidth = "24px";
export const labelOutlinePaddingLeft = "4px";
export const labelOutlinePaddingRight = "8px";
export const labelOutlineBorderWidth = "1px";
export const labelOutlineBorderStyle = "solid";
export const labelOutlineFontSize = "15px";
export const labelOutlineFontWeight = 400;
export const labelOutlineFontLineHeight = "18px";
export const labelOutlineFontLetterSpacing = "0.2px";
export const labelOutlineIconMarginRight = "4px";
export const labelOutlineErrorTextColor = "var(--color-status-error, #ff5555)";
export const labelOutlineErrorBorderColor = "var(--color-status-error, #ff5555)";
export const labelOutlineSuccessTextColor = "var(--color-status-success, #2ca853)";
export const labelOutlineSuccessBorderColor = "var(--color-status-success, #2ca853)";
export const labelOutlineInfoTextColor = "var(--color-status-info, #1a73e8)";
export const labelOutlineInfoBorderColor = "var(--color-status-info, #1a73e8)";
export const labelOutlineWarningTextColor = "var(--color-status-warning, #ff9419)";
export const labelOutlineWarningBorderColor = "var(--color-status-warning, #ff9419)";
export const labelOutlineNeutralTextColor = "var(--color-status-neutral, #5e6071)";
export const labelOutlineNeutralBorderColor = "var(--color-status-neutral, #5e6071)";
export const labelOutlineMagentaTextColor = "var(--color-accent-magenta, #bc00b8)";
export const labelOutlineMagentaBorderColor = "var(--color-accent-magenta, #bc00b8)";
export const labelOutlinePurpleTextColor = "var(--color-accent-purple, #7e00ed)";
export const labelOutlinePurpleBorderColor = "var(--color-accent-purple, #7e00ed)";
export const labelOutlineTealTextColor = "var(--color-accent-teal, #00adc7)";
export const labelOutlineTealBorderColor = "var(--color-accent-teal, #00adc7)";
export const labelOnlyTextPaddingHorizontal = "8px";
export const labelWithIconPaddingLeft = "4px";
export const labelWithIconPaddingRight = "8px";
export const labelWithIconSpacing = "4px";
export const labelOnlyIconPadding = "2px";
export const labelOnlyIconWidth = "24px";
