// @beeline/design-tokens/js/tokens/components/expansionPanel/expansionPanel.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	expansionPanelAccordionBorderRadius: "16px",
	expansionPanelAccordionBorderColor: "var(--color-border, rgba(25, 28, 52, 0.18))",
	expansionPanelAccordionBorderWidth: "1px",
	expansionPanelAccordionBackgroundColor: "var(--color-background-low, #ffffff)",
	expansionPanelHeaderPaddingVertical: "12px",
	expansionPanelHeaderPaddingHorizontal: "24px",
	expansionPanelHeaderSpacingHorizontal: "16px",
	expansionPanelHeaderTitleFontSize: "19px",
	expansionPanelHeaderTitleFontWeight: 500,
	expansionPanelHeaderTitleFontLineHeight: "24px",
	expansionPanelHeaderTitleFontLetterSpacing: "0.2px",
	expansionPanelHeaderTitleColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	expansionPanelHeaderSubtitleFontSize: "15px",
	expansionPanelHeaderSubtitleFontWeight: 400,
	expansionPanelHeaderSubtitleFontLineHeight: "18px",
	expansionPanelHeaderSubtitleFontLetterSpacing: "0.2px",
	expansionPanelHeaderSubtitleColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	expansionPanelHeaderDescriptionFontSize: "13px",
	expansionPanelHeaderDescriptionFontWeight: 400,
	expansionPanelHeaderDescriptionFontLineHeight: "16px",
	expansionPanelHeaderDescriptionFontLetterSpacing: "0.2px",
	expansionPanelHeaderDescriptionColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	expansionPanelHeaderBackgroundColor: "transparent",
	expansionPanelHeaderBackgroundColorHover: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	expansionPanelHeaderBackgroundColorPressed: "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))",
	expansionPanelHeaderBackgroundColorSelected: "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))",
	expansionPanelDividerColor: "var(--color-divider, rgba(25, 28, 52, 0.12))",
	expansionPanelDividerWidth: "1px",
	expansionPanelContentPaddingVertical: "16px",
	expansionPanelContentPaddingHorizontal: "24px",
};
