// @beeline/design-tokens/js/tokens/components/expansionPanel/expansionPanel.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const expansionPanelAccordionBorderRadius = "16px";
export const expansionPanelAccordionBorderColor = "var(--color-border, rgba(25, 28, 52, 0.18))";
export const expansionPanelAccordionBorderWidth = "1px";
export const expansionPanelAccordionBackgroundColor = "var(--color-background-low, #ffffff)";
export const expansionPanelHeaderPaddingVertical = "12px";
export const expansionPanelHeaderPaddingHorizontal = "24px";
export const expansionPanelHeaderSpacingHorizontal = "16px";
export const expansionPanelHeaderTitleFontSize = "19px";
export const expansionPanelHeaderTitleFontWeight = 500;
export const expansionPanelHeaderTitleFontLineHeight = "24px";
export const expansionPanelHeaderTitleFontLetterSpacing = "0.2px";
export const expansionPanelHeaderTitleColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const expansionPanelHeaderSubtitleFontSize = "15px";
export const expansionPanelHeaderSubtitleFontWeight = 400;
export const expansionPanelHeaderSubtitleFontLineHeight = "18px";
export const expansionPanelHeaderSubtitleFontLetterSpacing = "0.2px";
export const expansionPanelHeaderSubtitleColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const expansionPanelHeaderDescriptionFontSize = "13px";
export const expansionPanelHeaderDescriptionFontWeight = 400;
export const expansionPanelHeaderDescriptionFontLineHeight = "16px";
export const expansionPanelHeaderDescriptionFontLetterSpacing = "0.2px";
export const expansionPanelHeaderDescriptionColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const expansionPanelHeaderBackgroundColor = "transparent";
export const expansionPanelHeaderBackgroundColorHover = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const expansionPanelHeaderBackgroundColorPressed = "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))";
export const expansionPanelHeaderBackgroundColorSelected = "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))";
export const expansionPanelDividerColor = "var(--color-divider, rgba(25, 28, 52, 0.12))";
export const expansionPanelDividerWidth = "1px";
export const expansionPanelContentPaddingVertical = "16px";
export const expansionPanelContentPaddingHorizontal = "24px";
