// @beeline/design-system-react/build/components/DatePicker/Calendar/newUtils.js
import { d as dateFnsTz } from '../../../node_modules/date-fns-tz/index.js';
import isValid from '../../../node_modules/date-fns/esm/isValid/index.js';
import format from '../../../node_modules/date-fns/esm/format/index.js';

/** Преобразование UTC->TZ. Если timeZone не задана, возвращаем копию originalUTC. */
const convertUtcToTz = (originalUTC, timeZone) => {
    if (!timeZone) {
        return new Date(originalUTC);
    }
    return dateFnsTz.utcToZonedTime(originalUTC, timeZone);
};
/** Преобразование TZ->UTC. Если timeZone не задана, считаем, что zonedDate уже UTC. */
const convertTzToUtc = (zonedDate, timeZone) => {
    if (!timeZone) {
        return new Date(zonedDate);
    }
    return dateFnsTz.zonedTimeToUtc(zonedDate, timeZone);
};
/** Превращаем ISO-строку в Date(UTC). Если некорректно, возвращаем null. */
const getDateFromIsoString = (date) => {
    if (!date) {
        return null;
    }
    const isoDate = new Date(date);
    return isValid(isoDate) ? isoDate : null;
};
const isRangeMode = (value) => Array.isArray(value);
/** Возвращаем Date(UTC) для value. Если не получилось, вернём new Date() (UTC now). */
const getViewDate = (value) => {
    if (isRangeMode(value)) {
        const first = value[0];
        const parsed = getDateFromIsoString(first);
        return parsed || new Date();
    }
    const parsed = getDateFromIsoString(value);
    return parsed || new Date();
};
/** Форматируем Date(UTC) в нужном формате, напр. "yyyy-MM-dd'T'HH:mm:ss'Z'". */
const formatToISO = (dateUTC, outputDateFormat) => format(dateUTC, outputDateFormat);
/** Валидируем customViewDate (строка -> Date(UTC) или null). */
const validateCustomViewDate = (date) => {
    if (!date) {
        return null;
    }
    const iso = new Date(date);
    return isValid(iso) ? iso : null;
};
/** Проверяем, вышла ли checkedDate за пределы [minDate, maxDate], все в UTC. */
/**
 * Нормализуем дату к полуночи по UTC, сравнение выполняется только по году/месяцу/дню.
 * Это устраняет смещения по часовым поясам и off-by-one ошибки при проверке min/max.
 */
const normalizeToUtcMidnight = (d) => {
    if (!d || Number.isNaN(d.getTime())) {
        return null;
    }
    return new Date(Date.UTC(d.getUTCFullYear(), d.getUTCMonth(), d.getUTCDate()));
};
const hasDateLimit = (checkedDate, minDate, maxDate) => {
    const cd = normalizeToUtcMidnight(checkedDate);
    const min = normalizeToUtcMidnight(minDate);
    const max = normalizeToUtcMidnight(maxDate);
    if (!cd) {
        return false;
    }
    if (min && cd < min) {
        return true;
    }
    if (max && cd > max) {
        return true;
    }
    return false;
};
/** Сравниваем две даты (UTC) по году, месяцу, числу. */
const isSameDates = (d1, d2) => d1.getUTCFullYear() === d2.getUTCFullYear() &&
    d1.getUTCMonth() === d2.getUTCMonth() &&
    d1.getUTCDate() === d2.getUTCDate();
/** Проверяем валидность времени (val) в заданном диапазоне timeRange. Возвращаем true, если НЕвалидно. */
const isTimeValid = (val, timeRange = ['00:00', '23:59']) => {
    const [h, m] = [+val.hours, +val.minutes];
    const [minH, minM] = timeRange[0].split(':').map(Number);
    const [maxH, maxM] = timeRange[1].split(':').map(Number);
    // isInRange === true, если внутри диапазона
    const isInRange = (h > minH || (h === minH && m >= minM)) && (h < maxH || (h === maxH && m <= maxM));
    // Возвращаем true, если время вышло за диапазон => невалидно.
    return !isInRange;
};
/**
 * Возвращаем массив недель (каждая неделя – 7 чисел или null) для указанного локального года/месяца.
 * @param localYear Год
 * @param localMonth Месяц
 * @param showAdjacentMonthDays Объект с настройками отображения дней соседних месяцев
 */
const getWeeksOfMonth = (localYear, localMonth, showAdjacentMonthDays) => {
    // Локальная логика (в JS new Date(year, month, day) – локальная).
    // Но если нужно строгое ISO поведение, можно усложнить.
    // 1) найдём, какой день недели у первого числа (0=вс,1=пн,...)
    const firstDate = new Date(localYear, localMonth, 1);
    let startDay = firstDate.getDay(); // 0 (вс) ... 6 (сб)
    // Но в России обычно пн=1,... Вс=0 => возможно, стоит сдвигать.
    // Чтобы пн был первым (WEEKDAYS=["Пн","Вт",...,"Вс"]), сделаем небольшой трюк:
    // Если startDay=0 (вс), считаем, что это 7:
    if (startDay === 0) {
        startDay = 7;
    }
    // Сколько дней в месяце?
    const daysInMonth = new Date(localYear, localMonth + 1, 0).getDate();
    // Получаем информацию о предыдущем месяце
    const prevMonth = localMonth === 0 ? 11 : localMonth - 1;
    const prevYear = localMonth === 0 ? localYear - 1 : localYear;
    const daysInPrevMonth = new Date(prevYear, prevMonth + 1, 0).getDate();
    // Получаем информацию о следующем месяце
    const nextMonth = localMonth === 11 ? 0 : localMonth + 1;
    const nextYear = localMonth === 11 ? localYear + 1 : localYear;
    const weeks = [];
    let currentDay = 1;
    // считаем максимум 6 недель
    for (let w = 0; w < 6; w++) {
        const weekRow = Array.from({ length: 7 }).fill(null);
        for (let d = 0; d < 7; d++) {
            // Первая неделя (w=0): начнём с (startDay-1) как сдвиг
            if (w === 0 && d < startDay - 1) {
                // Дни предыдущего месяца
                if (showAdjacentMonthDays?.showPrevious) {
                    const prevMonthDay = daysInPrevMonth - (startDay - 2 - d);
                    weekRow[d] = { day: prevMonthDay, month: prevMonth, year: prevYear };
                }
                else {
                    weekRow[d] = null;
                }
            }
            else if (currentDay > daysInMonth) {
                // Дни следующего месяца
                if (showAdjacentMonthDays?.showNext) {
                    const nextMonthDay = currentDay - daysInMonth;
                    weekRow[d] = { day: nextMonthDay, month: nextMonth, year: nextYear };
                    currentDay++;
                }
                else {
                    weekRow[d] = null;
                }
            }
            else {
                // Дни текущего месяца
                weekRow[d] = { day: currentDay, month: localMonth, year: localYear };
                currentDay++;
            }
        }
        weeks.push(weekRow);
        // Если все дни текущего месяца уже отображены и не нужно показывать дни следующего месяца
        if (currentDay > daysInMonth && !showAdjacentMonthDays?.showNext) {
            break;
        }
        // Если все дни текущего месяца отображены и последняя неделя полностью заполнена
        if (w > 0 && currentDay > daysInMonth && weekRow.every(cell => cell !== null)) {
            break;
        }
    }
    return weeks;
};
/**
 * Сдвиг месяца влево/вправо (delta=-1 или +1) без мутаций исходной даты.
 * Сохраняет день месяца, если он существует в целевом месяце, иначе клемпит до последнего дня.
 * Возвращает новую дату в UTC.
 */
const shiftMonthUTC = (viewDateUTC, timeZone, delta, enableTimePicker) => {
    // Локальное представление для календарной арифметики
    const localView = enableTimePicker
        ? convertUtcToTz(viewDateUTC, timeZone)
        : new Date(Date.UTC(viewDateUTC.getUTCFullYear(), viewDateUTC.getUTCMonth(), viewDateUTC.getUTCDate()));
    const currentDay = localView.getDate();
    const targetMonthIndex = localView.getMonth() + delta;
    const targetYear = localView.getFullYear() + Math.floor(targetMonthIndex / 12);
    const normalizedMonth = ((targetMonthIndex % 12) + 12) % 12;
    const lastDay = new Date(targetYear, normalizedMonth + 1, 0).getDate();
    const targetDay = Math.min(currentDay, lastDay);
    if (enableTimePicker) {
        const targetLocal = new Date(targetYear, normalizedMonth, targetDay, localView.getHours(), localView.getMinutes(), 0, 0);
        return convertTzToUtc(targetLocal, timeZone);
    }
    return new Date(Date.UTC(targetYear, normalizedMonth, targetDay));
};

export { convertTzToUtc, convertUtcToTz, formatToISO, getDateFromIsoString, getViewDate, getWeeksOfMonth, hasDateLimit, isRangeMode, isSameDates, isTimeValid, shiftMonthUTC, validateCustomViewDate };
