// @beeline/design-system-react/build/components/DatePicker/Calendar/utils.js
import { d as dateFnsTz } from '../../../node_modules/date-fns-tz/index.js';
import { DAYS_IN_WEEK, ISO_FORMAT_PATTERN, ONE_DAY, SUNDAY_INDEX } from './constants.js';
import isValid from '../../../node_modules/date-fns/esm/isValid/index.js';

const isFilledDate = (value, options = {}) => {
    const { pattern = ISO_FORMAT_PATTERN, isPartial = false } = options;
    if (isRangeMode(value)) {
        return isPartial ? value.some(date => date && pattern.test(date)) : value.every(date => date && pattern.test(date));
    }
    return pattern.test(value);
};
const formatToISO = (date, outputDateFormat) => {
    const formatInTimeZone = (date, fmt, tz) => dateFnsTz.format(dateFnsTz.utcToZonedTime(date, tz), fmt, { timeZone: tz });
    return formatInTimeZone(date, outputDateFormat, 'UTC');
};
const getDateFromIsoString = (date) => {
    const isoDate = new Date(date);
    return isValid(isoDate) ? isoDate : null;
};
const isRangeMode = (value) => Array.isArray(value);
/**
 * Нормализуем дату к полуночи по UTC, чтобы сравнивать только по компонентам года/месяца/дня
 * и устранить off-by-one из-за разницы часовых поясов.
 */
const normalizeToUtcMidnight = (d) => {
    if (!d || Number.isNaN(d.getTime())) {
        return null;
    }
    return new Date(Date.UTC(d.getUTCFullYear(), d.getUTCMonth(), d.getUTCDate()));
};
const hasDateLimit = (checkedDate, minDate, maxDate) => {
    const cd = normalizeToUtcMidnight(checkedDate);
    const min = normalizeToUtcMidnight(minDate);
    const max = normalizeToUtcMidnight(maxDate);
    if (!cd) {
        return false;
    }
    else if (min && !max) {
        return cd < min;
    }
    else if (max && !min) {
        return cd > max;
    }
    else if (min && max) {
        return cd < min || cd > max;
    }
    return false;
};
const isValidDateString = (date) => {
    const parsedDate = Date.parse(date);
    return isFilledDate(date) && !Number.isNaN(parsedDate);
};
const isValidDateValue = (value, minDate, maxDate) => {
    if (isRangeMode(value)) {
        return (value.some(element => isValid(element)) &&
            value.some(isoDate => !hasDateLimit(getDateFromIsoString(isoDate), minDate, maxDate)));
    }
    return isValidDateString(value) && !hasDateLimit(getDateFromIsoString(value), minDate, maxDate);
};
// // Сюда может прийти
// 	08.08.2024 - обычная дата ( мы её вернули в UTC)
// 	08.08.2024 - обычная дата ( Часовой пояс)
// 	08.08.2024: +3:00 - обычная дата ( Часовой пояс)
// 	08.08.2024:Z - обычная дата ( Часовой пояс)
const getViewDate = (date) => {
    const viewDate = isRangeMode(date) ? date[0] : date;
    const currentDate = new Date();
    return getDateFromIsoString(viewDate) || currentDate;
};
const isSameDates = (firstDate, secondDate) => firstDate.getFullYear() === secondDate.getFullYear() &&
    firstDate.getMonth() === secondDate.getMonth() &&
    firstDate.getDate() === secondDate.getDate();
const isSunday = (dayIndex) => dayIndex === SUNDAY_INDEX;
const getFirstDayOfWeekInMonth = (year, month) => new Date(year, month, 1).getDay();
const countDaysInMonth = (year, month) => new Date(year, month + 1, 0).getDate();
const countWeeksInMonth = (firstMonthDay, totalDaysInMonth) => {
    const correctionIndex = isSunday(firstMonthDay) ? 6 : -1;
    return Math.ceil((firstMonthDay + totalDaysInMonth + correctionIndex) / DAYS_IN_WEEK);
};
const getDaysListOfMonth = (totalDaysInMonth) => Array.from({ length: totalDaysInMonth }, (empty, i) => i + 1);
const getWeeksOfMonth = (year, month) => {
    const weeksOfMonth = [];
    const firstDayOfWeekInMonth = getFirstDayOfWeekInMonth(year, month);
    const totalDaysInMonth = countDaysInMonth(year, month);
    const totalWeeksInMonth = countWeeksInMonth(firstDayOfWeekInMonth, totalDaysInMonth);
    /* Создаем массив всех дней месяца - [1, 2, 3, 4, 5, 6, 7, 8, 9, ...] */
    const daysListOfMonth = getDaysListOfMonth(totalDaysInMonth);
    for (let weekNumber = 1; weekNumber <= totalWeeksInMonth; weekNumber++) {
        /* Инициализируем массив текущей недели заполненный null'ами - [null x 7] */
        const currentWeek = new Array(DAYS_IN_WEEK).fill(null);
        const isFirstWeek = weekNumber === 1;
        /* Обрабатываем первую неделю месяца отдельно,
         * так как количество недель в месяце не кратко общему количеству дней
         */
        if (isFirstWeek) {
            /* Находим количество пустых (отсутствующих) дней первой недели месяца отображаемых в календаре */
            const totalEmptyDaysInFirstWeek = isSunday(firstDayOfWeekInMonth)
                ? DAYS_IN_WEEK - ONE_DAY
                : firstDayOfWeekInMonth - ONE_DAY;
            const totalDaysInCurrentWeek = isSunday(firstDayOfWeekInMonth)
                ? ONE_DAY
                : DAYS_IN_WEEK - totalEmptyDaysInFirstWeek;
            const daysListOfCurrentWeek = daysListOfMonth.splice(0, totalDaysInCurrentWeek);
            /* Заполняем массив текущей (первой) недели,
             * учитывая возможные пустые дни с начала месяца - [null, null, 1, 2, 3, 4, 5]
             */
            currentWeek.splice(totalEmptyDaysInFirstWeek, daysListOfCurrentWeek.length, ...daysListOfCurrentWeek);
        }
        else {
            const daysListOfCurrentWeek = daysListOfMonth.splice(0, DAYS_IN_WEEK);
            /* Для остальных недель учет оставшихся пустых дней последней недели
             * будет происходить автоматически - [27, 28, 29, 30, 31, null, null]
             */
            currentWeek.splice(0, daysListOfCurrentWeek.length, ...daysListOfCurrentWeek);
        }
        weeksOfMonth.push(currentWeek);
    }
    return weeksOfMonth;
};
const createArrayRange = (start, stop) => Array.from({ length: stop - start + 1 }, (value, index) => start + index);
const getWeeksOfMonths = (year, minYear = year - 1, maxYear = year + 1) => {
    const monthsNumbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11];
    return createArrayRange(minYear, maxYear).flatMap(currentYear => {
        const newMonthWeeks = monthsNumbers.map(monthNumber => ({
            year: currentYear,
            month: monthNumber,
            monthWeeks: getWeeksOfMonth(currentYear, monthNumber),
        }));
        return newMonthWeeks;
    });
};

export { formatToISO, getDateFromIsoString, getViewDate, getWeeksOfMonth, getWeeksOfMonths, hasDateLimit, isFilledDate, isRangeMode, isSameDates, isValidDateString, isValidDateValue };
