// @beeline/design-tokens/js/tokens/components/textarea/textarea.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	textareaTextMarginTop: "22px",
	textareaTextMarginBottom: "24px",
	textareaResizerMarginRight: "8px",
	textareaResizerMarginBottom: "8px",
	textareaSmallHeight: "92px",
	textareaSmallLabelMarginTop: "10px",
	textareaSmallLabelMarginTopFloated: "6px",
	textareaSmallLabelFontSize: "15px",
	textareaSmallLabelFontWeight: 400,
	textareaSmallLabelFontLineHeight: "18px",
	textareaSmallLabelFontLetterSpacing: "0.2px",
	textareaSmallLabelFloatedMarginTop: "6px",
	textareaSmallLabelFloatedFontSize: "13px",
	textareaSmallLabelFloatedFontLineHeight: "16px",
	textareaSmallTextHeight: "60px",
	textareaSmallWithoutLabelHeight: "80px",
	textareaSmallWithoutLabelTextMarginTop: "10px",
	textareaMediumHeight: "104px",
	textareaMediumTextHeight: "66px",
	textareaMediumLabelMarginTop: "14px",
	textareaMediumLabelMarginTopFloated: "6px",
	textareaMediumLabelFloatedMarginTop: "6px",
	textareaMediumWithoutLabelHeight: "96px",
	textareaMediumWithoutLabelTextMarginTop: "10px",
	textareaWithLabelTextMarginTop: "22px",
	textareaWithLabelSmallHeight: "92px",
	textareaWithLabelMediumHeight: "104px",
	textareaWithoutLabelSmallHeight: "80px",
	textareaWithoutLabelSmallTextMarginTop: "10px",
	textareaWithoutLabelMediumHeight: "96px",
	textareaWithoutLabelMediumTextMarginTop: "14px",
};
