// @beeline/design-tokens/js/tokens/components/textarea/textarea.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const textareaTextMarginTop = "22px";
export const textareaTextMarginBottom = "24px";
export const textareaResizerMarginRight = "8px";
export const textareaResizerMarginBottom = "8px";
export const textareaSmallHeight = "92px";
export const textareaSmallLabelMarginTop = "10px";
export const textareaSmallLabelMarginTopFloated = "6px";
export const textareaSmallLabelFontSize = "15px";
export const textareaSmallLabelFontWeight = 400;
export const textareaSmallLabelFontLineHeight = "18px";
export const textareaSmallLabelFontLetterSpacing = "0.2px";
export const textareaSmallLabelFloatedMarginTop = "6px";
export const textareaSmallLabelFloatedFontSize = "13px";
export const textareaSmallLabelFloatedFontLineHeight = "16px";
export const textareaSmallTextHeight = "60px";
export const textareaSmallWithoutLabelHeight = "80px";
export const textareaSmallWithoutLabelTextMarginTop = "10px";
export const textareaMediumHeight = "104px";
export const textareaMediumTextHeight = "66px";
export const textareaMediumLabelMarginTop = "14px";
export const textareaMediumLabelMarginTopFloated = "6px";
export const textareaMediumLabelFloatedMarginTop = "6px";
export const textareaMediumWithoutLabelHeight = "96px";
export const textareaMediumWithoutLabelTextMarginTop = "10px";
export const textareaWithLabelTextMarginTop = "22px";
export const textareaWithLabelSmallHeight = "92px";
export const textareaWithLabelMediumHeight = "104px";
export const textareaWithoutLabelSmallHeight = "80px";
export const textareaWithoutLabelSmallTextMarginTop = "10px";
export const textareaWithoutLabelMediumHeight = "96px";
export const textareaWithoutLabelMediumTextMarginTop = "14px";
