// @beeline/design-system-react/build/components/Chip/Chip.js
import { jsxs, jsx } from 'react/jsx-runtime';
import './Chip2.js';
import { NewDropdownMenu } from '../NewDropdownMenu/NewDropdownMenu.js';
import { PopoverContext } from '../Popover/Popover.js';
import '../Popover/PopoverHelper/PopoverHelper.js';
import '../Popover/PopoverOnboarding/PopoverOnboarding2.js';
import '../Icon/Icon.js';
import { classNames } from '../../helpers/classnames.js';
import '@beeline/design-tokens/js/iconfont/icons';
import '../Popover/PopoverOnboarding/Slides/Slide/Slide2.js';
import '../Button/Button.js';
import '../Button/ProgressButton/ProgressButton.js';
import { Typography } from '../Typography/Typography.js';
import '../Popover/PopoverOnboarding/Slides/Stepper/Stepper2.js';
import { DropDownContent } from '../Select/components/DropDownContent/index.js';
import { useOverflowText } from '../Select/components/hooks/useOverflowText.js';
import { OverflowCount } from '../Select/components/OverflowCount/index.js';
import { defaultRenderValue, defaultMakeOption } from '../Select/helpers.js';
import { Tooltip } from '../Tooltip/Tooltip.js';
import { forwardRef, useRef, useImperativeHandle, useState, useEffect, useMemo, isValidElement, cloneElement } from 'react';
import { useChip } from './useChip.js';

const Chip = forwardRef(({ values, label, active = false, disabled = false, dragged = false, multiple, filter, filterPlaceholder = 'Поиск', filterFunction, onClick, className, rootClassName, startAdornment, endAdornment, isOpen = false, onOutsideClick, onOpen, onClose, onEscape, onBlur, options = [], onSelect, dropdownElementID, dropdownClassName, dataTestId = 'Chip', selectAll = false, selectAllText = 'Выбрать все', onManualClickOption, loading, loadingText, loadingContent, makeOption, renderValue, showTooltip, noOptionsText = 'Нет результатов', enableKeyboardNavigation = false, isShowCountOfValues = false, isSelectMode = false, applicationRootElementID, overlayScroll = true, shouldRenderOverlay, overlayClassName, alignDropDown, ...props }, ref) => {
    const { chipRef, dropdownOpen, setDropdownOpen, handleClickOutside, selectClassName, privateValues, privateOptions, filterValue, optionsClassName, itemClassName, onMultiselectItem, onSelectItem, focusedIndex, handleSelectAll, optionsRef, getSelectAllStatus, filterRef, changeFilterValue, selectRef, handleOpen, } = useChip({
        isOpen,
        options,
        values,
        isSelectMode,
        onOpen,
        onClose,
        onOutsideClick,
        onSelect,
        filter,
        filterFunction,
        enableKeyboardNavigation,
        multiple,
        selectAll,
    });
    const counterWrapperRef = useRef(null);
    useImperativeHandle(ref, () => chipRef?.current);
    const selectAllIndex = -2;
    const [disableTooltip, setDisableTooltip] = useState(true);
    useEffect(() => {
        const shouldDisableTooltip = showTooltip ? chipRef?.current?.offsetWidth >= chipRef?.current?.scrollWidth : true;
        setDisableTooltip(shouldDisableTooltip);
    }, [chipRef, renderValue, values, showTooltip]);
    const dropdownMenuProps = {
        targetElement: chipRef,
        align: alignDropDown,
        verticalOffset: 1,
        applicationRootElementID,
        dropdownElementID,
        className: dropdownClassName,
        onBlur,
        isOpen: dropdownOpen,
        handleSetOpen: setDropdownOpen,
        onOutsideClick: handleClickOutside,
        overlayScroll,
        shouldRenderOverlay,
        overlayClassName,
    };
    const rootWrapClassName = classNames(selectClassName, Boolean(rootClassName) && rootClassName);
    const popoverContextValue = useMemo(() => ({
        applicationRootElementId: 'root',
        popoverRootElementID: 'chip-popover',
    }), []);
    const hiddenRef = useRef(null);
    const countRef = useRef(null);
    const { count, className: overflowClassName } = useOverflowText({
        hiddenRef,
        inputRef: counterWrapperRef,
    });
    const selectedLabel = renderValue ? renderValue(values) : defaultRenderValue(values);
    const counter = (jsxs("span", { className: classNames(overflowClassName, 'static'), ref: countRef, children: ["(+", count, ")"] }));
    const handleClick = (event) => {
        if (isSelectMode) {
            handleOpen?.();
        }
        else {
            onClick?.(event);
        }
    };
    return (jsxs("div", { className: rootWrapClassName, "data-testid": dataTestId, children: [jsx(PopoverContext.Provider, { value: popoverContextValue, children: jsx(Tooltip, { title: selectedLabel, disableHover: disableTooltip, children: jsxs("button", { "data-testid": "Chip-button", tabIndex: onClick ? 0 : -1, "aria-pressed": !!onClick, onClick: disabled ? undefined : handleClick, disabled: disabled, className: classNames('dsb_chip', active && 'dsb_chip--active', disabled && 'dsb_chip--disabled', startAdornment && 'dsb_chip--start-icon', endAdornment && 'dsb_chip--end-icon', onClick && 'dsb_chip--clickable', dragged && 'dsb_chip--dragged', multiple && 'dsb_chip--multiple', className), ...props, ref: chipRef, children: [startAdornment && (jsx("div", { className: "dsb_chip__adornment dsb_chip__start-icon", children: isValidElement(startAdornment)
                                    ? cloneElement(startAdornment, {
                                        size: 'small',
                                    })
                                    : startAdornment })), jsxs("div", { ref: counterWrapperRef, className: classNames(multiple && 'dsb_chip--multiple', 'ellipsis'), children: [jsx(Typography, { variant: "body3", className: "dsb_chip__title", children: selectedLabel || label }), isSelectMode && isShowCountOfValues && (jsx(OverflowCount, { inputRef: counterWrapperRef, hiddenRef: hiddenRef, values: values, countRef: countRef }))] }), isSelectMode && isShowCountOfValues && counter, endAdornment && (jsx("div", { className: "dsb_chip__adornment dsb_chip__end-icon", children: isValidElement(endAdornment)
                                    ? cloneElement(endAdornment, {
                                        size: 'small',
                                    })
                                    : endAdornment }))] }) }) }), isSelectMode && dropdownOpen && (jsx(NewDropdownMenu, { ...dropdownMenuProps, children: jsx(DropDownContent, { selectedItems: privateValues, optionsData: privateOptions, selectAll: selectAll, filter: filter, filterValue: filterValue, filterFunction: filterFunction, selectAllIndex: selectAllIndex, optionsClassName: optionsClassName, multiple: multiple, itemClassName: itemClassName, onMultiselectItem: onMultiselectItem, onManualClickOption: onManualClickOption, onSelectItem: onSelectItem, focusedIndex: focusedIndex, handleSelectAll: handleSelectAll, makeOption: makeOption, showTooltip: showTooltip, defaultMakeOption: defaultMakeOption, selectAllText: selectAllText, loading: loading, loadingText: loadingText, loadingContent: loadingContent, noOptionsText: noOptionsText, optionsRef: optionsRef, selectStatus: getSelectAllStatus(), filterPlaceholder: filterPlaceholder, filterRef: filterRef, changeFilterValue: changeFilterValue, dataTestId: dataTestId, selectRef: selectRef }) }))] }));
});

export { Chip };
