// @beeline/design-system-react/extracted-styles/esm/components/Popover/Popover.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { classNames } from '../../helpers/classnames.js';
import { PopupsContext } from '../../helpers/popupsContext.js';
import { useFloating } from '../../node_modules/@floating-ui/react/dist/floating-ui.react.js';
import React__default, { createContext, isValidElement, useState, useRef, useEffect, cloneElement, useContext, useLayoutEffect } from 'react';
import { createPortal } from 'react-dom';
import { offset, flip, shift, size } from '../../node_modules/@floating-ui/react-dom/dist/floating-ui.react-dom.js';
import { autoUpdate } from '../../node_modules/@floating-ui/dom/dist/floating-ui.dom.js';

const PopoverContext = createContext({
    applicationRootElementId: 'root',
    popoverRootElementID: 'dsb_popover',
});
const Popover = ({ open = false, content, disableHover = false, triggerMode = 'hover', closeMode = 'auto', closeOnScroll = false, onScroll, popoverClassName, portalClassName, children, noInteract, dataTestId = 'Popover', 'aria-describedby': ariaDescribedBy, attachToChildren = false, placement = 'bottom', transitionDuration = { enter: 400, exit: 200 }, }) => {
    if (!isValidElement(children)) {
        throw new Error('Popover children is not valid element!');
    }
    // Убрана проверка на обязательный ref, теперь он обрабатывается автоматически через cloneElement
    // if (!isForwardRef(children) && !children.ref) { ... }
    const enterDuration = typeof transitionDuration === 'number' ? transitionDuration : transitionDuration?.enter || 0;
    const exitDuration = typeof transitionDuration === 'number' ? transitionDuration : transitionDuration?.exit || 0;
    const [isVisible, setVisibility] = useState(open);
    const [wrappedContentRect, setWrappedContentRect] = useState(new DOMRect());
    // Animation states
    const [isMounted, setIsMounted] = useState(open);
    const [isOpen, setIsOpen] = useState(open);
    const targetRef = useRef(null);
    // Убираем useImperativeHandle, так как мы теперь сами управляем ref через cloneElement
    // useImperativeHandle(children.ref, () => targetRef.current);
    // Handle animation lifecycle
    useEffect(() => {
        if (isVisible) {
            setIsMounted(true);
            // Используем requestAnimationFrame для гарантии того, что элемент смонтирован
            // перед применением стилей анимации (opacity: 1).
            // Двойной requestAnimationFrame часто нужен для надежного срабатывания CSS-транзишнов
            // сразу после монтирования.
            requestAnimationFrame(() => {
                requestAnimationFrame(() => {
                    setIsOpen(true);
                });
            });
            return;
        }
        setIsOpen(false);
        if (exitDuration === 0) {
            setIsMounted(false);
            return;
        }
        const timer = setTimeout(() => {
            setIsMounted(false);
        }, exitDuration);
        return () => clearTimeout(timer);
    }, [isVisible, exitDuration]);
    const duration = isOpen ? enterDuration : exitDuration;
    const transitionStyle = enterDuration > 0 || exitDuration > 0
        ? {
            opacity: isOpen ? 1 : 0,
            visibility: isOpen ? 'visible' : 'hidden',
            transitionProperty: 'opacity, visibility',
            transitionDuration: `${duration}ms`,
            transitionTimingFunction: 'ease-in-out',
        }
        : {};
    // Handle hover events
    useEffect(() => {
        const targetEl = targetRef.current;
        const handleMouseEnter = () => {
            if (triggerMode === 'hover' || triggerMode === 'both') {
                setVisible(true);
            }
        };
        const handleMouseLeave = () => {
            if (triggerMode === 'hover' || triggerMode === 'both') {
                setVisible(false);
            }
        };
        const handleScroll = (event) => {
            if (!isVisible)
                return;
            // Вызываем пользовательский callback, если он предоставлен
            if (onScroll) {
                const shouldClose = onScroll(event);
                if (shouldClose) {
                    setVisible(false);
                }
                return; // Если есть onScroll callback, не выполняем остальную логику
            }
            // Автоматическое закрытие при скролле
            if (closeOnScroll) {
                setVisible(false);
                return;
            }
            // Старое поведение для hover режима
            if (triggerMode === 'hover' || triggerMode === 'both') {
                setVisible(false);
            }
        };
        if (disableHover || triggerMode === 'click') ;
        else {
            targetEl?.addEventListener('mouseenter', handleMouseEnter);
            targetEl?.addEventListener('mouseleave', handleMouseLeave);
        }
        // Добавляем слушатель события скролла если нужно
        if (closeOnScroll || onScroll || triggerMode === 'hover' || triggerMode === 'both') {
            window.addEventListener('scroll', handleScroll, true);
        }
        return () => {
            targetEl?.removeEventListener('mouseenter', handleMouseEnter);
            targetEl?.removeEventListener('mouseleave', handleMouseLeave);
            // Удаляем слушатель события скролла при размонтировании
            if (closeOnScroll || onScroll || triggerMode === 'hover' || triggerMode === 'both') {
                window.removeEventListener('scroll', handleScroll, true);
            }
        };
    }, [disableHover, triggerMode, children, isVisible, closeOnScroll, onScroll]);
    // Handle click events
    useEffect(() => {
        const targetEl = targetRef.current;
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const handleClick = (_event) => {
            if (triggerMode === 'click' || triggerMode === 'both') {
                if (closeMode === 'manual' && isVisible) {
                    return; // Только открываем, но не закрываем в ручном режиме
                }
                setVisible(!isVisible);
            }
        };
        const handleOutsideClick = (event) => {
            if (closeMode === 'manual') {
                return;
            }
            const popoverContentElements = document.querySelectorAll('.dsb_popover-content');
            // Проверяем, находится ли клик внутри любого из попапов
            const isClickInsideAnyContent = [...popoverContentElements].some(element => element.contains(event.target));
            // обработка разных режимов закрытия
            if (isVisible && targetEl && !targetEl.contains(event.target)) {
                if (closeMode === 'auto') {
                    // В режиме auto закрываем поповер при клике на любой элемент,
                    // который не является частью текущего поповера
                    setVisible(false);
                }
                else if (closeMode === 'outside-click' && !isClickInsideAnyContent) {
                    // Закрывайте только клики вне триггера и контента
                    setVisible(false);
                }
                else if (closeMode === 'content-click' && isClickInsideAnyContent) {
                    // Закрывать только по клику внутри контента
                    setVisible(false);
                }
            }
        };
        if (triggerMode === 'click' || triggerMode === 'both') {
            targetEl?.addEventListener('click', handleClick);
            document.addEventListener('click', handleOutsideClick);
        }
        return () => {
            targetEl?.removeEventListener('click', handleClick);
            document.removeEventListener('click', handleOutsideClick);
        };
    }, [triggerMode, closeMode, isVisible]);
    useEffect(() => {
        setVisible(open);
    }, [open]);
    const setVisible = (isVisible) => {
        const targetEl = targetRef.current;
        if (isVisible && targetEl) {
            setWrappedContentRect(targetEl.getBoundingClientRect());
        }
        setVisibility(isVisible);
    };
    return (jsxs(Fragment, { children: [cloneElement(children, {
                ...children.props,
                ref: (node) => {
                    // Сохраняем ref в наш внутренний targetRef
                    targetRef.current = node;
                    // Если у children уже был ref, вызываем и его
                    const childRef = children.ref;
                    if (typeof childRef === 'function') {
                        childRef(node);
                    }
                    else if (childRef) {
                        childRef.current = node;
                    }
                },
                'aria-haspopup': 'true',
                'aria-expanded': isVisible ? 'true' : 'false',
                'aria-controls': isVisible ? dataTestId : undefined,
            }), isMounted && (jsx(PopoverPortal, { "aria-describedby": ariaDescribedBy, noInteract: noInteract, content: content, wrappedContentRect: wrappedContentRect, className: popoverClassName, dataTestId: dataTestId, portalClassName: portalClassName, attachToChildren: attachToChildren, targetRef: targetRef, placement: placement, style: transitionStyle }))] }));
};
const PopoverPortal = ({ content, wrappedContentRect, className, portalClassName, noInteract = true, dataTestId = 'PopoverContent', attachToChildren = false, targetRef, 'aria-describedby': ariaDescribedBy, placement = 'bottom', style, }) => {
    const rootRef = React__default.useRef(null);
    const { applicationRootElementId, popoverRootElementID } = useContext(PopoverContext);
    const { applicationElementID, rootElement } = useContext(PopupsContext);
    const appElementID = applicationRootElementId || applicationElementID || 'root';
    // floating-ui: позиционирование относительно триггера
    const { refs, x, y, update } = useFloating({
        strategy: 'fixed',
        placement,
        middleware: [
            // TODO заменить на токен, когда он появится
            offset(8),
            flip({ crossAxis: true }),
            // TODO заменить на токен, когда он появится
            shift({ padding: 8, crossAxis: true, rootBoundary: 'viewport' }),
            size({
                apply({ availableWidth, availableHeight, elements }) {
                    Object.assign(elements.floating.style, {
                        maxHeight: `${availableHeight}px`,
                        maxWidth: `${availableWidth}px`,
                    });
                },
                padding: 8,
            }),
        ],
    });
    // Создаём контейнер портала (внешняя оболочка), к которой будут применяться координаты
    const elRef = useRef(null);
    if (!elRef.current) {
        const el = document.createElement('div');
        el.id = popoverRootElementID;
        el.className = classNames('dsb_popover', noInteract && 'dsb_popover__no-interact', portalClassName);
        el.style.position = 'fixed';
        elRef.current = el;
    }
    // Определяем элемент, к которому будет прикреплен портал
    // Если attachToChildren=true, используем targetRef (children)
    let mount;
    attachToChildren && targetRef?.current
        ? (mount = targetRef.current)
        : (mount = rootElement || document.querySelector(`#${appElementID}`));
    if (!mount) {
        throw new Error(`Application root element with ID "${appElementID}" not found`);
    }
    const el = elRef.current;
    el.classList.add('inverseTheme');
    // Привязываем элементы к floating-ui
    useLayoutEffect(() => {
        // reference: при наличии targetRef, иначе виртуальный элемент по wrappedContentRect
        if (targetRef?.current) {
            refs.setReference(targetRef.current);
        }
        else {
            const contextEl = mount instanceof ShadowRoot ? mount.host : mount;
            const virtualReference = {
                getBoundingClientRect: () => wrappedContentRect,
                contextElement: contextEl,
            };
            // setReference поддерживает VirtualElement через дженерик
            refs.setReference(virtualReference);
        }
        refs.setFloating(el);
        const cleanup = autoUpdate(refs.reference.current, el, update);
        return () => cleanup();
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [el, mount, targetRef, wrappedContentRect]);
    // Монтирование/демонтаж портала
    useLayoutEffect(() => {
        mount.append(el);
        return () => {
            el.remove();
        };
    }, [el, mount]);
    // Применяем координаты из floating-ui
    useLayoutEffect(() => {
        if (typeof x === 'number' && typeof y === 'number') {
            el.style.left = `${x}px`;
            el.style.top = `${y}px`;
        }
    }, [x, y, el]);
    const popoverContent = (jsx("div", { ref: rootRef, className: classNames('dsb_popover-content', className), "data-testid": dataTestId, role: "tooltip", id: dataTestId, "aria-hidden": !style || style.visibility === 'hidden', "aria-describedby": ariaDescribedBy, style: style, children: content }));
    return createPortal(popoverContent, el);
};

export { Popover, PopoverContext, PopoverPortal };
