// @beeline/design-tokens/js/tokens/components/timeline/timeline.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const timelineStepThumbBorderRadius = "50%";
export const timelineStepPastTrackBackgroundColor = "var(--color-background-brand, #fdd835)";
export const timelineStepPastThumbBackgroundColor = "var(--color-background-brand, #fdd835)";
export const timelineStepPastThumbWidth = "14px";
export const timelineStepPastThumbHeight = "14px";
export const timelineStepCurrentThumbBackgroundColor = "var(--color-background-brand, #fdd835)";
export const timelineStepCurrentThumbWidth = "28px";
export const timelineStepCurrentThumbHeight = "28px";
export const timelineStepCurrentThumbIconColor = "var(--color-text-dark-active, rgba(0, 0, 0, 0.87))";
export const timelineStepCurrentTrackBeforeThumbBackgroundColor = "var(--color-background-brand, #fdd835)";
export const timelineStepCurrentTrackAfterThumbBackgroundColor = "var(--color-status-neutral-background, #f1f1f3)";
export const timelineStepCurrentErrorThumbBackgroundColor = "var(--color-status-error, #ff5555)";
export const timelineStepCurrentErrorThumbIconColor = "var(--color-text-light-active, rgba(255, 255, 255, 0.87))";
export const timelineStepCurrentErrorDescriptionColor = "var(--color-status-error, #ff5555)";
export const timelineStepFutureTrackBackgroundColor = "var(--color-status-neutral-background, #f1f1f3)";
export const timelineStepFutureThumbBackgroundColor = "var(--color-status-neutral-background, #f1f1f3)";
export const timelineStepFutureThumbWidth = "14px";
export const timelineStepFutureThumbHeight = "14px";
export const timelineVerticalStepContentPaddingBottom = "24px";
export const timelineVerticalStepIndicatorWidth = "28px";
export const timelineVerticalStepIndicatorMarginRight = "12px";
export const timelineVerticalStepTitleFontSize = "17px";
export const timelineVerticalStepTitleFontLineHeight = "22px";
export const timelineVerticalStepTitleFontLetterSpacing = "0.2px";
export const timelineVerticalStepTitleMarginBottom = "8px";
export const timelineVerticalStepDescriptionFontSize = "15px";
export const timelineVerticalStepDescriptionFontWeight = 400;
export const timelineVerticalStepDescriptionFontLineHeight = "18px";
export const timelineVerticalStepDescriptionFontLetterSpacing = "0.2px";
export const timelineVerticalStepCaptionFontSize = "13px";
export const timelineVerticalStepCaptionFontWeight = 400;
export const timelineVerticalStepCaptionFontLineHeight = "16px";
export const timelineVerticalStepCaptionFontLetterSpacing = "0.2px";
export const timelineVerticalStepCaptionSpacing = "8px";
export const timelineVerticalStepTrackWidth = "6px";
export const timelineVerticalStepThumbPositionTop = "2px";
export const timelineVerticalStepPastTitleColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const timelineVerticalStepPastTitleFontWeight = 400;
export const timelineVerticalStepPastDescriptionColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const timelineVerticalStepPastCaptionColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const timelineVerticalStepCurrentTitleColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const timelineVerticalStepCurrentTitleFontWeight = 500;
export const timelineVerticalStepCurrentDescriptionColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const timelineVerticalStepCurrentCaptionColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const timelineVerticalStepFutureTitleColor = "var(--color-text-disabled, rgba(25, 28, 52, 0.48))";
export const timelineVerticalStepFutureTitleFontWeight = 400;
export const timelineVerticalStepFutureDescriptionColor = "var(--color-text-disabled, rgba(25, 28, 52, 0.48))";
export const timelineVerticalStepFutureCaptionColor = "var(--color-text-disabled, rgba(25, 28, 52, 0.48))";
export const timelineVerticalRollBarTextFontSize = "17px";
export const timelineVerticalRollBarTextFontWeight = 400;
export const timelineVerticalRollBarTextFontLineHeight = "22px";
export const timelineVerticalRollBarTextFontLetterSpacing = "0.2px";
export const timelineVerticalRollBarTextColor = "var(--color-text-link, #1a73e8)";
export const timelineVerticalRollBarIconSize = "20px";
export const timelineVerticalRollBarIconColor = "var(--color-text-link, #1a73e8)";
export const timelineVerticalRollBarIconSpacing = "4px";
export const timelineHorizontalStepNumberMarginBottom = "4px";
export const timelineHorizontalStepNumberFontSize = "10px";
export const timelineHorizontalStepNumberFontWeight = 700;
export const timelineHorizontalStepNumberFontLineHeight = "16px";
export const timelineHorizontalStepNumberFontLetterSpacing = "1.2px";
export const timelineHorizontalStepNumberColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const timelineHorizontalStepTitleFontSize = "19px";
export const timelineHorizontalStepTitleFontWeight = 500;
export const timelineHorizontalStepTitleFontLineHeight = "24px";
export const timelineHorizontalStepTitleFontLetterSpacing = "0.2px";
export const timelineHorizontalStepTitleColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const timelineHorizontalStepTitleMarginBottom = "16px";
export const timelineHorizontalStepIndicatorHeight = "28px";
export const timelineHorizontalStepIndicatorMarginBottom = "8px";
export const timelineHorizontalStepTrackHeight = "6px";
export const timelineHorizontalStepCurrentErrorDescriptionFontSize = "17px";
export const timelineHorizontalStepCurrentErrorDescriptionFontWeight = 400;
export const timelineHorizontalStepCurrentErrorDescriptionFontLineHeight = "22px";
export const timelineHorizontalStepCurrentErrorDescriptionFontLetterSpacing = "0.2px";
export const timelineHorizontalStagesFontSize = "17px";
export const timelineHorizontalStagesFontWeight = 400;
export const timelineHorizontalStagesFontLineHeight = "22px";
export const timelineHorizontalStagesFontLetterSpacing = "0.2px";
export const timelineHorizontalStagesColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
