// @beeline/design-tokens/js/tokens/components/timeline/timeline.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	timelineStepThumbBorderRadius: "50%",
	timelineStepPastTrackBackgroundColor: "var(--color-background-brand, #fdd835)",
	timelineStepPastThumbBackgroundColor: "var(--color-background-brand, #fdd835)",
	timelineStepPastThumbWidth: "14px",
	timelineStepPastThumbHeight: "14px",
	timelineStepCurrentThumbBackgroundColor: "var(--color-background-brand, #fdd835)",
	timelineStepCurrentThumbWidth: "28px",
	timelineStepCurrentThumbHeight: "28px",
	timelineStepCurrentThumbIconColor: "var(--color-text-dark-active, rgba(0, 0, 0, 0.87))",
	timelineStepCurrentTrackBeforeThumbBackgroundColor: "var(--color-background-brand, #fdd835)",
	timelineStepCurrentTrackAfterThumbBackgroundColor: "var(--color-status-neutral-background, #f1f1f3)",
	timelineStepCurrentErrorThumbBackgroundColor: "var(--color-status-error, #ff5555)",
	timelineStepCurrentErrorThumbIconColor: "var(--color-text-light-active, rgba(255, 255, 255, 0.87))",
	timelineStepCurrentErrorDescriptionColor: "var(--color-status-error, #ff5555)",
	timelineStepFutureTrackBackgroundColor: "var(--color-status-neutral-background, #f1f1f3)",
	timelineStepFutureThumbBackgroundColor: "var(--color-status-neutral-background, #f1f1f3)",
	timelineStepFutureThumbWidth: "14px",
	timelineStepFutureThumbHeight: "14px",
	timelineVerticalStepContentPaddingBottom: "24px",
	timelineVerticalStepIndicatorWidth: "28px",
	timelineVerticalStepIndicatorMarginRight: "12px",
	timelineVerticalStepTitleFontSize: "17px",
	timelineVerticalStepTitleFontLineHeight: "22px",
	timelineVerticalStepTitleFontLetterSpacing: "0.2px",
	timelineVerticalStepTitleMarginBottom: "8px",
	timelineVerticalStepDescriptionFontSize: "15px",
	timelineVerticalStepDescriptionFontWeight: 400,
	timelineVerticalStepDescriptionFontLineHeight: "18px",
	timelineVerticalStepDescriptionFontLetterSpacing: "0.2px",
	timelineVerticalStepCaptionFontSize: "13px",
	timelineVerticalStepCaptionFontWeight: 400,
	timelineVerticalStepCaptionFontLineHeight: "16px",
	timelineVerticalStepCaptionFontLetterSpacing: "0.2px",
	timelineVerticalStepCaptionSpacing: "8px",
	timelineVerticalStepTrackWidth: "6px",
	timelineVerticalStepThumbPositionTop: "2px",
	timelineVerticalStepPastTitleColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	timelineVerticalStepPastTitleFontWeight: 400,
	timelineVerticalStepPastDescriptionColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	timelineVerticalStepPastCaptionColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	timelineVerticalStepCurrentTitleColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	timelineVerticalStepCurrentTitleFontWeight: 500,
	timelineVerticalStepCurrentDescriptionColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	timelineVerticalStepCurrentCaptionColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	timelineVerticalStepFutureTitleColor: "var(--color-text-disabled, rgba(25, 28, 52, 0.48))",
	timelineVerticalStepFutureTitleFontWeight: 400,
	timelineVerticalStepFutureDescriptionColor: "var(--color-text-disabled, rgba(25, 28, 52, 0.48))",
	timelineVerticalStepFutureCaptionColor: "var(--color-text-disabled, rgba(25, 28, 52, 0.48))",
	timelineVerticalRollBarTextFontSize: "17px",
	timelineVerticalRollBarTextFontWeight: 400,
	timelineVerticalRollBarTextFontLineHeight: "22px",
	timelineVerticalRollBarTextFontLetterSpacing: "0.2px",
	timelineVerticalRollBarTextColor: "var(--color-text-link, #1a73e8)",
	timelineVerticalRollBarIconSize: "20px",
	timelineVerticalRollBarIconColor: "var(--color-text-link, #1a73e8)",
	timelineVerticalRollBarIconSpacing: "4px",
	timelineHorizontalStepNumberMarginBottom: "4px",
	timelineHorizontalStepNumberFontSize: "10px",
	timelineHorizontalStepNumberFontWeight: 700,
	timelineHorizontalStepNumberFontLineHeight: "16px",
	timelineHorizontalStepNumberFontLetterSpacing: "1.2px",
	timelineHorizontalStepNumberColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	timelineHorizontalStepTitleFontSize: "19px",
	timelineHorizontalStepTitleFontWeight: 500,
	timelineHorizontalStepTitleFontLineHeight: "24px",
	timelineHorizontalStepTitleFontLetterSpacing: "0.2px",
	timelineHorizontalStepTitleColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	timelineHorizontalStepTitleMarginBottom: "16px",
	timelineHorizontalStepIndicatorHeight: "28px",
	timelineHorizontalStepIndicatorMarginBottom: "8px",
	timelineHorizontalStepTrackHeight: "6px",
	timelineHorizontalStepCurrentErrorDescriptionFontSize: "17px",
	timelineHorizontalStepCurrentErrorDescriptionFontWeight: 400,
	timelineHorizontalStepCurrentErrorDescriptionFontLineHeight: "22px",
	timelineHorizontalStepCurrentErrorDescriptionFontLetterSpacing: "0.2px",
	timelineHorizontalStagesFontSize: "17px",
	timelineHorizontalStagesFontWeight: 400,
	timelineHorizontalStagesFontLineHeight: "22px",
	timelineHorizontalStagesFontLetterSpacing: "0.2px",
	timelineHorizontalStagesColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
};
