// @beeline/design-system-react/build/components/DatePicker/Calendar/views/CalendarRange/CalendarRange.js
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import '../../Calendar2.js';
import { usePrevious } from '../../../hooks/use-previous.js';
import { TimePicker } from '../../../TimePicker/TimePicker.js';
import { Tab } from '../../../../Tabs/Tab/Tab.js';
import { Tabs } from '../../../../Tabs/Tabs.js';
import { classNames } from '../../../../../helpers/classnames.js';
import { PopupsContext } from '../../../../../helpers/popupsContext.js';
import { Icons } from '@beeline/design-tokens/js/iconfont/icons';
import { useContext, useState, useRef, useEffect } from 'react';
import { CalendarMobileRange } from '../../CalendarMobile/CalendarMobileRange/CalendarMobileRange.js';
import { CalendarNavigationIcon } from '../../CalendarNavigationIcon/CalendarNavigationIcon.js';
import { Footer } from '../../components/Footer/Footer.js';
import { PresetPeriods } from '../../components/PresetPeriods/PresetPeriods.js';
import { MONTHS_NAMES } from '../../constants.js';
import { useOnOutsideCalendarClick } from '../../hooks.js';
import { validateCustomViewDate, getViewDate, convertUtcToTz, getDateFromIsoString, isRangeMode, isTimeValid, convertTzToUtc } from '../../newUtils.js';
import { CalendarDayView } from '../CalendarDayView/CalendarDayView.js';
import { buildLocalFromClick, toUTCFromLocal, selectionReducer, toOutputString, updateRangeTimeOnPick } from './rangeHelpers.js';

const CalendarRange = props => {
    const { value, minValue = '', maxValue = '', disabledStartDate = false, disabledEndDate = false, className, datepickerRef, onSelectDate, dataTestId = 'CalendarRange', multiselect, saveButtonTitle, onClickSelectDates, mobileView, readonly, onSaveClick, onOutsideClick, excludeDate, holidaysDate, customViewDate, initialViewDate, enableTimePicker, outputDateFormat = enableTimePicker ? "yyyy-MM-dd'T'HH:mm:ss'Z'" : 'yyyy-MM-dd', outOfRangeErrorText = 'Недопустимое время', timeRange = ['00:00', '23:59'], customizeDay, presetPeriods, onPresetPeriodSelect, timeZone, showAdjacentMonthDays, } = props;
    const { rootElement } = useContext(PopupsContext);
    const [calendarElement, setCalendarElement] = useState(null);
    useOnOutsideCalendarClick({
        rootElement,
        calendarElement,
        datepickerElement: datepickerRef?.current,
        onOutsideClick: () => {
            if (typeof onOutsideClick === 'function') {
                onOutsideClick();
            }
        },
    });
    // Поддерживаем обратную совместимость с customViewDate
    const customViewDateValue = validateCustomViewDate(customViewDate);
    const initialViewDateValue = initialViewDate ? new Date(initialViewDate) : null;
    // initialViewDate используется только когда нет value или value пустой
    const hasValidValue = value && value.some(v => v && v.trim() !== '');
    const finalInitialViewDate = hasValidValue
        ? getViewDate(value)
        : initialViewDateValue || customViewDateValue || new Date();
    const [viewDateUTC, setViewDateUTC] = useState(finalInitialViewDate);
    const effectiveTimeZone = timeZone || Intl.DateTimeFormat().resolvedOptions().timeZone;
    const prevValue = usePrevious(value);
    const internalChangeRef = useRef(false);
    useEffect(() => {
        // Изменения пришли извне — обновляем viewDate
        if (value !== prevValue && !internalChangeRef.current) {
            const hasValidValueInEffect = value && value.length > 0 && value.some(v => v && v.trim() !== '');
            if (hasValidValueInEffect) {
                const updatedUTC = getViewDate(value);
                const currentMonthStart = enableTimePicker
                    ? convertUtcToTz(viewDateUTC, effectiveTimeZone)
                    : new Date(Date.UTC(viewDateUTC.getUTCFullYear(), viewDateUTC.getUTCMonth(), viewDateUTC.getUTCDate()));
                currentMonthStart.setDate(1);
                if (enableTimePicker)
                    currentMonthStart.setHours(0, 0, 0, 0);
                const updatedLocal = enableTimePicker
                    ? convertUtcToTz(updatedUTC, effectiveTimeZone)
                    : new Date(Date.UTC(updatedUTC.getUTCFullYear(), updatedUTC.getUTCMonth(), updatedUTC.getUTCDate()));
                updatedLocal.setDate(1);
                if (enableTimePicker)
                    updatedLocal.setHours(0, 0, 0, 0);
                const currentYear = currentMonthStart.getFullYear();
                const currentMonth = currentMonthStart.getMonth();
                const nextMonthDate = new Date(currentMonthStart);
                nextMonthDate.setMonth(currentMonth + 1);
                const nextYear = nextMonthDate.getFullYear();
                const nextMonth = nextMonthDate.getMonth();
                const newYear = updatedLocal.getFullYear();
                const newMonth = updatedLocal.getMonth();
                const isVisible = (newYear === currentYear && newMonth === currentMonth) || (newYear === nextYear && newMonth === nextMonth);
                if (!isVisible) {
                    setViewDateUTC(updatedUTC);
                }
            }
        }
    }, [enableTimePicker, effectiveTimeZone, value, prevValue, viewDateUTC]);
    // min/max => UTC
    const minDateUTC = getDateFromIsoString(minValue);
    const maxDateUTC = getDateFromIsoString(maxValue);
    // selectedDate => UTC (single или массив)
    const selectedDate = isRangeMode(value)
        ? value.map(element => getDateFromIsoString(element))
        : getDateFromIsoString(value);
    // Это зачем?
    const isStartSelected = !Number.isNaN(Date.parse(value[0]));
    const isEndSelected = !Number.isNaN(Date.parse(value[1]));
    const [startTimeError, setStartTimeError] = useState(false);
    const [endTimeError, setEndTimeError] = useState(false);
    const getTimeFromValue = (index) => {
        let date;
        if (isRangeMode(value)) {
            date = value[index] ? getDateFromIsoString(value[index]) : undefined;
        }
        else if (index === 0 && value) {
            date = getDateFromIsoString(value);
        }
        // Если нет выбранной даты, используем дефолтные значения времени
        if (!date || Number.isNaN(date.getTime())) {
            const now = new Date();
            const local = enableTimePicker ? convertUtcToTz(now, effectiveTimeZone) : now;
            return {
                hours: local.getHours().toString().padStart(2, '0'),
                minutes: local.getMinutes().toString().padStart(2, '0'),
            };
        }
        const local = enableTimePicker
            ? convertUtcToTz(date, effectiveTimeZone)
            : new Date(Date.UTC(date.getUTCFullYear(), date.getUTCMonth(), date.getUTCDate()));
        return {
            hours: local.getHours().toString().padStart(2, '0'),
            minutes: local.getMinutes().toString().padStart(2, '0'),
        };
    };
    const [startTime, setStartTime] = useState(getTimeFromValue(0));
    const [endTime, setEndTime] = useState(getTimeFromValue(1));
    // Синхронизация времени с изменениями value
    useEffect(() => {
        if (internalChangeRef.current) {
            // Пропускаем синхронизацию при внутренних изменениях и сбрасываем флаг
            internalChangeRef.current = false;
            return;
        }
        if (enableTimePicker && value) {
            const newStartTime = getTimeFromValue(0);
            const newEndTime = getTimeFromValue(1);
            setStartTime(newStartTime);
            setEndTime(newEndTime);
        }
    }, [value, enableTimePicker, effectiveTimeZone]);
    const [activeTabIndex, setActiveTabIndex] = useState(0);
    if (mobileView) {
        return jsx(CalendarMobileRange, { ...props });
    }
    // ------------------ Форматирование UTC -> строку -----------------
    const toFormattedString = (dateUTC) => {
        if (!dateUTC || Number.isNaN(dateUTC.getTime())) {
            console.error('Получена некорректная дата:', dateUTC);
            return '';
        }
        return toOutputString(dateUTC, outputDateFormat);
    };
    const handleSelectDay = (day, month, year) => {
        internalChangeRef.current = true;
        if ((disabledStartDate && disabledEndDate) || readonly) {
            return;
        }
        const [startDate, endDate] = isRangeMode(selectedDate)
            ? selectedDate
            : [undefined, undefined];
        // Decide which time to apply for local date construction
        const isStartedRange = Boolean(startDate);
        const isFilledRange = Boolean(endDate);
        const isSelectingStartDate = !isStartedRange || (isStartedRange && isFilledRange);
        const timeToUse = enableTimePicker ? (isSelectingStartDate ? startTime : endTime) : { hours: '0', minutes: '0' };
        const localDate = buildLocalFromClick(day, month, year, +timeToUse.hours, +timeToUse.minutes);
        const newUTC = toUTCFromLocal(localDate, enableTimePicker, effectiveTimeZone);
        const newISO = enableTimePicker ? newUTC.toISOString() : toFormattedString(newUTC);
        // Single value mode fallback
        if (!isRangeMode(selectedDate)) {
            onSelectDate?.(newISO);
            return;
        }
        if (multiselect) {
            const newDate = new Date(year, month, day, viewDateUTC.getHours(), viewDateUTC.getMinutes());
            const validSelectedDate = selectedDate?.filter(date => date !== null);
            const indexDate = validSelectedDate.findIndex(date => date.getTime() === newDate.getTime());
            let newMultipleDates;
            if (indexDate === -1) {
                newMultipleDates = [...validSelectedDate, newDate];
            }
            else {
                validSelectedDate.splice(indexDate, 1);
                newMultipleDates = [...validSelectedDate];
            }
            onSelectDate?.(newMultipleDates.map(element => toFormattedString(element)));
            return;
        }
        const nextRange = selectionReducer({ startUTC: startDate, endUTC: endDate }, {
            type: 'dayClick',
            payload: {
                localDate,
                newUTC,
                enableTimePicker,
                disabledStartDate,
                disabledEndDate,
                outputDateFormat,
                originalStartStr: isRangeMode(value) ? value[0] : '',
                pickerStartTime: startTime,
                effectiveTimeZone,
            },
        });
        onSelectDate?.(nextRange);
    };
    // ------------------ Переключения месяц назад/вперёд ------------------
    const setPrevMonth = () => {
        const loc = enableTimePicker
            ? convertUtcToTz(viewDateUTC, effectiveTimeZone)
            : new Date(Date.UTC(viewDateUTC.getUTCFullYear(), viewDateUTC.getUTCMonth(), viewDateUTC.getUTCDate()));
        loc.setDate(1);
        if (enableTimePicker)
            loc.setHours(0, 0, 0, 0);
        loc.setMonth(loc.getMonth() - 1);
        setViewDateUTC(enableTimePicker ? convertTzToUtc(loc, effectiveTimeZone) : loc);
    };
    const setNextMonth = () => {
        const loc = enableTimePicker
            ? convertUtcToTz(viewDateUTC, effectiveTimeZone)
            : new Date(Date.UTC(viewDateUTC.getUTCFullYear(), viewDateUTC.getUTCMonth(), viewDateUTC.getUTCDate()));
        loc.setDate(1);
        if (enableTimePicker)
            loc.setHours(0, 0, 0, 0);
        loc.setMonth(loc.getMonth() + 1);
        setViewDateUTC(enableTimePicker ? convertTzToUtc(loc, effectiveTimeZone) : loc);
    };
    // ------------------ Обработка компонента TimePicker ------------------
    const handleTimePick = (val, index, setError, setTime) => {
        internalChangeRef.current = true;
        setTime(val);
        const invalid = isTimeValid(val, timeRange);
        setError(invalid);
        if (!enableTimePicker)
            return;
        const { newValue, newViewDateUTC } = updateRangeTimeOnPick({
            val,
            index,
            value: value,
            timeRange,
            enableTimePicker,
            effectiveTimeZone,
        });
        if (!invalid && newValue) {
            onSelectDate?.(newValue);
        }
    };
    const handleTimePickStart = val => handleTimePick(val, 0, setStartTimeError, setStartTime);
    const handleTimePickEnd = val => handleTimePick(val, 1, setEndTimeError, setEndTime);
    // ------------------ Рендер заголовка (локальные год/месяц) ------------------
    const localView = enableTimePicker
        ? convertUtcToTz(viewDateUTC, effectiveTimeZone)
        : new Date(Date.UTC(viewDateUTC.getUTCFullYear(), viewDateUTC.getUTCMonth(), viewDateUTC.getUTCDate()));
    localView.setDate(1);
    if (enableTimePicker)
        localView.setHours(0, 0, 0, 0);
    const localMonth = localView.getMonth();
    const localNextMonth = new Date(localView);
    localNextMonth.setMonth(localView.getMonth() + 1);
    const hasFooter = (onSaveClick || onClickSelectDates) && !readonly;
    const calendarContent = (jsxs(Fragment, { children: [jsxs("div", { className: classNames('dsb_calendar-panels', { multiselect: multiselect && !readonly }), children: [jsxs("div", { className: "dsb_calendar-panel", children: [jsxs("header", { className: "dsb_calendar-header", children: [jsx("span", { className: "dsb_calendar-navigation", children: jsx(CalendarNavigationIcon, { direction: Icons.NavArrowLeft, onClick: setPrevMonth, 
                                            // disabled={isDisabledMonthSwitch(1)}
                                            dataTestId: `${dataTestId}-prev-month` }) }), jsxs("p", { className: 'dsb_calendar-title', children: [jsx("span", { className: 'dsb_calendar-select-month', "data-test-id": `${dataTestId}-select-month`, children: MONTHS_NAMES[localMonth] }), jsx("span", { className: 'dsb_calendar-select-year', "data-test-id": `${dataTestId}-select-year`, children: localView.getFullYear() })] })] }), jsx(CalendarDayView, { minDate: minDateUTC, maxDate: maxDateUTC, viewDate: localView, disabledStartDate: disabledStartDate, disabledEndDate: disabledEndDate, selectedDate: selectedDate, onSelectDay: handleSelectDay, multiselect: multiselect, readonly: readonly, excludeDate: excludeDate, holidaysDate: holidaysDate, customizeDay: customizeDay, timeZone: effectiveTimeZone, showAdjacentMonthDays: showAdjacentMonthDays })] }), jsx("div", { className: "dsb_calendar-divider" }), jsxs("div", { className: "dsb_calendar-panel", children: [jsxs("header", { className: 'dsb_calendar-header dsb_calendar-header__right', children: [jsxs("p", { className: 'dsb_calendar-title', children: [jsx("span", { className: 'dsb_calendar-select-month', "data-test-id": `${dataTestId}-select-month`, children: MONTHS_NAMES[localNextMonth.getMonth()] }), jsx("span", { className: 'dsb_calendar-select-year', "data-test-id": `${dataTestId}-select-year`, children: localNextMonth.getFullYear() })] }), jsx("span", { className: "dsb_calendar-navigation", children: jsx(CalendarNavigationIcon, { direction: Icons.NavArrowRight, onClick: setNextMonth, 
                                            // disabled={isDisabledMonthSwitch(-1)} временное решение
                                            dataTestId: `${dataTestId}-next-month` }) })] }), jsx(CalendarDayView, { minDate: minDateUTC, maxDate: maxDateUTC, disabledEndDate: disabledEndDate, disabledStartDate: disabledStartDate, viewDate: localNextMonth, selectedDate: selectedDate, onSelectDay: handleSelectDay, multiselect: multiselect, readonly: readonly, excludeDate: excludeDate, holidaysDate: holidaysDate, customizeDay: customizeDay, timeZone: effectiveTimeZone, showAdjacentMonthDays: showAdjacentMonthDays })] })] }), enableTimePicker && (jsxs("div", { className: "dsb_calendar_timepickers", children: [jsx(TimePicker, { value: startTime, disabled: !isStartSelected, error: startTimeError, errorMessage: startTimeError && outOfRangeErrorText, onChange: handleTimePickStart, onBlur: val => {
                            setStartTimeError(isTimeValid(val, timeRange));
                        }, dataTestId: 'TimePicker1' }), jsx(TimePicker, { value: endTime, disabled: !isEndSelected, error: endTimeError, errorMessage: endTimeError && outOfRangeErrorText, onChange: handleTimePickEnd, onBlur: val => {
                            setEndTimeError(isTimeValid(val, timeRange));
                        }, dataTestId: 'TimePicker2' })] }))] }));
    const tabs = [
        {
            label: 'Интервал дат',
            content: calendarContent,
        },
        ...(presetPeriods
            ? [
                {
                    label: 'Предустановленный',
                    content: (jsx(PresetPeriods, { periods: presetPeriods, onSelect: range => {
                            onPresetPeriodSelect?.(range);
                            setActiveTabIndex(0);
                        }, dataTestId: `${dataTestId}-preset-periods` })),
                },
            ]
            : []),
    ];
    return (jsxs("div", { ref: setCalendarElement, className: classNames('dsb_calendar dsb_calendar-range', className), "data-testid": dataTestId, children: [presetPeriods?.length ? (jsx(Tabs, { selectedTabIndex: activeTabIndex, onChange: setActiveTabIndex, children: tabs.map(tab => (jsx(Tab, { label: tab.label, children: tab.content }, tab.label))) })) : (calendarContent), hasFooter && (jsx(Footer, { onSaveClick: onSaveClick, onClickSelectDates: onClickSelectDates, saveButtonTitle: saveButtonTitle }))] }));
};

export { CalendarRange };
