// @beeline/design-system-react/build/components/DatePicker/Calendar/views/CalendarRange/rangeHelpers.js
import { d as dateFnsTz } from '../../../../../node_modules/date-fns-tz/index.js';
import { formatToISO, convertUtcToTz, convertTzToUtc } from '../../newUtils.js';

/**
 * Предикаты для явной и читабельной ветвизации логики.
 */
const isValidDate = (d) => !!d && !Number.isNaN(d.getTime());
const hasStartedRange = (start) => isValidDate(start);
const hasFilledRange = (end) => isValidDate(end);
/**
 * Строит локальную дату из координат клика в календаре и опциональных часов/минут.
 */
const buildLocalFromClick = (day, month, year, hours, minutes) => {
    const h = typeof hours === 'number' ? hours : 0;
    const m = typeof minutes === 'number' ? minutes : 0;
    return new Date(year, month, day, h, m, 0, 0);
};
/**
 * Конвертирует локальную дату в представление UTC с учётом режима таймпикера.
 */
const toUTCFromLocal = (localDate, enableTimePicker, effectiveTimeZone) => {
    if (!enableTimePicker) {
        return localDate;
    }
    if (effectiveTimeZone) {
        const year = localDate.getFullYear();
        const month = localDate.getMonth() + 1;
        const day = localDate.getDate();
        const hours = localDate.getHours();
        const minutes = localDate.getMinutes();
        const seconds = localDate.getSeconds();
        const dateString = `${year}-${String(month).padStart(2, '0')}-${String(day).padStart(2, '0')} ${String(hours).padStart(2, '0')}:${String(minutes).padStart(2, '0')}:${String(seconds).padStart(2, '0')}`;
        return dateFnsTz.zonedTimeToUtc(dateString, effectiveTimeZone);
    }
    // Сохраняем текущую логику: собираем UTC напрямую из локальных компонентов даты/времени.
    return new Date(Date.UTC(localDate.getFullYear(), localDate.getMonth(), localDate.getDate(), localDate.getHours(), localDate.getMinutes(), 0, 0));
};
/**
 * Форматирует дату UTC в выходную строку по заданному формату.
 */
const toOutputString = (dateUTC, outputDateFormat) => {
    return formatToISO(dateUTC, outputDateFormat);
};
/**
 * Вычисляет следующий диапазон при клике по дню.
 * Возвращает кортеж строк [start, end]. Пустой конец представлен как ''.
 */
const computeNextRangeOnDayClick = (args) => {
    const { startUTC, endUTC, localDate, newUTC, enableTimePicker, disabledStartDate, disabledEndDate, outputDateFormat, originalStartStr, pickerStartTime, effectiveTimeZone, } = args;
    const started = hasStartedRange(startUTC);
    const filled = hasFilledRange(endUTC);
    const startStr = started && startUTC ? toOutputString(startUTC, outputDateFormat) : '';
    const endStr = filled && endUTC ? toOutputString(endUTC, outputDateFormat) : '';
    const newStr = enableTimePicker ? newUTC.toISOString() : toOutputString(newUTC, outputDateFormat);
    // Заблокирован только старт: разрешаем менять только конец
    if (disabledStartDate && !disabledEndDate) {
        if (started && startUTC && localDate.getTime() <= startUTC.getTime()) {
            return [startStr, startStr];
        }
        return [startStr, newStr];
    }
    // Заблокирован только конец: разрешаем менять только старт
    if (!disabledStartDate && disabledEndDate) {
        if (filled && endUTC && localDate.getTime() >= endUTC.getTime()) {
            return [endStr, endStr];
        }
        return [newStr, endStr];
    }
    // Первый выбор — устанавливаем начало, очищаем конец
    if (!started) {
        return [newStr, ''];
    }
    // Второй выбор — разрешаем порядок и равенство
    if (!filled) {
        if (!enableTimePicker) {
            // Сравниваем календарный день по выходной строке, чтобы избежать рассинхронизации TZ
            const startOut = toOutputString(startUTC, outputDateFormat);
            const newOut = toOutputString(newUTC, outputDateFormat);
            if (startOut === newOut) {
                return [newStr, newStr];
            }
            // Запасной порядок по меткам полуночи UTC
            const startMidnightUTC = new Date(Date.UTC(startUTC.getUTCFullYear(), startUTC.getUTCMonth(), startUTC.getUTCDate(), 0, 0, 0, 0));
            const newMidnightUTC = new Date(Date.UTC(newUTC.getUTCFullYear(), newUTC.getUTCMonth(), newUTC.getUTCDate(), 0, 0, 0, 0));
            if (newMidnightUTC.getTime() < startMidnightUTC.getTime()) {
                return [newStr, ''];
            }
            return [startStr, newStr];
        }
        // Включён таймпикер — сравниваем полные метки времени
        const startMs = startUTC.getTime();
        const newMs = newUTC.getTime();
        if (newMs === startMs) {
            return [newStr, newStr];
        }
        if (newMs < startMs) {
            if (pickerStartTime && effectiveTimeZone) {
                const correctedLocal = buildLocalFromClick(localDate.getDate(), localDate.getMonth(), localDate.getFullYear(), +pickerStartTime.hours, +pickerStartTime.minutes);
                const correctedUTC = toUTCFromLocal(correctedLocal, enableTimePicker, effectiveTimeZone);
                const correctedStr = enableTimePicker
                    ? correctedUTC.toISOString()
                    : toOutputString(correctedUTC, outputDateFormat);
                return [correctedStr, ''];
            }
            return [newStr, ''];
        }
        return [originalStartStr || startStr, newStr];
    }
    // Выбор нового старта — сбрасываем конец
    return [newStr, ''];
};
/**
 * Минимальный редьюсер выбора, делающий логику диапазона неизменяемой и явной.
 */
const selectionReducer = (state, action) => {
    switch (action.type) {
        case 'dayClick': {
            const { localDate, newUTC, enableTimePicker, disabledStartDate, disabledEndDate, outputDateFormat, originalStartStr, pickerStartTime, effectiveTimeZone, } = action.payload;
            return computeNextRangeOnDayClick({
                startUTC: state.startUTC,
                endUTC: state.endUTC,
                localDate,
                newUTC,
                enableTimePicker,
                disabledStartDate,
                disabledEndDate,
                outputDateFormat,
                originalStartStr,
                pickerStartTime,
                effectiveTimeZone,
            });
        }
        default:
            return ['', ''];
    }
};
/**
 * Обновляет время диапазона для границы (старт/конец) в режиме времени.
 * Возвращает новые ISO-строки и обновлённый viewDate при необходимости.
 */
const updateRangeTimeOnPick = (args) => {
    const { val, index, value, timeRange, enableTimePicker, effectiveTimeZone } = args;
    const invalid = false; // defer to component's isTimeValid to avoid duplicate logic here
    if (!enableTimePicker) {
        return { invalid };
    }
    const current = value[index];
    const utcDate = current ? new Date(current) : null;
    if (!utcDate || Number.isNaN(utcDate.getTime())) {
        return { invalid: true };
    }
    const loc = convertUtcToTz(utcDate, effectiveTimeZone);
    loc.setHours(+val.hours, +val.minutes, 0, 0);
    const newUTC = convertTzToUtc(loc, effectiveTimeZone);
    const next = [...value];
    next[index] = newUTC.toISOString();
    return { invalid, newValue: next, newViewDateUTC: newUTC };
};

export { buildLocalFromClick, computeNextRangeOnDayClick, hasFilledRange, hasStartedRange, isValidDate, selectionReducer, toOutputString, toUTCFromLocal, updateRangeTimeOnPick };
