// @beeline/design-tokens/js/tokens/components/navigationRail/navigationRail.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	navigationRailWidth: "56px",
	navigationRailPaddingVertical: "20px",
	navigationRailPaddingHorizontal: "8px", // Отступ от края контейнера до внутренних элементов, включает в себя 1px ширины бордера слева.
	navigationRailBackgroundColor: "var(--color-background-base, #ffffff)",
	navigationRailBorderLeftWidth: "1px",
	navigationRailBorderLeftColor: "var(--color-divider, rgba(25, 28, 52, 0.12))",
	navigationRailGap: "12px", // Вертикальный отступ между дочерними элементами меню, а также дивайдером и элементами меню.
	navigationRailItemHeight: "40px",
	navigationRailItemWidth: "40px",
	navigationRailItemPadding: "8px",
	navigationRailItemBorderRadius: "12px",
	navigationRailItemIconSize: "24px",
	navigationRailItemIconColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	navigationRailItemHoverBackgroundColor: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	navigationRailItemFocusedBorderWidth: "1px",
	navigationRailItemFocusedBorderColor: "var(--color-border-focus, #212121)",
	navigationRailItemPressedBackgroundColor: "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))",
	navigationRailItemActivatedIconColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	navigationRailItemActivatedBackgroundColor: "var(--color-background-base-activated, rgba(0, 0, 0, 0.12))",
	navigationRailItemDisabledOpacity: 0.48,
	navigationRailDividerContainerPadding: "8px",
};
