// @beeline/design-tokens/js/tokens/components/navigationRail/navigationRail.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const navigationRailWidth = "56px";
export const navigationRailPaddingVertical = "20px";
export const navigationRailPaddingHorizontal = "8px"; // Отступ от края контейнера до внутренних элементов, включает в себя 1px ширины бордера слева.
export const navigationRailBackgroundColor = "var(--color-background-base, #ffffff)";
export const navigationRailBorderLeftWidth = "1px";
export const navigationRailBorderLeftColor = "var(--color-divider, rgba(25, 28, 52, 0.12))";
export const navigationRailGap = "12px"; // Вертикальный отступ между дочерними элементами меню, а также дивайдером и элементами меню.
export const navigationRailItemHeight = "40px";
export const navigationRailItemWidth = "40px";
export const navigationRailItemPadding = "8px";
export const navigationRailItemBorderRadius = "12px";
export const navigationRailItemIconSize = "24px";
export const navigationRailItemIconColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const navigationRailItemHoverBackgroundColor = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const navigationRailItemFocusedBorderWidth = "1px";
export const navigationRailItemFocusedBorderColor = "var(--color-border-focus, #212121)";
export const navigationRailItemPressedBackgroundColor = "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))";
export const navigationRailItemActivatedIconColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const navigationRailItemActivatedBackgroundColor = "var(--color-background-base-activated, rgba(0, 0, 0, 0.12))";
export const navigationRailItemDisabledOpacity = 0.48;
export const navigationRailDividerContainerPadding = "8px";
