// @beeline/design-tokens/js/tokens/components/toolbar/toolbar.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const toolbarHeight = "56px";
export const toolbarPadding = "8px";
export const toolbarItemHeight = "40px";
export const toolbarItemPaddingHorizontal = "8px";
export const toolbarItemBorderRadius = "12px";
export const toolbarItemSpacing = "8px"; // Отступы внутри элемента тулбара: между иконкой, текстом и иконкой dropdown.
export const toolbarItemTextColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const toolbarItemTextFontSize = "15px";
export const toolbarItemTextFontWeight = 400;
export const toolbarItemTextFontLineHeight = "18px";
export const toolbarItemTextFontLetterSpacing = "0.2px";
export const toolbarItemIconSize = "24px";
export const toolbarItemDisabledOpacity = 0.48;
export const toolbarItemHoverBackgroundColor = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const toolbarItemActivatedBackgroundColor = "var(--color-background-base-activated, rgba(0, 0, 0, 0.12))";
export const toolbarDividerContainerHeight = "40px";
export const toolbarDividerContainerPaddingHorizontal = "8px";
export const toolbarCheckboxContainerPadding = "8px";
export const toolbarSettingsMarginLeft = "24px";
export const toolbarOnlyIconGap = "8px"; // Отступ между элементами тулбара в случае если используется только иконочный вид элементов.
