// @beeline/design-tokens/js/tokens/components/toolbar/toolbar.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	toolbarHeight: "56px",
	toolbarPadding: "8px",
	toolbarItemHeight: "40px",
	toolbarItemPaddingHorizontal: "8px",
	toolbarItemBorderRadius: "12px",
	toolbarItemSpacing: "8px", // Отступы внутри элемента тулбара: между иконкой, текстом и иконкой dropdown.
	toolbarItemTextColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	toolbarItemTextFontSize: "15px",
	toolbarItemTextFontWeight: 400,
	toolbarItemTextFontLineHeight: "18px",
	toolbarItemTextFontLetterSpacing: "0.2px",
	toolbarItemIconSize: "24px",
	toolbarItemDisabledOpacity: 0.48,
	toolbarItemHoverBackgroundColor: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	toolbarItemActivatedBackgroundColor: "var(--color-background-base-activated, rgba(0, 0, 0, 0.12))",
	toolbarDividerContainerHeight: "40px",
	toolbarDividerContainerPaddingHorizontal: "8px",
	toolbarCheckboxContainerPadding: "8px",
	toolbarSettingsMarginLeft: "24px",
	toolbarOnlyIconGap: "8px", // Отступ между элементами тулбара в случае если используется только иконочный вид элементов.
};
