// @beeline/design-tokens/js/tokens/components/floatingNav/floatingNav.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const floatingNavItemPaddingRight = "24px";
export const floatingNavItemPaddingTop = "8px";
export const floatingNavItemPaddingBottom = "8px";
export const floatingNavItemParentPaddingLeft = "20px";
export const floatingNavItemChildPaddingLeft = "40px";
export const floatingNavItemColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const floatingNavItemFontSize = "17px";
export const floatingNavItemFontWeight = 400;
export const floatingNavItemFontLineHeight = "22px";
export const floatingNavItemFontLetterSpacing = "0.2px";
export const floatingNavItemSelectedColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const floatingNavItemSelectedFontSize = "17px";
export const floatingNavItemSelectedFontWeight = 500;
export const floatingNavItemSelectedFontLineHeight = "22px";
export const floatingNavItemSelectedFontLetterSpacing = "0.2px";
export const floatingNavItemHoverColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const floatingNavItemFocusedBorderRadiusRight = "12px";
export const floatingNavItemFocusedBorderRadiusLeft = "0px";
export const floatingNavItemFocusedBorderWidth = "1px";
export const floatingNavItemFocusedBorderColor = "var(--color-border-focused, #202123)";
export const floatingNavIndicatorWidth = "4px";
export const floatingNavIndicatorBackgroundColor = "var(--color-background-brand, #fdd835)";
export const floatingNavIndicatorBorderRadiusRight = "4px";
export const floatingNavTrackColor = "var(--color-divider, rgba(25, 28, 52, 0.12))";
export const floatingNavTrackWidth = "1px";
