// @beeline/design-tokens/js/tokens/components/floatingNav/floatingNav.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	floatingNavItemPaddingRight: "24px",
	floatingNavItemPaddingTop: "8px",
	floatingNavItemPaddingBottom: "8px",
	floatingNavItemParentPaddingLeft: "20px",
	floatingNavItemChildPaddingLeft: "40px",
	floatingNavItemColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	floatingNavItemFontSize: "17px",
	floatingNavItemFontWeight: 400,
	floatingNavItemFontLineHeight: "22px",
	floatingNavItemFontLetterSpacing: "0.2px",
	floatingNavItemSelectedColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	floatingNavItemSelectedFontSize: "17px",
	floatingNavItemSelectedFontWeight: 500,
	floatingNavItemSelectedFontLineHeight: "22px",
	floatingNavItemSelectedFontLetterSpacing: "0.2px",
	floatingNavItemHoverColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	floatingNavItemFocusedBorderRadiusRight: "12px",
	floatingNavItemFocusedBorderRadiusLeft: "0px",
	floatingNavItemFocusedBorderWidth: "1px",
	floatingNavItemFocusedBorderColor: "var(--color-border-focused, #202123)",
	floatingNavIndicatorWidth: "4px",
	floatingNavIndicatorBackgroundColor: "var(--color-background-brand, #fdd835)",
	floatingNavIndicatorBorderRadiusRight: "4px",
	floatingNavTrackColor: "var(--color-divider, rgba(25, 28, 52, 0.12))",
	floatingNavTrackWidth: "1px",
};
