// @beeline/design-system-react/extracted-styles/esm/components/Tree/TreeContext.js
import { jsx } from 'react/jsx-runtime';
import { createContext, useContext, useState, useEffect } from 'react';
import { CheckboxStatus } from './Tree.types.js';

const TreeContext = createContext(null);
const useTreeContext = () => {
    const context = useContext(TreeContext);
    if (!context) {
        throw new Error('useTreeContext должен быть обернут в TreeProvider');
    }
    return context;
};
const TreeProvider = ({ children, callback }) => {
    const [selectedId, setSelectedId] = useState('');
    const [nodes, setNodes] = useState({});
    const [treeParams, setTreeParams] = useState({
        multiselect: false,
        amount: false,
        size: 'medium',
        expandOnArrow: true,
    });
    const _getSelectedNodes = () => {
        // возвращаем выбранные ноды для мультиселект режима
        const selectedNodes = Object.keys(nodes).filter(id => {
            return nodes[id].checked === CheckboxStatus.CHECKED;
        });
        return selectedNodes;
    };
    const toggleSingleSelect = (nodeId) => {
        _toggleSignleSelect(nodeId);
        callback([nodeId]);
    };
    const _toggleSignleSelect = (id) => {
        // выбор для синглселекта
        setSelectedId(id);
    };
    const changeTreeParams = (newParams) => {
        // обновляем параметры дерева, используется только в TreeNodesWrapper
        setTreeParams(newParams);
    };
    const registerNode = (node) => {
        // регистрируем ноду
        const needToToggle = node.checked === CheckboxStatus.CHECKED;
        node.checked = CheckboxStatus.UNCHECKED;
        setNodes(prevNodes => {
            const updatedNodes = { ...prevNodes };
            const parentIds = Object.keys(updatedNodes).filter(key => updatedNodes[key].children.includes(node.id)); // находим ID родителя проходясь по всем ключам
            const parentChecked = parentIds.some(parentId => updatedNodes[parentId].checked === CheckboxStatus.CHECKED); // checked статус родителя
            const checkedState = parentChecked && !node.disabled ? CheckboxStatus.CHECKED : node.checked; // состояние нашего элемента наследуется от родителя
            updatedNodes[node.id] = { ...node, checked: checkedState };
            _getParent(updatedNodes, node.id);
            return updatedNodes;
        });
        if (needToToggle) {
            _toggleCheckbox(node.id);
            _toggleSignleSelect(node.id);
        }
    };
    const deleteNode = nodeId => {
        setNodes(prevNodes => {
            const updatedNodes = { ...prevNodes };
            Object.keys(updatedNodes).map(node => {
                if (updatedNodes[node].children.includes(nodeId)) {
                    updatedNodes[node].children = updatedNodes[node].children.filter(child => child !== nodeId);
                    if (!updatedNodes[node]?.children) {
                        updatedNodes[node].checked = CheckboxStatus.UNCHECKED;
                    }
                }
            });
            delete updatedNodes[nodeId];
            return updatedNodes;
        });
    };
    const toggleCheckbox = (nodeId) => new Promise(resolve => {
        _toggleCheckbox(nodeId);
        resolve();
    }).then(() => {
        callback(_getSelectedNodes());
    });
    const _toggleCheckbox = (nodeId) => {
        // логика для мультиселекта
        setNodes(prevNodes => {
            const newNodeState = { ...prevNodes };
            const toggleChilds = (id, checked) => {
                // рекурсивно обновляем состояния у дочерних элементов
                const node = newNodeState[id];
                if (!node) {
                    return;
                }
                if (!node.disabled) {
                    node.checked = checked;
                }
                for (const childId of node.children) {
                    toggleChilds(childId, checked);
                }
            };
            const currentNode = newNodeState[nodeId];
            const newCheckedState = currentNode.checked === CheckboxStatus.UNCHECKED ? CheckboxStatus.CHECKED : CheckboxStatus.UNCHECKED; // состояние самого элемента
            toggleChilds(nodeId, newCheckedState); // обновляем у дочерних
            const updateParents = (childId) => {
                // рекурсивно обновляем у родителей
                const parents = Object.keys(newNodeState).filter(key => newNodeState[key].children.includes(childId));
                for (const parentId of parents) {
                    const parent = newNodeState[parentId];
                    const childStates = parent.children.map(id => newNodeState[id].checked);
                    if (childStates.every(state => state === CheckboxStatus.CHECKED)) {
                        parent.checked = CheckboxStatus.CHECKED;
                    }
                    else if (childStates.some(state => state === CheckboxStatus.CHECKED || state === CheckboxStatus.PARTIALLY)) {
                        parent.checked = CheckboxStatus.PARTIALLY;
                    }
                    else {
                        parent.checked = CheckboxStatus.UNCHECKED;
                    }
                    updateParents(parentId);
                }
            };
            updateParents(newNodeState[nodeId].children.length > 0 ? newNodeState[nodeId].children[0] : nodeId);
            return newNodeState;
        });
    };
    const toggleExpansion = (nodeId, event) => {
        // просто обновляем expanded для нашего элемента
        setNodes(prevNodes => ({
            ...prevNodes,
            [nodeId]: {
                ...prevNodes[nodeId],
                expanded: !prevNodes[nodeId].expanded,
            },
        }));
        // Вызываем callback если он существует
        if (event && treeParams.onExpandCallback) {
            treeParams.onExpandCallback(event, nodeId);
        }
    };
    const recursiveExpansion = (nodeId) => {
        // рекурсивно раскрываем все родительские ноды
        setNodes(prevNodes => {
            // debugger;
            const newNodeState = { ...prevNodes };
            newNodeState[nodeId].expanded = true;
            const updateParents = (childId) => {
                const parents = Object.keys(newNodeState).filter(key => newNodeState[key].children.includes(childId));
                for (const parentId of parents) {
                    newNodeState[parentId].expanded = true;
                    updateParents(parentId);
                }
            };
            updateParents(nodeId);
            // debugger;
            return newNodeState;
        });
    };
    const _getParent = (nodes, nodeId) => {
        return Object.keys(nodes).find(node => nodes[node]?.children.includes(nodeId));
    };
    const updateNodes = () => {
        setNodes(prevNodes => {
            const newNodeState = { ...prevNodes };
            const leafs = Object.keys(prevNodes).filter(key => prevNodes[key].children.length === 0); // ноды без детей
            const updateParentsState = nodeId => {
                const parentId = _getParent(newNodeState, nodeId);
                if (!parentId)
                    return;
                if (newNodeState[parentId].children.every(child => newNodeState[child].checked === CheckboxStatus.CHECKED)) {
                    newNodeState[parentId].checked = CheckboxStatus.CHECKED;
                }
                else if (newNodeState[parentId].children.some(child => newNodeState[child].checked != CheckboxStatus.UNCHECKED)) {
                    newNodeState[parentId].checked = CheckboxStatus.PARTIALLY;
                }
                else {
                    newNodeState[parentId].checked = CheckboxStatus.UNCHECKED;
                }
                updateParentsState(parentId);
            };
            for (const leaf of leafs) {
                updateParentsState(leaf);
            }
            // debugger;
            return newNodeState;
        });
    };
    useEffect(() => {
        updateNodes();
    }, [children]);
    return (jsx(TreeContext.Provider, { value: {
            selectedId,
            toggleSingleSelect,
            nodes,
            registerNode,
            toggleExpansion,
            toggleCheckbox,
            treeParams,
            changeTreeParams,
            recursiveExpansion,
            deleteNode,
        }, children: children }));
};

export { TreeProvider, useTreeContext };
