// @beeline/design-system-react/build/helpers/generate-metadata.js
import { validateSEOProps } from '../components/SEOHead/utils.js';

/**
 * Генерирует объект метаданных для Next.js App Router Metadata API
 *
 * Использует тот же базовый интерфейс SEOHeadBaseProps для унификации API
 *
 * @param props - SEO метаданные (совместимы с SEOHead компонентом)
 * @returns Объект Metadata для Next.js App Router
 *
 * @example
 * ```tsx
 * // app/page.tsx или app/layout.tsx
 * import { generateMetadata } from '@beeline/design-system-react/helpers/generate-metadata';
 *
 * export const metadata = generateMetadata({
 *   title: 'Заголовок страницы',
 *   description: 'Описание страницы',
 *   canonicalUrl: 'https://example.com/page',
 * });
 * ```
 */
function generateMetadata(props) {
    const { title, description, canonicalUrl, ogTitle, ogDescription, ogType = 'website', ogUrl, ogImage, twitterCard = 'summary_large_image', robots = 'index, follow', } = props;
    // Валидация в development режиме
    if (process.env.NODE_ENV !== 'production') {
        validateSEOProps({ title, description });
    }
    // Формируем robots директиву
    const robotsContent = Array.isArray(robots) ? robots.join(', ') : robots;
    // Используем fallback значения для OG тегов
    const finalOgTitle = ogTitle || title;
    const finalOgDescription = ogDescription || description;
    const finalOgUrl = ogUrl || canonicalUrl;
    // Формируем объект Metadata для Next.js
    const metadata = {
        title,
        description,
        alternates: {
            canonical: canonicalUrl,
        },
        robots: robotsContent,
        openGraph: {
            title: finalOgTitle,
            description: finalOgDescription,
            type: ogType,
            url: finalOgUrl,
            ...(ogImage && { images: [{ url: ogImage }] }),
        },
        twitter: {
            card: twitterCard,
        },
    };
    return metadata;
}

export { generateMetadata };
