// @beeline/design-tokens/js/tokens/components/button/button.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const buttonBorderRadius = "12px";
export const buttonBorderColorFocus = "var(--color-border-focused, #202123)";
export const buttonOpacityDisabled = 0.48;
export const buttonIconMarginText = "8px";
export const buttonContainedBackgroundColor = "var(--color-background-brand, #fdd835)";
export const buttonContainedBackgroundColorHover = "var(--color-background-brand-hover, #fdc435)";
export const buttonContainedBackgroundColorPressed = "var(--color-background-brand-pressed, #fdb435)";
export const buttonContainedLabelColor = "var(--color-text-black-active, rgba(9, 11, 22, 0.94))";
export const buttonContainedTextColor = "var(--color-text-black-active, rgba(9, 11, 22, 0.94))";
export const buttonContainedIconColor = "var(--color-text-black-active, rgba(9, 11, 22, 0.94))";
export const buttonContainedHoverBackgroundColor = "var(--color-background-brand-hover, #fdc435)";
export const buttonContainedPressedBackgroundColor = "var(--color-background-brand-pressed, #fdb435)";
export const buttonOutlineBorderWidth = "1px";
export const buttonOutlineBorderStyle = "solid";
export const buttonOutlineBorderColor = "var(--color-border, rgba(25, 28, 52, 0.18))";
export const buttonOutlineBackgroundColor = "transparent";
export const buttonOutlineBackgroundColorHover = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const buttonOutlineBackgroundColorPressed = "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))";
export const buttonOutlineLabelColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const buttonOutlineTextColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const buttonOutlineIconColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const buttonOutlineHoverBackgroundColor = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const buttonOutlinePressedBackgroundColor = "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))";
export const buttonOverlayBackgroundColor = "var(--color-button-overlay-background, #141622)";
export const buttonOverlayBackgroundColorHover = "var(--color-button-overlay-background-hover, #1f212d)";
export const buttonOverlayBackgroundColorPressed = "var(--color-button-overlay-background-pressed, #2a2c39)";
export const buttonOverlayLabelColor = "var(--color-text-active-inverse, #ffffff)";
export const buttonOverlayTextColor = "var(--color-text-active-inverse, #ffffff)";
export const buttonOverlayIconColor = "var(--color-text-active-inverse, #ffffff)";
export const buttonOverlayHoverBackgroundColor = "var(--color-button-overlay-background-hover, #1f212d)";
export const buttonOverlayPressedBackgroundColor = "var(--color-button-overlay-background-pressed, #2a2c39)";
export const buttonPlainBackgroundColor = "transparent";
export const buttonPlainBackgroundColorHover = "var(--color-button-plain-background-hover, rgba(26, 115, 232, 0.08))";
export const buttonPlainBackgroundColorPressed = "var(--color-button-plain-background-pressed, rgba(26, 115, 232, 0.12))";
export const buttonPlainLabelColor = "var(--color-button-plain, #1a73e8)";
export const buttonPlainTextColor = "var(--color-button-plain, #1a73e8)";
export const buttonPlainIconColor = "var(--color-button-plain, #1a73e8)";
export const buttonPlainHoverBackgroundColor = "var(--color-button-plain-background-hover, rgba(26, 115, 232, 0.08))";
export const buttonPlainPressedBackgroundColor = "var(--color-button-plain-background-pressed, rgba(26, 115, 232, 0.12))";
export const buttonSmallHeight = "40px";
export const buttonSmallIconSize = "18px";
export const buttonSmallLabelFontSize = "15px";
export const buttonSmallLabelFontWeight = 500;
export const buttonSmallLabelFontLineHeight = "20px";
export const buttonSmallLabelFontLetterSpacing = "0.2px";
export const buttonSmallTextFontSize = "15px";
export const buttonSmallTextFontWeight = 500;
export const buttonSmallTextFontLineHeight = "20px";
export const buttonSmallTextFontLetterSpacing = "0.2px";
export const buttonMediumHeight = "48px";
export const buttonMediumIconSize = "20px";
export const buttonMediumLabelFontSize = "17px";
export const buttonMediumLabelFontWeight = 500;
export const buttonMediumLabelFontLineHeight = "22px";
export const buttonMediumLabelFontLetterSpacing = "0.2px";
export const buttonMediumTextFontSize = "17px";
export const buttonMediumTextFontWeight = 500;
export const buttonMediumTextFontLineHeight = "22px";
export const buttonMediumTextFontLetterSpacing = "0.2px";
export const buttonLargeHeight = "56px";
export const buttonLargeIconSize = "24px";
export const buttonLargeLabelFontSize = "19px";
export const buttonLargeLabelFontWeight = 500;
export const buttonLargeLabelFontLineHeight = "24px";
export const buttonLargeLabelFontLetterSpacing = "0.2px";
export const buttonLargeTextFontSize = "19px";
export const buttonLargeTextFontWeight = 500;
export const buttonLargeTextFontLineHeight = "24px";
export const buttonLargeTextFontLetterSpacing = "0.2px";
export const buttonOnlyTextSmallPaddingHorizontal = "16px";
export const buttonOnlyTextMediumPaddingHorizontal = "20px";
export const buttonOnlyTextLargePaddingHorizontal = "24px";
export const buttonWithIconSpacing = "8px";
export const buttonWithIconSmallPaddingIconside = "12px";
export const buttonWithIconSmallPaddingTextside = "16px";
export const buttonWithIconMediumPaddingIconside = "16px";
export const buttonWithIconMediumPaddingTextside = "20px";
export const buttonWithIconLargePaddingIconside = "20px";
export const buttonWithIconLargePaddingTextside = "24px";
export const buttonOnlyIconSmallWidth = "40px";
export const buttonOnlyIconMediumWidth = "48px";
export const buttonOnlyIconLargeWidth = "56px";
export const buttonFocusedBorderColor = "var(--color-border-focused, #202123)";
export const buttonDisabledOpacity = 0.48;
