// @beeline/design-tokens/js/tokens/components/button/button.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	buttonBorderRadius: "12px",
	buttonBorderColorFocus: "var(--color-border-focused, #202123)",
	buttonOpacityDisabled: 0.48,
	buttonIconMarginText: "8px",
	buttonContainedBackgroundColor: "var(--color-background-brand, #fdd835)",
	buttonContainedBackgroundColorHover: "var(--color-background-brand-hover, #fdc435)",
	buttonContainedBackgroundColorPressed: "var(--color-background-brand-pressed, #fdb435)",
	buttonContainedLabelColor: "var(--color-text-black-active, rgba(9, 11, 22, 0.94))",
	buttonContainedTextColor: "var(--color-text-black-active, rgba(9, 11, 22, 0.94))",
	buttonContainedIconColor: "var(--color-text-black-active, rgba(9, 11, 22, 0.94))",
	buttonContainedHoverBackgroundColor: "var(--color-background-brand-hover, #fdc435)",
	buttonContainedPressedBackgroundColor: "var(--color-background-brand-pressed, #fdb435)",
	buttonOutlineBorderWidth: "1px",
	buttonOutlineBorderStyle: "solid",
	buttonOutlineBorderColor: "var(--color-border, rgba(25, 28, 52, 0.18))",
	buttonOutlineBackgroundColor: "transparent",
	buttonOutlineBackgroundColorHover: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	buttonOutlineBackgroundColorPressed: "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))",
	buttonOutlineLabelColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	buttonOutlineTextColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	buttonOutlineIconColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	buttonOutlineHoverBackgroundColor: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	buttonOutlinePressedBackgroundColor: "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))",
	buttonOverlayBackgroundColor: "var(--color-button-overlay-background, #141622)",
	buttonOverlayBackgroundColorHover: "var(--color-button-overlay-background-hover, #1f212d)",
	buttonOverlayBackgroundColorPressed: "var(--color-button-overlay-background-pressed, #2a2c39)",
	buttonOverlayLabelColor: "var(--color-text-active-inverse, #ffffff)",
	buttonOverlayTextColor: "var(--color-text-active-inverse, #ffffff)",
	buttonOverlayIconColor: "var(--color-text-active-inverse, #ffffff)",
	buttonOverlayHoverBackgroundColor: "var(--color-button-overlay-background-hover, #1f212d)",
	buttonOverlayPressedBackgroundColor: "var(--color-button-overlay-background-pressed, #2a2c39)",
	buttonPlainBackgroundColor: "transparent",
	buttonPlainBackgroundColorHover: "var(--color-button-plain-background-hover, rgba(26, 115, 232, 0.08))",
	buttonPlainBackgroundColorPressed: "var(--color-button-plain-background-pressed, rgba(26, 115, 232, 0.12))",
	buttonPlainLabelColor: "var(--color-button-plain, #1a73e8)",
	buttonPlainTextColor: "var(--color-button-plain, #1a73e8)",
	buttonPlainIconColor: "var(--color-button-plain, #1a73e8)",
	buttonPlainHoverBackgroundColor: "var(--color-button-plain-background-hover, rgba(26, 115, 232, 0.08))",
	buttonPlainPressedBackgroundColor: "var(--color-button-plain-background-pressed, rgba(26, 115, 232, 0.12))",
	buttonSmallHeight: "40px",
	buttonSmallIconSize: "18px",
	buttonSmallLabelFontSize: "15px",
	buttonSmallLabelFontWeight: 500,
	buttonSmallLabelFontLineHeight: "20px",
	buttonSmallLabelFontLetterSpacing: "0.2px",
	buttonSmallTextFontSize: "15px",
	buttonSmallTextFontWeight: 500,
	buttonSmallTextFontLineHeight: "20px",
	buttonSmallTextFontLetterSpacing: "0.2px",
	buttonMediumHeight: "48px",
	buttonMediumIconSize: "20px",
	buttonMediumLabelFontSize: "17px",
	buttonMediumLabelFontWeight: 500,
	buttonMediumLabelFontLineHeight: "22px",
	buttonMediumLabelFontLetterSpacing: "0.2px",
	buttonMediumTextFontSize: "17px",
	buttonMediumTextFontWeight: 500,
	buttonMediumTextFontLineHeight: "22px",
	buttonMediumTextFontLetterSpacing: "0.2px",
	buttonLargeHeight: "56px",
	buttonLargeIconSize: "24px",
	buttonLargeLabelFontSize: "19px",
	buttonLargeLabelFontWeight: 500,
	buttonLargeLabelFontLineHeight: "24px",
	buttonLargeLabelFontLetterSpacing: "0.2px",
	buttonLargeTextFontSize: "19px",
	buttonLargeTextFontWeight: 500,
	buttonLargeTextFontLineHeight: "24px",
	buttonLargeTextFontLetterSpacing: "0.2px",
	buttonOnlyTextSmallPaddingHorizontal: "16px",
	buttonOnlyTextMediumPaddingHorizontal: "20px",
	buttonOnlyTextLargePaddingHorizontal: "24px",
	buttonWithIconSpacing: "8px",
	buttonWithIconSmallPaddingIconside: "12px",
	buttonWithIconSmallPaddingTextside: "16px",
	buttonWithIconMediumPaddingIconside: "16px",
	buttonWithIconMediumPaddingTextside: "20px",
	buttonWithIconLargePaddingIconside: "20px",
	buttonWithIconLargePaddingTextside: "24px",
	buttonOnlyIconSmallWidth: "40px",
	buttonOnlyIconMediumWidth: "48px",
	buttonOnlyIconLargeWidth: "56px",
	buttonFocusedBorderColor: "var(--color-border-focused, #202123)",
	buttonDisabledOpacity: 0.48,
};
