// @beeline/design-tokens/js/tokens/globals/fonts/fonts.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	fontFamilyHeader: "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
	fontFamilyText: "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
	fontFamilyCode: "'Roboto Mono', monospace",
	fontFamilySystem: "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
	fontFamilyMono: "'Roboto Mono', monospace",
	fontLetterSpacingH1: "0.3px",
	fontLetterSpacingH2: "0.3px",
	fontLetterSpacingH3: "0.3px",
	fontLetterSpacingH4: "0.2px",
	fontLetterSpacingH5: "0.2px",
	fontLetterSpacingH6: "0.2px",
	fontLetterSpacingSubtitle1: "0.2px",
	fontLetterSpacingSubtitle2: "0.2px",
	fontLetterSpacingSubtitle3: "0.2px",
	fontLetterSpacingBody1: "0.2px",
	fontLetterSpacingBody2: "0.2px",
	fontLetterSpacingBody3: "0.2px",
	fontLetterSpacingCaption: "0.2px",
	fontLetterSpacingOverline: "1.2px",
	fontLetterSpacingProductName: "0px",
	fontLineHeightH1: "66px",
	fontLineHeightH2: "56px",
	fontLineHeightH3: "44px",
	fontLineHeightH4: "32px",
	fontLineHeightH5: "24px",
	fontLineHeightH6: "22px",
	fontLineHeightSubtitle1: "24px",
	fontLineHeightSubtitle2: "22px",
	fontLineHeightSubtitle3: "20px",
	fontLineHeightBody1: "24px",
	fontLineHeightBody2: "22px",
	fontLineHeightBody3: "18px",
	fontLineHeightCaption: "16px",
	fontLineHeightOverline: "16px",
	fontLineHeightProductName: "28px",
	fontSizeH1: "54px",
	fontSizeH2: "44px",
	fontSizeH3: "34px",
	fontSizeH4: "26px",
	fontSizeH5: "20px",
	fontSizeH6: "17px",
	fontSizeSubtitle1: "19px",
	fontSizeSubtitle2: "17px",
	fontSizeSubtitle3: "15px",
	fontSizeBody1: "19px",
	fontSizeBody2: "17px",
	fontSizeBody3: "15px",
	fontSizeCaption: "13px",
	fontSizeOverline: "10px",
	fontSizeProductName: "25px",
	fontVariantNumericTableNumbers: "tabular-nums",
	fontWeightMedium: 500,
	fontWeightRegular: 400,
	fontWeightBold: 700,
	fontWeightH1: 500,
	fontWeightH2: 500,
	fontWeightH3: 500,
	fontWeightH4: 500,
	fontWeightH5: 500,
	fontWeightH6: 500,
	fontWeightSubtitle1: 500,
	fontWeightSubtitle2: 500,
	fontWeightSubtitle3: 500,
	fontWeightBody1: 400,
	fontWeightBody2: 400,
	fontWeightBody3: 400,
	fontWeightCaption: 400,
	fontWeightOverline: 700,
	fontWeightProductName: 500,
	h1: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "54px",
        "fontWeight": 500,
        "lineHeight": "66px",
        "letterSpacing": "0.3px"
	},
	h2: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "44px",
        "fontWeight": 500,
        "lineHeight": "56px",
        "letterSpacing": "0.3px"
	},
	h3: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "34px",
        "fontWeight": 500,
        "lineHeight": "44px",
        "letterSpacing": "0.3px"
	},
	h4: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "26px",
        "fontWeight": 500,
        "lineHeight": "32px",
        "letterSpacing": "0.2px"
	},
	h5: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "20px",
        "fontWeight": 500,
        "lineHeight": "24px",
        "letterSpacing": "0.2px"
	},
	h6: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "17px",
        "fontWeight": 500,
        "lineHeight": "22px",
        "letterSpacing": "0.2px"
	},
	subtitle1: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "19px",
        "fontWeight": 500,
        "lineHeight": "24px",
        "letterSpacing": "0.2px"
	},
	subtitle2: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "17px",
        "fontWeight": 500,
        "lineHeight": "22px",
        "letterSpacing": "0.2px"
	},
	subtitle3: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "15px",
        "fontWeight": 500,
        "lineHeight": "20px",
        "letterSpacing": "0.2px"
	},
	body1: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "19px",
        "fontWeight": 400,
        "lineHeight": "24px",
        "letterSpacing": "0.2px"
	},
	body2: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "17px",
        "fontWeight": 400,
        "lineHeight": "22px",
        "letterSpacing": "0.2px"
	},
	body3: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "15px",
        "fontWeight": 400,
        "lineHeight": "18px",
        "letterSpacing": "0.2px"
	},
	caption: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "13px",
        "fontWeight": 400,
        "lineHeight": "16px",
        "letterSpacing": "0.2px"
	},
	overline: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "10px",
        "fontWeight": 700,
        "lineHeight": "16px",
        "letterSpacing": "1.2px"
	},
	productName: {
        "fontFamily": "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontSize": "25px",
        "fontWeight": 500,
        "lineHeight": "28px",
        "letterSpacing": "0px"
	},
};
