// @beeline/design-tokens/js/tokens/components/bubbleChat/bubbleChat.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const bubbleChatGap = "8px";
export const bubbleChatPaddingHorizontal = "20px";
export const bubbleChatPaddingVertical = "20px";
export const bubbleChatTextColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const bubbleChatTextFontSize = "19px";
export const bubbleChatTextFontWeight = 400;
export const bubbleChatTextFontLineHeight = "24px";
export const bubbleChatTextFontLetterSpacing = "0.2px";
export const bubbleChatGroupGap = "24px";
export const bubbleChatAvatarSpacing = "12px";
export const bubbleChatIncomingBackgroundColor = "var(--color-background-brand, #fdd835)";
export const bubbleChatIncomingFirstBorderRadiusTopLeft = "2px";
export const bubbleChatIncomingFirstBorderRadiusTopRight = "2px";
export const bubbleChatOutgoingBackgroundColor = "var(--color-status-neutral-background, #f1f1f3)";
export const bubbleChatOutgoingFirstBorderRadiusTopLeft = "20px";
export const bubbleChatOutgoingFirstBorderRadiusTopRight = "2px";
export const bubbleChatOutgoingStatusIconSize = "18px";
export const bubbleChatOutgoingStatusIconColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const bubbleChatOutgoingStatusIconPositionRight = "8px";
export const bubbleChatOutgoingStatusIconPositionBottom = "8px";
export const bubbleChatOutgoingReadStatusIconColor = "var(--color-text-black-active, rgba(9, 11, 22, 0.94))";
export const bubbleChatFirstBorderRadiusBottom = "20px";
export const bubbleChatNextBorderRadius = "20px";
