// @beeline/design-tokens/js/tokens/components/bubbleChat/bubbleChat.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	bubbleChatGap: "8px",
	bubbleChatPaddingHorizontal: "20px",
	bubbleChatPaddingVertical: "20px",
	bubbleChatTextColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	bubbleChatTextFontSize: "19px",
	bubbleChatTextFontWeight: 400,
	bubbleChatTextFontLineHeight: "24px",
	bubbleChatTextFontLetterSpacing: "0.2px",
	bubbleChatGroupGap: "24px",
	bubbleChatAvatarSpacing: "12px",
	bubbleChatIncomingBackgroundColor: "var(--color-background-brand, #fdd835)",
	bubbleChatIncomingFirstBorderRadiusTopLeft: "2px",
	bubbleChatIncomingFirstBorderRadiusTopRight: "2px",
	bubbleChatOutgoingBackgroundColor: "var(--color-status-neutral-background, #f1f1f3)",
	bubbleChatOutgoingFirstBorderRadiusTopLeft: "20px",
	bubbleChatOutgoingFirstBorderRadiusTopRight: "2px",
	bubbleChatOutgoingStatusIconSize: "18px",
	bubbleChatOutgoingStatusIconColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	bubbleChatOutgoingStatusIconPositionRight: "8px",
	bubbleChatOutgoingStatusIconPositionBottom: "8px",
	bubbleChatOutgoingReadStatusIconColor: "var(--color-text-black-active, rgba(9, 11, 22, 0.94))",
	bubbleChatFirstBorderRadiusBottom: "20px",
	bubbleChatNextBorderRadius: "20px",
};
