// @beeline/design-tokens/js/tokens/components/popover/popover.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	popoverTargetSpacing: "8px", // Отступ от целевого элемента до поповера
	popoverPaddingHorizontal: "16px",
	popoverPaddingTop: "16px",
	popoverPaddingBottom: "8px",
	popoverBackgroundColor: "var(--color-background-low, #ffffff)",
	popoverBorderRadius: "16px",
	popoverShadow: "0px 6px 38px rgba(0, 0, 0, 0.16), 0px 0px 10px rgba(0, 0, 0, 0.08)",
	popoverHeaderColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	popoverHeaderFontSize: "20px",
	popoverHeaderFontWeight: 500,
	popoverHeaderFontLineHeight: "24px",
	popoverHeaderFontLetterSpacing: "0.2px",
	popoverTextColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	popoverTextMarginBottom: "8px",
	popoverHelperTextFontSize: "13px",
	popoverHelperTextFontWeight: 400,
	popoverHelperTextFontLineHeight: "16px",
	popoverHelperTextFontLetterSpacing: "0.2px",
	popoverOnboardingTextFontSize: "15px",
	popoverOnboardingTextFontWeight: 400,
	popoverOnboardingTextFontLineHeight: "18px",
	popoverOnboardingTextFontLetterSpacing: "0.2px",
	popoverOnboardingActionsSpacing: "8px",
	popoverOnboardingPaginationActiveColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	popoverOnboardingPaginationInactiveColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
};
