// @beeline/design-tokens/js/tokens/components/popover/popover.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const popoverTargetSpacing = "8px"; // Отступ от целевого элемента до поповера
export const popoverPaddingHorizontal = "16px";
export const popoverPaddingTop = "16px";
export const popoverPaddingBottom = "8px";
export const popoverBackgroundColor = "var(--color-background-low, #ffffff)";
export const popoverBorderRadius = "16px";
export const popoverShadow = "0px 6px 38px rgba(0, 0, 0, 0.16), 0px 0px 10px rgba(0, 0, 0, 0.08)";
export const popoverHeaderColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const popoverHeaderFontSize = "20px";
export const popoverHeaderFontWeight = 500;
export const popoverHeaderFontLineHeight = "24px";
export const popoverHeaderFontLetterSpacing = "0.2px";
export const popoverTextColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const popoverTextMarginBottom = "8px";
export const popoverHelperTextFontSize = "13px";
export const popoverHelperTextFontWeight = 400;
export const popoverHelperTextFontLineHeight = "16px";
export const popoverHelperTextFontLetterSpacing = "0.2px";
export const popoverOnboardingTextFontSize = "15px";
export const popoverOnboardingTextFontWeight = 400;
export const popoverOnboardingTextFontLineHeight = "18px";
export const popoverOnboardingTextFontLetterSpacing = "0.2px";
export const popoverOnboardingActionsSpacing = "8px";
export const popoverOnboardingPaginationActiveColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const popoverOnboardingPaginationInactiveColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
