// @beeline/design-tokens/js/tokens/components/progressIndicator/progressIndicator.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const progressIndicatorLinearHeight = "4px";
export const progressIndicatorLinearIndicatorBackgroundColor = "linear-gradient(90deg, #feca48 0%, #ff7d5d 19.46%, #b732a2 43.54%, #621a9f 70.39%, #030013 100%)";
export const progressIndicatorLinearIndicatorBorderRadius = "8px";
export const progressIndicatorLinearTrackBackgroundColor = "var(--color-divider, rgba(25, 28, 52, 0.12))";
export const progressIndicatorLinearTrackBorderRadius = "8px";
export const progressIndicatorCircleStandartWidth = "44px";
export const progressIndicatorCircleStandartHeight = "44px";
export const progressIndicatorCircleStandartStrokeWidth = "6px";
export const progressIndicatorCircleStandartIndicatorBorderRadius = "12px";
export const progressIndicatorCircleSmallWidth = "20px";
export const progressIndicatorCircleSmallHeight = "20px";
export const progressIndicatorCircleSmallStrokeWidth = "3px";
export const progressIndicatorCircleSmallIndicatorBorderRadius = "6px";
export const progressIndicatorCircleIndicatorBackgroundColor = "var(--color-background-brand, #fdd835)";
export const progressIndicatorCircleTrackBackgroundColor = "var(--color-divider, rgba(25, 28, 52, 0.12))";
export const progressIndicatorAnimatedGap = "6px";
export const progressIndicatorAnimatedCircleBackgroundColor = "linear-gradient(90deg, #feca48 0%, #ff7d5d 19.46%, #b732a2 43.54%, #621a9f 70.39%, #030013 100%)";
export const progressIndicatorAnimatedCircleBorderRadius = "50%";
export const progressIndicatorAnimatedCircleWidth = "12px";
export const progressIndicatorAnimatedCircleHeight = "12px";
