// @beeline/design-tokens/js/tokens/components/progressIndicator/progressIndicator.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	progressIndicatorLinearHeight: "4px",
	progressIndicatorLinearIndicatorBackgroundColor: "linear-gradient(90deg, #feca48 0%, #ff7d5d 19.46%, #b732a2 43.54%, #621a9f 70.39%, #030013 100%)",
	progressIndicatorLinearIndicatorBorderRadius: "8px",
	progressIndicatorLinearTrackBackgroundColor: "var(--color-divider, rgba(25, 28, 52, 0.12))",
	progressIndicatorLinearTrackBorderRadius: "8px",
	progressIndicatorCircleStandartWidth: "44px",
	progressIndicatorCircleStandartHeight: "44px",
	progressIndicatorCircleStandartStrokeWidth: "6px",
	progressIndicatorCircleStandartIndicatorBorderRadius: "12px",
	progressIndicatorCircleSmallWidth: "20px",
	progressIndicatorCircleSmallHeight: "20px",
	progressIndicatorCircleSmallStrokeWidth: "3px",
	progressIndicatorCircleSmallIndicatorBorderRadius: "6px",
	progressIndicatorCircleIndicatorBackgroundColor: "var(--color-background-brand, #fdd835)",
	progressIndicatorCircleTrackBackgroundColor: "var(--color-divider, rgba(25, 28, 52, 0.12))",
	progressIndicatorAnimatedGap: "6px",
	progressIndicatorAnimatedCircleBackgroundColor: "linear-gradient(90deg, #feca48 0%, #ff7d5d 19.46%, #b732a2 43.54%, #621a9f 70.39%, #030013 100%)",
	progressIndicatorAnimatedCircleBorderRadius: "50%",
	progressIndicatorAnimatedCircleWidth: "12px",
	progressIndicatorAnimatedCircleHeight: "12px",
};
