// @beeline/design-tokens/js/tokens/components/avatar/avatar.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const avatarCircularBorderRadius = "50%";
export const avatarSquircleBorderRadius = "12px";
export const avatarSmallWidth = "20px";
export const avatarSmallHeight = "20px";
export const avatarMediumWidth = "40px";
export const avatarMediumHeight = "40px";
export const avatarMediumIconSize = "20px";
export const avatarMediumTextFontSize = "15px";
export const avatarMediumTextFontWeight = 500;
export const avatarMediumTextFontLineHeight = "20px";
export const avatarMediumTextFontLetterSpacing = "0.2px";
export const avatarGreyBackgroundColor = "var(--color-status-neutral-background, #f1f1f3)";
export const avatarGreyContentColor = "var(--color-status-neutral, #5e6071)";
export const avatarRedBackgroundColor = "var(--color-status-error-background, #ffecef)";
export const avatarRedContentColor = "var(--color-status-error, #ff5555)";
export const avatarAmberBackgroundColor = "var(--color-status-warning-background, #fff4e1)";
export const avatarAmberContentColor = "var(--color-status-warning, #ff9419)";
export const avatarBlueBackgroundColor = "var(--color-status-info-background, #e3f2ff)";
export const avatarBlueContentColor = "var(--color-status-info, #1a73e8)";
export const avatarGreenBackgroundColor = "var(--color-status-success-background, #e7f6eb)";
export const avatarGreenContentColor = "var(--color-status-success, #2ca853)";
export const avatarMagentaBackgroundColor = "var(--color-accent-magenta-background, #fae4f7)";
export const avatarMagentaContentColor = "var(--color-accent-magenta, #bc00b8)";
export const avatarPurpleBackgroundColor = "var(--color-accent-purple-background, #f3e7fe)";
export const avatarPurpleContentColor = "var(--color-accent-purple, #7e00ed)";
export const avatarTealBackgroundColor = "var(--color-accent-teal-background, #e0f7fb)";
export const avatarTealContentColor = "var(--color-accent-teal, #00adc7)";
