// @beeline/design-tokens/js/tokens/components/avatar/avatar.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	avatarCircularBorderRadius: "50%",
	avatarSquircleBorderRadius: "12px",
	avatarSmallWidth: "20px",
	avatarSmallHeight: "20px",
	avatarMediumWidth: "40px",
	avatarMediumHeight: "40px",
	avatarMediumIconSize: "20px",
	avatarMediumTextFontSize: "15px",
	avatarMediumTextFontWeight: 500,
	avatarMediumTextFontLineHeight: "20px",
	avatarMediumTextFontLetterSpacing: "0.2px",
	avatarGreyBackgroundColor: "var(--color-status-neutral-background, #f1f1f3)",
	avatarGreyContentColor: "var(--color-status-neutral, #5e6071)",
	avatarRedBackgroundColor: "var(--color-status-error-background, #ffecef)",
	avatarRedContentColor: "var(--color-status-error, #ff5555)",
	avatarAmberBackgroundColor: "var(--color-status-warning-background, #fff4e1)",
	avatarAmberContentColor: "var(--color-status-warning, #ff9419)",
	avatarBlueBackgroundColor: "var(--color-status-info-background, #e3f2ff)",
	avatarBlueContentColor: "var(--color-status-info, #1a73e8)",
	avatarGreenBackgroundColor: "var(--color-status-success-background, #e7f6eb)",
	avatarGreenContentColor: "var(--color-status-success, #2ca853)",
	avatarMagentaBackgroundColor: "var(--color-accent-magenta-background, #fae4f7)",
	avatarMagentaContentColor: "var(--color-accent-magenta, #bc00b8)",
	avatarPurpleBackgroundColor: "var(--color-accent-purple-background, #f3e7fe)",
	avatarPurpleContentColor: "var(--color-accent-purple, #7e00ed)",
	avatarTealBackgroundColor: "var(--color-accent-teal-background, #e0f7fb)",
	avatarTealContentColor: "var(--color-accent-teal, #00adc7)",
};
