// @beeline/design-tokens/js/tokens/components/tree/tree.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	treeItemBorderRadius: "12px",
	treeItemColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	treeItemInnerGap: "16px", // Расстояние между внутренними элементами (например между иконкой и текстом)
	treeItemDisabledOpacity: 0.48,
	treeItemAmountFontSize: "13px",
	treeItemAmountFontColor: "var(--color-text-grey-inactive, rgba(25, 28, 52, 0.7))",
	treeItemSmallHeight: "32px",
	treeItemSmallNestingStepIndent: "36px", // Шаг отступа для дочерней ветки
	treeItemSmallNestingEntitiesIndent: "72px", // Шаг отступа для дочерних сущностей
	treeItemSmallFontSize: "15px",
	treeItemSmallFontWeight: 400,
	treeItemSmallFontLineHeight: "18px",
	treeItemSmallFontLetterSpacing: "0.2px",
	treeItemMediumHeight: "48px",
	treeItemMediumNestingStepIndent: "38px", // Шаг отступа для дочерней ветки
	treeItemMediumNestingEntitiesIndent: "72px", // Шаг отступа для дочерних сущностей
	treeItemMediumFontSize: "15px",
	treeItemMediumFontWeight: 400,
	treeItemMediumFontLineHeight: "18px",
	treeItemMediumFontLetterSpacing: "0.2px",
	treeItemLargeHeight: "56px",
	treeItemLargeNestingStepIndent: "40px", // Шаг отступа для дочерней ветки
	treeItemLargeNestingEntitiesIndent: "80px", // Шаг отступа для дочерних сущностей
	treeItemLargeFontSize: "17px",
	treeItemLargeFontWeight: 400,
	treeItemLargeFontLineHeight: "22px",
	treeItemLargeFontLetterSpacing: "0.2px",
	treeItemHoverBackgroundColor: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	treeItemPressedBackgroundColor: "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))",
	treeItemFocusedBackgroundColor: "var(--color-background-base-focused, rgba(25, 28, 52, 0.12))",
	treeItemSelectedBackgroundColor: "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))",
	treeItemDraggedShadow: "0px 6px 38px rgba(0, 0, 0, 0.16), 0px 0px 10px rgba(0, 0, 0, 0.08)",
	treeItemDraggedBackgroundColor: "var(--color-background-base-dragged, rgba(25, 28, 52, 0.08))",
};
