// @beeline/design-tokens/js/tokens/components/tree/tree.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const treeItemBorderRadius = "12px";
export const treeItemColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const treeItemInnerGap = "16px"; // Расстояние между внутренними элементами (например между иконкой и текстом)
export const treeItemDisabledOpacity = 0.48;
export const treeItemAmountFontSize = "13px";
export const treeItemAmountFontColor = "var(--color-text-grey-inactive, rgba(25, 28, 52, 0.7))";
export const treeItemSmallHeight = "32px";
export const treeItemSmallNestingStepIndent = "36px"; // Шаг отступа для дочерней ветки
export const treeItemSmallNestingEntitiesIndent = "72px"; // Шаг отступа для дочерних сущностей
export const treeItemSmallFontSize = "15px";
export const treeItemSmallFontWeight = 400;
export const treeItemSmallFontLineHeight = "18px";
export const treeItemSmallFontLetterSpacing = "0.2px";
export const treeItemMediumHeight = "48px";
export const treeItemMediumNestingStepIndent = "38px"; // Шаг отступа для дочерней ветки
export const treeItemMediumNestingEntitiesIndent = "72px"; // Шаг отступа для дочерних сущностей
export const treeItemMediumFontSize = "15px";
export const treeItemMediumFontWeight = 400;
export const treeItemMediumFontLineHeight = "18px";
export const treeItemMediumFontLetterSpacing = "0.2px";
export const treeItemLargeHeight = "56px";
export const treeItemLargeNestingStepIndent = "40px"; // Шаг отступа для дочерней ветки
export const treeItemLargeNestingEntitiesIndent = "80px"; // Шаг отступа для дочерних сущностей
export const treeItemLargeFontSize = "17px";
export const treeItemLargeFontWeight = 400;
export const treeItemLargeFontLineHeight = "22px";
export const treeItemLargeFontLetterSpacing = "0.2px";
export const treeItemHoverBackgroundColor = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const treeItemPressedBackgroundColor = "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))";
export const treeItemFocusedBackgroundColor = "var(--color-background-base-focused, rgba(25, 28, 52, 0.12))";
export const treeItemSelectedBackgroundColor = "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))";
export const treeItemDraggedShadow = "0px 6px 38px rgba(0, 0, 0, 0.16), 0px 0px 10px rgba(0, 0, 0, 0.08)";
export const treeItemDraggedBackgroundColor = "var(--color-background-base-dragged, rgba(25, 28, 52, 0.08))";
