// @beeline/design-system-react/extracted-styles/esm/components/DatePicker/Calendar/CalendarMobile/CalendarMobileMonths/CalendarMobileMonths.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { classNames } from '../../../../../helpers/classnames.js';
import { getBEMClasses } from '../../../../../helpers/get-bem.js';
import { useRef, useEffect } from 'react';
import { WEEKDAYS, MONTHS_NAMES } from '../../constants.js';
import { getWeeksOfMonths, isSameDates, isRangeMode, hasDateLimit } from '../../utils.js';

const [, calendarModifier, calendarBlock] = getBEMClasses('dsb_calendar-mobile');
const [monthsClassName, monthsModifier] = calendarBlock('months');
const CalendarMobileMonths = ({ minDate, maxDate, disabledStartDate, disabledEndDate, viewDate, selectedDate, onSelectDay, multiselect, readonly, }) => {
    const today = new Date();
    const viewYear = viewDate.getFullYear();
    const viewMonth = viewDate.getMonth();
    // Если maxDate не задан, показываем +10 лет вперед, чтобы можно было скроллить
    const defaultMaxYear = viewYear + 10;
    const calculatedMaxYear = maxDate ? maxDate.getFullYear() : defaultMaxYear;
    const monthsWeeks = getWeeksOfMonths(viewYear, minDate?.getFullYear(), calculatedMaxYear);
    const monthsRef = useRef(null);
    useEffect(() => {
        /**
         * Скролл до активного месяца, поиск в рамках контейнера месяцев а не глобального document,
         * работает при MicroFrontend и shadowDOM
         * */
        const id = `${viewYear}-${viewMonth}`;
        const monthsObj = monthsRef?.current;
        // В тестовой среде (jsdom) scrollIntoView может отсутствовать, поэтому вызываем безопасно
        const targetEl = monthsObj?.getElementsByClassName(id)[0];
        try {
            targetEl?.scrollIntoView?.();
        }
        catch {
            /* no-op: в окружениях без scrollIntoView просто пропускаем */
        }
    }, [viewYear, viewMonth]);
    const handleSelectDay = (day, month, year) => {
        if (day) {
            onSelectDay(day, month, year);
        }
    };
    const isCurrentDay = (day, month, year) => {
        if (day) {
            const checkedDate = new Date(year, month, day);
            return isSameDates(checkedDate, today);
        }
        return false;
    };
    const isSelectedDay = (day, month, year) => {
        if (day) {
            const checkedDate = new Date(year, month, day);
            /**
             * Расчёт стиля для контролла "День"
             * условие, что параметр "стартовая/конечная дата заблокирована для выбора" - disabledStartValue/disabledFinishValue может быть true
             */
            const getRangeModeSelectedDate = () => {
                if (disabledStartDate && disabledEndDate) {
                    return false;
                }
                if (disabledStartDate) {
                    return Array.isArray(selectedDate) && selectedDate.indexOf(checkedDate) === 1;
                }
                if (disabledEndDate) {
                    return Array.isArray(selectedDate) && selectedDate.indexOf(checkedDate) === 0;
                }
                return Array.isArray(selectedDate) && selectedDate?.some(date => date && isSameDates(date, checkedDate));
            };
            return isRangeMode(selectedDate)
                ? getRangeModeSelectedDate()
                : selectedDate && isSameDates(selectedDate, checkedDate);
        }
        return false;
    };
    const isDisabledDay = (day, month, year) => {
        if (day) {
            // Используем UTC-полуночь для корректного сравнения с min/max без сдвигов по TZ
            const checkedDate = new Date(Date.UTC(year, month, day));
            return hasDateLimit(checkedDate, minDate, maxDate);
        }
        return false;
    };
    const getRangeStatusOfDay = (day, month, year) => {
        if (multiselect) {
            return {};
        }
        if (day && isRangeMode(selectedDate)) {
            const [startDay, endDay] = selectedDate;
            const isFilledRange = startDay && endDay;
            const checkedDate = new Date(year, month, day);
            /**
       Даты без учёта времени, приравняв часы в 00:00:00:000
       */
            const startDayMidnight = startDay && new Date(new Date(startDay).setHours(0, 0, 0, 0));
            const endDayMidnight = endDay && new Date(new Date(endDay).setHours(0, 0, 0, 0));
            /**
       Расчёт стартовой даты
       + условие, что параметр "стартовая дата заблокирована для выбора" - disabledStartValue может быть true
       */
            const getStarting = () => {
                if (disabledStartDate) {
                    return false;
                }
                return isFilledRange && isSameDates(startDay, checkedDate);
            };
            /**
       Расчёт конечной даты
       + условие, что параметр "конечная дата заблокирована для выбора" - disabledFinishValue может быть true
       */
            const getEnding = () => {
                if (disabledEndDate) {
                    return false;
                }
                return isFilledRange && isSameDates(endDay, checkedDate);
            };
            return {
                starting: getStarting(),
                ranging: startDayMidnight && startDayMidnight <= checkedDate && endDayMidnight && endDayMidnight >= checkedDate,
                ending: getEnding(),
            };
        }
        return {};
    };
    return (jsxs("div", { className: classNames(calendarModifier('panel'), { readonly }), children: [jsx("table", { className: classNames(calendarModifier('dates'), calendarModifier('panel-header')), children: jsx("thead", { className: calendarModifier('weekdays'), children: jsx("tr", { children: WEEKDAYS.map(weekday => (jsx("th", { className: calendarModifier('weekday'), children: weekday }, weekday))) }) }) }), jsx("div", { ref: monthsRef, className: monthsClassName, children: monthsWeeks.map(({ month, year, monthWeeks }) => (jsxs("div", { id: `${year}-${month}`, className: classNames(monthsModifier('month'), `${year}-${month}`), children: [jsx("span", { className: monthsModifier('month--title'), children: `${MONTHS_NAMES[month]} ${year}` }), jsx("table", { className: calendarModifier('dates'), children: jsx("tbody", { className: calendarModifier('weeks'), children: monthWeeks.map((week, index) => (jsx("tr", { children: week.map((day, index) => (jsx("td", { className: classNames(calendarModifier('day-cell'), {
                                            ...getRangeStatusOfDay(day, month, year),
                                        }), children: jsx("span", { className: classNames(calendarModifier('day'), {
                                                today: isCurrentDay(day, month, year),
                                                disabled: isDisabledDay(day, month, year),
                                                selected: isSelectedDay(day, month, year),
                                                readonly: !!readonly,
                                                [calendarModifier('day__empty')]: day === null,
                                            }), onClick: () => handleSelectDay(day, month, year), children: day }) }, `day-${index}`))) }, `week-${index}`))) }) })] }, `${year}-${month}`))) })] }));
};

export { CalendarMobileMonths };
