// @beeline/design-system-react/build/components/Search/Search.js
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';
import './Search2.js';
import { Icons } from '@beeline/design-tokens/js/iconfont/icons';
import React__default, { useRef, useImperativeHandle, useState, useEffect } from 'react';
import { classNames } from '../../helpers/classnames.js';
import { Divider } from '../Divider/Divider.js';
import { IconButton } from '../IconButton/IconButton.js';
import { Controlled } from './Controlled.js';
import { SearchDropdown } from './Search.dropdown.js';
import { iconSize } from './Search.types.js';

const Search = React__default.forwardRef(({ fullWidth, size = 'medium', className, inputClassName, value, onSearch, onChange, onKeyUp, onClear, filterItems, selectedFilter, onFilterChange, dataTestId = 'Search', isAdaptive = false, maskConfig, ...props }, ref) => {
    const inputRef = useRef(null);
    useImperativeHandle(ref, () => inputRef.current);
    const rootRef = useRef(null);
    const categoriesRef = useRef(null);
    const [isOpen, setOpen] = useState(false);
    const handleOpen = () => setOpen(true);
    const handleClose = () => setOpen(false);
    const inputClasses = classNames('dsb_search', `dsb_search-size__${size}`, inputClassName);
    const wrapperClasses = classNames('dsb_search-wrapper', fullWidth && 'dsb_search-wrapper__full', className);
    const handleSearch = () => {
        onSearch?.(value);
    };
    const handleClear = () => {
        onSearch?.('');
        onClear?.();
        inputRef?.current.focus();
    };
    const handleChange = event => {
        onChange?.(event);
    };
    const handleKeyUp = event => {
        if (event.key === 'Enter') {
            handleSearch();
        }
        onKeyUp?.(event);
    };
    const makeCategorySelectComponent = () => {
        const rightIcon = (jsx(IconButton, { dataTestId: dataTestId + '-arrowBut', size: size, iconName: isOpen ? Icons.NavArrowUp : Icons.NavArrowDown, onClick: isOpen ? handleClose : handleOpen }));
        const dropdownMenuProps = {
            options: filterItems || [],
            parentRef: rootRef,
            values: selectedFilter ? [selectedFilter] : [],
            onChange: ([onChangeSelectedFilter]) => {
                onFilterChange?.(onChangeSelectedFilter ?? undefined);
                setOpen(false);
            },
            makeOption: item => jsx("span", { children: item.value }),
            onOutsideClick: () => {
                setOpen(false);
                onClear?.();
            },
            isOpen,
            handleSetOpen: setOpen,
            multiple: false,
            dataTestId: dataTestId + '-dropdown',
        };
        const classNameLabel = classNames({
            'dsb_search_categories_label-placeholder': !selectedFilter?.value,
        });
        return (jsxs("div", { role: "search", tabIndex: 0, className: "dsb_search_categories", ref: categoriesRef, onClick: handleOpen, onKeyDown: event => {
                if (event.key === 'Enter' || event.key === ' ') {
                    event.preventDefault();
                    handleOpen();
                }
            }, "data-testid": dataTestId + '-categories', children: [jsx(Divider, { className: "dsb_search_categories_vertical-divider" }), jsx("span", { className: classNameLabel, "data-testid": dataTestId + '-selectedFilter', children: selectedFilter?.value || 'Категории' }), rightIcon, isOpen && jsx(SearchDropdown, { ...dropdownMenuProps })] }));
    };
    const [leftCategory, setLeftCategory] = useState(categoriesRef?.current?.clientWidth || 0);
    useEffect(() => {
        setLeftCategory(categoriesRef?.current?.clientWidth || 0);
    }, [categoriesRef, selectedFilter]);
    const rightCloseButton = leftCategory ? leftCategory + 16 : 16;
    const dataTestInputId = `${dataTestId}-input`;
    const dataTestIconId = `${dataTestId}-butClose`;
    const inputProps = {
        onKeyUp: handleKeyUp,
        onChange: handleChange,
        className: inputClasses,
        autoComplete: 'off',
        type: 'search',
        style: {
            paddingRight: `${leftCategory ? leftCategory + 52 : 52}px`,
        },
        ...props,
    };
    const iconButtonProps = {
        iconName: Icons.Close,
        className: 'dsb_search-close-button',
        style: { right: `${rightCloseButton}px` },
        size: iconSize[size],
        onClick: handleClear,
    };
    return (jsxs("span", { "data-testid": dataTestId, className: wrapperClasses, ref: rootRef, children: [jsx(IconButton, { iconName: Icons.Search, className: "dsb_search-button", size: iconSize[size], onClick: handleSearch, dataTestId: dataTestId + '-butSearch' }), value === undefined ? (jsx(Controlled, { inputRef: inputRef, dataTestId: dataTestInputId, dataTestIconId: dataTestIconId, iconButton: iconButtonProps, ...inputProps })) : (jsxs(Fragment, { children: [jsx("input", { ref: inputRef, value: value, "data-testid": dataTestInputId, ...inputProps }), value && jsx(IconButton, { ...iconButtonProps, dataTestId: dataTestIconId })] })), filterItems && makeCategorySelectComponent()] }));
});

export { Search };
