// @beeline/design-tokens/js/tokens/components/datepicker/datepicker.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const datepickerPanelShadow = "0px 2px 10px rgba(0, 0, 0, 0.08), 0px 2px 8px rgba(0, 0, 0, 0.08)";
export const datepickerPanelBorderRadius = "12px";
export const datepickerPanelBackgroundColor = "var(--color-background-medium, #ffffff)";
export const datepickerPanelPaddingBottom = "16px";
export const datepickerHeaderHeight = "52px";
export const datepickerHeaderPaddingTop = "16px";
export const datepickerHeaderPaddingBottom = "12px";
export const datepickerHeaderPaddingLeft = "24px";
export const datepickerHeaderPaddingRight = "16px";
export const datepickerHeaderTitleFontSize = "15px";
export const datepickerHeaderTitleFontWeight = 500;
export const datepickerHeaderTitleFontLineHeight = "20px";
export const datepickerHeaderTitleFontLetterSpacing = "0.2px";
export const datepickerHeaderTitleColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const datepickerHeaderTitleGap = "4px"; // Отступ внутри заголовка между месяцем и годом, годом и кнопкой-стрелкой
export const datepickerHeaderArrowsGap = "20px";
export const datepickerContentPaddingHorizontal = "16px";
export const datepickerContentPaddingTop = "8px";
export const datepickerTimePaddingTop = "8px";
export const datepickerTimePaddingBottom = "12px";
export const datepickerTimePaddingHorizontal = "16px";
export const datepickerFooterPaddingTop = "12px";
export const datepickerFooterPaddingHorizontal = "16px";
export const datepickerWeekbarMarginBottom = "2px";
export const datepickerWeekbarDayNameWidth = "32px";
export const datepickerWeekbarDayNameHeight = "16px";
export const datepickerWeekbarDayNameFontSize = "13px";
export const datepickerWeekbarDayNameFontWeight = 400;
export const datepickerWeekbarDayNameFontLineHeight = "16px";
export const datepickerWeekbarDayNameFontLetterSpacing = "0.2px";
export const datepickerWeekbarDayNameColor = "var(--color-text-disabled, rgba(25, 28, 52, 0.48))";
export const datepickerBaseCellTextColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const datepickerBaseCellTextFontSize = "15px";
export const datepickerBaseCellTextFontWeight = 400;
export const datepickerBaseCellTextFontLineHeight = "18px";
export const datepickerBaseCellTextFontLetterSpacing = "0.2px";
export const datepickerBaseCellBaseHoverContentBackgroundColor = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const datepickerBaseCellSelectedColor = "var(--color-text-black-active, rgba(9, 11, 22, 0.94))";
export const datepickerBaseCellSelectedContentBackgroundColor = "var(--color-background-brand, #fdd835)";
export const datepickerBaseCellSelectedHoverContentBackgroundColor = "var(--color-background-brand-hover, #fdc435)";
export const datepickerBaseCellCurrentContentBorderWidth = "1px";
export const datepickerBaseCellCurrentContentBorderColor = "var(--color-border-focus, #212121)";
export const datepickerBaseCellOtherMonthColor = "var(--color-text-disabled, rgba(25, 28, 52, 0.48))";
export const datepickerBaseCellDisabledOpacity = 0.48;
export const datepickerBaseCellDisabledTextDecoration = "line-through";
export const datepickerDateCellWidth = "32px";
export const datepickerDateCellHeight = "32px";
export const datepickerDateCellPadding = "2px";
export const datepickerDateCellContentWidth = "28px";
export const datepickerDateCellContentHeight = "28px";
export const datepickerDateCellContentBorderRadius = "10px";
export const datepickerTextCellHeight = "36px";
export const datepickerTextCellWidth = "64px";
export const datepickerTextCellPaddingHorizontal = "4px";
export const datepickerTextCellBorderRadius = "12px";
export const datepickerRangeSelectedTrackBackgroundColor = "var(--color-accent-lemon-background, #fff7d7)";
export const datepickerDividerColor = "var(--color-divider, rgba(25, 28, 52, 0.12))";
export const datepickerDividerWidth = "1px";
