// @beeline/design-tokens/js/tokens/components/datepicker/datepicker.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	datepickerPanelShadow: "0px 2px 10px rgba(0, 0, 0, 0.08), 0px 2px 8px rgba(0, 0, 0, 0.08)",
	datepickerPanelBorderRadius: "12px",
	datepickerPanelBackgroundColor: "var(--color-background-medium, #ffffff)",
	datepickerPanelPaddingBottom: "16px",
	datepickerHeaderHeight: "52px",
	datepickerHeaderPaddingTop: "16px",
	datepickerHeaderPaddingBottom: "12px",
	datepickerHeaderPaddingLeft: "24px",
	datepickerHeaderPaddingRight: "16px",
	datepickerHeaderTitleFontSize: "15px",
	datepickerHeaderTitleFontWeight: 500,
	datepickerHeaderTitleFontLineHeight: "20px",
	datepickerHeaderTitleFontLetterSpacing: "0.2px",
	datepickerHeaderTitleColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	datepickerHeaderTitleGap: "4px", // Отступ внутри заголовка между месяцем и годом, годом и кнопкой-стрелкой
	datepickerHeaderArrowsGap: "20px",
	datepickerContentPaddingHorizontal: "16px",
	datepickerContentPaddingTop: "8px",
	datepickerTimePaddingTop: "8px",
	datepickerTimePaddingBottom: "12px",
	datepickerTimePaddingHorizontal: "16px",
	datepickerFooterPaddingTop: "12px",
	datepickerFooterPaddingHorizontal: "16px",
	datepickerWeekbarMarginBottom: "2px",
	datepickerWeekbarDayNameWidth: "32px",
	datepickerWeekbarDayNameHeight: "16px",
	datepickerWeekbarDayNameFontSize: "13px",
	datepickerWeekbarDayNameFontWeight: 400,
	datepickerWeekbarDayNameFontLineHeight: "16px",
	datepickerWeekbarDayNameFontLetterSpacing: "0.2px",
	datepickerWeekbarDayNameColor: "var(--color-text-disabled, rgba(25, 28, 52, 0.48))",
	datepickerBaseCellTextColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	datepickerBaseCellTextFontSize: "15px",
	datepickerBaseCellTextFontWeight: 400,
	datepickerBaseCellTextFontLineHeight: "18px",
	datepickerBaseCellTextFontLetterSpacing: "0.2px",
	datepickerBaseCellBaseHoverContentBackgroundColor: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	datepickerBaseCellSelectedColor: "var(--color-text-black-active, rgba(9, 11, 22, 0.94))",
	datepickerBaseCellSelectedContentBackgroundColor: "var(--color-background-brand, #fdd835)",
	datepickerBaseCellSelectedHoverContentBackgroundColor: "var(--color-background-brand-hover, #fdc435)",
	datepickerBaseCellCurrentContentBorderWidth: "1px",
	datepickerBaseCellCurrentContentBorderColor: "var(--color-border-focus, #212121)",
	datepickerBaseCellOtherMonthColor: "var(--color-text-disabled, rgba(25, 28, 52, 0.48))",
	datepickerBaseCellDisabledOpacity: 0.48,
	datepickerBaseCellDisabledTextDecoration: "line-through",
	datepickerDateCellWidth: "32px",
	datepickerDateCellHeight: "32px",
	datepickerDateCellPadding: "2px",
	datepickerDateCellContentWidth: "28px",
	datepickerDateCellContentHeight: "28px",
	datepickerDateCellContentBorderRadius: "10px",
	datepickerTextCellHeight: "36px",
	datepickerTextCellWidth: "64px",
	datepickerTextCellPaddingHorizontal: "4px",
	datepickerTextCellBorderRadius: "12px",
	datepickerRangeSelectedTrackBackgroundColor: "var(--color-accent-lemon-background, #fff7d7)",
	datepickerDividerColor: "var(--color-divider, rgba(25, 28, 52, 0.12))",
	datepickerDividerWidth: "1px",
};
