// @beeline/design-system-react/extracted-styles/esm/components/DatePicker/Calendar/Calendar.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { TimePicker } from '../TimePicker/TimePicker.js';
import { classNames } from '../../../helpers/classnames.js';
import { PopupsContext } from '../../../helpers/popupsContext.js';
import { Icons } from '@beeline/design-tokens/js/iconfont/icons';
import { useContext, useState, useRef, useEffect } from 'react';
import { usePrevious } from '../hooks/use-previous.js';
import { CalendarMobile } from './CalendarMobile/CalendarMobile.js';
import { CalendarNavigationIcon } from './CalendarNavigationIcon/CalendarNavigationIcon.js';
import { CalendarPanel } from './components/CalendarPanel/CalendarPanel.js';
import { Footer } from './components/Footer/Footer.js';
import { PANELS, MONTHS_NAMES } from './constants.js';
import { useOnOutsideCalendarClick } from './hooks.js';
import { validateCustomViewDate, getViewDate, convertUtcToTz, getDateFromIsoString, isRangeMode, convertTzToUtc, shiftMonthUTC, isTimeValid, formatToISO } from './newUtils.js';

const Calendar = props => {
    const { value, minValue = '', maxValue = '', disabledStartDate = false, disabledEndDate = false, className, datepickerRef, onSelectDate, onOutsideClick, onClickSelectDates, dataTestId = 'Calendar', multiselect, onSaveClick, saveButtonTitle, mobileView, readonly, excludeDate, customViewDate, initialViewDate, holidaysDate, enableTimePicker, outputDateFormat = enableTimePicker ? "yyyy-MM-dd'T'HH:mm:ss'Z'" : 'yyyy-MM-dd', timeRange = ['00:00', '23:59'], outOfRangeErrorText = 'Недопустимое время', timeZone, customizeDay, showAdjacentMonthDays = {
        showPrevious: false,
        showNext: false,
    }, allowAdjacentMonthSelection, } = props;
    const { rootElement } = useContext(PopupsContext);
    const [calendarElement, setCalendarElement] = useState(null);
    const [panel, setPanel] = useState(PANELS.DAYS);
    // Поддерживаем обратную совместимость с customViewDate
    const customViewDateValue = validateCustomViewDate(customViewDate);
    const initialViewDateValue = initialViewDate ? new Date(initialViewDate) : null;
    // initialViewDate используется только когда нет value или value пустой
    const hasValidValue = value && value !== '';
    const finalInitialViewDate = hasValidValue
        ? getViewDate(value)
        : initialViewDateValue || customViewDateValue || getViewDate(value);
    const [viewDateUTC, setViewDateUTC] = useState(finalInitialViewDate);
    const effectiveTimeZone = timeZone || Intl.DateTimeFormat().resolvedOptions().timeZone;
    // Запоминаем предыдущее значение
    const prevValue = usePrevious(value);
    // Флаг для отслеживания внутренних изменений от таймпикера
    const internalChangeRef = useRef(false);
    useEffect(() => {
        if (value !== prevValue) {
            const currentHasValidValue = value && value !== '';
            // Если есть валидное value, всегда обновляем viewDateUTC на его основе
            // Если нет валидного value, используем initialViewDate или customViewDate только если их не было раньше
            if (currentHasValidValue) {
                const updatedUTC = getViewDate(value);
                setViewDateUTC(updatedUTC);
            }
            else if (!initialViewDateValue && !customViewDateValue) {
                const updatedUTC = getViewDate(value);
                setViewDateUTC(updatedUTC);
            }
            // Обновляем timepickerValue только если изменение пришло извне (не от таймпикера)
            if (enableTimePicker && !internalChangeRef.current) {
                const dateForTimePicker = currentHasValidValue
                    ? getViewDate(value)
                    : initialViewDateValue || customViewDateValue || getViewDate(value);
                const userTz = convertUtcToTz(dateForTimePicker, effectiveTimeZone);
                setTimepickerValue({
                    hours: userTz.getHours().toString().padStart(2, '0'),
                    minutes: userTz.getMinutes().toString().padStart(2, '0'),
                });
            }
            // Сбрасываем флаг после обработки
            internalChangeRef.current = false;
        }
    }, [enableTimePicker, effectiveTimeZone, value, prevValue, initialViewDateValue, customViewDateValue]);
    // min/max => UTC
    const minDateUTC = getDateFromIsoString(minValue);
    const maxDateUTC = getDateFromIsoString(maxValue);
    // selectedDate => UTC (single или массив)
    const selectedDate = isRangeMode(value)
        ? value.map(element => getDateFromIsoString(element))
        : getDateFromIsoString(value);
    // Если enableTimePicker включён, работаем через конвертацию с effectiveTimeZone,
    // иначе используем UTC-компоненты (без смещения)
    const userTzInitial = enableTimePicker
        ? convertUtcToTz(viewDateUTC, effectiveTimeZone)
        : new Date(Date.UTC(viewDateUTC.getUTCFullYear(), viewDateUTC.getUTCMonth(), viewDateUTC.getUTCDate()));
    const [timepickerValue, setTimepickerValue] = useState({
        hours: userTzInitial.getHours().toString().padStart(2, '0'),
        minutes: userTzInitial.getMinutes().toString().padStart(2, '0'),
    });
    const [timepickerError, setTimepickerError] = useState(false);
    // это дл мая мобильной версии календаря
    useOnOutsideCalendarClick({
        calendarElement,
        datepickerElement: datepickerRef?.current,
        disabled: mobileView,
        onOutsideClick,
        rootElement,
    });
    if (mobileView) {
        // TODO переписать мобильный календарь
        return jsx(CalendarMobile, { ...props });
    }
    const handleRef = (node) => {
        setCalendarElement(node);
    };
    // ------------------ Переходы между панелями ------------------
    const setDaysPanel = () => setPanel(PANELS.DAYS);
    const setMonthsPanel = () => setPanel(PANELS.MONTHS);
    const setYearsPanel = () => setPanel(PANELS.YEARS);
    const showCalendarTypesPanel = () => {
        if (panel === PANELS.YEARS || panel === PANELS.MONTHS) {
            setDaysPanel();
        }
        if (panel === PANELS.DAYS) {
            setYearsPanel();
        }
    };
    // ------------------ Форматирование UTC -> строку -----------------
    const toFormattedString = (dateUTC) => {
        if (!dateUTC || Number.isNaN(dateUTC.getTime())) {
            console.error('Получена некорректная дата:', dateUTC);
            return '';
        }
        return formatToISO(dateUTC, outputDateFormat);
    };
    // ------------------ Вспомогательные функции (данные/вычисления) ------------------
    const getLocalFromUTC = (utc) => enableTimePicker
        ? convertUtcToTz(utc, effectiveTimeZone)
        : new Date(Date.UTC(utc.getUTCFullYear(), utc.getUTCMonth(), utc.getUTCDate()));
    const buildLocalDateFromClick = (day, month, year, hours, minutes) => {
        const baseLocal = getLocalFromUTC(viewDateUTC);
        const targetYear = year ?? baseLocal.getFullYear();
        const targetMonth = month ?? baseLocal.getMonth();
        const targetHours = hours ?? (enableTimePicker || multiselect ? +timepickerValue.hours : 0);
        const targetMinutes = minutes ?? (enableTimePicker || multiselect ? +timepickerValue.minutes : 0);
        return new Date(targetYear, targetMonth, day, targetHours, targetMinutes, 0, 0);
    };
    // ------------------ Выбор дня ------------------
    const handleSelectDay = (day, month, year) => {
        // Если режим readonly или оба диапазона заблокированы – не обрабатываем клик
        if ((disabledStartDate && disabledEndDate) || readonly) {
            return;
        }
        // Если переданы month и year, это означает клик по дню соседнего месяца
        if (month !== undefined && year !== undefined) {
            // Переключаем viewDate на соседний месяц
            const newViewDate = new Date(Date.UTC(year, month, 1));
            setViewDateUTC(newViewDate);
        }
        // Строим локальную дату клика без мутаций базовой даты
        const localDate = buildLocalDateFromClick(day, month, year);
        // Обратно получаем UTC
        // при выключенном TimePicker не производим дополнительную конвертацию
        const newUTC = enableTimePicker ? convertTzToUtc(localDate, effectiveTimeZone) : localDate;
        const newISO = enableTimePicker ? newUTC.toISOString() : toFormattedString(newUTC);
        // одно значение
        if (!isRangeMode(selectedDate)) {
            onSelectDate?.(newISO);
            return;
        }
        // массив
        if (multiselect) {
            const validSelectedDate = selectedDate.filter(date => date !== null);
            const indexDate = validSelectedDate.findIndex(date => date.getDate() === newUTC.getDate());
            if (indexDate === -1) {
                validSelectedDate.push(newUTC);
            }
            else {
                validSelectedDate.splice(indexDate, 1);
            }
            const newArrayDate = validSelectedDate.map(element => toFormattedString(element));
            onSelectDate?.(newArrayDate);
            return;
        }
        // Если range
        let newDateRange;
        const [startDate, endDate] = selectedDate;
        const isStartedRange = Boolean(startDate);
        const isFilledRange = Boolean(endDate);
        const localizedStartDate = startDate ? toFormattedString(startDate) : '';
        const localizedEndDate = endDate ? toFormattedString(endDate) : '';
        if (disabledStartDate && !disabledEndDate) {
            // нельзя менять старт, только конец
            newDateRange =
                isStartedRange && localDate <= startDate
                    ? [localizedStartDate, localizedStartDate]
                    : [localizedStartDate, newISO];
            onSelectDate?.(newDateRange);
            return;
        }
        if (!disabledStartDate && disabledEndDate) {
            // нельзя менять конец, только старт
            newDateRange =
                isFilledRange && localDate >= endDate ? [localizedEndDate, localizedEndDate] : [newISO, localizedEndDate];
            onSelectDate?.(newDateRange);
            return;
        }
        if (!isStartedRange) {
            // Если ещё не выбрана начальная
            newDateRange = [newISO, ''];
            onSelectDate?.(newDateRange);
            return;
        }
        if (!isFilledRange) {
            // Начальная есть, конечной нет
            const isNewDateSmaller = startDate && newUTC < startDate;
            const isEquivalent = endDate && newUTC.getTime() === endDate.getTime();
            const isNeedUpdate = isNewDateSmaller || isEquivalent;
            newDateRange = isNeedUpdate ? [newISO, localizedEndDate] : [localizedStartDate, newISO];
            onSelectDate?.(newDateRange);
            return;
        }
        newDateRange = [newISO, ''];
        onSelectDate?.(newDateRange);
    };
    // ------------------ Выбор месяца  и года ------------------
    const handleSelectMonth = (month) => {
        // Переход к целевому месяцу с коррекцией переполнения дня (например, 31 → 28/29)
        const currentMonth = viewDateUTC.getUTCMonth();
        const delta = month - currentMonth; // может быть отрицательным/положительным
        const newUTC = shiftMonthUTC(viewDateUTC, effectiveTimeZone, delta, enableTimePicker);
        setViewDateUTC(newUTC);
        setDaysPanel();
    };
    const handleSelectYear = (year) => {
        setViewDateUTC(new Date(Date.UTC(year, viewDateUTC.getUTCMonth(), viewDateUTC.getUTCDate(), viewDateUTC.getUTCHours(), viewDateUTC.getUTCMinutes())));
        setDaysPanel();
    };
    // ------------------ Переключения месяц назад/вперёд ------------------
    const setPrevMonth = () => {
        const newUTC = shiftMonthUTC(viewDateUTC, effectiveTimeZone, -1, enableTimePicker);
        setViewDateUTC(newUTC);
    };
    const setNextMonth = () => {
        const newUTC = shiftMonthUTC(viewDateUTC, effectiveTimeZone, +1, enableTimePicker);
        setViewDateUTC(newUTC);
    };
    // ------------------ Обработка компонента TimePicker ------------------
    const handleTimePick = (val) => {
        setTimepickerValue(val);
        const invalid = isTimeValid(val, timeRange);
        setTimepickerError(invalid);
        // Обрабатываем только если TimePicker включён
        if (enableTimePicker) {
            // 1) viewDateUTC -> TZ
            const loc = convertUtcToTz(viewDateUTC, effectiveTimeZone);
            // Создаём новую дату с обновлёнными часами/минутами без мутаций
            const updatedLocal = new Date(loc.getFullYear(), loc.getMonth(), loc.getDate(), +val.hours, +val.minutes, 0, 0);
            // 2) TZ -> UTC (реальный момент времени для внутреннего состояния/рендера)
            const newUTC = convertTzToUtc(updatedLocal, effectiveTimeZone);
            setViewDateUTC(newUTC);
            // 3) Вызываем onSelectDate с реальным UTC ISO
            if (!invalid) {
                internalChangeRef.current = true;
                onSelectDate?.(newUTC.toISOString());
            }
        }
    };
    // ------------------ Рендер заголовка (локальные год/месяц) ------------------
    const localView = enableTimePicker
        ? convertUtcToTz(viewDateUTC, effectiveTimeZone)
        : new Date(Date.UTC(viewDateUTC.getUTCFullYear(), viewDateUTC.getUTCMonth(), viewDateUTC.getUTCDate()));
    const localYear = localView.getFullYear();
    const localMonth = localView.getMonth();
    const hasFooter = (onSaveClick || onClickSelectDates) && !readonly;
    return (jsxs("div", { ref: handleRef, className: classNames('dsb_calendar', className), "data-testid": dataTestId, "data-allow-adjacent-selection": allowAdjacentMonthSelection ? 'true' : 'false', children: [jsxs("div", { className: classNames('dsb_calendar-panel', { multiselect: hasFooter }), children: [jsxs("header", { className: 'dsb_calendar-header', children: [jsxs("p", { className: 'dsb_calendar-title', children: [jsx("span", { className: 'dsb_calendar-select-month', "data-testid": `${dataTestId}-select-month`, children: MONTHS_NAMES[localMonth] }), jsx("span", { className: 'dsb_calendar-select-year', "data-testid": `${dataTestId}-select-year`, children: localYear }), jsx(CalendarNavigationIcon, { direction: panel === PANELS.DAYS ? Icons.ArrowDropDown : Icons.ArrowDropUp, onClick: showCalendarTypesPanel, dataTestId: `${dataTestId}-back` })] }), jsxs("span", { className: "dsb_calendar-navigation", children: [jsx(CalendarNavigationIcon, { direction: Icons.NavArrowLeft, onClick: setPrevMonth, 
                                        // disabled={isDisabledMonthSwitch(1)} временное решение
                                        dataTestId: `${dataTestId}-prev-month` }), jsx(CalendarNavigationIcon, { direction: Icons.NavArrowRight, onClick: setNextMonth, 
                                        // disabled={isDisabledMonthSwitch(-1)} временное решение
                                        dataTestId: `${dataTestId}-next-month` })] })] }), jsx(CalendarPanel, { panel: panel, minDate: minDateUTC, maxDate: maxDateUTC, timeZone: effectiveTimeZone, viewDate: viewDateUTC, disabledStartDate: disabledStartDate, disabledEndDate: disabledEndDate, selectedDate: selectedDate, multiselect: multiselect, readonly: readonly, excludeDate: excludeDate, holidaysDate: holidaysDate, customizeDay: customizeDay, showAdjacentMonthDays: showAdjacentMonthDays, allowAdjacentMonthSelection: allowAdjacentMonthSelection, setMonthsPanel: setMonthsPanel, setDaysPanel: setDaysPanel, selectDay: handleSelectDay, selectMonth: handleSelectMonth, selectYear: handleSelectYear }), enableTimePicker && panel === PANELS.DAYS && (jsx("div", { className: "dsb_calendar_timepickers", children: jsx(TimePicker, { error: timepickerError, errorMessage: timepickerError && outOfRangeErrorText, value: timepickerValue, onChange: handleTimePick }) }))] }), hasFooter && (jsx(Footer, { onSaveClick: onSaveClick, onClickSelectDates: onClickSelectDates, saveButtonTitle: saveButtonTitle }))] }));
};

export { Calendar };
