// @beeline/design-tokens/js/tokens/components/table/table.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const tableHeaderCellPaddingVertical = "18px";
export const tableHeaderCellPaddingVerticalDense = "6px";
export const tableHeaderCellPaddingHorizontal = "24px";
export const tableHeaderRowScrollShadow = "0px 2px 10px rgba(0, 0, 0, 0.08), 0px 2px 8px rgba(0, 0, 0, 0.08)";
export const tableHeaderFontSize = "15px";
export const tableHeaderFontWeight = 500;
export const tableHeaderFontLineHeight = "20px";
export const tableHeaderFontLetterSpacing = "0.2px";
export const tableHeaderColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const tableBodyCellIconSpacing = "4px";
export const tableBodyCellPaddingVertical = "17px";
export const tableBodyCellPaddingVerticalDense = "6px";
export const tableBodyCellPaddingHorizontal = "24px";
export const tableBodyCellEditableHoverBackgroundColor = "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))";
export const tableBodyCellEditableHoverBorderColor = "var(--color-border-focus, #212121)";
export const tableBodyCellEditableFocusedBackgroundColor = "transparent";
export const tableBodyCellEditableFocusedBorderColor = "var(--color-border-focus, #212121)";
export const tableBodyCellEditableDisabledOpacity = 0.48;
export const tableBodyCellEditableInvalidBackgroundColor = "var(--color-control-background-error, rgba(255, 85, 85, 0.08))";
export const tableBodyCellEditableInvalidBorderColor = "var(--color-border-error, #ff5555)";
export const tableBodyRowHoverBackgroundColor = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const tableBodyRowPressedBackgroundColor = "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))";
export const tableBodyRowSelectedBackgroundColor = "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))";
export const tableBodyRowNestedSpacingLeftStep = "24px";
export const tableBodyFontSize = "15px";
export const tableBodyFontWeight = 400;
export const tableBodyFontLineHeight = "18px";
export const tableBodyFontLetterSpacing = "0.2px";
export const tableBodyColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const tableDividerColor = "var(--color-divider, rgba(25, 28, 52, 0.12))";
export const tableDividerWidth = "1px";
export const tableBorderColor = "var(--color-border, rgba(25, 28, 52, 0.18))";
export const tableBorderWidth = "1px";
export const tableBorderRadius = "12px";
