// @beeline/design-tokens/js/tokens/components/table/table.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	tableHeaderCellPaddingVertical: "18px",
	tableHeaderCellPaddingVerticalDense: "6px",
	tableHeaderCellPaddingHorizontal: "24px",
	tableHeaderRowScrollShadow: "0px 2px 10px rgba(0, 0, 0, 0.08), 0px 2px 8px rgba(0, 0, 0, 0.08)",
	tableHeaderFontSize: "15px",
	tableHeaderFontWeight: 500,
	tableHeaderFontLineHeight: "20px",
	tableHeaderFontLetterSpacing: "0.2px",
	tableHeaderColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	tableBodyCellIconSpacing: "4px",
	tableBodyCellPaddingVertical: "17px",
	tableBodyCellPaddingVerticalDense: "6px",
	tableBodyCellPaddingHorizontal: "24px",
	tableBodyCellEditableHoverBackgroundColor: "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))",
	tableBodyCellEditableHoverBorderColor: "var(--color-border-focus, #212121)",
	tableBodyCellEditableFocusedBackgroundColor: "transparent",
	tableBodyCellEditableFocusedBorderColor: "var(--color-border-focus, #212121)",
	tableBodyCellEditableDisabledOpacity: 0.48,
	tableBodyCellEditableInvalidBackgroundColor: "var(--color-control-background-error, rgba(255, 85, 85, 0.08))",
	tableBodyCellEditableInvalidBorderColor: "var(--color-border-error, #ff5555)",
	tableBodyRowHoverBackgroundColor: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	tableBodyRowPressedBackgroundColor: "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))",
	tableBodyRowSelectedBackgroundColor: "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))",
	tableBodyRowNestedSpacingLeftStep: "24px",
	tableBodyFontSize: "15px",
	tableBodyFontWeight: 400,
	tableBodyFontLineHeight: "18px",
	tableBodyFontLetterSpacing: "0.2px",
	tableBodyColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	tableDividerColor: "var(--color-divider, rgba(25, 28, 52, 0.12))",
	tableDividerWidth: "1px",
	tableBorderColor: "var(--color-border, rgba(25, 28, 52, 0.18))",
	tableBorderWidth: "1px",
	tableBorderRadius: "12px",
};
