// @beeline/design-system-react/extracted-styles/esm/components/Tabs/useTabs.js
import { useDebounceCallback } from '../../helpers/use-debounce-callback.js';
import { useEffect, useState, useCallback, useMemo, cloneElement } from 'react';

/**
 * инкапсулирует логику управления табами
 *
 * Работает так
 *
 * 1. changeTab - функция для переключения табов (переключай табы только ей), должна содержать в себе все вызовы, нужные при успешной смене таба
 * 2. enrichedTabs - массив компонентов Tab (передается ребенком в компонент), обогащенный данными из родителя. Это данные selected, реакция на клик и т.д
 * 3. body - тело текущей страницы. Ищет текущий выбранный таб и возвращает его children-a
 *
 * useEffect-ы в конце отвечают за управление выбранным табом "снаружи" и изначальную отрисовку доступного таба (чтобы при первом рендере не отрисовать контент задизейбленного таба)
 *
 * @returns данные нужные для отрисовки табов. Читай описание {@link UseTabsData}
 */
const useTabs = ({ onChange, selectedTabIndex, tabs }) => {
    // TODO: выпилить этот хук после удаления устаревших табов
    useEffect(() => {
        const tabWithSelected = tabs.filter(Boolean).find(tab => 'selected' in tab.props);
        if (tabWithSelected) {
            console.warn('[Tabs] Вы используете устареший пропс selected в дочернем компоненте Tab', '\nЧтобы убрать это ворнинг вы можете:', '\n1. Перевести компонент на новый синтаксис, он создан чтобы вам было проще:)', '\n о новом синтаксисе читайте на сторибуке (внутренняя сеть) https://ds-react.apps.mn-kp01.vimpelcom.ru/?path=/docs/components-tabs', '\n2. Импортировать компонент из папки deprecated и использовать старый синтаксис', '\n для этого замените import { Tab, Tabs } from "@beeline/design-tokens" на import { Tab, Tabs } from "@beeline/design-tokens/deprecated"');
        }
        // хук должен вызываться единожды
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
    const [_selectedTabIndex, setSelectedTabIndex] = useState(selectedTabIndex || 0);
    const [currentTabData, setCurrentTabData] = useState({
        tabBody: null,
        tabElement: null,
    });
    const [prevTabData, setPrevTabData] = useState({
        tabBody: null,
        tabElement: null,
    });
    // вызывает изменение таба
    const changeTab = useCallback((targetTabIndex) => {
        if (_selectedTabIndex !== targetTabIndex) {
            setSelectedTabIndex(targetTabIndex);
            onChange?.(targetTabIndex);
        }
    }, [onChange, _selectedTabIndex]);
    const [setPrevTabDataWithDebounce] = useDebounceCallback(setPrevTabData, 100);
    /**
     * обогощаем вложенные табы
     *
     * 1. слушаем клик по табу, для переключения таба без указания {@link selectedTabIndex} разработчиком
     * 2. указываем табу его индекс, если не передан пользователем
     * 3. указываем табу состояние selected
     */
    const enrichedTabs = useMemo(() => tabs.filter(Boolean).map((tabComponent, tabIndex) => {
        // если пользователь компонента передал индекс берем его, иначе бемер индекс элемента в массиве
        const currentTabIndex = tabComponent.props.index ?? tabIndex;
        // обманываем типизацию т.к знаем что на самом деле таб ждет InnerTabProps
        return cloneElement(tabComponent, {
            ...tabComponent.props,
            // передаем собственный обработчик клика
            onClick: () => {
                if (!tabComponent.props.disabled) {
                    // переключаем таб
                    changeTab(currentTabIndex);
                    // не забываем вызвать onClick, указанный пользователем компонента
                    tabComponent.props.onClick?.(tabComponent.props.value);
                }
            },
            index: currentTabIndex,
            // выбранным может быть только доступный таб, сравниваем индекс таба с текущим выбранным
            selected: !tabComponent.props.disabled && currentTabIndex === _selectedTabIndex,
            // передаем обработчик, который таб вызовет, когда будет выбран
            onSelected: (data) => {
                setCurrentTabData({ ...data, tabProps: tabComponent.props });
                setPrevTabDataWithDebounce(data);
            },
        });
    }), [tabs, changeTab, _selectedTabIndex, setPrevTabDataWithDebounce]);
    // при изменении пропса selectedTabIndex изменяем внутрений стейт
    useEffect(() => {
        if (typeof selectedTabIndex === 'number') {
            const tabsIndexes = [];
            const targetTab = enrichedTabs.find(tab => {
                tabsIndexes.push(tab.props.index);
                return tab.props.index === selectedTabIndex;
            });
            if (!targetTab) {
                console.warn(`[Tabs] Вы попытались выбрать таб с индексом ${selectedTabIndex}.`, 'Таб с таким индексом отсутствует.', `Доступные индексы: ${tabsIndexes.join(', ')}`);
            }
            if (targetTab && !targetTab.props.disabled) {
                changeTab(targetTab.props.index);
            }
        }
    }, [selectedTabIndex, changeTab, enrichedTabs]);
    // при инициализации устанавливаем активным первый доступный таб
    // иначе при загрузке на задизейблином табе будет пустой экран
    // хук намеренно вызывается только один раз, чтобы избежать коллизий при работе компонента
    useEffect(() => {
        // действие подразумевает самостоятельное изменение компонентом своего состояния
        // этого не должно происходить когда состоянием управляет пользователь (т.е когда selectedTabIndex передан)
        if (selectedTabIndex === undefined) {
            const firstEnabledChildIndex = enrichedTabs.find(child => !child.props.disabled)?.props.index;
            if (firstEnabledChildIndex) {
                changeTab(firstEnabledChildIndex);
            }
        }
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
    return {
        tabs: enrichedTabs,
        currentTabIndex: _selectedTabIndex,
        currentTabData,
        prevTabData,
    };
};

export { useTabs };
