// @beeline/design-system-react/build/components/Preview/components/WordGallery/WordGallery.js
import { jsxs, jsx } from 'react/jsx-runtime';
import './WordGallery2.js';
import { Card } from '../../../Card/Card.js';
import { useState, useRef, useEffect } from 'react';

const WordGallery = ({ selectedIndex, onClick, value }) => {
    const [pages, setPages] = useState([]);
    const [isLoaded, setIsLoaded] = useState(false);
    const galleryRef = useRef(null);
    const hiddenContainerRef = useRef(null);
    const scrollToSelected = (index) => {
        if (galleryRef.current) {
            const cards = galleryRef.current.querySelectorAll('.dsb_preview__gallery-word__frame');
            if (cards[index]) {
                const card = cards[index];
                const container = galleryRef.current;
                const cardTop = card.offsetTop;
                const cardHeight = card.offsetHeight;
                const containerHeight = container.offsetHeight;
                container.scrollTo({
                    top: cardTop - containerHeight / 2 + cardHeight / 2,
                    behavior: 'smooth',
                });
            }
        }
    };
    useEffect(() => {
        scrollToSelected(selectedIndex);
    }, [selectedIndex, isLoaded]);
    useEffect(() => {
        let isMounted = true;
        let abortController = null;
        const loadDocument = async () => {
            if (!hiddenContainerRef.current)
                return;
            setIsLoaded(false);
            setPages([]);
            try {
                abortController = new AbortController();
                const response = await fetch(value.src, { signal: abortController.signal });
                if (!response.ok) {
                    throw new Error(`Ошибка загрузки: ${response.statusText}`);
                }
                const arrayBuffer = await response.arrayBuffer();
                if (!isMounted || !hiddenContainerRef.current)
                    return;
                hiddenContainerRef.current.innerHTML = '';
                const docxPreview = await import('../../../../node_modules/docx-preview/dist/docx-preview.js');
                const renderFunction = docxPreview.renderDocument || docxPreview.default?.renderDocument;
                if (typeof renderFunction !== 'function')
                    return;
                const parsedDocument = await docxPreview.parseAsync(arrayBuffer);
                await renderFunction(parsedDocument, hiddenContainerRef.current, undefined, {
                    className: 'dsb_gallery_word_thumb',
                    inWrapper: true,
                    ignoreWidth: false,
                    ignoreHeight: false,
                    ignoreFonts: false,
                    breakPages: true,
                    experimental: false,
                });
                if (!isMounted || !hiddenContainerRef.current)
                    return;
                await new Promise(resolve => requestAnimationFrame(resolve));
                // Собираем стили из документа
                const styleElements = hiddenContainerRef.current.querySelectorAll('style');
                const styles = Array.from(styleElements)
                    .map(el => el.outerHTML)
                    .join('\n');
                const pageElements = hiddenContainerRef.current.querySelectorAll('section.dsb_gallery_word_thumb');
                const pagesData = Array.from(pageElements).map((page, index) => ({
                    index,
                    html: page.outerHTML,
                    styles,
                }));
                setPages(pagesData);
                setIsLoaded(true);
            }
            catch (err) {
                if (err instanceof Error && err.name !== 'AbortError' && isMounted) {
                    console.error('WordGallery: ошибка загрузки документа', err);
                }
            }
        };
        void loadDocument();
        return () => {
            isMounted = false;
            abortController?.abort();
        };
    }, [value.src]);
    return (jsxs("div", { className: "dsb_preview__gallery-word", ref: galleryRef, children: [jsx("div", { ref: hiddenContainerRef, className: "dsb_preview__gallery-word__hidden-container" }), isLoaded &&
                pages.map(page => (jsxs("div", { className: "dsb_preview__gallery-word__item", children: [jsx(Card, { border: page.index === selectedIndex ? 'brand' : 'default', elevation: "medium", className: "dsb_preview__gallery-word__frame", onClick: () => onClick(page.index), children: jsx("div", { className: "dsb_preview__gallery-word__frame__page", dangerouslySetInnerHTML: { __html: page.styles + page.html } }) }), jsx("p", { className: "dsb_preview__gallery-word__caption", children: page.index + 1 })] }, `page-wrapper_${page.index}`)))] }));
};

export { WordGallery };
