// @beeline/design-system-react/build/components/Select/helpers.js
import { jsx } from 'react/jsx-runtime';
import { Tooltip } from '../Tooltip/Tooltip.js';
import { Typography } from '../Typography/Typography.js';
import { shallowEqual } from '../../helpers/shallowEqual.js';
import { isValidElement } from 'react';
import { CheckboxType } from './Select.types.js';

/**
 * Сравнивает два значения с учетом compareBy
 */
// eslint-disable-next-line id-length
const compareValues = (a, b, compareBy) => {
    // Если передана функция сравнения, используем её
    if (typeof compareBy === 'function') {
        return compareBy(a, b);
    }
    // Если передан ключ для сравнения, сравниваем по этому ключу
    if (typeof compareBy === 'string') {
        return a[compareBy] === b[compareBy];
    }
    // По умолчанию используем shallowEqual
    return shallowEqual(a, b);
};
const getPreparedOptions = (options, filter, filterValue, filterFunction) => {
    if (filter && filterValue && filterFunction) {
        return filterFunction(options, filterValue);
    }
    if (filter && filterValue && !filterFunction) {
        return options.filter(opt => {
            const currentStringValue = opt?.value?.value ?? String(opt.value);
            return currentStringValue.toString().toLowerCase().includes(filterValue.toLowerCase());
        });
    }
    return options;
};
/**
 * Возвращает опции в порядке отображения при группировке (как в Options).
 * Без groups возвращает options как есть.
 */
const getOptionsInDisplayOrder = (options, groups) => {
    if (!groups || groups.length === 0) {
        return options;
    }
    return groups.flatMap(group => options.filter(opt => opt.value?.groupID === group.id));
};
const defaultRenderValue = (values) => values?.map(val => val?.value ?? val).join(', ');
const defaultMakeOption = (option, showTooltip) => {
    const value = option?.value ?? String(option);
    return (jsx(Tooltip, { title: value, disableHover: !showTooltip, children: jsx(Typography, { variant: "body2", ellipsis: true, children: value }) }));
};
const getLoading = (loadingContent) => isValidElement(loadingContent) ? (loadingContent) : (jsx("span", { style: { padding: '4px 16px 2px 16px' }, children: loadingContent || 'Загрузка...' }));
const getCheckboxStatus = (options, selectedValues, filterValue, filter, filterFunction, compareBy) => {
    const filteredOptions = getPreparedOptions(options, filter, filterValue, filterFunction);
    const enabledFilteredOptions = filteredOptions.filter(opt => !opt.disabled);
    const selectedFilteredValues = selectedValues.filter(val => enabledFilteredOptions.some(opt => compareValues(opt.value, val.value, compareBy)));
    if (enabledFilteredOptions.length === 0) {
        return CheckboxType.UNCHECKED;
    }
    if (selectedFilteredValues.length === enabledFilteredOptions.length) {
        return CheckboxType.CHECKED;
    }
    if (selectedFilteredValues.length === 0) {
        return CheckboxType.UNCHECKED;
    }
    return CheckboxType.PARTIALLY;
};

export { compareValues, defaultMakeOption, defaultRenderValue, getCheckboxStatus, getLoading, getOptionsInDisplayOrder, getPreparedOptions };
