// @beeline/design-system-react/extracted-styles/esm/components/Tabs/Tab/Tab.js
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';
import { getBEMClasses } from '../../../helpers/get-bem.js';
import React__default, { useRef, useImperativeHandle, useEffect, useMemo } from 'react';
import { classNames } from '../../../helpers/classnames.js';
import { Icon } from '../../Icon/Icon.js';

const [tabClassName, tabModifier, tabElement] = getBEMClasses('dsb_tab-new');
const [tabContentClassName] = tabElement('content');
const [tabIndicatorClassName] = tabElement('indicator');
const [tabLabelClassName] = tabElement('label');
const [tabIconClassName] = tabElement('icon');
const Tab = React__default.forwardRef((props, ref) => {
    const { className, value, label, onClick, iconName, children, selected, onSelected, disabled, routingComponent, ...rest } = props;
    const contentRef = useRef(null);
    const localRef = useRef(null);
    useImperativeHandle(ref, () => localRef.current);
    // вызываем колбек если таб выбран
    useEffect(() => {
        if (selected) {
            onSelected({
                tabBody: children,
                tabElement: contentRef.current,
            });
        }
    }, [selected, onSelected, children, label]);
    const buttonClassName = useMemo(() => classNames(tabClassName, {
        [tabModifier('icon-only')]: iconName && !label,
        [tabModifier('selected')]: selected,
    }, className), [iconName, label, selected, className]);
    const handlerClick = () => {
        // TODO: подумать всегда ли надо подкручивать скролл или необходимо предоставить выбор пользователю
        // при клике на таб скроллим вью пользователя к контенту
        // localRef?.current?.scrollIntoView();
        onClick?.(value);
    };
    const content = (jsxs("div", { className: tabContentClassName, ref: contentRef, children: [iconName && jsx(Icon, { iconName: iconName, className: tabIconClassName }), label && jsx("span", { className: tabLabelClassName, children: label }), jsx("div", { className: tabIndicatorClassName })] }));
    return (jsx(Fragment, { children: routingComponent ? (React__default.cloneElement(routingComponent, {
            className: buttonClassName,
            disabled: disabled,
            onClick: handlerClick,
            ref: localRef,
            'data-testid': 'tab',
            ...rest,
        }, content)) : (jsx("button", { "data-testid": "Tab", ref: localRef, className: buttonClassName, disabled: disabled, onClick: handlerClick, ...rest, children: content })) }));
});

export { Tab };
