// @beeline/design-tokens/js/tokens/components/fab/fab.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	fabColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	fabBackgroundColor: "var(--color-background-low, #ffffff)",
	fabShadow: "0px 6px 38px rgba(0, 0, 0, 0.16), 0px 0px 10px rgba(0, 0, 0, 0.08)",
	fabIconSize: "24px",
	fabTextFontSize: "17px",
	fabTextFontWeight: 500,
	fabTextFontLineHeight: "22px",
	fabTextFontLetterSpacing: "0.2px",
	fabHoverBackgroundColor: "var(--color-background-medium, #ffffff)",
	fabPressedBackgroundColor: "var(--color-background-high, #ffffff)",
	fabFocusedBorderColor: "var(--color-border-focus, #212121)",
	fabFocusedBorderWidth: "1px",
	fabDisabledOpacity: 0.48,
	fabStandardWidth: "56px",
	fabStandardHeight: "56px",
	fabStandardPadding: "16px",
	fabStandardBorderRadius: "50%",
	fabMiniWidth: "40px",
	fabMiniHeight: "40px",
	fabMiniPadding: "8px",
	fabMiniBorderRadius: "50%",
	fabExtendedHeight: "48px",
	fabExtendedPaddingLeft: "12px",
	fabExtendedPaddingRight: "20px",
	fabExtendedIconSpacing: "12px",
	fabExtendedBorderRadius: "24px",
};
