// @beeline/design-tokens/js/tokens/components/fab/fab.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const fabColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const fabBackgroundColor = "var(--color-background-low, #ffffff)";
export const fabShadow = "0px 6px 38px rgba(0, 0, 0, 0.16), 0px 0px 10px rgba(0, 0, 0, 0.08)";
export const fabIconSize = "24px";
export const fabTextFontSize = "17px";
export const fabTextFontWeight = 500;
export const fabTextFontLineHeight = "22px";
export const fabTextFontLetterSpacing = "0.2px";
export const fabHoverBackgroundColor = "var(--color-background-medium, #ffffff)";
export const fabPressedBackgroundColor = "var(--color-background-high, #ffffff)";
export const fabFocusedBorderColor = "var(--color-border-focus, #212121)";
export const fabFocusedBorderWidth = "1px";
export const fabDisabledOpacity = 0.48;
export const fabStandardWidth = "56px";
export const fabStandardHeight = "56px";
export const fabStandardPadding = "16px";
export const fabStandardBorderRadius = "50%";
export const fabMiniWidth = "40px";
export const fabMiniHeight = "40px";
export const fabMiniPadding = "8px";
export const fabMiniBorderRadius = "50%";
export const fabExtendedHeight = "48px";
export const fabExtendedPaddingLeft = "12px";
export const fabExtendedPaddingRight = "20px";
export const fabExtendedIconSpacing = "12px";
export const fabExtendedBorderRadius = "24px";
