// @beeline/design-system-react/build/components/NewDropdownMenu/NewDropdownMenu.js
import { jsx, Fragment } from 'react/jsx-runtime';
import './NewDropdownMenu2.js';
import { PopupsContext } from '../../helpers/popupsContext.js';
import { createContext, useContext, useState, useEffect, useLayoutEffect } from 'react';
import { useFloating, useDismiss, useInteractions, FloatingPortal, FloatingOverlay } from '../../node_modules/@floating-ui/react/dist/floating-ui.react.js';
import { flip, shift, offset, size } from '../../node_modules/@floating-ui/react-dom/dist/floating-ui.react-dom.js';
import { autoUpdate } from '../../node_modules/@floating-ui/dom/dist/floating-ui.dom.js';

const MENU_DEFAULT_OFFSET = 8;
const DropdownContext = createContext({});
const NewDropdownMenu = ({ verticalOffset = MENU_DEFAULT_OFFSET, horizontalOffset = 0, align = 'bottom-start', targetElement, className, applicationRootElementID: applicationRootElementIDProps, dropdownElementID: dropdownElementIDProps, children, onOutsideClick, onBlur, scrollable = true, minHeight, dataTestId = 'DropdownMenu', isOpen, handleSetOpen, overlayScroll = true, widthMode = 'widthParent', onOpen, onClose, shouldRenderOverlay = false, overlayClassName, ...rest }) => {
    const { applicationRootElementID: applicationRootElementIDContext, dropdownElementID: dropdownElementIDContext, rootElement: rootElementDDContext, } = useContext(DropdownContext);
    const { applicationElementID, rootElement: rootElementContext } = useContext(PopupsContext);
    const rootElementID = applicationRootElementIDProps || applicationRootElementIDContext || applicationElementID || 'root';
    let floatingPortalRootId = rootElementID;
    const element = rootElementDDContext || rootElementContext;
    const getRootElement = () => {
        const mountElement = element || document.getElementById(rootElementID) || document.body;
        if (mountElement instanceof ShadowRoot) {
            floatingPortalRootId = 'floating-ui-root';
            const floatingPortalRoot = mountElement.getElementById(floatingPortalRootId);
            if (!floatingPortalRoot) {
                const floatingContainer = document.createElement('div');
                floatingContainer.id = floatingPortalRootId;
                mountElement.append(floatingContainer);
                return floatingContainer;
            }
            const childrenArray = [...mountElement.children];
            const lastChild = childrenArray.at(-1);
            if (lastChild !== floatingPortalRoot) {
                floatingPortalRoot.remove();
                mountElement.append(floatingPortalRoot);
            }
            return floatingPortalRoot;
        }
        return mountElement;
    };
    const [rootElement, setRootElement] = useState(getRootElement());
    useEffect(() => {
        setRootElement(getRootElement());
    }, [rootElementID, element]);
    const customSizeMiddleware = () => size({
        apply: ({ rects, elements, availableHeight }) => {
            if (widthMode === 'widthParent' && rects && elements && availableHeight !== undefined) {
                Object.assign(elements.floating.style, {
                    maxHeight: `${availableHeight}px`,
                    width: `${rects.reference.width}px`,
                });
            }
            if (elements && availableHeight !== undefined) {
                Object.assign(elements.floating.style, {
                    maxHeight: `${availableHeight}px`,
                });
            }
        },
    });
    const autoUpdateHook = (referenceEl, floatingEl, update) => autoUpdate(referenceEl, floatingEl, update, { ancestorScroll: scrollable });
    const handleOpenChange = (nextOpen) => {
        if (nextOpen) {
            onOpen?.();
        }
        else {
            onClose?.();
        }
        handleSetOpen?.(nextOpen);
    };
    const { refs, floatingStyles, strategy, context, update } = useFloating({
        elements: { reference: targetElement.current },
        placement: align,
        middleware: [
            flip(),
            shift(),
            customSizeMiddleware(),
            offset({ mainAxis: verticalOffset, crossAxis: horizontalOffset }),
        ],
        open: isOpen,
        onOpenChange: handleOpenChange,
        whileElementsMounted: autoUpdateHook,
    });
    const handleOutsideClick = (event) => {
        if (targetElement.current && targetElement.current.contains(event.target)) {
            return false;
        }
        onOutsideClick?.(event);
        return true;
    };
    const dismiss = useDismiss(context, {
        outsidePress: handleOutsideClick,
    });
    const { getReferenceProps, getFloatingProps } = useInteractions([dismiss]);
    useLayoutEffect(() => {
        if (targetElement?.current) {
            refs.setReference(targetElement.current);
        }
    }, [refs, targetElement]);
    useLayoutEffect(() => {
        if (isOpen && targetElement?.current) {
            update();
        }
    }, [isOpen, targetElement, update]);
    if (!targetElement?.current) {
        return jsx(Fragment, {});
    }
    return (jsx(FloatingPortal, { root: rootElement, children: shouldRenderOverlay ? (jsx(FloatingOverlay, { className: overlayClassName, lockScroll: overlayScroll, children: jsx("div", { style: {
                    ...floatingStyles,
                    minWidth: '112px',
                    minHeight,
                }, tabIndex: -1, ref: refs.setFloating, className: className, contentEditable: false, "data-testid": dataTestId, onBlur: onBlur, ...rest, ...getFloatingProps(), children: children }) })) : (jsx("div", { style: {
                ...floatingStyles,
                minWidth: '112px',
                minHeight,
            }, tabIndex: -1, ref: refs.setFloating, className: className, contentEditable: false, "data-testid": dataTestId, onBlur: onBlur, ...rest, ...getFloatingProps(), children: children })) }));
};

export { DropdownContext, NewDropdownMenu };
