// @beeline/design-tokens/js/tokens/components/dialog/dialog.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const dialogWidth = "560px";
export const dialogBackgroundColor = "var(--color-background-high, #ffffff)";
export const dialogBackdropColor = "var(--color-background-backdrop, rgba(0, 0, 0, 0.48))";
export const dialogShadow = "0px 6px 38px rgba(0, 0, 0, 0.16), 0px 0px 10px rgba(0, 0, 0, 0.08)";
export const dialogBorderRadius = "12px";
export const dialogHeaderPaddingHorizontal = "24px";
export const dialogHeaderColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const dialogHeaderFontSize = "20px";
export const dialogHeaderFontWeight = 500;
export const dialogHeaderFontLineHeight = "24px";
export const dialogHeaderFontLetterSpacing = "0.2px";
export const dialogDescriptionPaddingHorizontal = "24px";
export const dialogDescriptionPaddingVertical = "16px";
export const dialogDescriptionTextColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const dialogDescriptionTextFontSize = "17px";
export const dialogDescriptionTextFontWeight = 400;
export const dialogDescriptionTextFontLineHeight = "22px";
export const dialogDescriptionTextFontLetterSpacing = "0.2px";
export const dialogDescriptionCheckboxMarginTop = "16px";
export const dialogDescriptionCheckboxLabelColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const dialogDescriptionCheckboxLabelFontSize = "15px";
export const dialogDescriptionCheckboxLabelFontWeight = 400;
export const dialogDescriptionCheckboxLabelFontLineHeight = "18px";
export const dialogDescriptionCheckboxLabelFontLetterSpacing = "0.2px";
export const dialogFooterPaddingHorizontal = "24px";
export const dialogScrollableHeight = "455px";
export const dialogScrollableHeaderPaddingVertical = "16px";
export const dialogScrollableHeaderBorderBottomColor = "var(--color-divider, rgba(25, 28, 52, 0.12))";
export const dialogScrollableHeaderBorderBottomWidth = "1px";
export const dialogScrollableFooterPaddingVertical = "16px";
export const dialogScrollableFooterBorderTopColor = "var(--color-divider, rgba(25, 28, 52, 0.12))";
export const dialogScrollableFooterBorderTopWidth = "1px";
export const dialogNonScrollableMinHeight = "126px";
export const dialogNonScrollableMaxHeight = "455px";
export const dialogNonScrollableHeaderPaddingTop = "24px";
export const dialogNonScrollableFooterPaddingBottom = "24px";
export const dialogNonScrollableSelectionsPaddingVertical = "16px";
export const dialogFullscreenBackgroundColor = "var(--color-background-base, #ffffff)";
