// @beeline/design-tokens/js/tokens/components/dialog/dialog.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	dialogWidth: "560px",
	dialogBackgroundColor: "var(--color-background-high, #ffffff)",
	dialogBackdropColor: "var(--color-background-backdrop, rgba(0, 0, 0, 0.48))",
	dialogShadow: "0px 6px 38px rgba(0, 0, 0, 0.16), 0px 0px 10px rgba(0, 0, 0, 0.08)",
	dialogBorderRadius: "12px",
	dialogHeaderPaddingHorizontal: "24px",
	dialogHeaderColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	dialogHeaderFontSize: "20px",
	dialogHeaderFontWeight: 500,
	dialogHeaderFontLineHeight: "24px",
	dialogHeaderFontLetterSpacing: "0.2px",
	dialogDescriptionPaddingHorizontal: "24px",
	dialogDescriptionPaddingVertical: "16px",
	dialogDescriptionTextColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	dialogDescriptionTextFontSize: "17px",
	dialogDescriptionTextFontWeight: 400,
	dialogDescriptionTextFontLineHeight: "22px",
	dialogDescriptionTextFontLetterSpacing: "0.2px",
	dialogDescriptionCheckboxMarginTop: "16px",
	dialogDescriptionCheckboxLabelColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	dialogDescriptionCheckboxLabelFontSize: "15px",
	dialogDescriptionCheckboxLabelFontWeight: 400,
	dialogDescriptionCheckboxLabelFontLineHeight: "18px",
	dialogDescriptionCheckboxLabelFontLetterSpacing: "0.2px",
	dialogFooterPaddingHorizontal: "24px",
	dialogScrollableHeight: "455px",
	dialogScrollableHeaderPaddingVertical: "16px",
	dialogScrollableHeaderBorderBottomColor: "var(--color-divider, rgba(25, 28, 52, 0.12))",
	dialogScrollableHeaderBorderBottomWidth: "1px",
	dialogScrollableFooterPaddingVertical: "16px",
	dialogScrollableFooterBorderTopColor: "var(--color-divider, rgba(25, 28, 52, 0.12))",
	dialogScrollableFooterBorderTopWidth: "1px",
	dialogNonScrollableMinHeight: "126px",
	dialogNonScrollableMaxHeight: "455px",
	dialogNonScrollableHeaderPaddingTop: "24px",
	dialogNonScrollableFooterPaddingBottom: "24px",
	dialogNonScrollableSelectionsPaddingVertical: "16px",
	dialogFullscreenBackgroundColor: "var(--color-background-base, #ffffff)",
};
