// @beeline/design-tokens/js/tokens/components/menu/menu.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	menuWidthMin: "112px",
	menuWidth: "280px",
	menuPaddingVertical: "8px",
	menuBorderRadius: "12px",
	menuBackgroundColor: "var(--color-background-medium, #ffffff)",
	menuShadow: "0px 4px 30px rgba(0, 0, 0, 0.1), 0px 0px 10px rgba(0, 0, 0, 0.1)",
	menuItemHeight: "46px",
	menuItemPaddingHorizontal: "16px",
	menuItemHoverBackgroundColor: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	menuItemDisabledOpacity: 0.48,
	menuItemMenuButtonSpacing: "8px",
	menuItemMenuButtonColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	menuItemMenuIndicatorSpacing: "8px",
	menuItemMenuIndicatorColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	menuItemIconSpacing: "8px",
	menuItemIconColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	menuItemIconSize: "24px",
	menuItemCommandSpacing: "8px",
	menuItemCommandColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	menuItemCommandFontSize: "17px",
	menuItemCommandFontWeight: 400,
	menuItemCommandFontLineHeight: "22px",
	menuItemCommandFontLetterSpacing: "0.2px",
};
