// @beeline/design-tokens/js/tokens/components/menu/menu.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const menuWidthMin = "112px";
export const menuWidth = "280px";
export const menuPaddingVertical = "8px";
export const menuBorderRadius = "12px";
export const menuBackgroundColor = "var(--color-background-medium, #ffffff)";
export const menuShadow = "0px 4px 30px rgba(0, 0, 0, 0.1), 0px 0px 10px rgba(0, 0, 0, 0.1)";
export const menuItemHeight = "46px";
export const menuItemPaddingHorizontal = "16px";
export const menuItemHoverBackgroundColor = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const menuItemDisabledOpacity = 0.48;
export const menuItemMenuButtonSpacing = "8px";
export const menuItemMenuButtonColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const menuItemMenuIndicatorSpacing = "8px";
export const menuItemMenuIndicatorColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const menuItemIconSpacing = "8px";
export const menuItemIconColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const menuItemIconSize = "24px";
export const menuItemCommandSpacing = "8px";
export const menuItemCommandColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const menuItemCommandFontSize = "17px";
export const menuItemCommandFontWeight = 400;
export const menuItemCommandFontLineHeight = "22px";
export const menuItemCommandFontLetterSpacing = "0.2px";
