// @beeline/design-tokens/js/tokens/components/formfield/formfield.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	formfieldPaddingHorizontal: "16px",
	formfieldBackgroundColor: "var(--color-control-background, rgba(25, 28, 52, 0.1))",
	formfieldBackgroundColorFocus: "transparent",
	formfieldBackgroundColorError: "var(--color-control-background-error, rgba(255, 85, 85, 0.08))",
	formfieldBorderWidth: "1px",
	formfieldBorderStyle: "solid",
	formfieldBorderColor: "transparent",
	formfieldBorderColorHover: "var(--color-border-focus, #212121)",
	formfieldBorderColorFocus: "var(--color-border-focus, #212121)",
	formfieldBorderColorError: "var(--color-border-error, #ff5555)",
	formfieldBorderRadius: "12px",
	formfieldDisabledLabelColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	formfieldDisabledTextColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	formfieldDisabledHelperTextColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	formfieldDisabledOpacity: 0.48,
	formfieldInvalidBackgroundColor: "var(--color-control-background-error, rgba(255, 85, 85, 0.08))",
	formfieldInvalidBorderColor: "var(--color-border-error, #ff5555)",
	formfieldFocusedBackgroundColor: "transparent",
	formfieldFocusedBorderColor: "var(--color-border-focus, #212121)",
	formfieldHoverBorderColor: "var(--color-border-focus, #212121)",
	formfieldLabelColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	formfieldLabelColorDisabled: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	formfieldPlaceholderColor: "var(--color-text-disabled, rgba(25, 28, 52, 0.48))",
	formfieldTextColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	formfieldTextColorDisabled: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	formfieldSubscriptHeight: "16px",
	formfieldSubscriptMarginTop: "4px",
	formfieldSubscriptFontSize: "13px",
	formfieldSubscriptFontWeight: 400,
	formfieldSubscriptFontLineHeight: "16px",
	formfieldSubscriptFontLetterSpacing: "0.2px",
	formfieldHintColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	formfieldHintColorDisabled: "var(--color-text-disabled, rgba(25, 28, 52, 0.48))",
	formfieldHelperTextColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	formfieldErrorTextColor: "var(--color-status-error, #ff5555)",
	formfieldSmallTextFontSize: "15px",
	formfieldSmallTextFontWeight: 400,
	formfieldSmallTextFontLineHeight: "18px",
	formfieldSmallTextFontLetterSpacing: "0.2px",
	formfieldMediumTextFontSize: "17px",
	formfieldMediumTextFontWeight: 400,
	formfieldMediumTextFontLineHeight: "22px",
	formfieldMediumTextFontLetterSpacing: "0.2px",
	formfieldMediumLabelFontSize: "17px",
	formfieldMediumLabelFontSizeFloated: "13px",
	formfieldMediumLabelFontWeight: 400,
	formfieldMediumLabelFontLineHeight: "22px",
	formfieldMediumLabelFontLineHeightFloated: "16px",
	formfieldMediumLabelFloatedFontSize: "13px",
	formfieldMediumLabelFloatedFontLineHeight: "16px",
};
