// @beeline/design-tokens/js/tokens/components/tab/tab.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const tabHeight = "48px";
export const tabPaddingHorizontal = "24px";
export const tabTextFontSize = "15px";
export const tabTextFontWeight = 500;
export const tabTextFontLineHeight = "20px";
export const tabTextFontLetterSpacing = "0.2px";
export const tabIconSize = "24px";
export const tabIconSpacing = "4px";
export const tabIndicatorHeight = "4px";
export const tabIndicatorMinWidth = "40px";
export const tabIndicatorBorderRadiusTopLeft = "4px";
export const tabIndicatorBorderRadiusTopRight = "4px";
export const tabHoverIndicatorColor = "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))";
export const tabPressedIndicatorColor = "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))";
export const tabFocusedBorderRadius = "12px";
export const tabFocusedBorderWidth = "1px";
export const tabFocusedBorderColor = "var(--color-border-focus, #212121)";
export const tabDisabledOpacity = 0.48;
export const tabInactiveTextColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const tabInactiveIconColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const tabInactiveHoverIndicatorColor = "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))";
export const tabInactivePressedIndicatorColor = "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))";
export const tabActiveTextColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const tabActiveIconColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const tabActiveIndicatorColor = "var(--color-background-brand, #fdd835)";
export const tabIconOnlyWidth = "64px";
export const tabIconOnlyPaddingHorizontal = "12px";
