// @beeline/design-tokens/js/tokens/components/tab/tab.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	tabHeight: "48px",
	tabPaddingHorizontal: "24px",
	tabTextFontSize: "15px",
	tabTextFontWeight: 500,
	tabTextFontLineHeight: "20px",
	tabTextFontLetterSpacing: "0.2px",
	tabIconSize: "24px",
	tabIconSpacing: "4px",
	tabIndicatorHeight: "4px",
	tabIndicatorMinWidth: "40px",
	tabIndicatorBorderRadiusTopLeft: "4px",
	tabIndicatorBorderRadiusTopRight: "4px",
	tabHoverIndicatorColor: "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))",
	tabPressedIndicatorColor: "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))",
	tabFocusedBorderRadius: "12px",
	tabFocusedBorderWidth: "1px",
	tabFocusedBorderColor: "var(--color-border-focus, #212121)",
	tabDisabledOpacity: 0.48,
	tabInactiveTextColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	tabInactiveIconColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	tabInactiveHoverIndicatorColor: "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))",
	tabInactivePressedIndicatorColor: "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))",
	tabActiveTextColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	tabActiveIconColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	tabActiveIndicatorColor: "var(--color-background-brand, #fdd835)",
	tabIconOnlyWidth: "64px",
	tabIconOnlyPaddingHorizontal: "12px",
};
