// @beeline/design-system-react/extracted-styles/esm/components/Tree/TreeNode/TreeNode.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { Avatar } from '../../Avatar/Avatar.js';
import { Checkbox } from '../../Checkbox/Checkbox.js';
import { Icon } from '../../Icon/Icon.js';
import { IconButton } from '../../IconButton/IconButton.js';
import { Tooltip } from '../../Tooltip/Tooltip.js';
import { Typography } from '../../Typography/Typography.js';
import { Icons } from '@beeline/design-tokens/js/iconfont/icons';
import React__default, { useRef, useState, useEffect } from 'react';
import { classNames } from '../../../helpers/classnames.js';
import { CheckboxStatus } from '../Tree.types.js';
import { useTreeContext } from '../TreeContext.js';

const TreeNode = ({ title, children, avatar, icon, style, className = '', id, disabled = false, checked = false, expanded = false, custom, }) => {
    const { selectedId, toggleSingleSelect, nodes, registerNode, toggleExpansion, toggleCheckbox, treeParams, recursiveExpansion, deleteNode, } = useTreeContext();
    const multiselect = treeParams.multiselect;
    const isSelected = id === selectedId && !multiselect;
    const amount = treeParams.amount;
    const size = treeParams.size;
    const expandOnArrow = treeParams.expandOnArrow; // общие параметры дерева из контекста
    const titleRef = useRef(null); // реф для отслеживания обрезан текст или нет
    const [isWrapped, setIsWrapped] = useState(false); // стейт для вывода тултипа
    const [firstRender, setFirstRender] = useState(true);
    const childrenWithProps = React__default.Children.map(children, child => {
        // для дочерних элементов прокидываем disabled
        if (React__default.isValidElement(child) && child.type === TreeNode) {
            return React__default.cloneElement(child, {
                disabled: disabled,
                ...child.props,
            });
        }
        return child;
    });
    useEffect(() => {
        if (!nodes[id]) {
            setFirstRender(true); // если нода до этого не была в контексте, т. е. отрендерилась в первый раз, меняем этот пропс
        }
        const node = {
            id: id,
            checked: checked ? CheckboxStatus.CHECKED : CheckboxStatus.UNCHECKED,
            disabled: disabled,
            expanded: firstRender ? expanded : nodes[id]?.expanded, // поменять тут если не нужно хранить состояние свернутости\развернутости вложенных элементов после закрытия родительского
            children: children
                ? React__default.Children.map(children, child => {
                    // собираем в массив все идшники дочерних компонентов
                    if (React__default.isValidElement(child) && child.type === TreeNode && child.props.id) {
                        return child.props.id;
                    }
                }).filter(Boolean)
                : [],
        };
        new Promise((resolve, reject) => {
            registerNode(node);
            resolve();
        })
            .then(() => {
            if (!firstRender) {
                return;
            }
            if (checked) {
                // если компонент рендерится в первый раз, то применяем параметры с пропсов
                recursiveExpansion(id); // тут мы считаем, что если элемент выбран то это значит что надо так же развернуть все элементы до него
            }
            else if (expanded) {
                recursiveExpansion(id); // поменять тут на toggleExpansion если рекурсивное раскрытие не требуется
            }
        })
            .finally(() => {
            setFirstRender(false);
        });
        return () => {
            deleteNode(id);
        };
    }, [children, checked]);
    // useEffect(() => {
    // if (!firstRender) {
    // 	return;
    // }
    // if (checked) {
    // 	// если компонент рендерится в первый раз, то применяем параметры с пропсов
    // 	recursiveExpansion(id); // тут мы считаем, что если элемент выбран то это значит что надо так же развернуть все элементы до него
    // } else if (expanded) {
    // 	recursiveExpansion(id); // поменять тут на toggleExpansion если рекурсивное раскрытие не требуется
    // }
    // }, [firstRender]);
    useEffect(() => {
        const checkWrap = () => {
            // проверяем обрезан ли текст
            setIsWrapped(titleRef.current.scrollWidth > titleRef.current.clientWidth);
        };
        checkWrap();
        window.addEventListener('resize', checkWrap); // на каждый ресайз будем проверять обрезан он или нет
        return () => {
            window.removeEventListener('resize', checkWrap);
        };
    }, [title]);
    const isFinalSingleselect = !multiselect && !React__default.Children.count(children) && !disabled && !className.includes('tree_title');
    const handleNodeClick = (event) => {
        if (
        // обрабатываем клик на ноду, в зависимости от режима мультиселетк или синглселетк
        isFinalSingleselect) {
            toggleSingleSelect(id);
        }
        if (!expandOnArrow && React__default.Children.count(children)) {
            // если нужно разворачивать ноду по клику не только на стрелку
            toggleExpansion(id, event);
        }
    };
    return (jsxs(Fragment, { children: [jsxs("li", { role: "treeitem", "data-testid": "TreeNode", className: classNames(`treenode-${size}`, Object.keys(children || {}).length === 0 && 'treenode-last', className, disabled && 'treenode-disable', isSelected && 'treenode-selected'), "aria-expanded": expanded, "aria-selected": isSelected, tabIndex: 0, onClick: disabled ? undefined : event => handleNodeClick(event), onKeyDown: disabled
                    ? undefined
                    : (event) => {
                        if (event.key === 'Enter' || event.key === ' ') {
                            event.preventDefault();
                            handleNodeClick();
                        }
                    }, style: style, children: [Object.keys(children || {}).length > 0 && (jsx(IconButton, { size: "large", onClick: (event) => {
                            event.stopPropagation();
                            toggleExpansion(id, event);
                        }, iconName: nodes[id]?.expanded ? Icons.NavArrowDown : Icons.NavArrowRight })), multiselect && !className.includes('tree_title') && (jsx("div", { role: "button", tabIndex: 0, onClick: (event) => event.stopPropagation(), onKeyDown: (event) => {
                            if (event.key === 'Enter' || event.key === ' ') {
                                event.preventDefault();
                                event.stopPropagation();
                            }
                        }, children: jsx(Checkbox, { onClick: () => {
                                toggleCheckbox(id);
                            }, onChange: () => { }, checked: !!(nodes[id]?.checked !== CheckboxStatus.UNCHECKED), disabled: disabled, type: nodes[id]?.checked === CheckboxStatus.PARTIALLY ? 'indeterminate' : 'checkbox' }) })), avatar && (jsx("div", { className: "treenode_aiwrapper", children: jsx(Avatar, { ...avatar, size: size === 'large' ? 'medium' : 'small', variant: 'circle', type: 'default' }) })), icon && (jsx("div", { className: "treenode_aiwrapper", children: jsx(Icon, { ...icon, size: size }) })), jsx(Tooltip, { title: title, disableHover: !isWrapped, children: jsx(Typography, { ref: titleRef, className: 'treenode_title', variant: 'body3', children: title }) }), jsxs("div", { className: "treenode_custom", role: "button", tabIndex: 0, onClick: (event) => event.stopPropagation(), onKeyDown: (event) => {
                            if (event.key === 'Enter' || event.key === ' ') {
                                event.preventDefault();
                                event.stopPropagation();
                            }
                        }, children: [custom, amount && Boolean(React__default.Children.count(children)) && (jsx("span", { className: "treenode_amount", children: React__default.Children.count(children) }))] })] }), jsx("div", { role: "group", className: classNames(`treenode_subnodes-${size}`, nodes[id]?.expanded && 'treenode_subnodes-expanded'), children: childrenWithProps })] }));
};

export { TreeNode };
