// @beeline/design-system-react/extracted-styles/esm/components/DatePicker/useDatePicker.js
import { useState, useRef, useEffect, useCallback } from 'react';
import { formatValue, getInputValue, isFilledDate, isRangeMode, ISO_FORMAT_PATTERN, EMPTY_DATE } from './utils.js';

const areDateValuesEqual = (a, b) => {
    if (typeof a === 'string' && typeof b === 'string') {
        return a === b;
    }
    if (Array.isArray(a) && Array.isArray(b)) {
        if (a.length !== b.length) {
            return false;
        }
        return a.every((item, index) => item === b[index]);
    }
    return false;
};
const useDatePicker = ({ value, minDate, maxDate, onChange, onOpen, onClose, onFocus, onSaveClick, mode, multiselect, disabledStartDate, disabledEndDate, prevValue, enableTimePicker, timeZone, datepickerRef, mobileView, dropdownClassName, onBlur, overlayScroll, shouldRenderOverlay, overlayClassName, alignDropDown, }) => {
    const [isOpen, setIsOpen] = useState(false);
    const isInternalChangeRef = useRef(false);
    const [inputValue, setInputValue] = useState(formatValue(value, enableTimePicker, timeZone, mode));
    useEffect(() => {
        const newInputValue = formatValue(value, enableTimePicker, timeZone, mode);
        if (isInternalChangeRef.current) {
            isInternalChangeRef.current = false;
            return;
        }
        setInputValue(prevInputValue => areDateValuesEqual(prevInputValue, newInputValue) ? prevInputValue : newInputValue);
    }, [value, enableTimePicker, mode, timeZone]);
    const open = useCallback(() => {
        setIsOpen(true);
        onOpen?.();
    }, [onOpen]);
    const close = useCallback(() => {
        setIsOpen(false);
        onClose?.();
    }, [onClose]);
    const toggle = useCallback(() => {
        setIsOpen(prev => !prev);
        isOpen ? onClose?.() : onOpen?.();
    }, [isOpen, onClose, onOpen]);
    const selectDate = useCallback((date) => {
        if (mode === 'single') {
            onChange(date);
            const updatedInputValue = getInputValue(date, enableTimePicker, timeZone);
            setInputValue(updatedInputValue);
        }
        if (mode === 'single' && !onSaveClick && !enableTimePicker) {
            close();
        }
        if (mode === 'range') {
            onChange(date);
            const updatedInputValue = getInputValue(date, enableTimePicker, timeZone);
            setInputValue(updatedInputValue);
        }
        const isFilledValue = isFilledDate(date, {
            pattern: ISO_FORMAT_PATTERN,
        });
        if (mode === 'range' && isFilledValue && !onSaveClick && !enableTimePicker) {
            close();
        }
        if (mode === 'rangeSplit' && multiselect) {
            onChange(date);
            return;
        }
        if (mode === 'rangeSplit' && isRangeMode(date) && isRangeMode(prevValue)) {
            const [startDate, endDate] = date;
            const setDate = (startDate, endDate) => {
                setInputValue([
                    getInputValue(startDate, enableTimePicker, timeZone),
                    getInputValue(endDate, enableTimePicker, timeZone),
                ]);
                onChange([startDate, endDate]);
            };
            if (disabledStartDate) {
                const [prevStartDate] = prevValue || date;
                setDate(prevStartDate, isFilledValue ? endDate : startDate);
                close();
                return;
            }
            if (disabledEndDate) {
                const [, prevEndDate] = prevValue || date;
                setDate(startDate, prevEndDate);
                close();
                return;
            }
            if (isFilledValue) {
                setDate(startDate, endDate);
                close();
            }
            else {
                setDate(startDate, EMPTY_DATE);
            }
        }
    }, [close, disabledEndDate, disabledStartDate, enableTimePicker, mode, multiselect, onChange, onSaveClick, prevValue]);
    const clearDate = useCallback(() => {
        // setValue(isRangeMode(value) ? ['', ''] : '');
        // onChange?.(isRangeMode(value) ? ['', ''] : '');
    }, [onChange]);
    const setMinDate = useCallback((date) => {
        // Реализуйте установку минимальной даты
    }, []);
    const setMaxDate = useCallback((date) => {
        // Реализуйте установку максимальной даты
    }, []);
    const validateDate = useCallback((date) => 
    // Реализуйте валидацию даты
    true, []);
    const formatDate = useCallback((date) => getInputValue(date, enableTimePicker, timeZone), [enableTimePicker, timeZone]);
    const handleChange = (value) => {
        // Устанавливаем флаг, что это внутреннее изменение
        isInternalChangeRef.current = true;
        onChange?.(value);
    };
    const handleInputFocus = (event) => {
        if (!isOpen) {
            open();
        }
        onFocus?.(event);
    };
    const dropdownMenuProps = {
        targetElement: datepickerRef,
        align: alignDropDown,
        // клик вне дропдауна должен закрыть только в десктопном варианте
        onOutsideClick: mobileView ? null : close,
        isOpen,
        handleSetOpen: setIsOpen,
        className: dropdownClassName,
        widthMode: 'widthContent',
        onBlur,
        verticalOffset: 1,
        overlayScroll,
        shouldRenderOverlay,
        overlayClassName,
    };
    const handleInputBlur = (event) => {
        onBlur?.(event);
        if (mode === 'range') {
            onClose?.();
        }
    };
    return {
        isOpen,
        inputValue,
        open,
        close,
        toggle,
        selectDate,
        validateDate,
        formatDate,
        clearDate,
        setMinDate,
        setMaxDate,
        setIsOpen,
        setInputValue,
        handleChange,
        handleInputFocus,
        handleInputBlur,
        dropdownMenuProps,
    };
};

export { useDatePicker };
