// @beeline/design-tokens/js/tokens/components/counter/counter.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const counterHeight = "16px";
export const counterMinWidth = "16px";
export const counterSmallHeight = "16px";
export const counterSmallMinWidth = "16px";
export const counterSmallBorderRadius = "8px";
export const counterMediumHeight = "20px";
export const counterMediumMinWidth = "20px";
export const counterMediumBorderRadius = "10px";
export const counterPaddingHorizontal = "4px";
export const counterBorderRadius = "8px";
export const counterBackgroundColor = "var(--color-background-brand, #fdd835)";
export const counterTextColor = "var(--color-text-black-active, rgba(9, 11, 22, 0.94))";
export const counterTextFontSize = "10px";
export const counterTextFontWeight = 700;
export const counterTextFontLineHeight = "16px";
export const counterTextFontLetterSpacing = "1.2px";
export const counterErrorBackgroundColor = "var(--color-status-error, #ff5555)";
export const counterErrorTextColor = "var(--color-text-active-inverse, #ffffff)";
