// @beeline/design-system-react/extracted-styles/esm/components/Preview/components/PDFPreview/PDFPreview.js
import { jsx } from 'react/jsx-runtime';
import { Progress } from '../../../Progress/Progress.js';
import { useState, useRef, useEffect } from 'react';
import '../../../../node_modules/react-pdf/dist/index.js';
import Document from '../../../../node_modules/react-pdf/dist/Document.js';
import Page from '../../../../node_modules/react-pdf/dist/Page.js';
import { GlobalWorkerOptions } from '../../../../node_modules/pdfjs-dist/build/pdf.js';

GlobalWorkerOptions.workerSrc = new URL('pdfjs-dist/build/pdf.worker.min.mjs', import.meta.url).toString();
const PDF_INITIAL_SCALE_FACTOR = 0.85;
const PDFPreview = ({ value, scale, onInitialScale, rotation, activePage }) => {
    const [totalPages, setTotalPages] = useState(0);
    const [isReady, setIsReady] = useState(false);
    const containerRef = useRef(null);
    const documentRef = useRef(null);
    const pagesRef = useRef([]);
    const [pageDimensions, setPageDimensions] = useState({ width: 0, height: 0 });
    const initialScaleCalculated = useRef(false);
    const initialScaleRef = useRef(null);
    const isDragging = useRef(false);
    const dragStart = useRef({ x: 0, y: 0, scrollTop: 0, translateX: 0 });
    const translateX = useRef(0);
    const getContentWidth = () => {
        if (!documentRef.current || !containerRef.current)
            return 0;
        return documentRef.current.scrollWidth;
    };
    const clampTranslateX = (tx) => {
        if (!containerRef.current)
            return 0;
        const contentWidth = getContentWidth();
        const containerWidth = containerRef.current.clientWidth;
        if (contentWidth <= containerWidth)
            return 0;
        const maxOffset = (contentWidth - containerWidth) / 2 + 120;
        return Math.max(-maxOffset, Math.min(maxOffset, tx));
    };
    const applyTranslateX = (tx) => {
        if (!documentRef.current)
            return;
        documentRef.current.style.transform = `translateX(${tx}px)`;
    };
    const handleMouseDown = (e) => {
        if (!containerRef.current || e.button !== 0)
            return;
        e.preventDefault();
        isDragging.current = true;
        dragStart.current = {
            x: e.clientX,
            y: e.clientY,
            scrollTop: containerRef.current.scrollTop,
            translateX: translateX.current,
        };
        containerRef.current.classList.add('dsb_preview__content-pdf-wrapper--dragging');
    };
    const scrollIntoPage = (pageNumber) => {
        if (pagesRef.current[pageNumber] && containerRef.current) {
            const pageElement = pagesRef.current[pageNumber];
            const container = containerRef.current;
            const pageRect = pageElement.getBoundingClientRect();
            const containerRect = container.getBoundingClientRect();
            const scrollTop = pageElement.offsetTop - containerRect.height / 2 + pageRect.height / 2 - 64;
            container.scrollTo({
                top: scrollTop,
                behavior: 'smooth',
            });
        }
    };
    const calculateInitialScale = () => {
        if (!containerRef.current || !documentRef.current || pageDimensions.width === 0 || pageDimensions.height === 0) {
            return;
        }
        const containerHeight = containerRef.current.clientHeight;
        let containerWidth = containerRef.current.clientWidth;
        // Учитываем перекрытие галереей слева и контролами справа
        const containerRect = containerRef.current.getBoundingClientRect();
        const previewEl = containerRef.current.closest('.dsb_preview');
        const galleryEl = previewEl?.querySelector('.dsb_preview__gallery-pdf, .dsb_preview__gallery-word');
        const controlsEl = previewEl?.querySelector('.dsb_preview__controls');
        if (galleryEl) {
            const overlap = galleryEl.getBoundingClientRect().right - containerRect.left;
            if (overlap > 0)
                containerWidth -= overlap;
        }
        if (controlsEl) {
            const overlap = containerRect.right - controlsEl.getBoundingClientRect().left;
            if (overlap > 0)
                containerWidth -= overlap;
        }
        const pageWidth = pageDimensions.width;
        const pageHeight = pageDimensions.height;
        const widthRatio = containerWidth / pageWidth;
        const heightRatio = containerHeight / pageHeight;
        const newRatio = Math.min(widthRatio, heightRatio) * PDF_INITIAL_SCALE_FACTOR;
        if (!initialScaleCalculated.current || Math.abs(newRatio - scale) > 0.001) {
            initialScaleRef.current = newRatio;
            onInitialScale?.(newRatio);
            initialScaleCalculated.current = true;
        }
    };
    const handleDocumentLoadSuccess = (document) => {
        setTotalPages(document.numPages);
        setIsReady(false);
        initialScaleRef.current = null;
        initialScaleCalculated.current = false;
        // Загружаем viewport всех страниц, чтобы найти максимальные размеры
        const pagePromises = Array.from({ length: document.numPages }, (_, i) => document.getPage(i + 1));
        void Promise.all(pagePromises).then(pages => {
            let maxWidth = 0;
            let maxHeight = 0;
            for (const page of pages) {
                const viewport = page.getViewport({ scale: 1 });
                maxWidth = Math.max(maxWidth, viewport.width);
                maxHeight = Math.max(maxHeight, viewport.height);
            }
            setPageDimensions({ width: maxWidth, height: maxHeight });
            calculateInitialScale();
        });
    };
    useEffect(() => {
        calculateInitialScale();
    }, [pageDimensions, containerRef.current]);
    // Показываем контент когда scale соответствует вычисленному initialScale
    useEffect(() => {
        if (initialScaleRef.current !== null && Math.abs(scale - initialScaleRef.current) < 0.01) {
            // Ждём применения scale перед показом
            requestAnimationFrame(() => {
                requestAnimationFrame(() => {
                    setIsReady(true);
                });
            });
        }
    }, [scale]);
    useEffect(() => {
        if (activePage > 0) {
            scrollIntoPage(activePage);
        }
    }, [activePage, scale, totalPages]);
    // Сброс translateX при изменении масштаба
    useEffect(() => {
        translateX.current = 0;
        applyTranslateX(0);
    }, [scale]);
    useEffect(() => {
        const onMouseMove = (e) => {
            if (!isDragging.current || !containerRef.current)
                return;
            e.preventDefault();
            const dx = e.clientX - dragStart.current.x;
            const dy = e.clientY - dragStart.current.y;
            // Вертикаль — через scroll
            containerRef.current.scrollTop = dragStart.current.scrollTop - dy;
            // Горизонталь — через CSS transform
            const newTx = clampTranslateX(dragStart.current.translateX + dx);
            translateX.current = newTx;
            applyTranslateX(newTx);
        };
        const onMouseUp = () => {
            if (!isDragging.current)
                return;
            isDragging.current = false;
            containerRef.current?.classList.remove('dsb_preview__content-pdf-wrapper--dragging');
        };
        window.addEventListener('mousemove', onMouseMove);
        window.addEventListener('mouseup', onMouseUp);
        return () => {
            window.removeEventListener('mousemove', onMouseMove);
            window.removeEventListener('mouseup', onMouseUp);
        };
    }, []);
    const setPageRef = (ref, index) => {
        if (ref) {
            pagesRef.current[index] = ref;
        }
    };
    const renderPages = () => Array.from({ length: totalPages }, (_, i) => (jsx(Page, { pageNumber: i + 1, scale: scale, loading: ' ', className: "dsb_preview__content-pdf__page", renderAnnotationLayer: false, renderTextLayer: false, inputRef: ref => setPageRef(ref, i) }, `page_${i + 1}`)));
    return (jsx("div", { className: "dsb_preview__content-pdf-wrapper", ref: containerRef, "aria-busy": !isReady, onMouseDown: handleMouseDown, children: jsx("div", { className: "dsb_preview__content-pdf", ref: documentRef, style: {
                opacity: isReady ? 1 : 0,
                visibility: isReady ? 'visible' : 'hidden',
                transition: 'opacity 0.15s ease-out',
            }, children: jsx(Document, { rotate: rotation, loading: jsx(Loader, {}), file: value.src, onLoadSuccess: handleDocumentLoadSuccess, children: renderPages() }) }) }));
};
const Loader = () => (jsx("div", { className: "dsb_preview__content-pdf__progress", role: "status", "aria-live": "polite", children: jsx(Progress, { shape: "circle", cycled: true }) }));

export { PDFPreview };
