// @beeline/design-tokens/js/tokens/components/navigationDrawer/navigationDrawer.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const navigationDrawerBackgroundColor = "var(--color-background-base, #ffffff)";
export const navigationDrawerPaddingVertical = "24px";
export const navigationDrawerExpandedWidth = "256px";
export const navigationDrawerExpandedTriggerButtonPaddingLeft = "32px";
export const navigationDrawerExpandedItemContainerPaddingRight = "16px";
export const navigationDrawerExpandedItemParentPaddingLeft = "32px";
export const navigationDrawerExpandedItemParentPaddingRight = "16px";
export const navigationDrawerFoldedWidth = "56px";
export const navigationDrawerFoldedTriggerButtonPaddingLeft = "16px";
export const navigationDrawerFoldedItemContainerPaddingRight = "0px";
export const navigationDrawerFoldedItemPaddingVertical = "12px";
export const navigationDrawerFoldedItemPaddingHorizontal = "16px";
export const navigationDrawerIndicatorWidth = "4px";
export const navigationDrawerIndicatorColor = "var(--color-background-brand, #fdd835)";
export const navigationDrawerIndicatorBorderRadiusTopright = "4px";
export const navigationDrawerIndicatorBorderRadiusBottomright = "4px";
export const navigationDrawerDividerContainerPadding = "32px";
export const navigationDrawerTitleHeight = "40px";
export const navigationDrawerTitlePaddingHorizontal = "32px";
export const navigationDrawerTitlePaddingVertical = "12px";
export const navigationDrawerTitleTextColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const navigationDrawerTitleTextFontSize = "10px";
export const navigationDrawerTitleTextFontWeight = 700;
export const navigationDrawerTitleTextFontLineHeight = "16px";
export const navigationDrawerTitleTextFontLetterSpacing = "1.2px";
export const navigationDrawerTriggerButtonPaddingBottom = "24px";
export const navigationDrawerListGap = "4px"; // Вертикальный отступ между элементами списка, а также между заголовком, дивайдером и элементами списка.
export const navigationDrawerListPaddingBottom = "24px";
export const navigationDrawerItemHeight = "48px";
export const navigationDrawerItemParentBorderRadiusTopright = "12px";
export const navigationDrawerItemParentBorderRadiusBottomright = "12px";
export const navigationDrawerItemChildBorderRadius = "12px";
export const navigationDrawerItemChildMarginLeft = "16px";
export const navigationDrawerItemChildPaddingLeft = "56px";
export const navigationDrawerItemChildPaddingRight = "16px";
export const navigationDrawerItemChildActivatedBackgroundColor = "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))";
export const navigationDrawerItemTextColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const navigationDrawerItemTextFontSize = "15px";
export const navigationDrawerItemTextFontWeight = 400;
export const navigationDrawerItemTextFontLineHeight = "18px";
export const navigationDrawerItemTextFontLetterSpacing = "0.2px";
export const navigationDrawerItemIconSize = "24px";
export const navigationDrawerItemIconSpacing = "16px";
export const navigationDrawerItemIconColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const navigationDrawerItemArrowSpacing = "8px";
export const navigationDrawerItemArrowColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const navigationDrawerItemIndicatorWidth = "4px";
export const navigationDrawerItemIndicatorBorderRadiusTopright = "4px";
export const navigationDrawerItemIndicatorBorderRadiusBottomright = "4px";
export const navigationDrawerItemHoverBackgroundColor = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const navigationDrawerItemFocusedBorderColor = "var(--color-border-focus, #212121)";
export const navigationDrawerItemPressedBackgroundColor = "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))";
export const navigationDrawerItemActivatedTextColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const navigationDrawerItemActivatedTextFontSize = "15px";
export const navigationDrawerItemActivatedTextFontWeight = 500;
export const navigationDrawerItemActivatedTextFontLineHeight = "20px";
export const navigationDrawerItemActivatedTextFontLetterSpacing = "0.2px";
export const navigationDrawerItemActivatedIconColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const navigationDrawerItemActivatedArrowColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const navigationDrawerItemActivatedIndicatorColor = "var(--color-background-brand, #fdd835)";
export const navigationDrawerItemDisabledOpacity = 0.48;
