// @beeline/design-tokens/js/tokens/components/navigationDrawer/navigationDrawer.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	navigationDrawerBackgroundColor: "var(--color-background-base, #ffffff)",
	navigationDrawerPaddingVertical: "24px",
	navigationDrawerExpandedWidth: "256px",
	navigationDrawerExpandedTriggerButtonPaddingLeft: "32px",
	navigationDrawerExpandedItemContainerPaddingRight: "16px",
	navigationDrawerExpandedItemParentPaddingLeft: "32px",
	navigationDrawerExpandedItemParentPaddingRight: "16px",
	navigationDrawerFoldedWidth: "56px",
	navigationDrawerFoldedTriggerButtonPaddingLeft: "16px",
	navigationDrawerFoldedItemContainerPaddingRight: "0px",
	navigationDrawerFoldedItemPaddingVertical: "12px",
	navigationDrawerFoldedItemPaddingHorizontal: "16px",
	navigationDrawerIndicatorWidth: "4px",
	navigationDrawerIndicatorColor: "var(--color-background-brand, #fdd835)",
	navigationDrawerIndicatorBorderRadiusTopright: "4px",
	navigationDrawerIndicatorBorderRadiusBottomright: "4px",
	navigationDrawerDividerContainerPadding: "32px",
	navigationDrawerTitleHeight: "40px",
	navigationDrawerTitlePaddingHorizontal: "32px",
	navigationDrawerTitlePaddingVertical: "12px",
	navigationDrawerTitleTextColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	navigationDrawerTitleTextFontSize: "10px",
	navigationDrawerTitleTextFontWeight: 700,
	navigationDrawerTitleTextFontLineHeight: "16px",
	navigationDrawerTitleTextFontLetterSpacing: "1.2px",
	navigationDrawerTriggerButtonPaddingBottom: "24px",
	navigationDrawerListGap: "4px", // Вертикальный отступ между элементами списка, а также между заголовком, дивайдером и элементами списка.
	navigationDrawerListPaddingBottom: "24px",
	navigationDrawerItemHeight: "48px",
	navigationDrawerItemParentBorderRadiusTopright: "12px",
	navigationDrawerItemParentBorderRadiusBottomright: "12px",
	navigationDrawerItemChildBorderRadius: "12px",
	navigationDrawerItemChildMarginLeft: "16px",
	navigationDrawerItemChildPaddingLeft: "56px",
	navigationDrawerItemChildPaddingRight: "16px",
	navigationDrawerItemChildActivatedBackgroundColor: "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))",
	navigationDrawerItemTextColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	navigationDrawerItemTextFontSize: "15px",
	navigationDrawerItemTextFontWeight: 400,
	navigationDrawerItemTextFontLineHeight: "18px",
	navigationDrawerItemTextFontLetterSpacing: "0.2px",
	navigationDrawerItemIconSize: "24px",
	navigationDrawerItemIconSpacing: "16px",
	navigationDrawerItemIconColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	navigationDrawerItemArrowSpacing: "8px",
	navigationDrawerItemArrowColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	navigationDrawerItemIndicatorWidth: "4px",
	navigationDrawerItemIndicatorBorderRadiusTopright: "4px",
	navigationDrawerItemIndicatorBorderRadiusBottomright: "4px",
	navigationDrawerItemHoverBackgroundColor: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	navigationDrawerItemFocusedBorderColor: "var(--color-border-focus, #212121)",
	navigationDrawerItemPressedBackgroundColor: "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))",
	navigationDrawerItemActivatedTextColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	navigationDrawerItemActivatedTextFontSize: "15px",
	navigationDrawerItemActivatedTextFontWeight: 500,
	navigationDrawerItemActivatedTextFontLineHeight: "20px",
	navigationDrawerItemActivatedTextFontLetterSpacing: "0.2px",
	navigationDrawerItemActivatedIconColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	navigationDrawerItemActivatedArrowColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	navigationDrawerItemActivatedIndicatorColor: "var(--color-background-brand, #fdd835)",
	navigationDrawerItemDisabledOpacity: 0.48,
};
