// @beeline/design-system-react/build/components/Autocomplete/Autocomplete.js
import { jsxs, jsx } from 'react/jsx-runtime';
import './Autocomplete2.js';
import { Button } from '../Button/Button.js';
import '../Button/ProgressButton/ProgressButton.js';
import { Chip } from '../Chip/Chip.js';
import { MaskField } from '../MaskField/MaskField.js';
import { Modal } from '../Modal/Modal.js';
import { NewDropdownMenu } from '../NewDropdownMenu/NewDropdownMenu.js';
import { Typography } from '../Typography/Typography.js';
import '../../helpers/popupsContext.js';
import { useDeviceSize } from '../../helpers/use-device-size.js';
import { classNames } from '../../helpers/classnames.js';
import { forwardRef, useImperativeHandle } from 'react';
import { Search } from '../Search/Search.js';
import { TextField } from '../TextField/TextField.js';
import { useAutocomplete } from './useAutocomplete.js';

const Autocomplete = forwardRef(({ size, value, onChange, defaultValue, dropdownClassName, dropdownElementID, onClose, onOpen, open, children, onFocus, onBlur, onSubmit, onClick, options, fullWidth, className, disabled, defaultGroupID, renderValue, onInputChange, onInputClear, onFilterChange, filterItems, selectedFilter, enableGrouping = false, makeOption, label, type, showTooltip = false, noOptionsText, noOptionsContents, loading, loadingText, loadingContent, hideDropDown = false, dataTestId = 'Autocomplete', enableKeyboardNavigation = false, maskConfig, overlayScroll = true, shouldRenderOverlay, overlayClassName, alignDropDown, mobileTitle = 'Поиск', mobileCaption = 'Результаты поиска', applicationRootElementID = 'root', classNameMobileModal, ...props }, ref) => {
    const { isExtraSmallDevice } = useDeviceSize();
    const { isOpen, innerRef, handleClose, setOpen, handleMaskInputOnChange, handleInputClick, handleInputBlur, handleInputFocus, inputValue, handleInputOnChange, handleInputOnClear, rootClassName, handleKeyDown, isSearch, optionsComponents, } = useAutocomplete({
        type,
        value,
        open,
        renderValue,
        enableKeyboardNavigation,
        onOpen,
        onClose,
        onChange,
        onFocus,
        onBlur,
        onInputChange,
        onInputClear,
        showTooltip,
        loading,
        loadingText,
        loadingContent,
        hideDropDown,
        options,
        noOptionsContents,
        noOptionsText,
        makeOption,
        className,
        isExtraSmallDevice,
    });
    useImperativeHandle(ref, () => innerRef.current);
    const dropdownMenuProps = {
        targetElement: innerRef,
        align: alignDropDown,
        verticalOffset: 1,
        applicationRootElementID,
        dropdownElementID,
        className: dropdownClassName,
        isOpen,
        handleSetOpen: setOpen,
        overlayScroll,
        shouldRenderOverlay,
        overlayClassName,
    };
    const makeDropdownView = (mobileOptions = false) => maskConfig ? (jsx(MaskField, { ...props, ref: innerRef, label: label, size: size, fullWidth: fullWidth || mobileOptions, maskConfig: maskConfig, onChange: handleMaskInputOnChange, onBlur: handleInputBlur, onClick: handleInputClick, onFocus: handleInputFocus, disabled: disabled, value: inputValue, contentEditable: true, dataTestId: dataTestId + '-textField' })) : (jsx(TextField, { ...props, ref: innerRef, label: label, size: size, fullWidth: fullWidth || mobileOptions, onChange: handleInputOnChange, onBlur: handleInputBlur, onFocus: handleInputFocus, onClick: handleInputClick, disabled: disabled, value: inputValue, contentEditable: true, dataTestId: dataTestId + '-textField' }));
    const makeSearchView = (mobileOptions = false) => {
        const filterByGroupProps = enableGrouping && !mobileOptions && !isExtraSmallDevice ? { filterItems, selectedFilter, onFilterChange } : {};
        return (jsx(Search, { ...props, ...filterByGroupProps, ref: innerRef, size: size, fullWidth: fullWidth || mobileOptions, onBlur: handleInputBlur, onFocus: handleInputFocus, onChange: handleInputOnChange, onClear: handleInputOnClear, disabled: disabled, value: inputValue, contentEditable: false, dataTestId: dataTestId + '-search' }));
    };
    const makeMobileView = () => (jsx(Modal, { className: classNameMobileModal, applicationRootElement: applicationRootElementID, open: true, children: jsxs("div", { className: "dsb__autocomplete-mobile", children: [jsx(Typography, { variant: "subtitle1", children: mobileTitle }), jsxs("div", { className: "dsb__autocomplete-mobile__component", children: [isSearch ? makeSearchView(true) : makeDropdownView(true), jsx(Button, { variant: "plain", onClick: () => {
                                handleClose();
                            }, children: "\u0417\u0430\u043A\u0440\u044B\u0442\u044C" })] }), jsx("div", { className: "dsb__autocomplete-mobile__chips", children: filterItems?.map(item => (jsx(Chip, { id: String(item.id), label: item.value, active: selectedFilter === item, onClick: () => onFilterChange(item) }))) }), jsx(Typography, { className: "dsb__autocomplete-mobile__caption", variant: "caption", children: mobileCaption }), optionsComponents] }) }));
    const makeOptionsView = () => isExtraSmallDevice ? (makeMobileView()) : (jsx(NewDropdownMenu, { ...dropdownMenuProps, children: optionsComponents }));
    return (jsxs("div", { "data-testid": dataTestId, className: classNames(rootClassName), onKeyDown: enableKeyboardNavigation ? handleKeyDown : () => { }, children: [isSearch ? makeSearchView() : makeDropdownView(), isOpen && !hideDropDown && makeOptionsView()] }));
});

export { Autocomplete };
