// @beeline/design-tokens/js/tokens/components/search/search.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const searchBackgroundColor = "var(--color-control-background, rgba(25, 28, 52, 0.1))";
export const searchBorderWidth = "1px";
export const searchBorderRadius = "12px";
export const searchBorderColor = "transparent";
export const searchOptionsBackgroundColor = "var(--color-background-medium, #ffffff)";
export const searchOptionsShadow = "0px 4px 30px rgba(0, 0, 0, 0.1), 0px 0px 10px rgba(0, 0, 0, 0.1)";
export const searchOptionsPaddingHorizontal = "16px";
export const searchFieldPaddingHorizontal = "16px";
export const searchFieldIconbuttonSpacing = "16px";
export const searchPlaceholderColor = "var(--color-text-disabled, rgba(25, 28, 52, 0.48))";
export const searchTextColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const searchTextFontSize = "17px";
export const searchTextFontWeight = 400;
export const searchTextFontLineHeight = "22px";
export const searchTextFontLetterSpacing = "0.2px";
export const searchCategoryPaddingHorizontal = "16px";
export const searchCategoryIconSpacing = "16px";
export const searchDividerHeight = "24px";
export const searchSmallHeight = "40px";
export const searchMediumHeight = "48px";
export const searchHoverBorderColor = "var(--color-border-focus, #212121)";
export const searchFocusedBackgroundColor = "transparent";
export const searchFocusedBorderColor = "var(--color-border-focus, #212121)";
