// @beeline/design-tokens/js/tokens/components/search/search.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	searchBackgroundColor: "var(--color-control-background, rgba(25, 28, 52, 0.1))",
	searchBorderWidth: "1px",
	searchBorderRadius: "12px",
	searchBorderColor: "transparent",
	searchOptionsBackgroundColor: "var(--color-background-medium, #ffffff)",
	searchOptionsShadow: "0px 4px 30px rgba(0, 0, 0, 0.1), 0px 0px 10px rgba(0, 0, 0, 0.1)",
	searchOptionsPaddingHorizontal: "16px",
	searchFieldPaddingHorizontal: "16px",
	searchFieldIconbuttonSpacing: "16px",
	searchPlaceholderColor: "var(--color-text-disabled, rgba(25, 28, 52, 0.48))",
	searchTextColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	searchTextFontSize: "17px",
	searchTextFontWeight: 400,
	searchTextFontLineHeight: "22px",
	searchTextFontLetterSpacing: "0.2px",
	searchCategoryPaddingHorizontal: "16px",
	searchCategoryIconSpacing: "16px",
	searchDividerHeight: "24px",
	searchSmallHeight: "40px",
	searchMediumHeight: "48px",
	searchHoverBorderColor: "var(--color-border-focus, #212121)",
	searchFocusedBackgroundColor: "transparent",
	searchFocusedBorderColor: "var(--color-border-focus, #212121)",
};
