// @beeline/design-tokens/js/tokens/components/select/select.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	selectPaddingVertical: "8px",
	selectBorderRadius: "12px",
	selectBackgroundColor: "var(--color-background-medium, #ffffff)",
	selectShadow: "0px 4px 30px rgba(0, 0, 0, 0.1), 0px 0px 10px rgba(0, 0, 0, 0.1)",
	selectItemPaddingRight: "16px",
	selectItemSingleRootPaddingLeft: "16px",
	selectItemSingleNestedPaddingLeft: "24px",
	selectItemMultiPaddingLeft: "16px",
	selectItemMultiCheckboxSpacing: "16px",
	selectItemEnabledBackgroundColor: "var(--color-background-medium, #ffffff)",
	selectItemFocusedBackgroundColor: "var(--color-background-base-focused, rgba(25, 28, 52, 0.12))",
	selectItemPressedBackgroundColor: "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))",
	selectItemHoverBackgroundColor: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	selectItemSelectedBackgroundColor: "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))",
	selectItemDisabledBackgroundColor: "var(--color-background-medium, #ffffff)",
	selectItemDisabledOpacity: 0.48,
	selectTitleHeight: "46px",
	selectTitlePaddingHorizontal: "16px",
	selectTitlePaddingVertical: "12px",
	selectTitleTextColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	selectTitleTextFontSize: "15px",
	selectTitleTextFontWeight: 500,
	selectTitleTextFontLineHeight: "20px",
	selectTitleTextFontLetterSpacing: "0.2px",
	selectSingleItemRootPaddingLeft: "16px",
	selectSingleItemNestedPaddingLeft: "24px",
	selectMultiItemPaddingLeft: "16px",
	selectMultiItemCheckboxSpacing: "16px",
};
