// @beeline/design-tokens/js/tokens/components/select/select.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const selectPaddingVertical = "8px";
export const selectBorderRadius = "12px";
export const selectBackgroundColor = "var(--color-background-medium, #ffffff)";
export const selectShadow = "0px 4px 30px rgba(0, 0, 0, 0.1), 0px 0px 10px rgba(0, 0, 0, 0.1)";
export const selectItemPaddingRight = "16px";
export const selectItemSingleRootPaddingLeft = "16px";
export const selectItemSingleNestedPaddingLeft = "24px";
export const selectItemMultiPaddingLeft = "16px";
export const selectItemMultiCheckboxSpacing = "16px";
export const selectItemEnabledBackgroundColor = "var(--color-background-medium, #ffffff)";
export const selectItemFocusedBackgroundColor = "var(--color-background-base-focused, rgba(25, 28, 52, 0.12))";
export const selectItemPressedBackgroundColor = "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))";
export const selectItemHoverBackgroundColor = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const selectItemSelectedBackgroundColor = "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))";
export const selectItemDisabledBackgroundColor = "var(--color-background-medium, #ffffff)";
export const selectItemDisabledOpacity = 0.48;
export const selectTitleHeight = "46px";
export const selectTitlePaddingHorizontal = "16px";
export const selectTitlePaddingVertical = "12px";
export const selectTitleTextColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const selectTitleTextFontSize = "15px";
export const selectTitleTextFontWeight = 500;
export const selectTitleTextFontLineHeight = "20px";
export const selectTitleTextFontLetterSpacing = "0.2px";
export const selectSingleItemRootPaddingLeft = "16px";
export const selectSingleItemNestedPaddingLeft = "24px";
export const selectMultiItemPaddingLeft = "16px";
export const selectMultiItemCheckboxSpacing = "16px";
