// @beeline/design-system-react/build/components/InlineEdit/InlineEditModal/InlineEditModal.js
import { jsx } from 'react/jsx-runtime';
import './InlineEditModal2.js';
import React__default, { useState, useCallback, useLayoutEffect } from 'react';
import ReactDOM__default from 'react-dom';

const PopoverContext = React__default.createContext({
    applicationRootElementId: 'root',
});
const InlineEditModal = ({ contentRect, children }) => {
    const [coords, setCoords] = useState({
        x: 0,
        y: 0,
    });
    const { applicationRootElementId } = React__default.useContext(PopoverContext);
    const MAX_HEIGHT = 56;
    const getXCoords = useCallback(() => {
        // Because property 'ref' does not exist on type ReactElement by default.
        const modalObj = children.ref.current;
        const bodyCoords = document.querySelector('body')?.getBoundingClientRect();
        if (contentRect && bodyCoords && modalObj) {
            const modalCoords = modalObj.getBoundingClientRect();
            /**
        Если по ширине вылазим за правый край, смещаемся влево на ширину формы
      */
            return bodyCoords.width - contentRect?.x < modalCoords.width
                ? bodyCoords.width - modalCoords.width
                : contentRect?.x;
        }
        return 0;
    }, [children, contentRect]);
    const getYCoords = useCallback(() => {
        if (contentRect) {
            return contentRect?.y - (MAX_HEIGHT / 2 - contentRect?.height / 2);
        }
        return 0;
    }, [contentRect]);
    useLayoutEffect(() => {
        setCoords({ x: getXCoords(), y: getYCoords() });
    }, [contentRect, getXCoords, getYCoords]);
    const mount = document.querySelector('#' + applicationRootElementId);
    if (!mount) {
        throw new Error(`Application root element with ID "${applicationRootElementId}" not found`);
    }
    const modal = (jsx("div", { "data-testid": "InlineEditModal", className: "dsb_inline-edit-modal", children: jsx("div", { className: "dsb_inline-edit-modal_positioner", style: {
                position: 'absolute',
                left: `${coords.x}px`,
                top: `${coords.y}px`,
            }, children: children }) }));
    return ReactDOM__default.createPortal(modal, mount);
};

export { InlineEditModal, PopoverContext };
