// @beeline/design-system-react/extracted-styles/esm/components/Typography/Typography.js
import { jsx } from 'react/jsx-runtime';
import { classNames } from '../../helpers/classnames.js';
import React__default, { forwardRef, createElement } from 'react';

const Typography = forwardRef(({ variant = 'body1', className, ellipsis = false, inactive = false, dataTestId = 'Typography', ...rest }, ref) => {
    const tagMap = {
        h1: 'h1',
        h2: 'h2',
        h3: 'h3',
        h4: 'h4',
        h5: 'h5',
        h6: 'h6',
        body1: 'p',
        body2: 'p',
        body3: 'p',
        subtitle1: 'span',
        subtitle2: 'span',
        subtitle3: 'span',
        caption: 'span',
        overline: 'span',
        productName: 'span',
        nativeLink: 'a',
    };
    if (variant === 'routedLink' && 'routingComponent' in rest) {
        const { routingComponent, linkState, ...restWithoutExtras } = rest;
        return React__default.cloneElement(routingComponent, {
            className: classNames('dsb_typography', `dsb_typography__${variant}`, ellipsis && 'dsb_typography_ellipsis', inactive && 'dsb_typography_inactive', className, linkState && `dsb_typography__${variant}-${linkState}`),
            ref,
            role: 'link',
            'aria-label': rest['aria-label'],
            ...restWithoutExtras,
        });
    }
    if (variant === 'contextualLink' && 'onHoverElement' in rest) {
        const { onHoverElement, linkState, ...restWithoutExtras } = rest;
        return React__default.cloneElement(onHoverElement, {
            children: (jsx("span", { className: classNames('dsb_typography', `dsb_typography__${variant}`, ellipsis && 'dsb_typography_ellipsis', inactive && 'dsb_typography_inactive', className, linkState && `dsb_typography__${variant}-${linkState}`), ref: ref, ...restWithoutExtras })),
        });
    }
    const getRole = variant => {
        if (variant.includes('h'))
            return 'heading';
        if (variant.includes('Link'))
            return 'link';
    };
    const { linkState, routingComponent: _routingComponent, onHoverElement: _onHoverElement, ...domProps } = rest;
    return createElement(tagMap[variant], {
        className: classNames('dsb_typography', `dsb_typography__${variant}`, ellipsis && 'dsb_typography_ellipsis', inactive && 'dsb_typography_inactive', className, linkState && `dsb_typography__${variant}-${linkState}`),
        ref,
        'data-testid': dataTestId,
        role: getRole(variant),
        'aria-level': domProps['aria-level'],
        ...domProps,
    });
});

export { Typography };
