// @beeline/design-tokens/js/tokens/components/pagination/pagination.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	paginationCellMinWidth: "40px",
	paginationCellHeight: "40px",
	paginationCellBorderRadius: "12px",
	paginationCellPadding: "8px",
	paginationCellColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	paginationCellFontSize: "17px",
	paginationCellFontWeight: 400,
	paginationCellFontLineHeight: "22px",
	paginationCellFontLetterSpacing: "0.2px",
	paginationCellIconSize: "24px",
	paginationCellMoreColor: "var(--color-text-disabled, rgba(25, 28, 52, 0.48))",
	paginationCellHoverBackgroundColor: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	paginationCellPressedBackgroundColor: "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))",
	paginationCellFocusedBorderColor: "var(--color-border-focused, #202123)",
	paginationCellFocusedBorderWidth: "1px",
	paginationCellSelectedColor: "var(--color-text-black-active, rgba(9, 11, 22, 0.94))",
	paginationCellSelectedBackgroundColor: "var(--color-background-brand, #fdd835)",
	paginationCellDisabledOpacity: 0.48,
	paginationGap: "8px",
};
