// @beeline/design-tokens/js/tokens/components/pagination/pagination.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const paginationCellMinWidth = "40px";
export const paginationCellHeight = "40px";
export const paginationCellBorderRadius = "12px";
export const paginationCellPadding = "8px";
export const paginationCellColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const paginationCellFontSize = "17px";
export const paginationCellFontWeight = 400;
export const paginationCellFontLineHeight = "22px";
export const paginationCellFontLetterSpacing = "0.2px";
export const paginationCellIconSize = "24px";
export const paginationCellMoreColor = "var(--color-text-disabled, rgba(25, 28, 52, 0.48))";
export const paginationCellHoverBackgroundColor = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const paginationCellPressedBackgroundColor = "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))";
export const paginationCellFocusedBorderColor = "var(--color-border-focused, #202123)";
export const paginationCellFocusedBorderWidth = "1px";
export const paginationCellSelectedColor = "var(--color-text-black-active, rgba(9, 11, 22, 0.94))";
export const paginationCellSelectedBackgroundColor = "var(--color-background-brand, #fdd835)";
export const paginationCellDisabledOpacity = 0.48;
export const paginationGap = "8px";
