// @beeline/design-system-react/build/components/NavigationDrawer/NavigationDrawer.js
import { jsxs, jsx, Fragment } from 'react/jsx-runtime';
import './NavigationDrawer2.js';
import { Divider } from '../Divider/Divider.js';
import { Icon } from '../Icon/Icon.js';
import { IconButton } from '../IconButton/IconButton.js';
import { Tooltip } from '../Tooltip/Tooltip.js';
import { Typography } from '../Typography/Typography.js';
import '../../helpers/popupsContext.js';
import { useDeviceSize } from '../../helpers/use-device-size.js';
import { classNames } from '../../helpers/classnames.js';
import { Icons } from '@beeline/design-tokens/js/iconfont/icons';
import { useState, useRef, useEffect, useCallback, useMemo } from 'react';
import { NavigationItemSkeleton } from './NavigationItemSkeleton.js';

const NavigationDrawer = ({ className, groups, isGroupDivider = true, isGroupTitle = true, active, bottom, onClickItem, isOpen = false, mobileOnOutsideClick, onOpen, onClose, disableMobileView = false, expandedItems: expandedItemsProp, onExpandedItemsChange, defaultExpandedItems = {}, }) => {
    // Определяем, является ли компонент контролируемым
    // Компонент считается контролируемым, если переданы onOpen или onClose колбэки
    const isControlled = onOpen !== undefined || onClose !== undefined;
    // Внутреннее состояние для неконтролируемого режима
    const [internalIsExpanded, setInternalIsExpanded] = useState(isOpen);
    // Используем внешнее состояние для контролируемого режима, внутреннее для неконтролируемого
    const isExpanded = isControlled ? isOpen : internalIsExpanded;
    // Функция для изменения состояния (работает только в неконтролируемом режиме)
    const setIsExpanded = (value) => {
        if (!isControlled) {
            setInternalIsExpanded(value);
        }
    };
    const [activeItem, setActiveItem] = useState(() => active ?? '');
    // Управление состоянием раскрытых вложенных пунктов
    const isExpandedItemsControlled = expandedItemsProp !== undefined && onExpandedItemsChange !== undefined;
    const [internalExpandedItems, setInternalExpandedItems] = useState(defaultExpandedItems);
    const expandedItems = isExpandedItemsControlled ? (expandedItemsProp ?? {}) : internalExpandedItems;
    // Состояние для отслеживания элементов, которые недавно завершили загрузку
    const [recentlyLoadedItems, setRecentlyLoadedItems] = useState(new Set());
    const timeoutRefs = useRef(new Map());
    const { isExtraSmallDevice } = useDeviceSize();
    useEffect(() => {
        if (!isExtraSmallDevice) {
            return;
        }
        if (!isControlled) {
            setInternalIsExpanded(isOpen);
        }
    }, [isExtraSmallDevice, isOpen, isControlled]);
    useEffect(() => {
        if (!isControlled) {
            setInternalIsExpanded(isOpen);
        }
    }, [isOpen, isControlled]);
    // В контролируемом режиме не вызываем колбэки при изменении isExpanded,
    // так как они должны вызываться только при действиях пользователя
    // В неконтролируемом режиме вызываем колбэки при изменении внутреннего состояния
    useEffect(() => {
        if (!isControlled) {
            if (internalIsExpanded) {
                onOpen?.();
            }
            else {
                onClose?.();
            }
        }
    }, [internalIsExpanded, isControlled, onOpen, onClose]);
    useEffect(() => {
        setActiveItem(active);
    }, [active]);
    // Состояние для отслеживания предыдущего состояния загрузки
    const prevLoadingState = useRef(new Map());
    // Отслеживаем изменения состояния загрузки элементов
    useEffect(() => {
        const checkLoadingItems = (items, parentPath = '') => {
            items.forEach(item => {
                const itemKey = `${parentPath}${item.path}`;
                const wasLoading = prevLoadingState.current.get(itemKey);
                const isCurrentlyLoading = item.isLoading;
                // Обновляем состояние загрузки
                prevLoadingState.current.set(itemKey, isCurrentlyLoading);
                // Если элемент был в состоянии загрузки, но теперь загрузка завершена
                if (wasLoading && !isCurrentlyLoading) {
                    // Элемент завершил загрузку - добавляем в список для анимации
                    setRecentlyLoadedItems(prev => new Set([...prev, itemKey]));
                    // Убираем из списка через 300ms (длительность анимации)
                    const existingTimeout = timeoutRefs.current.get(itemKey);
                    if (existingTimeout) {
                        clearTimeout(existingTimeout);
                    }
                    const timeout = setTimeout(() => {
                        setRecentlyLoadedItems(prev => {
                            const newSet = new Set(prev);
                            newSet.delete(itemKey);
                            return newSet;
                        });
                        timeoutRefs.current.delete(itemKey);
                    }, 300);
                    timeoutRefs.current.set(itemKey, timeout);
                }
                // Рекурсивно проверяем дочерние элементы
                if (item.children) {
                    checkLoadingItems(item.children, `${itemKey}-`);
                }
            });
        };
        // Проверяем все группы
        groups.forEach(group => checkLoadingItems(group.items));
        if (bottom) {
            bottom.forEach(group => checkLoadingItems(group.items));
        }
        // Очищаем таймауты при размонтировании
        return () => {
            timeoutRefs.current.forEach(timeout => clearTimeout(timeout));
            timeoutRefs.current.clear();
        };
    }, [groups, bottom]);
    const handleItemClick = useCallback((path, level = 0) => {
        // Активируем элемент, на который кликнули
        setActiveItem(path);
        onClickItem?.(path);
    }, [onClickItem]);
    const handleToggleExpanded = useCallback((path) => {
        const newExpanded = {
            ...expandedItems,
            [path]: !expandedItems[path],
        };
        if (isExpandedItemsControlled) {
            onExpandedItemsChange?.(newExpanded);
        }
        else {
            setInternalExpandedItems(newExpanded);
        }
    }, [expandedItems, isExpandedItemsControlled, onExpandedItemsChange]);
    const getItemList = useCallback((lists, prevItemsCount = -1, level = 0) => {
        const nodes = [];
        // считает сколько айтемов уже было обработано начиная с входной длины (нужна для сложения айтемов верхних и нижних груп)
        // это позволяет задать задержку анимации для каскадного эффекта при открытии меню
        let itemAnimationDelayCounter = prevItemsCount;
        if (lists) {
            for (const [index, { title, items }] of lists.entries()) {
                nodes.push(jsxs("div", { className: "list__wrapper", role: "menu", children: [isGroupTitle && title && level === 0 && (jsx("div", { className: "list__wrapper-title", children: jsx(Typography, { variant: "overline", children: title }) })), items?.map(item => {
                            itemAnimationDelayCounter++;
                            // Если элемент в состоянии загрузки, показываем скелетон
                            if (item.isLoading) {
                                return jsx(NavigationItemSkeleton, { isChild: level > 0, animated: true }, item.name);
                            }
                            const hasChildren = item.children && item.children.length > 0;
                            const isItemExpanded = expandedItems[item.path];
                            // Для дочерних элементов проверяем активность самого элемента
                            const isItemActive = item.path === activeItem;
                            // Для родительских элементов проверяем, активен ли какой-то из дочерних
                            const isChildActive = level === 0 && hasChildren
                                ? (() => {
                                    return item.children?.some(child => child.path === activeItem) || false;
                                })()
                                : false;
                            // Проверяем, нужно ли применить анимацию появления
                            const itemKey = level > 0 ? `${item.path}-${item.path}` : item.path;
                            const shouldAnimate = recentlyLoadedItems.has(itemKey);
                            const itemNodeKey = item.name;
                            const commonProps = {
                                className: classNames('list__wrapper-item', {
                                    ['list__wrapper-item--active']: (isItemActive || isChildActive) && !item.disableActive,
                                    ['list__wrapper-item--child']: level > 0,
                                    ['navigation-item-fade-in']: shouldAnimate,
                                }),
                                tabIndex: 0,
                                role: 'menuitem',
                                'aria-label': item.name,
                                'aria-selected': (isItemActive || isChildActive) && !item.disableActive,
                                ...(hasChildren && { 'aria-expanded': isItemExpanded }),
                            };
                            const content = (jsxs(Fragment, { children: [(isItemActive || isChildActive) && !item.disableActive && jsx("span", { className: "indicator" }), jsxs(Fragment, { children: [level === 0 && (jsx(Tooltip, { title: item.name, disableHover: isExpanded, children: jsx(Icon, { className: "icon", size: "large", iconName: item.icon }) })), jsx("span", { className: "list__wrapper-item-text", style: {
                                                    transitionDelay: isExpanded ? itemAnimationDelayCounter * 50 + 'ms' : '0ms',
                                                }, children: jsx(Typography, { className: "typography", variant: "body3", children: item.name }) }), hasChildren && (jsx(Icon, { className: classNames('expand-icon', {
                                                    ['expand-icon--expanded']: isItemExpanded,
                                                }), size: "medium", iconName: Icons.NavArrowDown }))] })] }));
                            const itemNodes = [];
                            if (item.href && !hasChildren) {
                                const linkProps = {
                                    ...commonProps,
                                    href: item.href,
                                    onClick: () => handleItemClick(item.path, level),
                                    ...(item.target && { target: item.target }),
                                    ...(item.rel && { rel: item.rel }),
                                };
                                itemNodes.push(jsx(Typography, { variant: "nativeLink", ...linkProps, children: content }, itemNodeKey));
                            }
                            else {
                                itemNodes.push(jsx("button", { ...commonProps, disabled: item.disabled, onClick: () => {
                                        if (hasChildren) {
                                            // Если Drawer свернут, раскрываем его по клику на родительский пункт
                                            if (!isExpanded) {
                                                if (isControlled) {
                                                    onOpen?.();
                                                }
                                                else {
                                                    setIsExpanded(true);
                                                }
                                            }
                                            handleToggleExpanded(item.path);
                                        }
                                        else {
                                            handleItemClick(item.path, level);
                                        }
                                    }, children: content }, itemNodeKey));
                            }
                            // Рендерим вложенные элементы всегда, а состояние сворачивания управляем классом
                            if (hasChildren && level < 1) {
                                const childGroups = [{ items: item.children }];
                                const childNodes = getItemList(childGroups, itemAnimationDelayCounter, level + 1);
                                itemNodes.push(jsx("div", { className: classNames('list__wrapper-children', {
                                        ['list__wrapper-children--collapsed']: !isItemExpanded || !isExpanded,
                                    }), children: childNodes }, `${item.name}-children`));
                            }
                            return itemNodes;
                        })] }, index));
                if (isGroupDivider && index !== lists.length - 1 && level === 0) {
                    nodes.push(jsx("div", { className: "list__divider", children: jsx(Divider, {}) }, 'divider' + index));
                }
            }
        }
        return nodes;
    }, [activeItem, handleItemClick, handleToggleExpanded, isExpanded, isGroupDivider, isGroupTitle, expandedItems]);
    // считает общее кол-во айтемов в основной группе. Нужно для расчета задержки анимации появления айтемов buttonGroup
    const groupsItemsCount = useMemo(() => {
        let totalItems = 0;
        for (const group of groups) {
            totalItems += group.items.length;
        }
        return totalItems;
    }, [groups]);
    const topGroups = useMemo(() => getItemList(groups), [groups, getItemList]);
    const bottomGroups = useMemo(() => getItemList(bottom, groupsItemsCount), [bottom, getItemList, groupsItemsCount]);
    const handleTriggerButton = () => {
        if (isControlled) {
            // В контролируемом режиме вызываем соответствующий колбэк
            if (isExpanded) {
                onClose?.();
            }
            else {
                onOpen?.();
            }
        }
        else {
            // В неконтролируемом режиме изменяем внутреннее состояние
            setInternalIsExpanded(prev => !prev);
        }
    };
    return (jsxs(Fragment, { children: [isExtraSmallDevice && isExpanded && !disableMobileView && (jsx("div", { className: "dsb-navigation-drawer-backdrop", onClick: mobileOnOutsideClick })), jsx("div", { "data-testid": "NavigationDrawer", className: classNames('dsb-navigation-drawer', { ['dsb-navigation-drawer--expanded']: isExpanded }, [className], isExtraSmallDevice && !disableMobileView && 'dsb-navigation-drawer-mobile'), children: jsxs("nav", { className: classNames('dsb-navigation-drawer__container'), children: [(!isExtraSmallDevice || disableMobileView) && (jsx("div", { className: "trigger-button", children: jsx(IconButton, { size: "large", iconName: isExpanded ? Icons.MenuOpen : Icons.Menu, onClick: handleTriggerButton }) })), jsx("div", { className: "list-top", children: groups && jsx("div", { className: classNames('list'), children: topGroups }) }), jsx("div", { className: "list-bottom", children: bottom && (jsxs("div", { className: classNames('list'), children: [jsx("div", { className: "list__divider", children: jsx(Divider, {}) }), bottomGroups] })) })] }) })] }));
};

export { NavigationDrawer };
