// @beeline/design-system-react/extracted-styles/esm/components/Search/Search.dropdown.js
import { jsx } from 'react/jsx-runtime';
import { NewDropdownMenu } from '../NewDropdownMenu/NewDropdownMenu.js';
import { useRef } from 'react';
import { classNames } from '../../helpers/classnames.js';
import { getBEMClasses } from '../../helpers/get-bem.js';

const selectElement = getBEMClasses('dsb__select')[2];
const [optionsClassName, , optionsElement] = selectElement('options');
const [itemClassName] = optionsElement('item');
const SearchDropdown = ({ multiple, values, onChange, options, onOutsideClick, makeOption, parentRef, isOpen, handleSetOpen, dataTestId = 'SearchDropdown', }) => {
    const optionsRef = useRef(null);
    const onSelectItem = (item) => (event) => {
        onChange([item]);
        handleSetOpen(false);
        // надо ли?
        event.stopPropagation();
    };
    const onMultiselectItem = (item) => (event) => {
        const itemIndex = values.findIndex(value => item.id === value.id);
        const isAddition = itemIndex === -1;
        const updItems = isAddition ? [...values, item] : values.filter((_value, i) => itemIndex !== i);
        onChange(updItems);
        // надо ли?
        event.stopPropagation();
    };
    const makeItem = (item, isSelected) => {
        const className = classNames(itemClassName, {
            selected: isSelected,
        });
        const onClickHandler = multiple ? onMultiselectItem : onSelectItem;
        return (jsx("div", { role: "button", tabIndex: 0, className: className, onClick: onClickHandler(item), onKeyDown: event => {
                if (event.key === 'Enter' || event.key === ' ') {
                    event.preventDefault();
                }
            }, children: makeOption(item, isSelected) }, item.id.toString()));
    };
    const makeOptions = (selectedItems, options) => (jsx("div", { className: optionsClassName, ref: optionsRef, children: options?.map(item => makeItem(item, selectedItems.some(i => i.id === item.id))) }));
    const optionsComponents = makeOptions(values, options);
    const MenuProps = {
        targetElement: parentRef,
        align: 'bottom-start',
        verticalOffset: 0,
        isOpen,
        handleSetOpen: handleSetOpen,
        className: 'dsb_options',
        onOutsideClick,
    };
    return (jsx(NewDropdownMenu, { "data-testid": dataTestId, ...MenuProps, children: optionsComponents }));
};

export { SearchDropdown };
