// @beeline/design-tokens/js/tokens/components/inlineAlert/inlineAlert.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	inlineAlertPaddingHorizontal: "16px",
	inlineAlertPaddingVertical: "8px",
	inlineAlertBorderRadius: "8px",
	inlineAlertTextColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	inlineAlertTextFontSize: "15px",
	inlineAlertTextFontWeight: 400,
	inlineAlertTextFontLineHeight: "18px",
	inlineAlertTextFontLetterSpacing: "0.2px",
	inlineAlertIconSize: "20px",
	inlineAlertIconSpacing: "16px",
	inlineAlertNeutralBackgroundColor: "var(--color-status-neutral-background, #f1f1f3)",
	inlineAlertNeutralIconColor: "var(--color-status-neutral, #5e6071)",
	inlineAlertErrorBackgroundColor: "var(--color-status-error-background, #ffecef)",
	inlineAlertErrorIconColor: "var(--color-status-error, #ff5555)",
	inlineAlertWarningBackgroundColor: "var(--color-status-warning-background, #fff4e1)",
	inlineAlertWarningIconColor: "var(--color-status-warning, #ff9419)",
	inlineAlertInfoBackgroundColor: "var(--color-status-info-background, #e3f2ff)",
	inlineAlertInfoIconColor: "var(--color-status-info, #1a73e8)",
	inlineAlertSuccessBackgroundColor: "var(--color-status-success-background, #e7f6eb)",
	inlineAlertSuccessIconColor: "var(--color-status-success, #2ca853)",
};
