// @beeline/design-tokens/js/tokens/components/inlineAlert/inlineAlert.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const inlineAlertPaddingHorizontal = "16px";
export const inlineAlertPaddingVertical = "8px";
export const inlineAlertBorderRadius = "8px";
export const inlineAlertTextColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const inlineAlertTextFontSize = "15px";
export const inlineAlertTextFontWeight = 400;
export const inlineAlertTextFontLineHeight = "18px";
export const inlineAlertTextFontLetterSpacing = "0.2px";
export const inlineAlertIconSize = "20px";
export const inlineAlertIconSpacing = "16px";
export const inlineAlertNeutralBackgroundColor = "var(--color-status-neutral-background, #f1f1f3)";
export const inlineAlertNeutralIconColor = "var(--color-status-neutral, #5e6071)";
export const inlineAlertErrorBackgroundColor = "var(--color-status-error-background, #ffecef)";
export const inlineAlertErrorIconColor = "var(--color-status-error, #ff5555)";
export const inlineAlertWarningBackgroundColor = "var(--color-status-warning-background, #fff4e1)";
export const inlineAlertWarningIconColor = "var(--color-status-warning, #ff9419)";
export const inlineAlertInfoBackgroundColor = "var(--color-status-info-background, #e3f2ff)";
export const inlineAlertInfoIconColor = "var(--color-status-info, #1a73e8)";
export const inlineAlertSuccessBackgroundColor = "var(--color-status-success-background, #e7f6eb)";
export const inlineAlertSuccessIconColor = "var(--color-status-success, #2ca853)";
