// @beeline/design-system-react/extracted-styles/esm/components/Preview/components/WordPreview/WordPreview.js
import { jsx, jsxs } from 'react/jsx-runtime';
import { InlineAlert } from '../../../InlineAlert/InlineAlert.js';
import { Progress } from '../../../Progress/Progress.js';
import { useRef, useState, useEffect } from 'react';

const WORD_INITIAL_WIDTH_FACTOR = 0.7;
const WordPreview = ({ value, scale, onInitialScale, activePage = 0, onPagesLoaded, }) => {
    const containerRef = useRef(null);
    const contentRef = useRef(null);
    const [loading, setLoading] = useState(true);
    const [isReady, setIsReady] = useState(false);
    const [error, setError] = useState(null);
    const [totalPages, setTotalPages] = useState(0);
    const initialScaleCalculated = useRef(false);
    const onInitialScaleRef = useRef(onInitialScale);
    const onPagesLoadedRef = useRef(onPagesLoaded);
    const documentWidth = useRef(0);
    const documentHeight = useRef(0);
    const initialScaleRef = useRef(null);
    const isDragging = useRef(false);
    const dragStart = useRef({ x: 0, y: 0, scrollTop: 0, translateX: 0 });
    const translateX = useRef(0);
    const getContentWidth = () => {
        if (!contentRef.current)
            return 0;
        return contentRef.current.offsetWidth;
    };
    const clampTranslateX = (tx) => {
        if (!containerRef.current)
            return 0;
        const contentWidth = getContentWidth();
        const containerWidth = containerRef.current.clientWidth;
        if (contentWidth <= containerWidth)
            return 0;
        const maxOffset = (contentWidth - containerWidth) / 2 + 120;
        return Math.max(-maxOffset, Math.min(maxOffset, tx));
    };
    const applyTranslateX = (tx) => {
        if (!contentRef.current)
            return;
        contentRef.current.style.transform = `translateX(${tx}px)`;
    };
    const handleMouseDown = (e) => {
        if (!containerRef.current || e.button !== 0)
            return;
        e.preventDefault();
        isDragging.current = true;
        dragStart.current = {
            x: e.clientX,
            y: e.clientY,
            scrollTop: containerRef.current.scrollTop,
            translateX: translateX.current,
        };
        containerRef.current.classList.add('dsb_preview__content-docx-wrapper--dragging');
    };
    useEffect(() => {
        onInitialScaleRef.current = onInitialScale;
    }, [onInitialScale]);
    useEffect(() => {
        onPagesLoadedRef.current = onPagesLoaded;
    }, [onPagesLoaded]);
    useEffect(() => {
        let isMounted = true;
        let abortController = null;
        const loadDocument = async () => {
            if (!containerRef.current || !contentRef.current) {
                return;
            }
            setLoading(true);
            setIsReady(false);
            setError(null);
            initialScaleCalculated.current = false;
            documentWidth.current = 0;
            documentHeight.current = 0;
            initialScaleRef.current = null;
            try {
                abortController = new AbortController();
                const response = await fetch(value.src, { signal: abortController.signal });
                if (!response.ok) {
                    throw new Error(`Ошибка загрузки: ${response.statusText}`);
                }
                const arrayBuffer = await response.arrayBuffer();
                if (!isMounted)
                    return;
                contentRef.current.innerHTML = '';
                contentRef.current.style.width = 'auto';
                contentRef.current.style.transform = '';
                const docxPreview = await import('../../../../node_modules/docx-preview/dist/docx-preview.js');
                const renderFunction = docxPreview.renderDocument || docxPreview.default?.renderDocument;
                if (typeof renderFunction !== 'function') {
                    throw new Error('Функция рендеринга docx-preview не найдена');
                }
                const parsedDocument = await docxPreview.parseAsync(arrayBuffer);
                await renderFunction(parsedDocument, contentRef.current, undefined, {
                    className: 'dsb_preview__word-content',
                    inWrapper: true,
                    ignoreWidth: false,
                    ignoreHeight: false,
                    ignoreFonts: false,
                    breakPages: true,
                    experimental: false,
                });
                if (!isMounted || !containerRef.current || !contentRef.current)
                    return;
                // Ждем обновления DOM
                await new Promise(resolve => requestAnimationFrame(resolve));
                const wrapperElement = contentRef.current.querySelector('.dsb_preview__word-content-wrapper');
                if (wrapperElement && containerRef.current) {
                    // Убираем серый фон, установленный библиотекой
                    wrapperElement.style.background = 'transparent';
                    const contentWidth = wrapperElement.offsetWidth || 595;
                    const contentHeight = wrapperElement.offsetHeight || 842;
                    documentWidth.current = contentWidth;
                    documentHeight.current = contentHeight;
                    let containerWidth = containerRef.current.clientWidth - 48; // вычитаем padding wrapper
                    // Учитываем перекрытие галереей слева и контролами справа
                    const containerRect = containerRef.current.getBoundingClientRect();
                    const previewEl = containerRef.current.closest('.dsb_preview');
                    const galleryEl = previewEl?.querySelector('.dsb_preview__gallery-pdf, .dsb_preview__gallery-word');
                    const controlsEl = previewEl?.querySelector('.dsb_preview__controls');
                    if (galleryEl) {
                        const overlap = galleryEl.getBoundingClientRect().right - containerRect.left;
                        if (overlap > 0)
                            containerWidth -= overlap;
                    }
                    if (controlsEl) {
                        const overlap = containerRect.right - controlsEl.getBoundingClientRect().left;
                        if (overlap > 0)
                            containerWidth -= overlap;
                    }
                    const newRatio = (containerWidth / contentWidth) * WORD_INITIAL_WIDTH_FACTOR;
                    if (!initialScaleCalculated.current && newRatio > 0 && isFinite(newRatio)) {
                        initialScaleRef.current = newRatio;
                        onInitialScaleRef.current?.(newRatio);
                        initialScaleCalculated.current = true;
                    }
                    // Подсчитываем количество страниц
                    const pages = contentRef.current.querySelectorAll('section.dsb_preview__word-content');
                    const pagesCount = pages.length;
                    setTotalPages(pagesCount);
                    onPagesLoadedRef.current?.(pagesCount);
                }
                setLoading(false);
            }
            catch (err) {
                if (err instanceof Error && err.name !== 'AbortError' && isMounted) {
                    setError(err);
                    setLoading(false);
                }
            }
        };
        void loadDocument();
        return () => {
            isMounted = false;
            abortController?.abort();
        };
    }, [value.src]);
    useEffect(() => {
        const onMouseMove = (e) => {
            if (!isDragging.current || !containerRef.current)
                return;
            e.preventDefault();
            const dx = e.clientX - dragStart.current.x;
            const dy = e.clientY - dragStart.current.y;
            // Вертикаль — через scroll
            containerRef.current.scrollTop = dragStart.current.scrollTop - dy;
            // Горизонталь — через CSS transform
            const newTx = clampTranslateX(dragStart.current.translateX + dx);
            translateX.current = newTx;
            applyTranslateX(newTx);
        };
        const onMouseUp = () => {
            if (!isDragging.current)
                return;
            isDragging.current = false;
            containerRef.current?.classList.remove('dsb_preview__content-docx-wrapper--dragging');
        };
        window.addEventListener('mousemove', onMouseMove);
        window.addEventListener('mouseup', onMouseUp);
        return () => {
            window.removeEventListener('mousemove', onMouseMove);
            window.removeEventListener('mouseup', onMouseUp);
        };
    }, []);
    // Прокрутка к выбранной странице
    useEffect(() => {
        if (!containerRef.current || !contentRef.current || loading || totalPages === 0)
            return;
        const pages = contentRef.current.querySelectorAll('section.dsb_preview__word-content');
        if (pages[activePage]) {
            const page = pages[activePage];
            const container = containerRef.current;
            const pageTop = page.offsetTop * scale;
            container.scrollTo({
                top: pageTop,
                behavior: 'smooth',
            });
        }
    }, [activePage, totalPages, loading, scale]);
    // Сброс translateX при изменении масштаба
    useEffect(() => {
        translateX.current = 0;
        applyTranslateX(0);
    }, [scale]);
    useEffect(() => {
        if (!contentRef.current || !containerRef.current || loading || documentWidth.current === 0)
            return;
        const wrapperElement = contentRef.current.querySelector('.dsb_preview__word-content-wrapper');
        if (!wrapperElement)
            return;
        const originalWidth = documentWidth.current;
        const originalHeight = documentHeight.current;
        const scaledWidth = originalWidth * scale;
        const scaledHeight = originalHeight * scale;
        // Компенсируем разницу layout размера через отрицательный margin
        const widthDiff = originalWidth - scaledWidth;
        const heightDiff = originalHeight - scaledHeight;
        wrapperElement.style.transform = `scale(${scale})`;
        wrapperElement.style.transformOrigin = 'top left';
        wrapperElement.style.marginRight = `-${widthDiff}px`;
        wrapperElement.style.marginBottom = `-${heightDiff}px`;
        // Ширина contentRef = scaledWidth, центрирование через flex (align-items: center)
        contentRef.current.style.width = `${scaledWidth}px`;
        // Показываем контент после первого применения scale
        if (initialScaleRef.current !== null && !isReady) {
            requestAnimationFrame(() => {
                requestAnimationFrame(() => {
                    setIsReady(true);
                });
            });
        }
    }, [scale, loading, isReady]);
    if (error) {
        return (jsx("div", { className: "dsb_preview__content-docx-error", ref: containerRef, role: "status", "aria-live": "polite", children: jsxs(InlineAlert, { type: "error", children: ["\u041D\u0435 \u0443\u0434\u0430\u043B\u043E\u0441\u044C \u0437\u0430\u0433\u0440\u0443\u0437\u0438\u0442\u044C \u0434\u043E\u043A\u0443\u043C\u0435\u043D\u0442 Word. ", error.message] }) }));
    }
    return (jsxs("div", { className: "dsb_preview__content-docx-wrapper", ref: containerRef, "aria-busy": loading, onMouseDown: handleMouseDown, children: [loading && (jsx("div", { className: "dsb_preview__content-docx__progress", children: jsx(Progress, { shape: "circle", cycled: true }) })), jsx("div", { className: "dsb_preview__content-docx", ref: contentRef, style: {
                    opacity: isReady ? 1 : 0,
                    visibility: isReady ? 'visible' : 'hidden',
                    transition: 'opacity 0.15s ease-out',
                } })] }));
};

export { WordPreview };
