// @beeline/design-system-react/extracted-styles/esm/components/Sidesheet/Sidesheet.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { Icons } from '@beeline/design-tokens/js/iconfont';
import { classNames } from '../../helpers/classnames.js';
import { IconButton } from '../IconButton/IconButton.js';
import { Typography } from '../Typography/Typography.js';

const Sidesheet = props => {
    const { mode = 'default', elevation = 'coplanar', placement = 'right', size = 'default', content, children, hasDivider = true, hasOverlay = false, isOpen, onClose, title, className, titleNoWrap = true, contentPadding, } = props;
    const isModalMode = mode === 'modal';
    const isLeftAlign = placement === 'left';
    const isRightAlign = placement === 'right';
    const isCoplanarLayout = elevation === 'coplanar';
    const isAboveLayout = elevation === 'above';
    const containerClass = classNames('dsb_sidesheet', {
        dsb_sidesheet_open: isOpen,
        'align-left': isLeftAlign,
        'align-right': isRightAlign,
    }, className);
    const sheetClass = classNames({
        dsb_sidesheet__sheet: true,
        coplanar: isCoplanarLayout && isOpen,
        'align-left': isLeftAlign && isOpen,
        'align-right': isRightAlign && isOpen,
    });
    const elevationContainerClass = classNames({
        dsb_sidesheet__sheet__elevation: true,
        'is-open': isOpen,
        'align-left': isLeftAlign,
        'align-right': isRightAlign,
        coplanar: isCoplanarLayout,
        'size-large': size === 'large',
        above: isAboveLayout,
        'has-shadow': isModalMode,
    });
    const sheetContentWrapperClass = classNames({
        'dsb_sidesheet__sheet__elevation__content-wrapper': true,
        'has-divider': hasDivider,
        'align-left': isLeftAlign,
        'align-right': isRightAlign,
        coplanar: isCoplanarLayout,
        'size-large': size === 'large',
        above: isAboveLayout,
    });
    const contentHeaderClass = 'dsb_sidesheet__sheet__elevation__content-wrapper__header';
    const sidesheetContentClass = 'dsb_sidesheet__sheet__elevation__content-wrapper__content';
    const contentClass = classNames({
        dsb_sidesheet__content: true,
        modal: isOpen && isModalMode,
        overlay: hasOverlay,
    });
    const titleClass = classNames({
        dsb_sidesheet__title: true,
        'dsb_sidesheet__title_no-wrap': titleNoWrap,
    });
    const handleOverlayClick = () => {
        if (isOpen && isModalMode && onClose) {
            onClose();
        }
    };
    const isStringTitle = typeof title === 'string';
    // Создаем кастомные стили для паддингов
    const customPaddingStyle = {};
    if (contentPadding) {
        if (Array.isArray(contentPadding)) {
            customPaddingStyle.padding = `${contentPadding[0]}px ${contentPadding[1]}px ${contentPadding[2]}px ${contentPadding[3]}px`;
        }
        else {
            customPaddingStyle.padding = contentPadding;
        }
    }
    return (jsxs("div", { className: containerClass, "data-testid": "Sidesheet", role: "dialog", "aria-modal": true, children: [jsxs("div", { className: contentClass, onClick: handleOverlayClick, role: "presentation", onKeyDown: (event) => {
                    if (event.key === 'Enter' || event.key === ' ') {
                        handleOverlayClick();
                    }
                }, children: [children, isOpen && isModalMode && hasOverlay && (jsx("div", { className: "dsb_sidesheet__overlay-bg", onClick: onClose, role: "button", tabIndex: 0, onKeyDown: (event) => {
                            if (event.key === 'Enter' || event.key === ' ') {
                                event.preventDefault();
                                onClose?.();
                            }
                        } }))] }), jsx("div", { className: sheetClass, children: jsx("div", { className: elevationContainerClass, children: jsxs("div", { className: sheetContentWrapperClass, style: customPaddingStyle, children: [jsxs("div", { className: contentHeaderClass, children: [isStringTitle ? (jsx(Typography, { className: titleClass, variant: "h5", children: title })) : (title), jsx(IconButton, { iconName: Icons.Close, size: "large", onClick: onClose, className: "dsb_sidesheet__close-button" })] }), jsx("div", { className: sidesheetContentClass, children: content })] }) }) })] }));
};

export { Sidesheet };
