// @beeline/design-system-react/extracted-styles/esm/components/Select/components/Item/index.js
import { jsxs, Fragment, jsx } from 'react/jsx-runtime';
import { Checkbox } from '../../../Checkbox/Checkbox.js';
import { CheckboxType } from '../../Select.types.js';
import { classNames } from '../../../../helpers/classnames.js';

const Item = ({ optionId, option, isSelected, multiple, selectAll, itemClassName, onMultiselectItem, onManualClickOption, onSelectItem, focusedIndex, optionIndex, handleSelectAll, makeOption, showTooltip, defaultMakeOption, makeOptionWrapper, }) => {
    const isDisabled = option.disabled;
    // Формируем базовый класс с выделением выбранной опции
    const baseClass = classNames(itemClassName, {
        selected: isSelected === CheckboxType.CHECKED,
        disabled: isDisabled,
    });
    // Добавляем класс 'focused' если текущий индекс совпадает с focusedIndex
    const finalClassName = classNames(baseClass, focusedIndex === optionIndex && 'focused');
    const onClickHandler = multiple ? onMultiselectItem : onSelectItem;
    const onChangeAction = typeof onManualClickOption === 'function' ? () => onManualClickOption(option.value) : onClickHandler(option);
    const wrapperKey = option.id;
    const wrapperProps = {
        key: option.id,
        id: optionId,
        role: 'option',
        tabIndex: -1,
        'aria-selected': isSelected === CheckboxType.CHECKED,
        'aria-disabled': isDisabled,
        className: finalClassName,
        onClick: isDisabled ? undefined : selectAll ? handleSelectAll : onChangeAction,
        ...(selectAll ? { 'data-testid': 'select-all' } : {}),
        children: (jsxs(Fragment, { children: [multiple && (jsx(Checkbox, { checked: isSelected !== CheckboxType.UNCHECKED, type: isSelected === CheckboxType.PARTIALLY ? 'indeterminate' : 'checkbox', onClick: event => event.stopPropagation(), onChange: event => event.stopPropagation(), className: "dsb__select__options__item__checkbox", disabled: isDisabled })), selectAll ? (jsx("span", { children: option.value })) : makeOption ? (makeOption(option.value, isSelected === CheckboxType.CHECKED)) : (defaultMakeOption(option.value, showTooltip))] })),
    };
    // Рендерим кастомный враппер если передан
    if (makeOptionWrapper) {
        return makeOptionWrapper(option.value, isSelected === CheckboxType.CHECKED, { key: wrapperKey, ...wrapperProps });
    }
    // Дефолтный рендер
    return jsx("div", { ...wrapperProps }, wrapperKey);
};

export { Item };
