// @beeline/design-tokens/js/tokens/components/textfieldwithchips/textfieldwithchips.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const textfieldwithchipsBackgroundColor = "var(--color-control-background, rgba(25, 28, 52, 0.1))";
export const textfieldwithchipsBackgroundColorHover = "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))";
export const textfieldwithchipsBackgroundColorError = "var(--color-control-background-error, rgba(255, 85, 85, 0.08))";
export const textfieldwithchipsBackgroundColorErrorHover = "rgba(255, 85, 85, 0.14)";
export const textfieldwithchipsBackgroundColorDisabledHover = "var(--color-control-background, rgba(25, 28, 52, 0.1))";
export const textfieldwithchipsBorderRadius = "12px";
export const textfieldwithchipsBorderWidth = "1px";
export const textfieldwithchipsBorderColorHover = "var(--color-border-focus, #212121)";
export const textfieldwithchipsBorderColorError = "var(--color-border-error, #ff5555)";
export const textfieldwithchipsBorderColorFocused = "var(--color-border-focus, #212121)";
export const textfieldwithchipsOpacityDisabled = 0.48;
export const textfieldwithchipsDropdownPaddingVertical = "8px";
export const textfieldwithchipsDropdownMaxHeight = "480px";
export const textfieldwithchipsDropdownBackgroundColor = "var(--color-background-medium, #ffffff)";
export const textfieldwithchipsDropdownShadow = "0px 4px 30px rgba(0, 0, 0, 0.1), 0px 0px 10px rgba(0, 0, 0, 0.1)";
export const textfieldwithchipsDropdownOptionPaddingVertical = "12px";
export const textfieldwithchipsDropdownOptionPaddingHorizontal = "16px";
export const textfieldwithchipsDropdownOptionBackgroundColorHover = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const textfieldwithchipsDropdownOptionBackgroundColorFocused = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const textfieldwithchipsContentMarginLeft = "16px";
export const textfieldwithchipsContentGap = "8px";
export const textfieldwithchipsContentMinHeight = "32px";
export const textfieldwithchipsContentInputColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const textfieldwithchipsContentInputMinWidth = "8px";
export const textfieldwithchipsContentLabelLeft = "16px";
export const textfieldwithchipsContentLabelFontSize = "17px";
export const textfieldwithchipsContentLabelFontSizeSmall = "13px";
export const textfieldwithchipsBottomTextLeft = "16px";
export const textfieldwithchipsBottomTextFontFamily = "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif";
export const textfieldwithchipsBottomTextFontSize = "13px";
export const textfieldwithchipsBottomTextColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const textfieldwithchipsBottomTextColorError = "var(--color-status-error, #ff5555)";
export const textfieldwithchipsSmallHeight = "40px";
export const textfieldwithchipsSmallPaddingVertical = "4px";
export const textfieldwithchipsMediumHeight = "48px";
export const textfieldwithchipsMediumPaddingVertical = "8px";
export const textfieldwithchipsLargeHeight = "56px";
export const textfieldwithchipsLargePaddingVertical = "12px";
