// @beeline/design-tokens/js/tokens/components/textfieldwithchips/textfieldwithchips.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	textfieldwithchipsBackgroundColor: "var(--color-control-background, rgba(25, 28, 52, 0.1))",
	textfieldwithchipsBackgroundColorHover: "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))",
	textfieldwithchipsBackgroundColorError: "var(--color-control-background-error, rgba(255, 85, 85, 0.08))",
	textfieldwithchipsBackgroundColorErrorHover: "rgba(255, 85, 85, 0.14)",
	textfieldwithchipsBackgroundColorDisabledHover: "var(--color-control-background, rgba(25, 28, 52, 0.1))",
	textfieldwithchipsBorderRadius: "12px",
	textfieldwithchipsBorderWidth: "1px",
	textfieldwithchipsBorderColorHover: "var(--color-border-focus, #212121)",
	textfieldwithchipsBorderColorError: "var(--color-border-error, #ff5555)",
	textfieldwithchipsBorderColorFocused: "var(--color-border-focus, #212121)",
	textfieldwithchipsOpacityDisabled: 0.48,
	textfieldwithchipsDropdownPaddingVertical: "8px",
	textfieldwithchipsDropdownMaxHeight: "480px",
	textfieldwithchipsDropdownBackgroundColor: "var(--color-background-medium, #ffffff)",
	textfieldwithchipsDropdownShadow: "0px 4px 30px rgba(0, 0, 0, 0.1), 0px 0px 10px rgba(0, 0, 0, 0.1)",
	textfieldwithchipsDropdownOptionPaddingVertical: "12px",
	textfieldwithchipsDropdownOptionPaddingHorizontal: "16px",
	textfieldwithchipsDropdownOptionBackgroundColorHover: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	textfieldwithchipsDropdownOptionBackgroundColorFocused: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	textfieldwithchipsContentMarginLeft: "16px",
	textfieldwithchipsContentGap: "8px",
	textfieldwithchipsContentMinHeight: "32px",
	textfieldwithchipsContentInputColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	textfieldwithchipsContentInputMinWidth: "8px",
	textfieldwithchipsContentLabelLeft: "16px",
	textfieldwithchipsContentLabelFontSize: "17px",
	textfieldwithchipsContentLabelFontSizeSmall: "13px",
	textfieldwithchipsBottomTextLeft: "16px",
	textfieldwithchipsBottomTextFontFamily: "'Beeline Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif",
	textfieldwithchipsBottomTextFontSize: "13px",
	textfieldwithchipsBottomTextColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	textfieldwithchipsBottomTextColorError: "var(--color-status-error, #ff5555)",
	textfieldwithchipsSmallHeight: "40px",
	textfieldwithchipsSmallPaddingVertical: "4px",
	textfieldwithchipsMediumHeight: "48px",
	textfieldwithchipsMediumPaddingVertical: "8px",
	textfieldwithchipsLargeHeight: "56px",
	textfieldwithchipsLargePaddingVertical: "12px",
};
