// @beeline/design-system-react/extracted-styles/esm/components/DatePicker/Calendar/views/CalendarDayView/CalendarDayView.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { classNames } from '../../../../../helpers/classnames.js';
import { WEEKDAYS } from '../../constants.js';
import { convertUtcToTz, getWeeksOfMonth, isRangeMode, isSameDates, hasDateLimit } from '../../newUtils.js';

const CalendarDayView = ({ minDate, maxDate, disabledStartDate, disabledEndDate, viewDate, selectedDate, onSelectDay, multiselect, excludeDate, holidaysDate, customizeDay, readonly, timeZone, showAdjacentMonthDays, allowAdjacentMonthSelection, }) => {
    // Переведём viewDate (UTC) в локальное время
    const userTzDate = convertUtcToTz(viewDate, timeZone);
    const localYear = userTzDate.getFullYear();
    const localMonth = userTzDate.getMonth();
    // Для «сегодня»
    const now = new Date(); // Локальное «сейчас» (или UTC -> TZ, на ваше усмотрение)
    const todayLocalYear = now.getFullYear();
    const todayLocalMonth = now.getMonth();
    const todayDay = now.getDate();
    // Список недель (локальный год/месяц)
    const monthWeeks = getWeeksOfMonth(localYear, localMonth, showAdjacentMonthDays);
    const handleSelectDay = (dayInfo) => {
        if (!dayInfo || !dayInfo.day || readonly) {
            return;
        }
        // Проверяем, является ли день соседним месяцем
        const isAdjacentMonth = dayInfo.month !== localMonth || dayInfo.year !== localYear;
        // Если это день соседнего месяца и клики по соседним месяцам не разрешены, не обрабатываем клик
        if (isAdjacentMonth && !allowAdjacentMonthSelection) {
            return;
        }
        onSelectDay(dayInfo.day, dayInfo.month, dayInfo.year);
    };
    const isCurrentDay = (dayInfo) => {
        if (!dayInfo || !dayInfo.day) {
            return false;
        }
        return dayInfo.year === todayLocalYear && dayInfo.month === todayLocalMonth && dayInfo.day === todayDay;
    };
    const toLocalMidnight = (date) => {
        if (!date) {
            return null;
        }
        // Если время равно полуночи по UTC, считаем, что это "Date Only" значение (например, "2023-08-15"),
        // и не применяем смещение часового пояса, чтобы избежать сдвига на предыдущий день (например, в NY).
        const isUtcMidnight = date.getUTCHours() === 0 &&
            date.getUTCMinutes() === 0 &&
            date.getUTCSeconds() === 0 &&
            date.getUTCMilliseconds() === 0;
        if (isUtcMidnight) {
            return new Date(Date.UTC(date.getUTCFullYear(), date.getUTCMonth(), date.getUTCDate()));
        }
        try {
            // Используем Intl для безопасного извлечения компонентов даты в целевом часовом поясе
            // Это позволяет избежать проблем со смещением date-fns-tz и зависимостями от системного часового пояса
            const formatter = new Intl.DateTimeFormat('en-US', {
                timeZone: timeZone || 'UTC',
                year: 'numeric',
                month: 'numeric',
                day: 'numeric',
            });
            const parts = formatter.formatToParts(date);
            const getPart = (type) => {
                const part = parts.find(p => p.type === type);
                return part ? parseInt(part.value, 10) : 0;
            };
            return new Date(Date.UTC(getPart('year'), getPart('month') - 1, getPart('day')));
        }
        catch (e) {
            // Фолбэк, если Intl не работает (например, некорректная таймзона)
            const local = convertUtcToTz(date, timeZone);
            return new Date(Date.UTC(local.getFullYear(), local.getMonth(), local.getDate()));
        }
    };
    const isSelectedDay = (dayInfo) => {
        // dayInfo -- это один день в календаре, мы их перебираем
        if (!dayInfo || !dayInfo.day) {
            return false;
        }
        // Для корректного сравнения берем UTC year/month из viewDate,
        // но создаём new Date() через Date.UTC(...) чтобы получить
        // одну и ту же дату (без смещения).
        const checkedDate = new Date(Date.UTC(dayInfo.year, dayInfo.month, dayInfo.day));
        if (!isRangeMode(selectedDate)) {
            if (!selectedDate) {
                return false;
            }
            const sDate = toLocalMidnight(new Date(selectedDate));
            return sDate ? isSameDates(checkedDate, sDate) : false;
        }
        // мультипл + рендж
        if (disabledStartDate && disabledEndDate) {
            return false;
        }
        if (disabledStartDate) {
            const end = selectedDate[1];
            const localEnd = end ? toLocalMidnight(end) : null;
            return localEnd ? isSameDates(localEnd, checkedDate) : false;
        }
        if (disabledEndDate) {
            const start = selectedDate[0];
            const localStart = start ? toLocalMidnight(start) : null;
            return localStart ? isSameDates(localStart, checkedDate) : false;
        }
        return selectedDate?.some(date => {
            const localDate = toLocalMidnight(date);
            return localDate && isSameDates(localDate, checkedDate);
        });
    };
    const isDisabledDay = (dayInfo) => {
        if (!dayInfo || !dayInfo.day) {
            return false;
        }
        // Формируем проверяемую дату как полночь по UTC из календарных Y/M/D, чтобы избежать ошибок сдвига
        const localCheckedDate = new Date(dayInfo.year, dayInfo.month, dayInfo.day, 0, 0, 0);
        const checkedUTC = new Date(Date.UTC(dayInfo.year, dayInfo.month, dayInfo.day));
        if (excludeDate?.(localCheckedDate)) {
            return true;
        }
        if (hasDateLimit(checkedUTC, minDate, maxDate)) {
            return true;
        }
        return false;
    };
    const isHolidayDay = (dayInfo) => {
        if (dayInfo?.day && holidaysDate) {
            const checkedDate = new Date(dayInfo.year, dayInfo.month, dayInfo.day);
            return holidaysDate(checkedDate);
        }
        return false;
    };
    // Определяем «статус» дней (start / end / range) в режиме range
    const getRangeStatusOfDay = (dayInfo) => {
        if (multiselect) {
            // В мультивыборе не подсвечиваем «промежуток» между датами
            return {};
        }
        if (!dayInfo || !dayInfo.day || !isRangeMode(selectedDate)) {
            return {};
        }
        const [startDay, endDay] = selectedDate;
        const isFilledRange = startDay && endDay;
        const checkedDate = new Date(Date.UTC(dayInfo.year, dayInfo.month, dayInfo.day));
        const startDayMidnight = startDay ? toLocalMidnight(startDay) : null;
        const endDayMidnight = endDay ? toLocalMidnight(endDay) : null;
        const getStarting = () => {
            if (disabledStartDate) {
                return false;
            }
            return isFilledRange && startDayMidnight && isSameDates(startDayMidnight, checkedDate);
        };
        const getEnding = () => {
            if (disabledEndDate) {
                return false;
            }
            return isFilledRange && endDayMidnight && isSameDates(endDayMidnight, checkedDate);
        };
        return {
            starting: getStarting(),
            ranging: startDayMidnight && startDayMidnight <= checkedDate && endDayMidnight && endDayMidnight >= checkedDate,
            ending: getEnding(),
        };
    };
    const getCustomization = (dayInfo) => {
        if (!dayInfo || !dayInfo.day || !customizeDay) {
            return null;
        }
        const dateToCustomize = new Date(dayInfo.year, dayInfo.month, dayInfo.day);
        return customizeDay(dateToCustomize);
    };
    return (jsxs("table", { className: 'dsb_calendar-dates', children: [jsx("thead", { className: 'dsb_calendar-weekdays', children: jsx("tr", { children: WEEKDAYS.map(weekday => (jsx("th", { className: 'dsb_calendar-weekday', children: weekday }, weekday))) }) }), jsx("tbody", { className: 'dsb_calendar-weeks', children: monthWeeks.map((week, weekIndex) => (jsx("tr", { children: week.map((dayInfo, dayIndex) => {
                        const disabled = isDisabledDay(dayInfo);
                        const holiday = isHolidayDay(dayInfo);
                        const selected = isSelectedDay(dayInfo);
                        const today = isCurrentDay(dayInfo);
                        const rangeStatus = getRangeStatusOfDay(dayInfo);
                        const customization = getCustomization(dayInfo);
                        const isAdjacentMonth = dayInfo && (dayInfo.month !== localMonth || dayInfo.year !== localYear);
                        return (jsx("td", { className: classNames('dsb_calendar-day-cell', {
                                ...rangeStatus,
                            }), children: jsx("span", { "data-testid": dayInfo ? `day-${dayInfo.year}-${dayInfo.month + 1}-${dayInfo.day}` : undefined, className: classNames('dsb_calendar-day', {
                                    today,
                                    disabled,
                                    holiday,
                                    selected,
                                    ['dsb_calendar-day__empty']: !dayInfo || dayInfo.day === null,
                                    readonly,
                                    'adjacent-month': isAdjacentMonth,
                                    ...rangeStatus,
                                }, customization?.className), onClick: () => handleSelectDay(dayInfo), children: customization?.content || (dayInfo && dayInfo.day) }) }, `day-${dayIndex}`));
                    }) }, `week-${weekIndex}`))) })] }));
};

export { CalendarDayView };
