// @beeline/design-tokens/js/tokens/components/list/list.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const listPaddingVertical = "12px";
export const listTitleColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const listTitleFontSize = "17px";
export const listTitleFontWeight = 400;
export const listTitleFontLineHeight = "22px";
export const listTitleFontLetterSpacing = "0.2px";
export const listTextColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const listTextFontSize = "15px";
export const listTextFontWeight = 400;
export const listTextFontLineHeight = "18px";
export const listTextFontLetterSpacing = "0.2px";
export const listOverlineColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const listOverlineFontSize = "10px";
export const listOverlineFontWeight = 700;
export const listOverlineFontLineHeight = "16px";
export const listOverlineFontLetterSpacing = "1.2px";
export const listCaptionColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const listCaptionSpacing = "8px";
export const listCaptionFontSize = "13px";
export const listCaptionFontWeight = 400;
export const listCaptionFontLineHeight = "16px";
export const listCaptionFontLetterSpacing = "0.2px";
