// @beeline/design-tokens/js/tokens/components/breadcrumbs/breadcrumbs.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	breadcrumbsGap: "8px", // Горизонтальный отступ между элементами хлебных крошек и разделяющими их иконками-стрелками, а также между строками.
	breadcrumbsItemHeight: "18px",
	breadcrumbsItemBorderRadius: "4px",
	breadcrumbsItemTextPaddingHorizontal: "4px",
	breadcrumbsItemTextFontSize: "15px",
	breadcrumbsItemTextFontWeight: 400,
	breadcrumbsItemTextFontLineHeight: "18px",
	breadcrumbsItemTextFontLetterSpacing: "0.2px",
	breadcrumbsItemParentStandardTextColor: "var(--color-button-plain, #1a73e8)",
	breadcrumbsItemParentCollapsedWidth: "18px",
	breadcrumbsItemParentCollapsedIconColor: "var(--color-button-plain, #1a73e8)",
	breadcrumbsItemParentHoverBackgroundColor: "var(--color-button-plain-background-hover, rgba(26, 115, 232, 0.08))",
	breadcrumbsItemParentPressedBackgroundColor: "var(--color-button-plain-background-pressed, rgba(26, 115, 232, 0.12))",
	breadcrumbsItemCurrentTextColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	breadcrumbsItemCurrentIconColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	breadcrumbsItemCurrentHoverBackgroundColor: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	breadcrumbsItemCurrentPressedBackgroundColor: "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))",
};
