// @beeline/design-tokens/js/tokens/components/breadcrumbs/breadcrumbs.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const breadcrumbsGap = "8px"; // Горизонтальный отступ между элементами хлебных крошек и разделяющими их иконками-стрелками, а также между строками.
export const breadcrumbsItemHeight = "18px";
export const breadcrumbsItemBorderRadius = "4px";
export const breadcrumbsItemTextPaddingHorizontal = "4px";
export const breadcrumbsItemTextFontSize = "15px";
export const breadcrumbsItemTextFontWeight = 400;
export const breadcrumbsItemTextFontLineHeight = "18px";
export const breadcrumbsItemTextFontLetterSpacing = "0.2px";
export const breadcrumbsItemParentStandardTextColor = "var(--color-button-plain, #1a73e8)";
export const breadcrumbsItemParentCollapsedWidth = "18px";
export const breadcrumbsItemParentCollapsedIconColor = "var(--color-button-plain, #1a73e8)";
export const breadcrumbsItemParentHoverBackgroundColor = "var(--color-button-plain-background-hover, rgba(26, 115, 232, 0.08))";
export const breadcrumbsItemParentPressedBackgroundColor = "var(--color-button-plain-background-pressed, rgba(26, 115, 232, 0.12))";
export const breadcrumbsItemCurrentTextColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const breadcrumbsItemCurrentIconColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const breadcrumbsItemCurrentHoverBackgroundColor = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const breadcrumbsItemCurrentPressedBackgroundColor = "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))";
