// @beeline/design-tokens/js/tokens/components/switcher/switcher.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const switcherOpacityDisabled = 0.48;
export const switcherTrackWidth = "44px";
export const switcherTrackHeight = "24px";
export const switcherTrackBackgroundColor = "var(--color-control-background, rgba(25, 28, 52, 0.1))";
export const switcherTrackBackgroundColorHover = "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))";
export const switcherTrackBackgroundColorPressed = "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))";
export const switcherTrackBorderWidth = "1px";
export const switcherTrackBorderColorFocus = "var(--color-border-focus, #212121)";
export const switcherTrackBorderRadius = "10px";
export const switcherThumbWidth = "16px";
export const switcherThumbHeight = "16px";
export const switcherThumbBackgroundColor = "var(--color-background-white, #ffffff)";
export const switcherThumbMarginLeft = "4px";
export const switcherThumbBorderRadius = "50%";
export const switcherThumbShadow = "0px 2px 10px rgba(0, 0, 0, 0.08), 0px 2px 8px rgba(0, 0, 0, 0.08)";
export const switcherThumbHoleWidth = "6px";
export const switcherFocusedTrackBorderColor = "var(--color-border-focus, #212121)";
export const switcherFocusedTrackBorderWidth = "1px";
export const switcherDisabledOpacity = 0.48;
export const switcherBaseThumbWidth = "16px";
export const switcherBaseThumbMarginLeft = "4px";
export const switcherBaseThumbBorderRadius = "50%";
export const switcherBaseThumbShadow = "0px 2px 10px rgba(0, 0, 0, 0.08), 0px 2px 8px rgba(0, 0, 0, 0.08)";
export const switcherBaseThumbHoleWidth = "6px";
export const switcherBaseTrackBackgroundColor = "var(--color-control-background, rgba(25, 28, 52, 0.1))";
export const switcherBaseHoverTrackBackgroundColor = "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))";
export const switcherBasePressedTrackBackgroundColor = "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))";
export const switcherSelectedTrackBackgroundColor = "var(--color-background-brand, #fdd835)";
export const switcherSelectedTrackBackgroundColorHover = "var(--color-background-brand-hover, #fdc435)";
export const switcherSelectedTrackBackgroundColorPressed = "var(--color-background-brand-pressed, #fdb435)";
export const switcherSelectedThumbWidth = "6px";
export const switcherSelectedThumbMarginRight = "8px";
export const switcherSelectedThumbBorderRadius = "6px";
export const switcherSelectedThumbShadow = [{"x":"0","y":"0","blur":"4","color":"rgba(0, 0, 0, 0.12)"}];
export const switcherSelectedHoverTrackBackgroundColor = "var(--color-background-brand-hover, #fdc435)";
export const switcherSelectedPressedTrackBackgroundColor = "var(--color-background-brand-pressed, #fdb435)";
