// @beeline/design-tokens/js/tokens/components/switcher/switcher.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	switcherOpacityDisabled: 0.48,
	switcherTrackWidth: "44px",
	switcherTrackHeight: "24px",
	switcherTrackBackgroundColor: "var(--color-control-background, rgba(25, 28, 52, 0.1))",
	switcherTrackBackgroundColorHover: "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))",
	switcherTrackBackgroundColorPressed: "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))",
	switcherTrackBorderWidth: "1px",
	switcherTrackBorderColorFocus: "var(--color-border-focus, #212121)",
	switcherTrackBorderRadius: "10px",
	switcherThumbWidth: "16px",
	switcherThumbHeight: "16px",
	switcherThumbBackgroundColor: "var(--color-background-white, #ffffff)",
	switcherThumbMarginLeft: "4px",
	switcherThumbBorderRadius: "50%",
	switcherThumbShadow: "0px 2px 10px rgba(0, 0, 0, 0.08), 0px 2px 8px rgba(0, 0, 0, 0.08)",
	switcherThumbHoleWidth: "6px",
	switcherFocusedTrackBorderColor: "var(--color-border-focus, #212121)",
	switcherFocusedTrackBorderWidth: "1px",
	switcherDisabledOpacity: 0.48,
	switcherBaseThumbWidth: "16px",
	switcherBaseThumbMarginLeft: "4px",
	switcherBaseThumbBorderRadius: "50%",
	switcherBaseThumbShadow: "0px 2px 10px rgba(0, 0, 0, 0.08), 0px 2px 8px rgba(0, 0, 0, 0.08)",
	switcherBaseThumbHoleWidth: "6px",
	switcherBaseTrackBackgroundColor: "var(--color-control-background, rgba(25, 28, 52, 0.1))",
	switcherBaseHoverTrackBackgroundColor: "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))",
	switcherBasePressedTrackBackgroundColor: "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))",
	switcherSelectedTrackBackgroundColor: "var(--color-background-brand, #fdd835)",
	switcherSelectedTrackBackgroundColorHover: "var(--color-background-brand-hover, #fdc435)",
	switcherSelectedTrackBackgroundColorPressed: "var(--color-background-brand-pressed, #fdb435)",
	switcherSelectedThumbWidth: "6px",
	switcherSelectedThumbMarginRight: "8px",
	switcherSelectedThumbBorderRadius: "6px",
	switcherSelectedThumbShadow: [{"x":"0","y":"0","blur":"4","color":"rgba(0, 0, 0, 0.12)"}],
	switcherSelectedHoverTrackBackgroundColor: "var(--color-background-brand-hover, #fdc435)",
	switcherSelectedPressedTrackBackgroundColor: "var(--color-background-brand-pressed, #fdb435)",
};
