// @beeline/design-tokens/js/tokens/components/notifications/notifications.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	notificationsTargetSpacing: "0px", // Отступ от целевого элемента до поповера
	notificationsPanelBorderRadius: "12px",
	notificationsPanelShadow: "0px 4px 30px rgba(0, 0, 0, 0.1), 0px 0px 10px rgba(0, 0, 0, 0.1)",
	notificationsPanelBackgroundColor: "var(--color-background-medium, #ffffff)",
	notificationsPanelWidth: "440px",
	notificationsPanelMaxHeightTablet: "90vh",
	notificationsPanelMaxHeightDesktop: "75vh",
	notificationsPanelMaxHeightWide: "55vh",
	notificationsPanelFooterPaddingVertical: "12px",
	notificationsPanelFooterPaddingHorizontal: "24px",
	notificationsPanelFooterActionsGap: "16px",
	notificationsPanelDividerColor: "var(--color-divider, rgba(25, 28, 52, 0.12))",
	notificationsPanelDividerHeight: "1px",
	notificationsItemPaddingVertical: "12px",
	notificationsItemPaddingHorizontal: "24px",
	notificationsItemIndicatorSpacingTop: "24px",
	notificationsItemIndicatorSpacingLeft: "16px",
	notificationsItemIndicatorWidth: "8px",
	notificationsItemIndicatorHeight: "8px",
	notificationsItemIndicatorBorderRadius: "50%",
	notificationsItemIndicatorBackgroundColor: "var(--color-text-link, #1a73e8)",
	notificationsItemUnreadBackgroundColor: "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))",
	notificationsItemHoverBackgroundColor: "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))",
	notificationsItemFocusedBackgroundColor: "var(--color-background-base-focused, rgba(25, 28, 52, 0.12))",
	notificationsItemPressedBackgroundColor: "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))",
};
