// @beeline/design-tokens/js/tokens/components/notifications/notifications.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const notificationsTargetSpacing = "0px"; // Отступ от целевого элемента до поповера
export const notificationsPanelBorderRadius = "12px";
export const notificationsPanelShadow = "0px 4px 30px rgba(0, 0, 0, 0.1), 0px 0px 10px rgba(0, 0, 0, 0.1)";
export const notificationsPanelBackgroundColor = "var(--color-background-medium, #ffffff)";
export const notificationsPanelWidth = "440px";
export const notificationsPanelMaxHeightTablet = "90vh";
export const notificationsPanelMaxHeightDesktop = "75vh";
export const notificationsPanelMaxHeightWide = "55vh";
export const notificationsPanelFooterPaddingVertical = "12px";
export const notificationsPanelFooterPaddingHorizontal = "24px";
export const notificationsPanelFooterActionsGap = "16px";
export const notificationsPanelDividerColor = "var(--color-divider, rgba(25, 28, 52, 0.12))";
export const notificationsPanelDividerHeight = "1px";
export const notificationsItemPaddingVertical = "12px";
export const notificationsItemPaddingHorizontal = "24px";
export const notificationsItemIndicatorSpacingTop = "24px";
export const notificationsItemIndicatorSpacingLeft = "16px";
export const notificationsItemIndicatorWidth = "8px";
export const notificationsItemIndicatorHeight = "8px";
export const notificationsItemIndicatorBorderRadius = "50%";
export const notificationsItemIndicatorBackgroundColor = "var(--color-text-link, #1a73e8)";
export const notificationsItemUnreadBackgroundColor = "var(--color-background-base-selected, rgba(25, 28, 52, 0.05))";
export const notificationsItemHoverBackgroundColor = "var(--color-background-base-hover, rgba(25, 28, 52, 0.08))";
export const notificationsItemFocusedBackgroundColor = "var(--color-background-base-focused, rgba(25, 28, 52, 0.12))";
export const notificationsItemPressedBackgroundColor = "var(--color-background-base-pressed, rgba(25, 28, 52, 0.12))";
