// @beeline/design-system-react/build/components/PhoneInput/CountrySelect/Countries.const.js
const getImagePath = (imageName) => `/pictures/flags/${imageName}.svg`;
const CountriesData = {
    UZ: {
        nameCode: 'UZ',
        name: 'Узбекистан',
        phoneCode: '+998',
        flag: getImagePath('UZ'),
    },
    NP: {
        nameCode: 'NP',
        name: 'Непал',
        phoneCode: '+977',
        flag: getImagePath('NP'),
    },
    AD: {
        nameCode: 'AD',
        name: 'Андорра',
        phoneCode: '+376',
        flag: getImagePath('AD'),
    },
    EE: {
        nameCode: 'EE',
        name: 'Эстония',
        phoneCode: '+372',
        flag: getImagePath('EE'),
    },
    KW: {
        nameCode: 'KW',
        name: 'Кувейт',
        phoneCode: '+965',
        flag: getImagePath('KW'),
    },
    PY: {
        nameCode: 'PY',
        name: 'Парагвай',
        phoneCode: '+595',
        flag: getImagePath('PY'),
    },
    AE: {
        nameCode: 'AE',
        name: 'Объединённые Арабские Эмираты',
        phoneCode: '+971',
        flag: getImagePath('AE'),
    },
    EG: {
        nameCode: 'EG',
        name: 'Египет',
        phoneCode: '+20',
        flag: getImagePath('EG'),
    },
    KY: {
        nameCode: 'KY',
        name: 'Каймановы острова',
        phoneCode: '+1',
        flag: getImagePath('KY'),
    },
    QA: {
        nameCode: 'QA',
        name: 'Катар',
        phoneCode: '+974',
        flag: getImagePath('QA'),
    },
    AF: {
        nameCode: 'AF',
        name: 'Афганистан',
        phoneCode: '+93',
        flag: getImagePath('AF'),
    },
    EH: {
        nameCode: 'EH',
        name: 'Западная Сахара',
        phoneCode: '+212',
        flag: getImagePath('EH'),
    },
    KZ: {
        nameCode: 'KZ',
        name: 'Казахстан',
        phoneCode: '+7',
        flag: getImagePath('KZ'),
    },
    RE: {
        nameCode: 'RE',
        name: 'Реюньон',
        phoneCode: '+262',
        flag: getImagePath('RE'),
    },
    AG: {
        nameCode: 'AG',
        name: 'Антигуа и Барбуда',
        phoneCode: '+1',
        flag: getImagePath('AG'),
    },
    ER: {
        nameCode: 'ER',
        name: 'Эритрея',
        phoneCode: '+291',
        flag: getImagePath('ER'),
    },
    LA: {
        nameCode: 'LA',
        name: 'Лаос',
        phoneCode: '+856',
        flag: getImagePath('LA'),
    },
    RO: {
        nameCode: 'RO',
        name: 'Румыния',
        phoneCode: '+40',
        flag: getImagePath('RO'),
    },
    AI: {
        nameCode: 'AI',
        name: 'Ангилья',
        phoneCode: '+1',
        flag: getImagePath('AI'),
    },
    ES: {
        nameCode: 'ES',
        name: 'Испания',
        phoneCode: '+34',
        flag: getImagePath('ES'),
    },
    LB: {
        nameCode: 'LB',
        name: 'Ливан',
        phoneCode: '+961',
        flag: getImagePath('LB'),
    },
    RS: {
        nameCode: 'RS',
        name: 'Сербия',
        phoneCode: '+381',
        flag: getImagePath('RS'),
    },
    AL: {
        nameCode: 'AL',
        name: 'Албания',
        phoneCode: '+355',
        flag: getImagePath('AL'),
    },
    ET: {
        nameCode: 'ET',
        name: 'Эфиопия',
        phoneCode: '+251',
        flag: getImagePath('ET'),
    },
    LC: {
        nameCode: 'LC',
        name: 'Сент-Люсия',
        phoneCode: '+1',
        flag: getImagePath('LC'),
    },
    RU: {
        nameCode: 'RU',
        name: 'Россия',
        phoneCode: '+7',
        flag: getImagePath('RU'),
    },
    AM: {
        nameCode: 'AM',
        name: 'Армения',
        phoneCode: '+374',
        flag: getImagePath('AM'),
    },
    FI: {
        nameCode: 'FI',
        name: 'Финляндия',
        phoneCode: '+358',
        flag: getImagePath('FI'),
    },
    LI: {
        nameCode: 'LI',
        name: 'Лихтенштейн',
        phoneCode: '+423',
        flag: getImagePath('LI'),
    },
    RW: {
        nameCode: 'RW',
        name: 'Руанда',
        phoneCode: '+250',
        flag: getImagePath('RW'),
    },
    AO: {
        nameCode: 'AO',
        name: 'Ангола',
        phoneCode: '+244',
        flag: getImagePath('AO'),
    },
    FJ: {
        nameCode: 'FJ',
        name: 'Фиджи',
        phoneCode: '+679',
        flag: getImagePath('FJ'),
    },
    LK: {
        nameCode: 'LK',
        name: 'Шри-Ланка',
        phoneCode: '+94',
        flag: getImagePath('LK'),
    },
    SA: {
        nameCode: 'SA',
        name: 'Саудовская Аравия',
        phoneCode: '+966',
        flag: getImagePath('SA'),
    },
    AQ: {
        nameCode: 'AQ',
        name: 'Антарктида',
        phoneCode: '+672',
        flag: getImagePath('AQ'),
    },
    FK: {
        nameCode: 'FK',
        name: 'Фолклендские острова',
        phoneCode: '+500',
        flag: getImagePath('FK'),
    },
    LR: {
        nameCode: 'LR',
        name: 'Либерия',
        phoneCode: '+231',
        flag: getImagePath('LR'),
    },
    SB: {
        nameCode: 'SB',
        name: 'Соломоновы острова',
        phoneCode: '+677',
        flag: getImagePath('SB'),
    },
    AR: {
        nameCode: 'AR',
        name: 'Аргентина',
        phoneCode: '+54',
        flag: getImagePath('AR'),
    },
    FM: {
        nameCode: 'FM',
        name: 'Федеративные Штаты Микронезии',
        phoneCode: '+691',
        flag: getImagePath('FM'),
    },
    LS: {
        nameCode: 'LS',
        name: 'Лесото',
        phoneCode: '+266',
        flag: getImagePath('LS'),
    },
    SC: {
        nameCode: 'SC',
        name: 'Сейшельские Острова',
        phoneCode: '+248',
        flag: getImagePath('SC'),
    },
    AS: {
        nameCode: 'AS',
        name: 'Американское Самоа',
        phoneCode: '+1',
        flag: getImagePath('AS'),
    },
    FO: {
        nameCode: 'FO',
        name: 'Фарерские острова',
        phoneCode: '+298',
        flag: getImagePath('FO'),
    },
    LT: {
        nameCode: 'LT',
        name: 'Литва',
        phoneCode: '+370',
        flag: getImagePath('LT'),
    },
    SD: {
        nameCode: 'SD',
        name: 'Судан',
        phoneCode: '+249',
        flag: getImagePath('SD'),
    },
    AT: {
        nameCode: 'AT',
        name: 'Австрия',
        phoneCode: '+43',
        flag: getImagePath('AT'),
    },
    FR: {
        nameCode: 'FR',
        name: 'Франция',
        phoneCode: '+33',
        flag: getImagePath('FR'),
    },
    LU: {
        nameCode: 'LU',
        name: 'Люксембург',
        phoneCode: '+352',
        flag: getImagePath('LU'),
    },
    SE: {
        nameCode: 'SE',
        name: 'Швеция',
        phoneCode: '+46',
        flag: getImagePath('SE'),
    },
    AU: {
        nameCode: 'AU',
        name: 'Австралия',
        phoneCode: '+61',
        flag: getImagePath('AU'),
    },
    GA: {
        nameCode: 'GA',
        name: 'Габон',
        phoneCode: '+241',
        flag: getImagePath('GA'),
    },
    LV: {
        nameCode: 'LV',
        name: 'Латвия',
        phoneCode: '+371',
        flag: getImagePath('LV'),
    },
    SG: {
        nameCode: 'SG',
        name: 'Сингапур',
        phoneCode: '+65',
        flag: getImagePath('SG'),
    },
    AW: {
        nameCode: 'AW',
        name: 'Аруба',
        phoneCode: '+297',
        flag: getImagePath('AW'),
    },
    'GB-ENG': {
        nameCode: 'GB-ENG',
        name: 'Англия',
        phoneCode: '+44',
        flag: getImagePath('GB_ENG'),
    },
    LY: {
        nameCode: 'LY',
        name: 'Ливия',
        phoneCode: '+218',
        flag: getImagePath('LY'),
    },
    SH: {
        nameCode: 'SH',
        name: 'Остров Святой Елены',
        phoneCode: '+290',
        flag: getImagePath('SH'),
    },
    AX: {
        nameCode: 'AX',
        name: 'Аландские острова',
        phoneCode: '+358',
        flag: getImagePath('AX'),
    },
    'GB-NIR': {
        nameCode: 'GB-NIR',
        name: 'Северная Ирландия',
        phoneCode: '+44',
        flag: getImagePath('GB_NIR'),
    },
    MA: {
        nameCode: 'MA',
        name: 'Марокко',
        phoneCode: '+212',
        flag: getImagePath('MA'),
    },
    SI: {
        nameCode: 'SI',
        name: 'Словения',
        phoneCode: '+386',
        flag: getImagePath('SI'),
    },
    AZ: {
        nameCode: 'AZ',
        name: 'Азербайджан',
        phoneCode: '+994',
        flag: getImagePath('AZ'),
    },
    'GB-SCT': {
        nameCode: 'GB-SCT',
        name: 'Шотландия',
        phoneCode: '+44',
        flag: getImagePath('GB_SCT'),
    },
    MC: {
        nameCode: 'MC',
        name: 'Монако',
        phoneCode: '+377',
        flag: getImagePath('MC'),
    },
    SJ: {
        nameCode: 'SJ',
        name: 'Шпицберген и Ян-Майен',
        phoneCode: '+47',
        flag: getImagePath('SJ'),
    },
    BA: {
        nameCode: 'BA',
        name: 'Босния и Герцеговина',
        phoneCode: '+387',
        flag: getImagePath('BA'),
    },
    'GB-UKM': {
        nameCode: 'GB-UKM',
        name: 'Соединённое Королевство',
        phoneCode: '+44',
        flag: getImagePath('GB_UKM'),
    },
    MD: {
        nameCode: 'MD',
        name: 'Молдова',
        phoneCode: '+373',
        flag: getImagePath('MD'),
    },
    SK: {
        nameCode: 'SK',
        name: 'Словакия',
        phoneCode: '+421',
        flag: getImagePath('SK'),
    },
    BB: {
        nameCode: 'BB',
        name: 'Барбадос',
        phoneCode: '+1',
        flag: getImagePath('BB'),
    },
    'GB-WLS': {
        nameCode: 'GB-WLS',
        name: 'Уэльс',
        phoneCode: '+44',
        flag: getImagePath('GB_WLS'),
    },
    ME: {
        nameCode: 'ME',
        name: 'Черногория',
        phoneCode: '+382',
        flag: getImagePath('ME'),
    },
    SL: {
        nameCode: 'SL',
        name: 'Сьерра-Леоне',
        phoneCode: '+232',
        flag: getImagePath('SL'),
    },
    BD: {
        nameCode: 'BD',
        name: 'Бангладеш',
        phoneCode: '+880',
        flag: getImagePath('BD'),
    },
    GD: {
        nameCode: 'GD',
        name: 'Гренада',
        phoneCode: '+1',
        flag: getImagePath('GD'),
    },
    MF: {
        nameCode: 'MF',
        name: 'Сен-Мартен',
        phoneCode: '+590',
        flag: getImagePath('MF'),
    },
    SM: {
        nameCode: 'SM',
        name: 'Сан-Марино',
        phoneCode: '+378',
        flag: getImagePath('SM'),
    },
    BE: {
        nameCode: 'BE',
        name: 'Бельгия',
        phoneCode: '+32',
        flag: getImagePath('BE'),
    },
    GE: {
        nameCode: 'GE',
        name: 'Грузия',
        phoneCode: '+995',
        flag: getImagePath('GE'),
    },
    MG: {
        nameCode: 'MG',
        name: 'Мадагаскар',
        phoneCode: '+261',
        flag: getImagePath('MG'),
    },
    SN: {
        nameCode: 'SN',
        name: 'Сенегал',
        phoneCode: '+221',
        flag: getImagePath('SN'),
    },
    BF: {
        nameCode: 'BF',
        name: 'Буркина-Фасо',
        phoneCode: '+226',
        flag: getImagePath('BF'),
    },
    GF: {
        nameCode: 'GF',
        name: 'Французская Гвиана',
        phoneCode: '+594',
        flag: getImagePath('GF'),
    },
    MH: {
        nameCode: 'MH',
        name: 'Маршалловы Острова',
        phoneCode: '+692',
        flag: getImagePath('MH'),
    },
    SO: {
        nameCode: 'SO',
        name: 'Сомали',
        phoneCode: '+252',
        flag: getImagePath('SO'),
    },
    BG: {
        nameCode: 'BG',
        name: 'Болгария',
        phoneCode: '+359',
        flag: getImagePath('BG'),
    },
    GG: {
        nameCode: 'GG',
        name: 'Гернси',
        phoneCode: '+44',
        flag: getImagePath('GG'),
    },
    MK: {
        nameCode: 'MK',
        name: 'Северная Македония',
        phoneCode: '+389',
        flag: getImagePath('MK'),
    },
    SR: {
        nameCode: 'SR',
        name: 'Суринам',
        phoneCode: '+597',
        flag: getImagePath('SR'),
    },
    BH: {
        nameCode: 'BH',
        name: 'Бахрейн',
        phoneCode: '+973',
        flag: getImagePath('BH'),
    },
    GH: {
        nameCode: 'GH',
        name: 'Гана',
        phoneCode: '+233',
        flag: getImagePath('GH'),
    },
    ML: {
        nameCode: 'ML',
        name: 'Мали',
        phoneCode: '+223',
        flag: getImagePath('ML'),
    },
    SS: {
        nameCode: 'SS',
        name: 'Южный Судан',
        phoneCode: '+211',
        flag: getImagePath('SS'),
    },
    BI: {
        nameCode: 'BI',
        name: 'Бурунди',
        phoneCode: '+257',
        flag: getImagePath('BI'),
    },
    GI: {
        nameCode: 'GI',
        name: 'Гибралтар',
        phoneCode: '+350',
        flag: getImagePath('GI'),
    },
    MM: {
        nameCode: 'MM',
        name: 'Мьянма',
        phoneCode: '+95',
        flag: getImagePath('MM'),
    },
    ST: {
        nameCode: 'ST',
        name: 'Сан-Томе и Принсипи',
        phoneCode: '+239',
        flag: getImagePath('ST'),
    },
    BJ: {
        nameCode: 'BJ',
        name: 'Бенин',
        phoneCode: '+229',
        flag: getImagePath('BJ'),
    },
    GL: {
        nameCode: 'GL',
        name: 'Гренландия',
        phoneCode: '+299',
        flag: getImagePath('GL'),
    },
    MN: {
        nameCode: 'MN',
        name: 'Монголия',
        phoneCode: '+976',
        flag: getImagePath('MN'),
    },
    SV: {
        nameCode: 'SV',
        name: 'Сальвадор',
        phoneCode: '+503',
        flag: getImagePath('SV'),
    },
    BL: {
        nameCode: 'BL',
        name: 'Сен-Бартелеми',
        phoneCode: '+590',
        flag: getImagePath('BL'),
    },
    GM: {
        nameCode: 'GM',
        name: 'Гамбия',
        phoneCode: '+220',
        flag: getImagePath('GM'),
    },
    MO: {
        nameCode: 'MO',
        name: 'Макао',
        phoneCode: '+853',
        flag: getImagePath('MO'),
    },
    SX: {
        nameCode: 'SX',
        name: 'Синт-Мартен',
        phoneCode: '+1',
        flag: getImagePath('SX'),
    },
    BM: {
        nameCode: 'BM',
        name: 'Бермуды',
        phoneCode: '+1',
        flag: getImagePath('BM'),
    },
    GN: {
        nameCode: 'GN',
        name: 'Гвинея',
        phoneCode: '+224',
        flag: getImagePath('GN'),
    },
    MP: {
        nameCode: 'MP',
        name: 'Северные Марианские острова',
        phoneCode: '+1',
        flag: getImagePath('MP'),
    },
    SY: {
        nameCode: 'SY',
        name: 'Сирия',
        phoneCode: '+963',
        flag: getImagePath('SY'),
    },
    BN: {
        nameCode: 'BN',
        name: 'Бруней-Даруссалам',
        phoneCode: '+673',
        flag: getImagePath('BN'),
    },
    GP: {
        nameCode: 'GP',
        name: 'Гваделупа',
        phoneCode: '+590',
        flag: getImagePath('GP'),
    },
    MQ: {
        nameCode: 'MQ',
        name: 'Мартиника',
        phoneCode: '+596',
        flag: getImagePath('MQ'),
    },
    SZ: {
        nameCode: 'SZ',
        name: 'Эсватини',
        phoneCode: '+268',
        flag: getImagePath('SZ'),
    },
    BO: {
        nameCode: 'BO',
        name: 'Боливия',
        phoneCode: '+591',
        flag: getImagePath('BO'),
    },
    TC: {
        nameCode: 'TC',
        name: 'Тёркс и Кайкос',
        phoneCode: '+1',
        flag: getImagePath('TC'),
    },
    'BQ-BO': {
        nameCode: 'BQ-BO',
        name: 'Бонайре',
        phoneCode: '+599',
        flag: getImagePath('BQ_BO'),
    },
    GQ: {
        nameCode: 'GQ',
        name: 'Экваториальная Гвинея',
        phoneCode: '+240',
        flag: getImagePath('GQ'),
    },
    MR: {
        nameCode: 'MR',
        name: 'Мавритания',
        phoneCode: '+222',
        flag: getImagePath('MR'),
    },
    TD: {
        nameCode: 'TD',
        name: 'Чад',
        phoneCode: '+235',
        flag: getImagePath('TD'),
    },
    'BQ-SA': {
        nameCode: 'BQ-SA',
        name: 'Саба',
        phoneCode: '+599',
        flag: getImagePath('BQ_SA'),
    },
    GR: {
        nameCode: 'GR',
        name: 'Греция',
        phoneCode: '+30',
        flag: getImagePath('GR'),
    },
    MS: {
        nameCode: 'MS',
        name: 'Монтсеррат',
        phoneCode: '+1',
        flag: getImagePath('MS'),
    },
    TF: {
        nameCode: 'TF',
        name: 'Французские Южные и Антарктические территории',
        phoneCode: '+262',
        flag: getImagePath('TF'),
    },
    'BQ-SE': {
        nameCode: 'BQ-SE',
        name: 'Синт-Эстатиус',
        phoneCode: '+599',
        flag: getImagePath('BQ_SE'),
    },
    MT: {
        nameCode: 'MT',
        name: 'Мальта',
        phoneCode: '+356',
        flag: getImagePath('MT'),
    },
    TG: {
        nameCode: 'TG',
        name: 'Того',
        phoneCode: '+228',
        flag: getImagePath('TG'),
    },
    BR: {
        nameCode: 'BR',
        name: 'Бразилия',
        phoneCode: '+55',
        flag: getImagePath('BR'),
    },
    GT: {
        nameCode: 'GT',
        name: 'Гватемала',
        phoneCode: '+502',
        flag: getImagePath('GT'),
    },
    MU: {
        nameCode: 'MU',
        name: 'Маврикий',
        phoneCode: '+230',
        flag: getImagePath('MU'),
    },
    TH: {
        nameCode: 'TH',
        name: 'Таиланд',
        phoneCode: '+66',
        flag: getImagePath('TH'),
    },
    BS: {
        nameCode: 'BS',
        name: 'Багамы',
        phoneCode: '+1',
        flag: getImagePath('BS'),
    },
    GU: {
        nameCode: 'GU',
        name: 'Гуам',
        phoneCode: '+1',
        flag: getImagePath('GU'),
    },
    MV: {
        nameCode: 'MV',
        name: 'Мальдивы',
        phoneCode: '+960',
        flag: getImagePath('MV'),
    },
    TJ: {
        nameCode: 'TJ',
        name: 'Таджикистан',
        phoneCode: '+992',
        flag: getImagePath('TJ'),
    },
    BT: {
        nameCode: 'BT',
        name: 'Бутан',
        phoneCode: '+975',
        flag: getImagePath('BT'),
    },
    GW: {
        nameCode: 'GW',
        name: 'Гвинея-Бисау',
        phoneCode: '+245',
        flag: getImagePath('GW'),
    },
    MW: {
        nameCode: 'MW',
        name: 'Малави',
        phoneCode: '+265',
        flag: getImagePath('MW'),
    },
    TK: {
        nameCode: 'TK',
        name: 'Токелау',
        phoneCode: '+690',
        flag: getImagePath('TK'),
    },
    BV: {
        nameCode: 'BV',
        name: 'Остров Буве',
        phoneCode: '+47',
        flag: getImagePath('BV'),
    },
    GY: {
        nameCode: 'GY',
        name: 'Гайана',
        phoneCode: '+592',
        flag: getImagePath('GY'),
    },
    MX: {
        nameCode: 'MX',
        name: 'Мексика',
        phoneCode: '+52',
        flag: getImagePath('MX'),
    },
    TL: {
        nameCode: 'TL',
        name: 'Тимор-Лесте',
        phoneCode: '+670',
        flag: getImagePath('TL'),
    },
    BW: {
        nameCode: 'BW',
        name: 'Ботсвана',
        phoneCode: '+267',
        flag: getImagePath('BW'),
    },
    HK: {
        nameCode: 'HK',
        name: 'Гонконг',
        phoneCode: '+852',
        flag: getImagePath('HK'),
    },
    MY: {
        nameCode: 'MY',
        name: 'Малайзия',
        phoneCode: '+60',
        flag: getImagePath('MY'),
    },
    TM: {
        nameCode: 'TM',
        name: 'Туркменистан',
        phoneCode: '+993',
        flag: getImagePath('TM'),
    },
    BY: {
        nameCode: 'BY',
        name: 'Беларусь',
        phoneCode: '+375',
        flag: getImagePath('BY'),
    },
    HM: {
        nameCode: 'HM',
        name: 'Остров Херд и острова Макдональд',
        phoneCode: '+672',
        flag: getImagePath('HM'),
    },
    MZ: {
        nameCode: 'MZ',
        name: 'Мозамбик',
        phoneCode: '+258',
        flag: getImagePath('MZ'),
    },
    TN: {
        nameCode: 'TN',
        name: 'Тунис',
        phoneCode: '+216',
        flag: getImagePath('TN'),
    },
    BZ: {
        nameCode: 'BZ',
        name: 'Белиз',
        phoneCode: '+501',
        flag: getImagePath('BZ'),
    },
    HN: {
        nameCode: 'HN',
        name: 'Гондурас',
        phoneCode: '+504',
        flag: getImagePath('HN'),
    },
    NA: {
        nameCode: 'NA',
        name: 'Намибия',
        phoneCode: '+264',
        flag: getImagePath('NA'),
    },
    TO: {
        nameCode: 'TO',
        name: 'Тонга',
        phoneCode: '+676',
        flag: getImagePath('TO'),
    },
    CA: {
        nameCode: 'CA',
        name: 'Канада',
        phoneCode: '+1',
        flag: getImagePath('CA'),
    },
    HR: {
        nameCode: 'HR',
        name: 'Хорватия',
        phoneCode: '+385',
        flag: getImagePath('HR'),
    },
    NC: {
        nameCode: 'NC',
        name: 'Новая Каледония',
        phoneCode: '+687',
        flag: getImagePath('NC'),
    },
    TR: {
        nameCode: 'TR',
        name: 'Турция',
        phoneCode: '+90',
        flag: getImagePath('TR'),
    },
    CC: {
        nameCode: 'CC',
        name: 'Кокосовые (Килинг) острова',
        phoneCode: '+61',
        flag: getImagePath('CC'),
    },
    HT: {
        nameCode: 'HT',
        name: 'Гаити',
        phoneCode: '+509',
        flag: getImagePath('HT'),
    },
    NE: {
        nameCode: 'NE',
        name: 'Нигер',
        phoneCode: '+227',
        flag: getImagePath('NE'),
    },
    TT: {
        nameCode: 'TT',
        name: 'Тринидад и Тобаго',
        phoneCode: '+1',
        flag: getImagePath('TT'),
    },
    CD: {
        nameCode: 'CD',
        name: 'Демократическая Республика Конго',
        phoneCode: '+243',
        flag: getImagePath('CD'),
    },
    HU: {
        nameCode: 'HU',
        name: 'Венгрия',
        phoneCode: '+36',
        flag: getImagePath('HU'),
    },
    NF: {
        nameCode: 'NF',
        name: 'Остров Норфолк',
        phoneCode: '+672',
        flag: getImagePath('NF'),
    },
    TV: {
        nameCode: 'TV',
        name: 'Тувалу',
        phoneCode: '+688',
        flag: getImagePath('TV'),
    },
    CF: {
        nameCode: 'CF',
        name: 'Центральноафриканская Республика',
        phoneCode: '+236',
        flag: getImagePath('CF'),
    },
    ID: {
        nameCode: 'ID',
        name: 'Индонезия',
        phoneCode: '+62',
        flag: getImagePath('ID'),
    },
    NG: {
        nameCode: 'NG',
        name: 'Нигерия',
        phoneCode: '+234',
        flag: getImagePath('NG'),
    },
    TW: {
        nameCode: 'TW',
        name: 'Тайвань',
        phoneCode: '+886',
        flag: getImagePath('TW'),
    },
    CG: {
        nameCode: 'CG',
        name: 'Республика Конго',
        phoneCode: '+242',
        flag: getImagePath('CG'),
    },
    IE: {
        nameCode: 'IE',
        name: 'Ирландия',
        phoneCode: '+353',
        flag: getImagePath('IE'),
    },
    NI: {
        nameCode: 'NI',
        name: 'Никарагуа',
        phoneCode: '+505',
        flag: getImagePath('NI'),
    },
    TZ: {
        nameCode: 'TZ',
        name: 'Танзания',
        phoneCode: '+255',
        flag: getImagePath('TZ'),
    },
    CH: {
        nameCode: 'CH',
        name: 'Швейцария',
        phoneCode: '+41',
        flag: getImagePath('CH'),
    },
    IL: {
        nameCode: 'IL',
        name: 'Израиль',
        phoneCode: '+972',
        flag: getImagePath('IL'),
    },
    NL: {
        nameCode: 'NL',
        name: 'Нидерланды',
        phoneCode: '+31',
        flag: getImagePath('NL'),
    },
    UA: {
        nameCode: 'UA',
        name: 'Украина',
        phoneCode: '+380',
        flag: getImagePath('UA'),
    },
    CI: {
        nameCode: 'CI',
        name: 'Кот-д’Ивуар',
        phoneCode: '+225',
        flag: getImagePath('CI'),
    },
    IM: {
        nameCode: 'IM',
        name: 'Остров Мэн',
        phoneCode: '+44',
        flag: getImagePath('IM'),
    },
    NO: {
        nameCode: 'NO',
        name: 'Норвегия',
        phoneCode: '+47',
        flag: getImagePath('NO'),
    },
    UG: {
        nameCode: 'UG',
        name: 'Уганда',
        phoneCode: '+256',
        flag: getImagePath('UG'),
    },
    CK: {
        nameCode: 'CK',
        name: 'Острова Кука',
        phoneCode: '+682',
        flag: getImagePath('CK'),
    },
    IN: {
        nameCode: 'IN',
        name: 'Индия',
        phoneCode: '+91',
        flag: getImagePath('IN'),
    },
    NR: {
        nameCode: 'NR',
        name: 'Науру',
        phoneCode: '+674',
        flag: getImagePath('NR'),
    },
    UM: {
        nameCode: 'UM',
        name: 'Внешние малые острова США',
        phoneCode: '+1',
        flag: getImagePath('UM'),
    },
    CL: {
        nameCode: 'CL',
        name: 'Чили',
        phoneCode: '+56',
        flag: getImagePath('CL'),
    },
    INT: {
        nameCode: 'INT',
        name: 'Международный',
        phoneCode: '+882',
        flag: getImagePath('INT'),
    },
    NU: {
        nameCode: 'NU',
        name: 'Ниуэ',
        phoneCode: '+683',
        flag: getImagePath('NU'),
    },
    US: {
        nameCode: 'US',
        name: 'Соединённые Штаты',
        phoneCode: '+1',
        flag: getImagePath('US'),
    },
    CM: {
        nameCode: 'CM',
        name: 'Камерун',
        phoneCode: '+237',
        flag: getImagePath('CM'),
    },
    IO: {
        nameCode: 'IO',
        name: 'Британская территория в Индийском океане',
        phoneCode: '+246',
        flag: getImagePath('IO'),
    },
    NZ: {
        nameCode: 'NZ',
        name: 'Новая Зеландия (Аотеароа)',
        phoneCode: '+64',
        flag: getImagePath('NZ'),
    },
    UY: {
        nameCode: 'UY',
        name: 'Уругвай',
        phoneCode: '+598',
        flag: getImagePath('UY'),
    },
    CN: {
        nameCode: 'CN',
        name: 'Китай',
        phoneCode: '+86',
        flag: getImagePath('CN'),
    },
    IQ: {
        nameCode: 'IQ',
        name: 'Ирак',
        phoneCode: '+964',
        flag: getImagePath('IQ'),
    },
    OM: {
        nameCode: 'OM',
        name: 'Оман',
        phoneCode: '+968',
        flag: getImagePath('OM'),
    },
    VA: {
        nameCode: 'VA',
        name: 'Ватикан',
        phoneCode: '+39',
        flag: getImagePath('VA'),
    },
    CR: {
        nameCode: 'CR',
        name: 'Коста-Рика',
        phoneCode: '+506',
        flag: getImagePath('CR'),
    },
    IS: {
        nameCode: 'IS',
        name: 'Исландия',
        phoneCode: '+354',
        flag: getImagePath('IS'),
    },
    PA: {
        nameCode: 'PA',
        name: 'Панама',
        phoneCode: '+507',
        flag: getImagePath('PA'),
    },
    VC: {
        nameCode: 'VC',
        name: 'Сент-Винсент и Гренадины',
        phoneCode: '+1',
        flag: getImagePath('VC'),
    },
    CU: {
        nameCode: 'CU',
        name: 'Куба',
        phoneCode: '+53',
        flag: getImagePath('CU'),
    },
    IT: {
        nameCode: 'IT',
        name: 'Италия',
        phoneCode: '+39',
        flag: getImagePath('IT'),
    },
    PE: {
        nameCode: 'PE',
        name: 'Перу',
        phoneCode: '+51',
        flag: getImagePath('PE'),
    },
    VE: {
        nameCode: 'VE',
        name: 'Венесуэла',
        phoneCode: '+58',
        flag: getImagePath('VE'),
    },
    CV: {
        nameCode: 'CV',
        name: 'Кабо-Верде',
        phoneCode: '+238',
        flag: getImagePath('CV'),
    },
    JE: {
        nameCode: 'JE',
        name: 'Джерси',
        phoneCode: '+44',
        flag: getImagePath('JE'),
    },
    PF: {
        nameCode: 'PF',
        name: 'Французская Полинезия',
        phoneCode: '+689',
        flag: getImagePath('PF'),
    },
    VG: {
        nameCode: 'VG',
        name: 'Виргинские острова (Британские)',
        phoneCode: '+1',
        flag: getImagePath('VG'),
    },
    CW: {
        nameCode: 'CW',
        name: 'Кюрасао',
        phoneCode: '+599',
        flag: getImagePath('CW'),
    },
    JM: {
        nameCode: 'JM',
        name: 'Ямайка',
        phoneCode: '+1',
        flag: getImagePath('JM'),
    },
    PG: {
        nameCode: 'PG',
        name: 'Папуа — Новая Гвинея',
        phoneCode: '+675',
        flag: getImagePath('PG'),
    },
    CX: {
        nameCode: 'CX',
        name: 'Остров Рождества',
        phoneCode: '+61',
        flag: getImagePath('CX'),
    },
    JO: {
        nameCode: 'JO',
        name: 'Иордания',
        phoneCode: '+962',
        flag: getImagePath('JO'),
    },
    PH: {
        nameCode: 'PH',
        name: 'Филиппины',
        phoneCode: '+63',
        flag: getImagePath('PH'),
    },
    VN: {
        nameCode: 'VN',
        name: 'Вьетнам',
        phoneCode: '+84',
        flag: getImagePath('VN'),
    },
    CY: {
        nameCode: 'CY',
        name: 'Кипр',
        phoneCode: '+357',
        flag: getImagePath('CY'),
    },
    JP: {
        nameCode: 'JP',
        name: 'Япония',
        phoneCode: '+81',
        flag: getImagePath('JP'),
    },
    PK: {
        nameCode: 'PK',
        name: 'Пакистан',
        phoneCode: '+92',
        flag: getImagePath('PK'),
    },
    VU: {
        nameCode: 'VU',
        name: 'Вануату',
        phoneCode: '+678',
        flag: getImagePath('VU'),
    },
    CZ: {
        nameCode: 'CZ',
        name: 'Чехия',
        phoneCode: '+420',
        flag: getImagePath('CZ'),
    },
    KE: {
        nameCode: 'KE',
        name: 'Кения',
        phoneCode: '+254',
        flag: getImagePath('KE'),
    },
    PL: {
        nameCode: 'PL',
        name: 'Польша',
        phoneCode: '+48',
        flag: getImagePath('PL'),
    },
    WF: {
        nameCode: 'WF',
        name: 'Уоллис и Футуна',
        phoneCode: '+681',
        flag: getImagePath('WF'),
    },
    DE: {
        nameCode: 'DE',
        name: 'Германия',
        phoneCode: '+49',
        flag: getImagePath('DE'),
    },
    KG: {
        nameCode: 'KG',
        name: 'Киргизия',
        phoneCode: '+996',
        flag: getImagePath('KG'),
    },
    PM: {
        nameCode: 'PM',
        name: 'Сен-Пьер и Микелон',
        phoneCode: '+508',
        flag: getImagePath('PM'),
    },
    WS: {
        nameCode: 'WS',
        name: 'Самоа',
        phoneCode: '+685',
        flag: getImagePath('WS'),
    },
    DJ: {
        nameCode: 'DJ',
        name: 'Джибути',
        phoneCode: '+253',
        flag: getImagePath('DJ'),
    },
    KH: {
        nameCode: 'KH',
        name: 'Камбоджа',
        phoneCode: '+855',
        flag: getImagePath('KH'),
    },
    YE: {
        nameCode: 'YE',
        name: 'Йемен',
        phoneCode: '+967',
        flag: getImagePath('YE'),
    },
    DK: {
        nameCode: 'DK',
        name: 'Дания',
        phoneCode: '+45',
        flag: getImagePath('DK'),
    },
    KI: {
        nameCode: 'KI',
        name: 'Кирибати',
        phoneCode: '+686',
        flag: getImagePath('KI'),
    },
    PN: {
        nameCode: 'PN',
        name: 'Острова Питкэрн',
        phoneCode: '+64',
        flag: getImagePath('PN'),
    },
    'YT-UNF': {
        nameCode: 'YT-UNF',
        name: 'Майотта',
        phoneCode: '+262',
        flag: getImagePath('YT_UNF'),
    },
    DM: {
        nameCode: 'DM',
        name: 'Доминика',
        phoneCode: '+1',
        flag: getImagePath('DM'),
    },
    KM: {
        nameCode: 'KM',
        name: 'Коморы',
        phoneCode: '+269',
        flag: getImagePath('KM'),
    },
    PR: {
        nameCode: 'PR',
        name: 'Пуэрто-Рико',
        phoneCode: '+1',
        flag: getImagePath('PR'),
    },
    ZA: {
        nameCode: 'ZA',
        name: 'Южно-Африканская Республика',
        phoneCode: '+27',
        flag: getImagePath('ZA'),
    },
    DO: {
        nameCode: 'DO',
        name: 'Доминиканская Республика',
        phoneCode: '+1',
        flag: getImagePath('DO'),
    },
    KN: {
        nameCode: 'KN',
        name: 'Сент-Китс и Невис',
        phoneCode: '+1',
        flag: getImagePath('KN'),
    },
    PS: {
        nameCode: 'PS',
        name: 'Государство Палестина',
        phoneCode: '+970',
        flag: getImagePath('PS'),
    },
    ZM: {
        nameCode: 'ZM',
        name: 'Замбия',
        phoneCode: '+260',
        flag: getImagePath('ZM'),
    },
    DZ: {
        nameCode: 'DZ',
        name: 'Алжир',
        phoneCode: '+213',
        flag: getImagePath('DZ'),
    },
    KP: {
        nameCode: 'KP',
        name: 'КНДР (Северная Корея)',
        phoneCode: '+850',
        flag: getImagePath('KP'),
    },
    PT: {
        nameCode: 'PT',
        name: 'Португалия',
        phoneCode: '+351',
        flag: getImagePath('PT'),
    },
    ZW: {
        nameCode: 'ZW',
        name: 'Зимбабве',
        phoneCode: '+263',
        flag: getImagePath('ZW'),
    },
    EC: {
        nameCode: 'EC',
        name: 'Эквадор',
        phoneCode: '+593',
        flag: getImagePath('EC'),
    },
    KR: {
        nameCode: 'KR',
        name: 'Республика Корея (Южная Корея)',
        phoneCode: '+82',
        flag: getImagePath('KR'),
    },
    PW: {
        nameCode: 'PW',
        name: 'Палау',
        phoneCode: '+680',
        flag: getImagePath('PW'),
    },
};

export { CountriesData };
