// @beeline/design-tokens/js/tokens/components/selectionControl/selectionControl.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const selectionControlWidth = "24px";
export const selectionControlHeight = "24px";
export const selectionControlOpacityDisabled = 0.48;
export const selectionControlBackgroundColor = "transparent";
export const selectionControlBackgroundColorHover = "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))";
export const selectionControlBackgroundColorPressed = "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))";
export const selectionControlBorderWidth = "1px";
export const selectionControlBorderStyle = "solid";
export const selectionControlBorderColor = "var(--color-border, rgba(25, 28, 52, 0.18))";
export const selectionControlBorderColorFocus = "var(--color-border-focus, #212121)";
export const selectionControlCheckmarkColor = "transparent";
export const selectionControlCheckmarkColorHover = "var(--color-text-disabled, rgba(25, 28, 52, 0.48))";
export const selectionControlCheckmarkColorPressed = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const selectionControlHoverBackgroundColor = "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))";
export const selectionControlHoverCheckmarkColor = "var(--color-text-disabled, rgba(25, 28, 52, 0.48))";
export const selectionControlPressedBackgroundColor = "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))";
export const selectionControlPressedCheckmarkColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const selectionControlFocusedBorderColor = "var(--color-border-focus, #212121)";
export const selectionControlDisabledOpacity = 0.48;
export const selectionControlSelectedBackgroundColor = "var(--color-background-brand, #fdd835)";
export const selectionControlSelectedBackgroundColorHover = "var(--color-background-brand-hover, #fdc435)";
export const selectionControlSelectedBackgroundColorPressed = "var(--color-background-brand-pressed, #fdb435)";
export const selectionControlSelectedBorderColor = "transparent";
export const selectionControlSelectedBorderColorFocus = "var(--color-border-focus, #212121)";
export const selectionControlSelectedCheckmarkColor = "var(--color-text-black-active, rgba(9, 11, 22, 0.94))";
export const selectionControlSelectedHoverBackgroundColor = "var(--color-background-brand-hover, #fdc435)";
export const selectionControlSelectedPressedBackgroundColor = "var(--color-background-brand-pressed, #fdb435)";
export const selectionControlSelectedFocusedBorderColor = "var(--color-border-focus, #212121)";
