// @beeline/design-tokens/js/tokens/components/selectionControl/selectionControl.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	selectionControlWidth: "24px",
	selectionControlHeight: "24px",
	selectionControlOpacityDisabled: 0.48,
	selectionControlBackgroundColor: "transparent",
	selectionControlBackgroundColorHover: "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))",
	selectionControlBackgroundColorPressed: "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))",
	selectionControlBorderWidth: "1px",
	selectionControlBorderStyle: "solid",
	selectionControlBorderColor: "var(--color-border, rgba(25, 28, 52, 0.18))",
	selectionControlBorderColorFocus: "var(--color-border-focus, #212121)",
	selectionControlCheckmarkColor: "transparent",
	selectionControlCheckmarkColorHover: "var(--color-text-disabled, rgba(25, 28, 52, 0.48))",
	selectionControlCheckmarkColorPressed: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	selectionControlHoverBackgroundColor: "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))",
	selectionControlHoverCheckmarkColor: "var(--color-text-disabled, rgba(25, 28, 52, 0.48))",
	selectionControlPressedBackgroundColor: "var(--color-control-background-pressed, rgba(25, 28, 52, 0.24))",
	selectionControlPressedCheckmarkColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	selectionControlFocusedBorderColor: "var(--color-border-focus, #212121)",
	selectionControlDisabledOpacity: 0.48,
	selectionControlSelectedBackgroundColor: "var(--color-background-brand, #fdd835)",
	selectionControlSelectedBackgroundColorHover: "var(--color-background-brand-hover, #fdc435)",
	selectionControlSelectedBackgroundColorPressed: "var(--color-background-brand-pressed, #fdb435)",
	selectionControlSelectedBorderColor: "transparent",
	selectionControlSelectedBorderColorFocus: "var(--color-border-focus, #212121)",
	selectionControlSelectedCheckmarkColor: "var(--color-text-black-active, rgba(9, 11, 22, 0.94))",
	selectionControlSelectedHoverBackgroundColor: "var(--color-background-brand-hover, #fdc435)",
	selectionControlSelectedPressedBackgroundColor: "var(--color-background-brand-pressed, #fdb435)",
	selectionControlSelectedFocusedBorderColor: "var(--color-border-focus, #212121)",
};
