// @beeline/design-system-react/build/components/Button/ProgressButton/utils.js
import { PROGRESS_BUTTON_STROKE_WIDTH, DEFAULT_SETTINGS } from './consts.js';

/** Рассчёт размеров контейнера и радиуса границ для отрисовки в нём svg */
const calcSettings = (buttonRef) => {
    const buttonObj = buttonRef?.current;
    if (buttonObj) {
        const buttonObjRect = buttonObj.getBoundingClientRect();
        return {
            width: buttonObjRect.width,
            height: buttonObjRect.height,
            startPos: buttonObj.offsetWidth / 2,
            strokeWidth: PROGRESS_BUTTON_STROKE_WIDTH,
            borderRadius: 12,
        };
    }
    return DEFAULT_SETTINGS;
};
const calcPathDefinition = (settings) => `
	M ${settings.startPos} ${settings.strokeWidth} 
	H${settings.width - settings.borderRadius}
	a${settings.borderRadius},${settings.borderRadius} 0 0 1 ${settings.borderRadius},${settings.borderRadius}
	V${settings.height - settings.borderRadius}
	a${settings.borderRadius},${settings.borderRadius} 0 0 1 -${settings.borderRadius},${settings.borderRadius}
	H${settings.borderRadius + settings.strokeWidth}
	a${settings.borderRadius},${settings.borderRadius} 0 0 1 -${settings.borderRadius},-${settings.borderRadius}
	V${settings.borderRadius + settings.strokeWidth}
	a${settings.borderRadius},${settings.borderRadius} 0 0 1 ${settings.borderRadius},-${settings.borderRadius}
	H ${settings.startPos}
`;
/**
 * проверяет является ли переданный объект оьектом ошибки
 *
 * @param err потенциальный объект ошибки
 */
const isErrorObject = (err) => !!err && typeof err === 'object' && Object.prototype.toString.call(err) === '[object Error]';

export { calcPathDefinition, calcSettings, isErrorObject };
