// @beeline/design-system-react/extracted-styles/esm/components/PhoneInput/Flags/index.js
import SvgAd from './AD.js';
import SvgAe from './AE.js';
import SvgAf from './AF.js';
import SvgAg from './AG.js';
import SvgAi from './AI.js';
import SvgAl from './AL.js';
import SvgAm from './AM.js';
import SvgAo from './AO.js';
import SvgAq from './AQ.js';
import SvgAr from './AR.js';
import SvgAs from './AS.js';
import SvgAt from './AT.js';
import SvgAu from './AU.js';
import SvgAw from './AW.js';
import SvgAx from './AX.js';
import SvgAz from './AZ.js';
import SvgBa from './BA.js';
import SvgBb from './BB.js';
import SvgBd from './BD.js';
import SvgBe from './BE.js';
import SvgBf from './BF.js';
import SvgBg from './BG.js';
import SvgBh from './BH.js';
import SvgBi from './BI.js';
import SvgBj from './BJ.js';
import SvgBl from './BL.js';
import SvgBm from './BM.js';
import SvgBn from './BN.js';
import SvgBo from './BO.js';
import SvgBqBo from './BQ-BO.js';
import SvgBqSa from './BQ-SA.js';
import SvgBqSe from './BQ-SE.js';
import SvgBr from './BR.js';
import SvgBs from './BS.js';
import SvgBt from './BT.js';
import SvgBv from './BV.js';
import SvgBw from './BW.js';
import SvgBy from './BY.js';
import SvgBz from './BZ.js';
import SvgCa from './CA.js';
import SvgCc from './CC.js';
import SvgCd from './CD.js';
import SvgCf from './CF.js';
import SvgCg from './CG.js';
import SvgCh from './CH.js';
import SvgCi from './CI.js';
import SvgCk from './CK.js';
import SvgCl from './CL.js';
import SvgCm from './CM.js';
import SvgCn from './CN.js';
import SvgCo from './CO.js';
import SvgCr from './CR.js';
import SvgCu from './CU.js';
import SvgCv from './CV.js';
import SvgCw from './CW.js';
import SvgCx from './CX.js';
import SvgCy from './CY.js';
import SvgCz from './CZ.js';
import SvgDe from './DE.js';
import SvgDj from './DJ.js';
import SvgDk from './DK.js';
import SvgDm from './DM.js';
import SvgDo from './DO.js';
import SvgDz from './DZ.js';
import SvgEc from './EC.js';
import SvgEe from './EE.js';
import SvgEg from './EG.js';
import SvgEh from './EH.js';
import SvgEr from './ER.js';
import SvgEs from './ES.js';
import SvgEt from './ET.js';
import SvgFi from './FI.js';
import SvgFj from './FJ.js';
import SvgFk from './FK.js';
import SvgFm from './FM.js';
import SvgFo from './FO.js';
import SvgFr from './FR.js';
import SvgGa from './GA.js';
import SvgGbEng from './GB-ENG.js';
import SvgGbNir from './GB-NIR.js';
import SvgGbSct from './GB-SCT.js';
import SvgGbUkm from './GB-UKM.js';
import SvgGbWls from './GB-WLS.js';
import SvgGd from './GD.js';
import SvgGe from './GE.js';
import SvgGf from './GF.js';
import SvgGg from './GG.js';
import SvgGh from './GH.js';
import SvgGi from './GI.js';
import SvgGl from './GL.js';
import SvgGm from './GM.js';
import SvgGn from './GN.js';
import SvgGp from './GP.js';
import SvgGq from './GQ.js';
import SvgGr from './GR.js';
import SvgGs from './GS.js';
import SvgGt from './GT.js';
import SvgGu from './GU.js';
import SvgGw from './GW.js';
import SvgGy from './GY.js';
import SvgHk from './HK.js';
import SvgHm from './HM.js';
import SvgHn from './HN.js';
import SvgHr from './HR.js';
import SvgHt from './HT.js';
import SvgHu from './HU.js';
import SvgId from './ID.js';
import SvgIe from './IE.js';
import SvgIl from './IL.js';
import SvgIm from './IM.js';
import SvgIn from './IN.js';
import SvgInt from './INT.js';
import SvgIo from './IO.js';
import SvgIq from './IQ.js';
import SvgIr from './IR.js';
import SvgIs from './IS.js';
import SvgIt from './IT.js';
import SvgJe from './JE.js';
import SvgJm from './JM.js';
import SvgJo from './JO.js';
import SvgJp from './JP.js';
import SvgKe from './KE.js';
import SvgKg from './KG.js';
import SvgKh from './KH.js';
import SvgKi from './KI.js';
import SvgKm from './KM.js';
import SvgKn from './KN.js';
import SvgKp from './KP.js';
import SvgKr from './KR.js';
import SvgKw from './KW.js';
import SvgKy from './KY.js';
import SvgKz from './KZ.js';
import SvgLa from './LA.js';
import SvgLb from './LB.js';
import SvgLc from './LC.js';
import SvgLi from './LI.js';
import SvgLk from './LK.js';
import SvgLr from './LR.js';
import SvgLs from './LS.js';
import SvgLt from './LT.js';
import SvgLu from './LU.js';
import SvgLv from './LV.js';
import SvgLy from './LY.js';
import SvgMa from './MA.js';
import SvgMc from './MC.js';
import SvgMd from './MD.js';
import SvgMe from './ME.js';
import SvgMf from './MF.js';
import SvgMg from './MG.js';
import SvgMh from './MH.js';
import SvgMk from './MK.js';
import SvgMl from './ML.js';
import SvgMm from './MM.js';
import SvgMn from './MN.js';
import SvgMo from './MO.js';
import SvgMp from './MP.js';
import SvgMq from './MQ.js';
import SvgMr from './MR.js';
import SvgMs from './MS.js';
import SvgMt from './MT.js';
import SvgMu from './MU.js';
import SvgMv from './MV.js';
import SvgMw from './MW.js';
import SvgMx from './MX.js';
import SvgMy from './MY.js';
import SvgMz from './MZ.js';
import SvgNa from './NA.js';
import SvgNc from './NC.js';
import SvgNe from './NE.js';
import SvgNf from './NF.js';
import SvgNg from './NG.js';
import SvgNi from './NI.js';
import SvgNl from './NL.js';
import SvgNo from './NO.js';
import SvgNp from './NP.js';
import SvgNr from './NR.js';
import SvgNu from './NU.js';
import SvgNz from './NZ.js';
import SvgOm from './OM.js';
import SvgPa from './PA.js';
import SvgPe from './PE.js';
import SvgPf from './PF.js';
import SvgPg from './PG.js';
import SvgPh from './PH.js';
import SvgPk from './PK.js';
import SvgPl from './PL.js';
import SvgPm from './PM.js';
import SvgPn from './PN.js';
import SvgPr from './PR.js';
import SvgPs from './PS.js';
import SvgPt from './PT.js';
import SvgPw from './PW.js';
import SvgPy from './PY.js';
import SvgQa from './QA.js';
import SvgRe from './RE.js';
import SvgRo from './RO.js';
import SvgRs from './RS.js';
import SvgRu from './RU.js';
import SvgRw from './RW.js';
import SvgSa from './SA.js';
import SvgSb from './SB.js';
import SvgSc from './SC.js';
import SvgSd from './SD.js';
import SvgSe from './SE.js';
import SvgSg from './SG.js';
import SvgSh from './SH.js';
import SvgSi from './SI.js';
import SvgSj from './SJ.js';
import SvgSk from './SK.js';
import SvgSl from './SL.js';
import SvgSm from './SM.js';
import SvgSn from './SN.js';
import SvgSo from './SO.js';
import SvgSr from './SR.js';
import SvgSs from './SS.js';
import SvgSt from './ST.js';
import SvgSv from './SV.js';
import SvgSx from './SX.js';
import SvgSy from './SY.js';
import SvgSz from './SZ.js';
import SvgTc from './TC.js';
import SvgTd from './TD.js';
import SvgTf from './TF.js';
import SvgTg from './TG.js';
import SvgTh from './TH.js';
import SvgTj from './TJ.js';
import SvgTk from './TK.js';
import SvgTl from './TL.js';
import SvgTm from './TM.js';
import SvgTn from './TN.js';
import SvgTo from './TO.js';
import SvgTr from './TR.js';
import SvgTt from './TT.js';
import SvgTv from './TV.js';
import SvgTw from './TW.js';
import SvgTz from './TZ.js';
import SvgUa from './UA.js';
import SvgUg from './UG.js';
import SvgUm from './UM.js';
import SvgUs from './US.js';
import SvgUy from './UY.js';
import SvgUz from './UZ.js';
import SvgVa from './VA.js';
import SvgVc from './VC.js';
import SvgVe from './VE.js';
import SvgVg from './VG.js';
import SvgVi from './VI.js';
import SvgVn from './VN.js';
import SvgVu from './VU.js';
import SvgWf from './WF.js';
import SvgWs from './WS.js';
import SvgYe from './YE.js';
import SvgYtUnf from './YT-UNF.js';
import SvgZa from './ZA.js';
import SvgZm from './ZM.js';
import SvgZw from './ZW.js';

const FlagIconsByCode = {
    AD: SvgAd,
    AE: SvgAe,
    AF: SvgAf,
    AG: SvgAg,
    AI: SvgAi,
    AL: SvgAl,
    AM: SvgAm,
    AO: SvgAo,
    AQ: SvgAq,
    AR: SvgAr,
    AS: SvgAs,
    AT: SvgAt,
    AU: SvgAu,
    AW: SvgAw,
    AX: SvgAx,
    AZ: SvgAz,
    BA: SvgBa,
    BB: SvgBb,
    BD: SvgBd,
    BE: SvgBe,
    BF: SvgBf,
    BG: SvgBg,
    BH: SvgBh,
    BI: SvgBi,
    BJ: SvgBj,
    BL: SvgBl,
    BM: SvgBm,
    BN: SvgBn,
    BO: SvgBo,
    'BQ-BO': SvgBqBo,
    'BQ-SA': SvgBqSa,
    'BQ-SE': SvgBqSe,
    BR: SvgBr,
    BS: SvgBs,
    BT: SvgBt,
    BV: SvgBv,
    BW: SvgBw,
    BY: SvgBy,
    BZ: SvgBz,
    CA: SvgCa,
    CC: SvgCc,
    CD: SvgCd,
    CF: SvgCf,
    CG: SvgCg,
    CH: SvgCh,
    CI: SvgCi,
    CK: SvgCk,
    CL: SvgCl,
    CM: SvgCm,
    CN: SvgCn,
    CO: SvgCo,
    CR: SvgCr,
    CU: SvgCu,
    CV: SvgCv,
    CW: SvgCw,
    CX: SvgCx,
    CY: SvgCy,
    CZ: SvgCz,
    DE: SvgDe,
    DJ: SvgDj,
    DK: SvgDk,
    DM: SvgDm,
    DO: SvgDo,
    DZ: SvgDz,
    EC: SvgEc,
    EE: SvgEe,
    EG: SvgEg,
    EH: SvgEh,
    ER: SvgEr,
    ES: SvgEs,
    ET: SvgEt,
    FI: SvgFi,
    FJ: SvgFj,
    FK: SvgFk,
    FM: SvgFm,
    FO: SvgFo,
    FR: SvgFr,
    GA: SvgGa,
    'GB-ENG': SvgGbEng,
    'GB-NIR': SvgGbNir,
    'GB-SCT': SvgGbSct,
    'GB-UKM': SvgGbUkm,
    'GB-WLS': SvgGbWls,
    GD: SvgGd,
    GE: SvgGe,
    GF: SvgGf,
    GG: SvgGg,
    GH: SvgGh,
    GI: SvgGi,
    GL: SvgGl,
    GM: SvgGm,
    GN: SvgGn,
    GP: SvgGp,
    GQ: SvgGq,
    GR: SvgGr,
    GS: SvgGs,
    GT: SvgGt,
    GU: SvgGu,
    GW: SvgGw,
    GY: SvgGy,
    HK: SvgHk,
    HM: SvgHm,
    HN: SvgHn,
    HR: SvgHr,
    HT: SvgHt,
    HU: SvgHu,
    ID: SvgId,
    IE: SvgIe,
    IL: SvgIl,
    IM: SvgIm,
    IN: SvgIn,
    INT: SvgInt,
    IO: SvgIo,
    IQ: SvgIq,
    IR: SvgIr,
    IS: SvgIs,
    IT: SvgIt,
    JE: SvgJe,
    JM: SvgJm,
    JO: SvgJo,
    JP: SvgJp,
    KE: SvgKe,
    KG: SvgKg,
    KH: SvgKh,
    KI: SvgKi,
    KM: SvgKm,
    KN: SvgKn,
    KP: SvgKp,
    KR: SvgKr,
    KW: SvgKw,
    KY: SvgKy,
    KZ: SvgKz,
    LA: SvgLa,
    LB: SvgLb,
    LC: SvgLc,
    LI: SvgLi,
    LK: SvgLk,
    LR: SvgLr,
    LS: SvgLs,
    LT: SvgLt,
    LU: SvgLu,
    LV: SvgLv,
    LY: SvgLy,
    MA: SvgMa,
    MC: SvgMc,
    MD: SvgMd,
    ME: SvgMe,
    MF: SvgMf,
    MG: SvgMg,
    MH: SvgMh,
    MK: SvgMk,
    ML: SvgMl,
    MM: SvgMm,
    MN: SvgMn,
    MO: SvgMo,
    MP: SvgMp,
    MQ: SvgMq,
    MR: SvgMr,
    MS: SvgMs,
    MT: SvgMt,
    MU: SvgMu,
    MV: SvgMv,
    MW: SvgMw,
    MX: SvgMx,
    MY: SvgMy,
    MZ: SvgMz,
    NA: SvgNa,
    NC: SvgNc,
    NE: SvgNe,
    NF: SvgNf,
    NG: SvgNg,
    NI: SvgNi,
    NL: SvgNl,
    NO: SvgNo,
    NP: SvgNp,
    NR: SvgNr,
    NU: SvgNu,
    NZ: SvgNz,
    OM: SvgOm,
    PA: SvgPa,
    PE: SvgPe,
    PF: SvgPf,
    PG: SvgPg,
    PH: SvgPh,
    PK: SvgPk,
    PL: SvgPl,
    PM: SvgPm,
    PN: SvgPn,
    PR: SvgPr,
    PS: SvgPs,
    PT: SvgPt,
    PW: SvgPw,
    PY: SvgPy,
    QA: SvgQa,
    RE: SvgRe,
    RO: SvgRo,
    RS: SvgRs,
    RU: SvgRu,
    RW: SvgRw,
    SA: SvgSa,
    SB: SvgSb,
    SC: SvgSc,
    SD: SvgSd,
    SE: SvgSe,
    SG: SvgSg,
    SH: SvgSh,
    SI: SvgSi,
    SJ: SvgSj,
    SK: SvgSk,
    SL: SvgSl,
    SM: SvgSm,
    SN: SvgSn,
    SO: SvgSo,
    SR: SvgSr,
    SS: SvgSs,
    ST: SvgSt,
    SV: SvgSv,
    SX: SvgSx,
    SY: SvgSy,
    SZ: SvgSz,
    TC: SvgTc,
    TD: SvgTd,
    TF: SvgTf,
    TG: SvgTg,
    TH: SvgTh,
    TJ: SvgTj,
    TK: SvgTk,
    TL: SvgTl,
    TM: SvgTm,
    TN: SvgTn,
    TO: SvgTo,
    TR: SvgTr,
    TT: SvgTt,
    TV: SvgTv,
    TW: SvgTw,
    TZ: SvgTz,
    UA: SvgUa,
    UG: SvgUg,
    UM: SvgUm,
    US: SvgUs,
    UY: SvgUy,
    UZ: SvgUz,
    VA: SvgVa,
    VC: SvgVc,
    VE: SvgVe,
    VG: SvgVg,
    VI: SvgVi,
    VN: SvgVn,
    VU: SvgVu,
    WF: SvgWf,
    WS: SvgWs,
    YE: SvgYe,
    'YT-UNF': SvgYtUnf,
    ZA: SvgZa,
    ZM: SvgZm,
    ZW: SvgZw,
};

export { FlagIconsByCode };
