// @beeline/design-tokens/js/tokens/components/slider/slider.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	sliderTrackBackgroundColor: "var(--color-background-brand, #fdd835)",
	sliderTrackBorderRadius: "12px",
	sliderTrackHoverBackgroundColor: "var(--color-background-brand-hover, #fdc435)",
	sliderRailBackgroundColor: "var(--color-control-background, rgba(25, 28, 52, 0.1))",
	sliderRailBorderRadius: "12px",
	sliderRailHeight: "6px",
	sliderRailHoverBackgroundColor: "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))",
	sliderThumbBackgroundColor: "var(--color-background-brand, #fdd835)",
	sliderThumbBorderRadius: "50%",
	sliderThumbBorderWidth: "7px",
	sliderThumbBorderColor: "var(--color-background-white, #ffffff)",
	sliderThumbWidth: "24px",
	sliderThumbHeight: "24px",
	sliderThumbShadow: "0px 2px 10px rgba(0, 0, 0, 0.08), 0px 2px 8px rgba(0, 0, 0, 0.08)",
	sliderThumbHoverBackgroundColor: "var(--color-background-brand-hover, #fdc435)",
	sliderThumbPressedBackgroundColor: "var(--color-background-brand-pressed, #fdb435)",
	sliderMarksMarginTop: "16px",
	sliderMarksFontSize: "15px",
	sliderMarksFontWeight: 400,
	sliderMarksFontLineHeight: "18px",
	sliderMarksFontLetterSpacing: "0.2px",
	sliderMarksColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	sliderMarksSelectedFontSize: "15px",
	sliderMarksSelectedFontWeight: 500,
	sliderMarksSelectedFontLineHeight: "20px",
	sliderMarksSelectedFontLetterSpacing: "0.2px",
	sliderMarksSelectedColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	sliderDisabledOpacity: 0.48,
};
