// @beeline/design-tokens/js/tokens/components/slider/slider.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const sliderTrackBackgroundColor = "var(--color-background-brand, #fdd835)";
export const sliderTrackBorderRadius = "12px";
export const sliderTrackHoverBackgroundColor = "var(--color-background-brand-hover, #fdc435)";
export const sliderRailBackgroundColor = "var(--color-control-background, rgba(25, 28, 52, 0.1))";
export const sliderRailBorderRadius = "12px";
export const sliderRailHeight = "6px";
export const sliderRailHoverBackgroundColor = "var(--color-control-background-hover, rgba(25, 28, 52, 0.14))";
export const sliderThumbBackgroundColor = "var(--color-background-brand, #fdd835)";
export const sliderThumbBorderRadius = "50%";
export const sliderThumbBorderWidth = "7px";
export const sliderThumbBorderColor = "var(--color-background-white, #ffffff)";
export const sliderThumbWidth = "24px";
export const sliderThumbHeight = "24px";
export const sliderThumbShadow = "0px 2px 10px rgba(0, 0, 0, 0.08), 0px 2px 8px rgba(0, 0, 0, 0.08)";
export const sliderThumbHoverBackgroundColor = "var(--color-background-brand-hover, #fdc435)";
export const sliderThumbPressedBackgroundColor = "var(--color-background-brand-pressed, #fdb435)";
export const sliderMarksMarginTop = "16px";
export const sliderMarksFontSize = "15px";
export const sliderMarksFontWeight = 400;
export const sliderMarksFontLineHeight = "18px";
export const sliderMarksFontLetterSpacing = "0.2px";
export const sliderMarksColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const sliderMarksSelectedFontSize = "15px";
export const sliderMarksSelectedFontWeight = 500;
export const sliderMarksSelectedFontLineHeight = "20px";
export const sliderMarksSelectedFontLetterSpacing = "0.2px";
export const sliderMarksSelectedColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const sliderDisabledOpacity = 0.48;
