// @beeline/design-tokens/js/tokens/components/stepper/stepper.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	stepperStepPaddingHorizontal: "24px",
	stepperStepTitleFontSize: "17px",
	stepperStepTitleFontWeight: 400,
	stepperStepTitleFontLineHeight: "22px",
	stepperStepTitleFontLetterSpacing: "0.2px",
	stepperStepTextColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	stepperStepTextFontSize: "13px",
	stepperStepTextFontWeight: 400,
	stepperStepTextFontLineHeight: "16px",
	stepperStepTextFontLetterSpacing: "0.2px",
	stepperStepNumberWidth: "24px",
	stepperStepNumberHeight: "24px",
	stepperStepNumberBorderRadius: "50%",
	stepperStepNumberSpacing: "12px",
	stepperStepIconSize: "24px",
	stepperStepIconSpacing: "12px",
	stepperStepActiveTitleColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	stepperStepActiveNumberColor: "var(--color-text-black-active, rgba(9, 11, 22, 0.94))",
	stepperStepActiveNumberBackgroundColor: "var(--color-background-brand, #fdd835)",
	stepperStepVisitedTitleColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	stepperStepVisitedNumberColor: "var(--color-text-active, rgba(9, 11, 22, 0.94))",
	stepperStepVisitedNumberBorderColor: "var(--color-border-focus, #212121)",
	stepperStepNotVisitedTitleColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	stepperStepNotVisitedNumberColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	stepperStepNotVisitedNumberBorderColor: "var(--color-border, rgba(25, 28, 52, 0.18))",
	stepperStepDisabledOpacity: 0.48,
	stepperStepSuccessIconColor: "var(--color-status-success, #2ca853)",
	stepperStepErrorIconColor: "var(--color-status-error, #ff5555)",
	stepperVerticalStepIndicatorWidth: "2px",
	stepperVerticalStepIndicatorBorderRadiusTopRight: "4px",
	stepperVerticalStepIndicatorBorderRadiusBottomRight: "4px",
	stepperHorizontalStepHeight: "56px",
	stepperHorizontalStepPaddingBottom: "2px",
	stepperHorizontalStepIndicatorHeight: "2px",
	stepperHorizontalStepIndicatorBorderRadiusTopLeft: "4px",
	stepperHorizontalStepIndicatorBorderRadiusTopRight: "4px",
};
