// @beeline/design-tokens/js/tokens/components/stepper/stepper.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const stepperStepPaddingHorizontal = "24px";
export const stepperStepTitleFontSize = "17px";
export const stepperStepTitleFontWeight = 400;
export const stepperStepTitleFontLineHeight = "22px";
export const stepperStepTitleFontLetterSpacing = "0.2px";
export const stepperStepTextColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const stepperStepTextFontSize = "13px";
export const stepperStepTextFontWeight = 400;
export const stepperStepTextFontLineHeight = "16px";
export const stepperStepTextFontLetterSpacing = "0.2px";
export const stepperStepNumberWidth = "24px";
export const stepperStepNumberHeight = "24px";
export const stepperStepNumberBorderRadius = "50%";
export const stepperStepNumberSpacing = "12px";
export const stepperStepIconSize = "24px";
export const stepperStepIconSpacing = "12px";
export const stepperStepActiveTitleColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const stepperStepActiveNumberColor = "var(--color-text-black-active, rgba(9, 11, 22, 0.94))";
export const stepperStepActiveNumberBackgroundColor = "var(--color-background-brand, #fdd835)";
export const stepperStepVisitedTitleColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const stepperStepVisitedNumberColor = "var(--color-text-active, rgba(9, 11, 22, 0.94))";
export const stepperStepVisitedNumberBorderColor = "var(--color-border-focus, #212121)";
export const stepperStepNotVisitedTitleColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const stepperStepNotVisitedNumberColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const stepperStepNotVisitedNumberBorderColor = "var(--color-border, rgba(25, 28, 52, 0.18))";
export const stepperStepDisabledOpacity = 0.48;
export const stepperStepSuccessIconColor = "var(--color-status-success, #2ca853)";
export const stepperStepErrorIconColor = "var(--color-status-error, #ff5555)";
export const stepperVerticalStepIndicatorWidth = "2px";
export const stepperVerticalStepIndicatorBorderRadiusTopRight = "4px";
export const stepperVerticalStepIndicatorBorderRadiusBottomRight = "4px";
export const stepperHorizontalStepHeight = "56px";
export const stepperHorizontalStepPaddingBottom = "2px";
export const stepperHorizontalStepIndicatorHeight = "2px";
export const stepperHorizontalStepIndicatorBorderRadiusTopLeft = "4px";
export const stepperHorizontalStepIndicatorBorderRadiusTopRight = "4px";
