// @beeline/design-tokens/js/tokens/components/textfield/textfield.common.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

module.exports = {
	textfieldInputHeight: "20px",
	textfieldIconColor: "var(--color-text-inactive, rgba(25, 28, 52, 0.7))",
	textfieldIconColorDisabled: "var(--color-text-disabled, rgba(25, 28, 52, 0.48))",
	textfieldIconColorError: "var(--color-status-error, #ff5555)",
	textfieldIconLeftMarginRight: "8px",
	textfieldIconRightMarginLeft: "8px",
	textfieldIconSpacing: "8px",
	textfieldIconbuttonSpacing: "8px",
	textfieldDisabledIconColor: "var(--color-text-disabled, rgba(25, 28, 52, 0.48))",
	textfieldInvalidIconColor: "var(--color-status-error, #ff5555)",
	textfieldSmallHeight: "40px",
	textfieldSmallIconSize: "18px",
	textfieldSmallTextMarginTop: "10px",
	textfieldSmallTextMarginBottom: "10px",
	textfieldMediumHeight: "48px",
	textfieldMediumIconSize: "20px",
	textfieldMediumTextMarginTop: "22px",
	textfieldMediumTextMarginBottom: "6px",
	textfieldMediumLabelMarginTop: "14px",
	textfieldMediumLabelMarginTopFloated: "6px",
	textfieldMediumLabelFloatedMarginTop: "6px",
	textfieldMediumWithoutLabelTextMarginTop: "14px",
	textfieldMediumWithoutLabelTextMarginBottom: "14px",
	textfieldLargeHeight: "56px",
	textfieldLargeIconSize: "24px",
	textfieldLargeTextMarginTop: "26px",
	textfieldLargeTextMarginBottom: "10px",
	textfieldLargeTextFontSize: "17px",
	textfieldLargeTextFontWeight: 400,
	textfieldLargeTextFontLineHeight: "22px",
	textfieldLargeTextFontLetterSpacing: "0.2px",
	textfieldLargeLabelMarginTop: "18px",
	textfieldLargeLabelMarginTopFloated: "8px",
	textfieldLargeLabelMarginBottomFloated: "2px",
	textfieldLargeLabelFontSize: "17px",
	textfieldLargeLabelFontSizeFloated: "13px",
	textfieldLargeLabelFontWeight: 400,
	textfieldLargeLabelFontLineHeight: "22px",
	textfieldLargeLabelFontLineHeightFloated: "16px",
	textfieldLargeLabelFloatedMarginTop: "8px",
	textfieldLargeLabelFloatedMarginBottom: "2px",
	textfieldLargeLabelFloatedFontSize: "13px",
	textfieldLargeLabelFloatedFontLineHeight: "16px",
	textfieldLargeWithoutLabelTextMarginTop: "18px",
	textfieldLargeWithoutLabelTextMarginBottom: "18px",
	textfieldWithLabelMediumTextMarginTop: "22px",
	textfieldWithLabelMediumTextMarginBottom: "6px",
	textfieldWithLabelLargeTextMarginTop: "26px",
	textfieldWithLabelLargeTextMarginBottom: "10px",
	textfieldWithoutLabelMediumTextMarginVertical: "14px",
	textfieldWithoutLabelLargeTextMarginVertical: "18px",
};
