// @beeline/design-tokens/js/tokens/components/textfield/textfield.esm.js
/**
 * Do not edit directly
 * Generated on Tue, 21 Apr 2026 09:05:43 GMT
 */

export const textfieldInputHeight = "20px";
export const textfieldIconColor = "var(--color-text-inactive, rgba(25, 28, 52, 0.7))";
export const textfieldIconColorDisabled = "var(--color-text-disabled, rgba(25, 28, 52, 0.48))";
export const textfieldIconColorError = "var(--color-status-error, #ff5555)";
export const textfieldIconLeftMarginRight = "8px";
export const textfieldIconRightMarginLeft = "8px";
export const textfieldIconSpacing = "8px";
export const textfieldIconbuttonSpacing = "8px";
export const textfieldDisabledIconColor = "var(--color-text-disabled, rgba(25, 28, 52, 0.48))";
export const textfieldInvalidIconColor = "var(--color-status-error, #ff5555)";
export const textfieldSmallHeight = "40px";
export const textfieldSmallIconSize = "18px";
export const textfieldSmallTextMarginTop = "10px";
export const textfieldSmallTextMarginBottom = "10px";
export const textfieldMediumHeight = "48px";
export const textfieldMediumIconSize = "20px";
export const textfieldMediumTextMarginTop = "22px";
export const textfieldMediumTextMarginBottom = "6px";
export const textfieldMediumLabelMarginTop = "14px";
export const textfieldMediumLabelMarginTopFloated = "6px";
export const textfieldMediumLabelFloatedMarginTop = "6px";
export const textfieldMediumWithoutLabelTextMarginTop = "14px";
export const textfieldMediumWithoutLabelTextMarginBottom = "14px";
export const textfieldLargeHeight = "56px";
export const textfieldLargeIconSize = "24px";
export const textfieldLargeTextMarginTop = "26px";
export const textfieldLargeTextMarginBottom = "10px";
export const textfieldLargeTextFontSize = "17px";
export const textfieldLargeTextFontWeight = 400;
export const textfieldLargeTextFontLineHeight = "22px";
export const textfieldLargeTextFontLetterSpacing = "0.2px";
export const textfieldLargeLabelMarginTop = "18px";
export const textfieldLargeLabelMarginTopFloated = "8px";
export const textfieldLargeLabelMarginBottomFloated = "2px";
export const textfieldLargeLabelFontSize = "17px";
export const textfieldLargeLabelFontSizeFloated = "13px";
export const textfieldLargeLabelFontWeight = 400;
export const textfieldLargeLabelFontLineHeight = "22px";
export const textfieldLargeLabelFontLineHeightFloated = "16px";
export const textfieldLargeLabelFloatedMarginTop = "8px";
export const textfieldLargeLabelFloatedMarginBottom = "2px";
export const textfieldLargeLabelFloatedFontSize = "13px";
export const textfieldLargeLabelFloatedFontLineHeight = "16px";
export const textfieldLargeWithoutLabelTextMarginTop = "18px";
export const textfieldLargeWithoutLabelTextMarginBottom = "18px";
export const textfieldWithLabelMediumTextMarginTop = "22px";
export const textfieldWithLabelMediumTextMarginBottom = "6px";
export const textfieldWithLabelLargeTextMarginTop = "26px";
export const textfieldWithLabelLargeTextMarginBottom = "10px";
export const textfieldWithoutLabelMediumTextMarginVertical = "14px";
export const textfieldWithoutLabelLargeTextMarginVertical = "18px";
